SAMPLES, = glob_wildcards("data/{sample}_R1_001.fastq.gz")

rule all:
	input:
		expand("data/FASTQC/{sample}_{read}_001_fastqc.zip", sample=SAMPLES, read=["R1", "R2"]),
		expand("trimmed_data/FASTQC/{sample}_{read}_trimmed_fastqc.zip", sample=SAMPLES, read=["R1", "R2"]),
		"data/multiqc_report.html",
		"trimmed_data/multiqc_report.html",,
#		expand("metaxa2/{sample}", sample=SAMPLES),
#		expand("metaxa2/{sample}.taxonomy.txt", sample=SAMPLES),
#		expand("metaxa2/{sample}_level_6.txt", sample=SAMPLES),
		"resfinder_db/resfinder.1.bt2",
		expand("mapped_reads/{sample}.bam", sample=SAMPLES),
		expand("sorted_reads/{sample}.bam", sample=SAMPLES),
		expand("sorted_reads/{sample}.bam.bai", sample=SAMPLES),
		expand("resfinder_out/{sample}_counts", sample=SAMPLES),
		"CARD/protein_fasta_protein_homolog_model.dmnd",
		"CARD/CARD_results"

rule fastqc_raw:
	input:
		"data/{sample}_{read}_001.fastq.gz"
	output:
		"data/FASTQC/{sample}_{read}_001_fastqc.zip"
	message:
		"-- Quality check of raw data with Fastqc --"
	conda:
		"envs/QC_env.yml"
	threads:
		2
	shell:
              "fastqc --quiet -t {threads} --outdir data/FASTQC -f fastq {input}"

rule multiqc_raw:
	input:
		expand("data/FASTQC/{sample}_{read}_001_fastqc.zip", sample=SAMPLES, read=["R1", "R2"])
	output:
		"data/multiqc_report.html"
	message:
		"-- Running MultiQC for raw data --"
	conda:
		"envs/QC_env.yml"
	shell:
		"multiqc -f --interactive --quiet data/ -o data/"

rule cutadapt:
	input:
		fw= "data/{sample}_R1_001.fastq.gz",
		rv= "data/{sample}_R2_001.fastq.gz"
	output:
		fw= "trimmed_data/{sample}_R1_trimmed.fastq.gz",
		rv= "trimmed_data/{sample}_R2_trimmed.fastq.gz",
		log= "trimmed_data/{sample}.trimmed.txt"
	message:
		"-- Running Cutadapt --"
	conda:
		"envs/cutadapt.yml"
	shell:
		"cutadapt -a CTGTCTCTTATACACATCT -A CTGTCTCTTATACACATCT -O 10 -m 30 -q 20 \
                                {input.fw} {input.rv} -o {output.fw} -p {output.rv} > {output.log}"

rule fastqc_trim:
	input:
		"trimmed_data/{sample}_{read}_trimmed.fastq.gz"
	output:
		"trimmed_data/FASTQC/{sample}_{read}_trimmed_fastqc.zip"
	message:
		"-- Quality check of trimmed data with Fastqc --"
	conda:
		"envs/QC_env.yml"
	threads:
		2
	shell:
		"fastqc --quiet -t {threads} --outdir trimmed_data/FASTQC -f fastq {input}"

rule multiqc_trim:
	input:
		expand("trimmed_data/FASTQC/{sample}_{read}_trimmed_fastqc.zip", sample=SAMPLES, read=["R1", "R2"])
	output:
		"trimmed_data/multiqc_report.html"
	message:
		"-- Running MultiQC for trimmed data--"
	conda:
		"envs/QC_env.yml"
	shell:
		"multiqc -f --interactive --quiet trimmed_data/ -o trimmed_data/"

#rule metaxa2_1:
#	input:
#		fw= expand("trimmed_data/{sample}_R1_trimmed.fastq.gz", sample=SAMPLES),
#		rv= expand("trimmed_data/{sample}_R2_trimmed.fastq.gz", sample=SAMPLES)	
#	output:
#		expand("metaxa2/{sample}", sample=SAMPLES)
#	message:
#		"-- Running metaxa2.1 --"
#	conda:
#		"envs/metaxa.yml"
#	threads:
#		2
#	shell:
#		"metaxa2 -1 {input.fw} -2 {input.rv} -z gzip -o {output} --align-none --plus --graphical F -t {threads}"

#rule metaxa2_2:
#	input:
#		expand("metaxa2/{sample}.taxonomy", sample=SAMPLES)
#	output:
#		"metaxa2/{sample}_level_6.txt"
#	message:
#		"-- Running metaxa2_2 --"
#	conda:
#		"envs/metaxa.yml"
#	threads:
#		1
#	shell:
#		"metaxa2_ttt -i {input} -o {output}"
#
#rule metaxa2_3:
#	input:
#		"metaxa2/{sample}_level_6.txt"
#	output:
#		"metaxa2/metaxa_genus.txt"
#	message: 
#		"-- Running metaxa2_2 --"
#	conda:
#		"envs/metaxa.yml"
#	threads:
#		1
#	shell:
#		"metaxa2_dc -o {output} {input}"

# Before rule "resfinder_db" create and remove duplicates from resfinder.fasta
#cat *.fsa > resfinder.fasta
#seqkit rmdup -s resfinder.fasta > unique_resfinder.fasta -d removed
#[INFO] 18 duplicated records removed
#mv unique_resfinder.fasta resfinder.fasta

rule resfinder_db:
	input:
		fasta= "resfinder_db/resfinder.fasta"
	output:
		indexed_db= "resfinder_db/resfinder.1.bt2"
	message:
		"-- ResFinder db --"
	conda:
		"envs/bowtie2.yml"
	threads:
		1
	shell:
		"bowtie2-build {input.fasta} resfinder_db/resfinder"

rule resfinder_mapping:
	input:
		fw= expand("trimmed_data/{sample}_R1_trimmed.fastq.gz", sample=SAMPLES),
		rv= expand("trimmed_data/{sample}_R2_trimmed.fastq.gz", sample=SAMPLES),
		indexed_db= "resfinder_db/resfinder.1.bt2"
	output:
		"mapped_reads/{sample}.bam"
	message:
		"-- Mapping w/ ResFinder --"
	conda:
		"envs/bowtie2.yml"
	threads:
		1
	shell:
		"bowtie2 -x resfinder_db/resfinder -1 {input.fw} -2 {input.rv} -p {threads} -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 | samtools view -Shu - > {output}"
# add filter step

rule resfinder_sorting:
	input:
		"mapped_reads/{sample}.bam"
	output:
		"sorted_reads/{sample}.bam"
	message:
		"-- Sorting reads --"
	conda:
		"envs/bowtie2.yml"
	threads:
		1
	shell:
		"samtools sort -T sorted_reads/{wildcards.sample} "
		"-O bam {input} > {output}"

rule resfinder_indexing:
	input:
		"sorted_reads/{sample}.bam"
	output:
		"sorted_reads/{sample}.bam.bai"
	message:
		"-- Indexing mapped reads --" 	
	conda:
		"envs/bowtie2.yml"      
	threads:
		1
	shell:
		"samtools index {input}"

rule resfinder_results:
	input:
		bam= expand("sorted_reads/{sample}.bam", sample=SAMPLES),
		bai= expand("sorted_reads/{sample}.bam.bai", sample=SAMPLES)
	output:
		"resfinder_out/{sample}_counts"
	message:
		"-- Preparing resfinder result files --"
	threads:
		1
	shell:
		"echo -e {wildcards.sample} > {output}"

# Combine results into matrix outside Snakemake pipeline 
# because it gets complex with escape charachters in the pipeline - see batch job in README.md.

rule fastq_to_fasta:
	input:
		expand("trimmed_data/{sample}_{read}_trimmed.fastq.gz", sample=SAMPLES, read=["R1", "R2"])
	output:
		temp(expand("fasta/{sample}_{read}.fasta.gz"), sample=SAMPLES, read=["R1", "R2"])
	message:
		"-- Converting fastq to FASTA --""
	conda:
		"envs/seqkit.yml"
	threads:
		1
	shell:
		"seqkit fq2fa {input} -o {output}"

rule diamond_db:
	input:
		"CARD/protein_fasta_protein_homolog_model.fasta"
	output:
		"CARD/protein_fasta_protein_homolog_model.dmnd"
	message:
		"-- Creating DIAMOND database --"
	conda:
		"envs/diamond.yml"
	threads:
		1
	shell:
		"diamond makedb --in {input} -d CARD/protein_fasta_protein_homolog_model"

rule run_diamond:
	input:
		fasta= expand("fasta/{sample}_{read}.fasta.gz", sample=SAMPLES, read=["R1", "R2"]),
		db= "CARD/protein_fasta_protein_homolog_model.dmnd"
	output:
		"CARD/CARD_results"
	message:
		"-- Running DIAMOND --"
	conda:
		"envs/diamond.yml"
	threads:
		2
	shell:
		"gunzip -c {input.fasta} | diamond blastx -d CARD/protein_fasta_protein_homolog_model -q - -o {output} --threads {threads}"		# Possible filter options for matches: -evalue 1e-50; also -max_target_seqs n; query-cover n; -id n

