---
title: "R_scripts"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Set working directory
```{r}
setwd("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles")
```
## Load libraries
```{r, warning=FALSE, error=FALSE, message=FALSE}
library(phyloseq)
#source("https://raw.githubusercontent.com/mahendra-mariadassou/phyloseq-extended/master/load-extra-functions.R")
library(dplyr)
library(viridis)
library(stringr)
library(vegan)
library(RColorBrewer)
library(BiocManager)
BiocManager::install("microbiome")
library(microbiome)
library(microbiomeutilities)
library(ggplot2)
library(knitr)
library(ggpubr)
library(pheatmap)
library(ape)
library(multcomp)
```
## Load metadata
```{r}
metadata <-read.table("metadata.txt", sep="\t", header = T, row.names = 1, fill = 1)
```
# Metaxa2
```{r}
# Load data
metaxa_genus <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaxa_genus.txt")

# Create OTU table
OTU_metaxa <- metaxa_genus[,-1]

# Create tax table
tax_table_metaxa <- data.frame(str_split_fixed(data.frame(metaxa_genus) [,1], ";", 6))
colnames(tax_table_metaxa) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus")

# Combine into phyloseq object
metaxa_PHY <- phyloseq(otu_table(OTU_metaxa, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaxa)), sample_data(metadata))

# Exclude Unknown
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown"))
```
## Add SSU counts into metadata
```{r}
metadata$SSU_counts <- sample_sums(metaxa_PHY)

# Create extra column for Eucaryota
metaxa_PHY_temp <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown", "Bacteria", "Archaea", "Mitochondria", "Chloroplast"))
metadata$SSU_eucaryota <- sample_sums(metaxa_PHY_temp)
```
## Filter samples
```{r}
# With low quality (BFH24_S142, BH63_S118, FH10_S171)
metaxa_PHY = subset_samples(metaxa_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Feces samples (BH20_S88, BH22_S89, BH24_S90, BH25_S91)
metaxa_PHY_ww = subset_samples(metaxa_PHY, name!="BH20_S88" & name!="BH22_S89" & name!="BH24_S90" & name!="BH25_S91")

# Suspicious samples (very high ARG level)
metaxa_PHY_clean = subset_samples(metaxa_PHY, name != "BH02_S77" & name != "BH27_S92" & name != "BH30_S94")

# Clear taxa that is not present in any of the remaining samples
any(taxa_sums(metaxa_PHY_ww) == 0)
metaxa_PHY_ww <- prune_taxa(taxa_sums(metaxa_PHY_ww) > 0, metaxa_PHY_ww)
any(taxa_sums(metaxa_PHY_ww) == 0)
```
## Ordinate Metaxa2 result to study whether the taxonomy can be explained by the variable "country"
```{r}
# Change into relative data
metaxa_rel_PHY <- transform_sample_counts(metaxa_PHY_ww, function(x) x/sum(x))

# Ordinate and plot data
metaxa_rel_PHY_ord <- ordinate(metaxa_rel_PHY, method = "PCoA", distance = "horn")
p1 <- plot_ordination(metaxa_rel_PHY, metaxa_rel_PHY_ord, color = "country", label = "name")
metaxa.p1 <- p1 + scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) + geom_point(colour = "black", 
    pch = 21, size = 2, alpha = 0.5) + stat_ellipse(level = 0.9, linetype = 1) +
    theme_minimal() + labs(title = "Metaxa2")
metaxa.p1

# Test significance using pair-wise adonis
metaxa_temp <- subset_samples(metaxa_PHY_ww, (country == "Benin" | country == "Finland"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

metaxa_temp <- subset_samples(metaxa_PHY_ww, (country == "Burkina Faso" | country == "Finland"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

metaxa_temp <- subset_samples(metaxa_PHY_ww, (country == "Benin" | country == "Burkina Faso"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

# The explanatory variable "country" explains some (~16.7 %) of the difference in the grouping of taxonomy in the comparison between of Benin and Finland.
```
# Testing whether the taxonomic composition is grouped significantly by location in each country, Metaxa2
```{r}
Benin <- subset_samples(metaxa_PHY_ww, (country == "Benin"))
Benin_dist <- vegdist(t(otu_table(Benin)), dist = "horn")
adonis(Benin_dist ~ location, data = data.frame(sample_data(Benin)), permutations = 99999)

BF <- subset_samples(metaxa_PHY_ww, (country == "Burkina Faso"))
BF_dist <- vegdist(t(otu_table(BF)), dist = "horn")
adonis(BF_dist ~ location, data = data.frame(sample_data(BF)), permutations = 99999)

Finland <- subset_samples(metaxa_PHY_ww, (country == "Finland"))
Finland_dist <- vegdist(t(otu_table(Finland)), dist = "horn")
adonis(Finland_dist ~ location, data = data.frame(sample_data(Finland)), permutations = 99999)

# In Benin the taxonomic profile is explained by  26.3 % by the location. Whereas in Burkina Faso the same percentage is  24.2 %. In Finland the grouping by location is not significant.
```

# Metaphlan3
```{r}
# Modify data in command-line to match sample names in metadata file
## tr '|' ';' <merged_abundance_table.txt > mod_merged_abundance_table.txt
## sed -i 's/_profile//g' mod_merged_abundance_table.txt
## sed -i 's/BFH38-A_S156/BFH38.A_S156/g' mod_merged_abundance_table.txt
## sed -i 's/BFH38-B_S157/BFH38.B_S157/g' mod_merged_abundance_table.txt
## sed -i 's/BH34-A_S98/BH34.A_S98/g' mod_merged_abundance_table.txt
## sed -i 's/BH34-B_S99/BH34.B_S99/g' mod_merged_abundance_table.txt

# Load data
metaphlan <- as.matrix(read.table("mod_merged_abundance_table.txt", fill = 1, header = T, check.names = F))

# Exclude the first column "NCBI_tax_id"
metaphlan <- subset(metaphlan, select = -c(NCBI_tax_id))

# Create OTU table
OTU_metaphlan <- metaphlan[,-1]

# Change values as numeric
class(OTU_metaphlan) <- "numeric"

# Create tax_table
tax_table_metaphlan <- data.frame(str_split_fixed(data.frame(metaphlan) [,1], ";", 7))
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# Clean "*__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

# Combine into phyloseq object
metaphlan_PHY <- phyloseq(otu_table(OTU_metaphlan, taxa_are_rows = TRUE), tax_table(as.matrix(tax_table_metaphlan)), sample_data(metadata))
```
## Filter samples
```{r}
# Virus
metaphlan_PHY <- subset_taxa(metaphlan_PHY, Kingdom != "Virus")

# With low quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_PHY = subset_samples(metaphlan_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Feces samples (BH20_S88, BH22_S89, BH24_S90, BH25_S91)
metaphlan_PHY_ww = subset_samples(metaphlan_PHY, name!="BH20_S88" & name!="BH22_S89" & name!="BH24_S90" & name!="BH25_S91")

# Suspicious samples (very high ARG level)
metaphlan_PHY_clean = subset_samples(metaphlan_PHY, name != "BH02_S77" & name != "BH27_S92" & name != "BH30_S94")

# Clear taxa that is not present in any of the remaining samples
any(taxa_sums(metaphlan_PHY_ww) == 0)
metaphlan_PHY_ww <- prune_taxa(taxa_sums(metaphlan_PHY_ww) > 0, metaphlan_PHY_ww)
any(taxa_sums(metaphlan_PHY_ww) == 0)
```
## Ordinate Metaphlan3 results to study wheather the taxonomy can be explained by the variable "country"
```{r}
# Ordinate and plot data
metaphlan_PHY_ord <- ordinate(metaphlan_PHY_ww, method = "PCoA", distance = "horn")
p2 <- plot_ordination(metaphlan_PHY_ww, metaphlan_PHY_ord, color = "country", label = "sample_type")
metaphlan.p2 <- p2 + scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) + 
    geom_point(colour = "black", pch = 21, size = 2, alpha = 0.5) + stat_ellipse(level = 0.9, linetype = 1) +
    theme_minimal() + labs(title = "Metaphlan3")
metaphlan.p2

# Test significance using pair-wise adonis
metaphlan_temp <- subset_samples(metaphlan_PHY_ww, (country == "Benin" | country == "Finland"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY_ww, (country == "Burkina Faso" | country == "Finland"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY_ww, (country == "Burkina Faso" | country == "Benin"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

# The explanatory variable "country" explains less than 10 % of the grouping in every pair wise comparison.
# There seems to be a small group of samples differing from other ones. They represent mostly other than ww samples. Let's see if this grouping is significant.

metaphlan_temp <- subset_samples(metaphlan_PHY_ww, (sample_type == "waste water" | sample_type == "soil"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ sample_type, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY_ww, (sample_type == "waste water" | sample_type == "river water"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ sample_type, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

# Not really. Whether the sample is river water or waste water seems to explain 4.6 % of the grouping.
```
# Testing wheter the taxonomic composition is grouped significantly by location in each country
```{r}
Benin <- subset_samples(metaphlan_PHY_ww, (country == "Benin"))
Benin_dist <- vegdist(t(otu_table(Benin)), dist = "horn")
adonis(Benin_dist ~ location, data = data.frame(sample_data(Benin)), permutations = 99999)

BF <- subset_samples(metaphlan_PHY_ww, (country == "Burkina Faso"))
BF_dist <- vegdist(t(otu_table(BF)), dist = "horn")
adonis(BF_dist ~ location, data = data.frame(sample_data(BF)), permutations = 99999)

Finland <- subset_samples(metaphlan_PHY_ww, (country == "Finland"))
Finland_dist <- vegdist(t(otu_table(Finland)), dist = "horn")
adonis(Finland_dist ~ location, data = data.frame(sample_data(Finland)), permutations = 99999)

# In Benin the taxonomy profile is explained by  29.7 % by the location. Whereas in Burkina Faso the same percentage is  22.2 %. In Finland the grouping by location is not significant.
```

# ResFinder
## Load data
```{r}
# Modify mapping output file "ARG_genemat.txt" in command line to match sample names in metadata file
## sed 's/BFH38-A_S156/BFH38.A_S156/g' ARG_genemat.txt > mod_ARG_genemat.txt
## sed -i 's/BFH38-B_S157/BFH38.B_S157/g' mod_ARG_genemat.txt
## sed -i 's/BH34-A_S98/BH34.A_S98/g' mod_ARG_genemat.txt
## sed -i 's/BH34-B_S99/BH34.B_S99/g' mod_ARG_genemat.txt

ARG_genemat <-as.matrix(read.table("mod_ARG_genemat.txt", header= T, check.names = F, row.names = 1))
```
## Load ResFinder gene lengths
```{r}
# Modify "resfinder.fasta" file so that only hits remain
## seqkit grep -f ARG_genes.txt resfinder.fasta > filtered_resfinder.fasta
# Check if there are correct number of lines
## grep ">" filtered_resfinder.fasta | wc -l
# Print out the gene lengths of these genes into file "lengths_resfinder.txt"
## cat filtered_resfinder.fasta | awk '$0 ~ ">" {if (NR > 1) {print c;} c=0;printf substr($0,2,100) "\t"; } $0 !~ ">" {c+=length($0);} END { print c; }' > lengths_resfinder.txt
# Reorder in excel to match with file "ARG_genemat_norm"
lengths_resfinder <-as.matrix(read.table("lengths_resfinder.txt", header= F, check.names = T, row.names = 1))
colnames(lengths_resfinder) <- c("Length")
```
## Normalization with gene lengths
```{r}
# Divide by ResFinder hit gene lengths
ARG_length_norm <- ARG_genemat/lengths_resfinder[, 1]

# Divide by SSU counts and normalize to bacterial 16S rRNA length (1550)
ARG_genemat_norm <- t(t(ARG_length_norm)/metadata$SSU_counts) * 1550

# Check if correct:
identical(ARG_genemat[2020, 4]/metadata$SSU_counts[4], ARG_genemat_norm[2020, 4])

# Save and load again to exclude row.names
write.table(ARG_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)

ARG_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=NULL)
ARG_genemat_norm$row.names<-NULL
```
## Normalize ARG counts to SSU
### Do not run if the above is run
```{r}
#ARG_genemat_norm <- t(t(ARG_genemat)/metadata$SSU_counts)

# Check if correct:
#identical(ARG_genemat[2020, 4]/metadata$SSU_counts[10], ARG_genemat_norm[2020, 4])

# Save and load again to exclude row.names
#write.table(ARG_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)

#ARG_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=NULL)
#ARG_genemat_norm$row.names<-NULL
```
## Create phyloseq object
```{r}
# Create tax table
tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_resfinder.txt", header=FALSE, sep=";")
colnames(tax_table_resfinder) <- c("Gene", "Class")

# Combine to phyloseq object
resfinder_PHY <- phyloseq(otu_table(ARG_genemat_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_resfinder)))
```
## Filter samples
```{r}
# With low quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY = subset_samples(resfinder_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Feces samples (BH20_S88, BH22_S89, BH24_S90, BH25_S91)
resfinder_PHY_ww = subset_samples(resfinder_PHY, name!="BH20_S88" & name!="BH22_S89" & name!="BH24_S90" & name!="BH25_S91")

# Suspicious samples (very high ARG level) (BH02_S77, BH27_S92, BH30_S94)
resfinder_PHY_clean = subset_samples(resfinder_PHY_ww, name != "BH02_S77" & name != "BH27_S92" & name != "BH30_S94")

# Check if there are taxa that are not present in any of the remaining samples
any(taxa_sums(resfinder_PHY_clean) == 0)
resfinder_PHY_clean <- prune_taxa(taxa_sums(resfinder_PHY_clean) > 0, resfinder_PHY_clean)
any(taxa_sums(resfinder_PHY_clean) == 0)
```
## Ordinate ResFinder mapping results to study wheather the taxonomy can be explained by the variable "country"
```{r}
# Ordinate and plot data
resfinder_PHY_ord <- ordinate(resfinder_PHY_clean, method = "PCoA", distance = "horn")
p3 <- plot_ordination(resfinder_PHY_clean, resfinder_PHY_ord, color = "country", label = "name")
resfinder.p3 <- p3 + scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) + 
    geom_point(colour = "black", pch = 21, size = 2, alpha = 0.5) + stat_ellipse(level = 0.9, linetype = 1) +
    theme_minimal() + labs(title = "ResFinder")
resfinder.p3

# Test significance between all pairs
resfinder_temp <- subset_samples(resfinder_PHY_clean, (country == "Finland" | country == "Benin"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

resfinder_temp <- subset_samples(resfinder_PHY_clean, (country == "Burkina Faso" | country == "Benin"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

resfinder_temp <- subset_samples(resfinder_PHY_clean, (country == "Finland" | country == "Burkina Faso"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

# The explanatory variable "country" explains some (~11.6 %) of the difference in the grouping of taxonomy in the comparison between of Benin and Finland and some (~12.9%) between Burkina Faso and Finland.
```
# Testing whether the (alpha diversities in?) resistomes are grouped significantly by location in each country
```{r}
Benin <- subset_samples(resfinder_PHY_ww, (country == "Benin"))
Benin_dist <- vegdist(t(otu_table(Benin)), dist = "horn")
adonis(Benin_dist ~ location, data = data.frame(sample_data(Benin)), permutations = 99999)

BF <- subset_samples(resfinder_PHY_ww, (country == "Burkina Faso"))
BF_dist <- vegdist(t(otu_table(BF)), dist = "horn")
adonis(BF_dist ~ location, data = data.frame(sample_data(BF)), permutations = 99999)

Finland <- subset_samples(resfinder_PHY_ww, (country == "Finland"))
Finland_dist <- vegdist(t(otu_table(Finland)), dist = "horn")
adonis(Finland_dist ~ location, data = data.frame(sample_data(Finland)), permutations = 99999)

# In Benin the ARG profile is explained by 21.2 % by the location. Whereas in Burkina Faso the same percentage is 20.5 %. In Finland the grouping by location is not significant.
```
# Diversity indexes by Metaxa2
```{r, results='hide'}
# Take a look at the indexes (Can't work with the metaxa_rel_PHY, since it has "double" = non-integer data.)
metaxa_tab <-microbiome::alpha(metaxa_PHY_ww, index = "all")
kable(head(metaxa_tab))

# Create data table
metaxa.meta <- meta(metaxa_PHY_ww)
kable(head(metaxa.meta))

# Add indexes to data table
## Shannon
metaxa.meta$diversity_shannon <- metaxa_tab$diversity_shannon
## Gini Simpson
metaxa.meta$diversity_gini_simpson <- metaxa_tab$diversity_gini_simpson
## Inverse Simpson
metaxa.meta$diversity_inverse_simpson <- metaxa_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(metaxa.meta$diversity_shannon)
shapiro.test(metaxa.meta$diversity_shannon) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaxa.meta$diversity_shannon)

hist(metaxa.meta$diversity_gini_simpson)
shapiro.test(metaxa.meta$diversity_gini_simpson) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaxa.meta$diversity_gini_simpson)

hist(metaxa.meta$diversity_inverse_simpson)
shapiro.test(metaxa.meta$diversity_inverse_simpson)
qqnorm(metaxa.meta$diversity_inverse_simpson)

# Create a list of pairwise comparisons
div.country <- levels(metaxa.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])
print(country.pairs)

# Plot
## Shannon
p4 <- ggviolin(metaxa.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p4 <- p4 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p4)

## Gini Simpson
p5 <- ggviolin(metaxa.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p5 <- p5 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p5)

# The probability that two individuals randomly selected from a sample will belong to same species is higher in Finland than in Benin.

## Inverse Simpson
p6 <- ggviolin(metaxa.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p6 <- p6 + stat_compare_means(comparisons = country.pairs, method = "t.test") 
print(p6)

# Inverse simpson index shows significant results. The alpha diversity is the highest in Benin compared to Burkina Faso and Finland. Of Burkina Faso and Finland the diversity is higher in Burkina Faso. 
```
# Diversity indexes by Metaphlan3
```{r}
# Shannon
## Take a look at the indexes
metaphlan_tab <-microbiome::alpha(metaphlan_PHY_ww, index = "diversity_shannon")
kable(head(metaphlan_tab))

## Create data table
metaphlan.meta <- meta(metaphlan_PHY_ww)
kable(head(metaphlan.meta))

## Add indexes to data table
metaphlan.meta$diversity_shannon <- metaphlan_tab$diversity_shannon

# Check whether the distribution of the diversity is normal
hist(metaphlan.meta$diversity_shannon)
shapiro.test(metaphlan.meta$diversity_shannon) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaphlan.meta$diversity_shannon)

# Plot
p7 <- ggviolin(metaphlan.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p7 <- p7 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p7)

# Gini Simpson
## Take a look at the indexes
metaphlan_tab <-microbiome::alpha(metaphlan_PHY_ww, index = "diversity_gini_simpson")
kable(head(metaphlan_tab))

## Add indexes to data table
metaphlan.meta$diversity_gini_simpson <- metaphlan_tab$diversity_gini_simpson

# Check whether the distribution of the diversity is normal
hist(metaphlan.meta$diversity_gini_simpson)
shapiro.test(metaphlan.meta$diversity_gini_simpson) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaphlan.meta$diversity_gini_simpson)

# Create a list of pairwise comparisons
div.country <- levels(metaphlan.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])
print(country.pairs)

# Plot
p8 <- ggviolin(metaphlan.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p8 <- p8 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p8)

# Inverse Simpson
## Take a look at the indexes
metaphlan_tab <-microbiome::alpha(metaphlan_PHY_ww, index = "diversity_inverse_simpson")
kable(head(metaphlan_tab))

## Add indexes to data table
metaphlan.meta$diversity_inverse_simpson <- metaphlan_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(metaphlan.meta$diversity_inverse_simpson)
shapiro.test(metaphlan.meta$diversity_inverse_simpson) 
qqnorm(metaphlan.meta$diversity_inverse_simpson)

# Plot
p9 <- ggviolin(metaphlan.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p9 <- p9 + stat_compare_means(comparisons = country.pairs, method = "t.test") 
print(p9)

# Inverse Simpson index shows weakly significant results. There are no significant differences in the alpha diversities of the countries taxonomical composition.
```
# Diversity indexes by ResFinder
```{r}
# Shannon
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY_clean, index = "diversity_shannon")
kable(head(resfinder_tab))

## Create data table
resfinder.meta <- meta(resfinder_PHY_clean)
kable(head(resfinder.meta))

## Add indexes to data table
resfinder.meta$diversity_shannon <- resfinder_tab$diversity_shannon

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_shannon)
shapiro.test(resfinder.meta$diversity_shannon) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(resfinder.meta$diversity_shannon)

# Create a list of pairwise comparisons
div.country <- levels(resfinder.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])
print(country.pairs)

# Plot
p10 <- ggviolin(resfinder.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p10 <- p10 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p10)

# Gini Simpson
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY_clean, index = "diversity_gini_simpson")
kable(head(resfinder_tab))

## Add indexes to data table
resfinder.meta$diversity_gini_simpson <- resfinder_tab$diversity_gini_simpson

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_gini_simpson)
shapiro.test(resfinder.meta$diversity_gini_simpson) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(resfinder.meta$diversity_gini_simpson)

# Plot
p11 <- ggviolin(resfinder.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p11 <- p11 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p11)

# Inverse Simpson
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY_clean, index = "diversity_inverse_simpson")
kable(head(resfinder_tab))

## Add indexes to data table
resfinder.meta$diversity_inverse_simpson <- resfinder_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_inverse_simpson)
shapiro.test(resfinder.meta$diversity_inverse_simpson) 
qqnorm(resfinder.meta$diversity_inverse_simpson)

# Plot
p12 <- ggviolin(resfinder.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p12 <- p12 + stat_compare_means(comparisons = country.pairs, method = "t.test") 
print(p12)

# The alpha diversity of resistance genes detected by ResFinder seems slightly higher in Burkina Faso than in Benin and in Finland.
```

```{r}
# Plot by individual sample
nb.cols <- 14
mycolors <- colorRampPalette(brewer.pal(8, "Dark2"))(nb.cols)

loc <- plot_richness(resfinder_PHY_clean, measures = c("Shannon", "Simpson"), 
                     color="location", shape = "country")

resfinder.in <- loc + scale_fill_manual(values=mycolors) + 
  geom_boxplot() +
  geom_point(size=3) + theme_bw() +
  theme(axis.text.x = element_text(angle = 90, size =5))
resfinder.in
```
# Taxonomic composition by Metaxa2 using microbiome package
```{r}
# Load data
metaxa.com <- metaxa_rel_PHY
taxic_metaxa <- as.data.frame(metaxa.com@tax_table) 

# Add OTU ids
taxic_metaxa$OTU <- rownames(taxic_metaxa) 
colnames(taxic_metaxa)

# Convert into a matrix.
taxmat_metaxa <- as.matrix(taxic_metaxa)

# Convert into phyloseq compatible file.
new.tax_metaxa <- tax_table(taxmat_metaxa)  

# Incroporate into new phyloseq object
tax_table(metaxa.com) <- new.tax_metaxa

# Class level
metaxa_top5class <- aggregate_top_taxa(metaxa.com, top = 5, "Class")

#metaxa_top5class.rel <- microbiome::transform(metaxa_top5class, "compositional")

metaxa_top5class.p13 <- plot_composition(metaxa_top5class, group_by = "country") + 
  theme(legend.position = "bottom") + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, size = 8), axis.title = element_blank()) +
    ggtitle("Relative abundance of top 5 classes, Metaxa2")

metaxa_top5class.p13 + 
    scale_fill_manual(values = c("#1B9E77","#E6AB02","#7570B3","#E7298A","#66A61E", "#FFFFFF"))
```
# Top 15 genera merged by location, Metaxa2
```{r}
metaxa_PHY_genus <- tax_glom(metaxa_rel_PHY, "Genus")

metaxa_PHY_genus_abun <- prune_taxa(names(sort(taxa_sums(metaxa_PHY_genus), TRUE)[1:15]), 
    metaxa_PHY_genus)

# Transform_1
metaxa_PHY_genus_abun_trans <- transform_sample_counts(metaxa_PHY_genus_abun, function(x) x / sum(x))

# Merge by location
metaxa_mrg_loc <- merge_samples(metaxa_PHY_genus_abun_trans, "location")

# Transform_2
metaxa_mrg_loc_trans <- transform_sample_counts(metaxa_mrg_loc, function(x) x / sum(x))

# Normalize by sample number / location
otu_table(metaxa_mrg_loc_trans) <- otu_table(metaxa_mrg_loc_trans)[, ]/as.matrix(table(sample_data(metaxa_PHY)$location))[, 
    1]

# Set colors
nb.cols <- 15
mycolors <- colorRampPalette(brewer.pal(8, "Paired"))(nb.cols)

# Facet labs
country.labs <- c("1" = "Benin", "2" = "Burkina Faso", "3" = "Finland")

# Plot
metaxa.p14 <- plot_bar(metaxa_mrg_loc_trans, fill = "Genus", title = "15 most abundant genera, Metaxa2") + 
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country = country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=55, hjust=1, size = 10),
        axis.title.x = element_blank()) +
        ylab("Merged relative abundance (%)") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 8), keywidth = 1.5, keyheight = 1.5, title = "Genus"))
metaxa.p14
```
# Top 15 genera merged by location, Metaphlan3
```{r}
metaphlan_PHY_genus <- tax_glom(metaphlan_PHY_ww, "Genus")

metaphlan_PHY_genus_abun <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_genus), TRUE)[1:15]), 
    metaphlan_PHY_genus)

# Transform_1
metaphlan_PHY_genus_abun_trans <- transform_sample_counts(metaphlan_PHY_genus_abun, function(x) x / sum(x))

# Merge by location
metaphlan_mrg_loc <- merge_samples(metaphlan_PHY_genus_abun_trans, "location")

# Transform_2
metaphlan_mrg_loc_trans <- transform_sample_counts(metaphlan_mrg_loc, function(x) x / sum(x))

# Normalize by sample number / location
otu_table(metaphlan_mrg_loc_trans) <- otu_table(metaphlan_mrg_loc_trans)[, ]/as.matrix(table(sample_data(metaphlan_PHY_ww)$location))[, 
    1]

# Set colors
nb.cols <- 15
mycolors <- colorRampPalette(brewer.pal(8, "Paired"))(nb.cols)
# Facet labs
country.labs <- c("1" = "Benin", "2" = "Burkina Faso", "3" = "Finland")

# Plot
metaphlan.p16 <- plot_bar(metaphlan_mrg_loc_trans, fill = "Genus", title = "15 most abundant genera, Metaphlan3") + 
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country = country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=55, hjust=1, size = 10),
        axis.title.x = element_blank()) +
        ylab("Relative abundance (%)") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 8), keywidth = 1.5, keyheight = 1.5, title = "Genus"))
metaphlan.p16
```
# DRAFT Heatmap of Metaxa2 results
```{r}
heat.metaphlan <- plot_taxa_heatmap(metaphlan_PHY_genus,
                                 subset.top = 50,
                                 VariableA = "country",
                                 heatcolors = rev(brewer.pal(12, "GnBu")),
                                 transformation = "log10",
                                 cluster_cols = T)
heat.metaphlan
```
# Sum of the relative abundances of ResFinder results by location
```{r}
# Average 16S counts by location
lo_data<-data.frame(SUM=round(mean(sample_data(resfinder_PHY_clean)$SSU_counts)*(sample_sums(resfinder_PHY_clean))), location=as.factor(sample_data(resfinder_PHY_clean)$location))

plot(lo_data$SUM~lo_data$location, cex.axis=0.5, srt=45, las=2, xlab=NULL)
# There are outliers

# Testing models to support the data
## Linear
model.ln<-lm(SUM~location, data=lo_data)
summary(model.ln)
shapiro.test(lo_data$SUM)
plot(model.ln)

# The p value is > 0.05 and differs significantly from normal distribution

## Log
model.logln<-lm(log(SUM)~location, data=lo_data)
summary(model.logln)
shapiro.test(log(lo_data$SUM))
plot(model.logln)

# The p value is > 0.05 and differs significantly from normal distribution

## Poisson
model.pois = glm(SUM ~ location , data = lo_data, family = poisson)
summary(model.pois)
plot(model.pois)

# P value <2e-16, Residual deviance: 1495738  on 68  degrees of freedom, AIC: 1496754

## Quasipoisson
model.qpois<-glm(SUM~location, data=lo_data, family="quasipoisson")
summary(model.qpois)
plot(model.qpois)

# Residual deviance: 1495738  on 68  degrees of freedom, AIC: NA

## Negative binomial
model.nb <- glm.nb(SUM~location, data=lo_data)
summary(model.nb)
plot(model.nb)

# Residual deviance:  90.286  on 68  degrees of freedom, AIC: 1915.4, theta, 1.594, Std. Err.:  0.228

# Summarize and compare models
data.frame(linear=coef(model.ln),
                 loglinear=exp(coef(model.logln)),
                 poisson=exp(coef(model.pois)),
                 qpoisson=exp(coef(model.qpois)),
                 negbin=exp(coef(model.nb)))
# Critical value
qchisq(0.95, df.residual(model.pois))

deviance(model.ln)
deviance(model.logln)
deviance(model.pois)
deviance(model.qpois)
deviance(model.nb)

# None of the models fits the data. But the negative binomial is the vbest of the bad ones.
```

```{r}
# Plot using neg. binom. model
fit <- glm.nb(SUM ~ location, data = lo_data, link = log)

lo_data <- cbind(lo_data, Mean = predict(fit, newdata = lo_data, type = "response"), SE = predict(fit, newdata = lo_data, type = "response", se.fit = T)$se.fit)

nb.cols <- 14
mycolors <- colorRampPalette(brewer.pal(8, "Dark2"))(nb.cols)

ARG.sum.lo <- ggplot(lo_data, aes(x = location, y = Mean)) +
        scale_fill_manual(values = mycolors) +
    geom_line() + geom_jitter(data = lo_data, aes(x = location, y = SUM, color = location), 
    size = 2.3, alpha = 0.5, width = 0.3) + 
    geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.6, lwd = 0.6) + 
    geom_point(size = 0.9) +
    theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    labs(y = "Sum abundance/16S", 
    x = "") + guides(color = FALSE, alpha = FALSE) + labs(title = "ARGs") +
  scale_y_continuous(trans = "log10")
ARG.sum.lo
```

```{r}
# Extract outliers

lo_data<-data.frame(SUM=round(mean(sample_data(resfinder_PHY_clean)$SSU_counts)*(sample_sums(resfinder_PHY_clean))), location=as.factor(sample_data(resfinder_PHY_clean)$location))

boxplot(lo_data$SUM)
boxplot(lo_data$SUM, plot=FALSE)$out
outliers <- boxplot(lo_data$SUM, plot=FALSE)$out
print(outliers)

# Exclude outliers
## where they at
lo_data[which(lo_data$SUM %in% outliers),]
## remove
lo_data <- lo_data[-which(lo_data$SUM %in% outliers),]
boxplot(lo_data$SUM)

# New  plot
fit <- glm.nb(SUM ~ location, data = lo_data, link = log)

lo_data <- cbind(lo_data, Mean = predict(fit, newdata = lo_data, type = "response"), SE = predict(fit, newdata = lo_data, type = "response", se.fit = T)$se.fit)

nb.cols <- 14
mycolors <- colorRampPalette(brewer.pal(8, "Dark2"))(nb.cols)

ARG.sum.lo_outl <- ggplot(lo_data, aes(x = location, y = Mean)) +
    scale_fill_manual(values = mycolors) +
    geom_line() + geom_jitter(data = lo_data, aes(x = location, y = SUM, color = location), 
    size = 2.3, alpha = 0.5, width = 0.3) + 
    geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.6, lwd = 0.6) + 
    geom_point(size = 0.9) +
    theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    labs(y = "Sum abundance/16S", 
    x = "") + guides(color = FALSE, alpha = FALSE) + labs(title = "ARGs") + scale_y_log10()
ARG.sum.lo_outl
```
# Sum of the relative abundances of ResFinder results by country
```{r, results='hide'}
# Average 16S counts bu location
co_data<-data.frame(SUM=round(mean(sample_data(resfinder_PHY_clean)$SSU_counts)*(sample_sums(resfinder_PHY_clean))), country=as.factor(sample_data(resfinder_PHY_clean)$country))

plot(co_data$SUM~co_data$country, cex.axis=0.5, srt=45, las=2, xlab=NULL)

# Testing models to support the data
## Linear
model.ln<-lm(SUM~country, data=co_data)
summary(model.ln)
shapiro.test(lo_data$SUM)
plot(model.ln)

# The p value is > 0.05 and differs significantly from normal distribution

## Log
model.logln<-lm(log(SUM)~country, data=co_data)
summary(model.logln)
shapiro.test(log(co_data$SUM))
plot(model.logln)

# P value 0.7534 and data not normally distributed.

## Poisson
model.pois = glm(SUM ~ country , data = co_data, family = poisson)
summary(model.pois)
plot(model.pois)

# P value <2e-16, Residual deviance: 77293085  on 86  degrees of freedom, AIC: 77294188.

## Quasipoisson
model.qpois<-glm(SUM~country, data=co_data, family="quasipoisson")
summary(model.qpois)
plot(model.qpois)

# P value <2e-16, Residual deviance: 77293085  on 86  degrees of freedom AIC: NA

## Negative binomial
model.nb <- glm.nb(SUM~country, data=co_data)
summary(model.nb)
plot(model.nb)

# P value < 2e-16, Residual deviance: 113.98  on 86  degrees of freedom, AIC: 2248, Theta:  0.4747 Std. Err.:  0.0585 

1/model.nb$theta
-2*(logLik(model.pois)-logLik(model.nb))


# Summarize and compare models
data.frame(linear=coef(model.ln),
                 loglinear=exp(coef(model.logln)),
                 poisson=exp(coef(model.pois)),
                 qpoisson=exp(coef(model.qpois)),
                 negbin=exp(coef(model.nb)))
# Critical value
qchisq(0.95, df.residual(model.nb))

deviance(model.ln)
deviance(model.logln)
deviance(model.pois)
deviance(model.qpois)
deviance(model.nb)

# None of the models fits the data.
```

```{r}
# Plot using neg. binom. model
co_data<-data.frame(SUM=round(mean(sample_data(resfinder_PHY_clean)$SSU_counts)*(sample_sums(resfinder_PHY_clean))), country=as.factor(sample_data(resfinder_PHY_clean)$country))

fit <- glm.nb(SUM ~ country, data = co_data, link = log)

co_data <- cbind(co_data, Mean = predict(fit, newdata = co_data, type = "response"), SE = predict(fit, newdata = co_data, type = "response", se.fit = T)$se.fit)

ARG.sum.co <- ggplot(co_data, aes(x = country, y = Mean)) +
        scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) +
    geom_line() + geom_jitter(data = co_data, aes(x = country, y = SUM, color = country), 
    size = 2.3, alpha = 0.5, width = 0.3) + 
    geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.6, lwd = 0.6) + 
    geom_point(size = 0.9) +
    theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    labs(y = "Sum abundance/16S", 
    x = "") + guides(color = FALSE, alpha = FALSE) + labs(title = "ARGs") + scale_y_log10()
ARG.sum.co
```
# Extract outliers
```{r}
co_data<-data.frame(SUM=round(mean(sample_data(resfinder_PHY_clean)$SSU_counts)*(sample_sums(resfinder_PHY_clean))), country=as.factor(sample_data(resfinder_PHY_clean)$country))

boxplot(co_data$SUM)
boxplot(co_data$SUM, plot=FALSE)$out
outliers <- boxplot(co_data$SUM, plot=FALSE)$out
print(outliers)

# Exclude outliers
## where they at
co_data[which(co_data$SUM %in% outliers),]
## remove
co_data <- co_data[-which(co_data$SUM %in% outliers),]
boxplot(co_data$SUM)

# New  plot
fit <- glm.nb(SUM ~ country, data = co_data, link = log)

co_data <- cbind(co_data, Mean = predict(fit, newdata = co_data, type = "response"), SE = predict(fit, newdata = co_data, type = "response", se.fit = T)$se.fit)

ARG.sum.co_outl <- ggplot(co_data, aes(x = country, y = Mean)) +
    scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) +
    geom_line() + geom_jitter(data = co_data, aes(x = country, y = SUM, color = country), 
    size = 2.3, alpha = 0.5, width = 0.3) + 
    geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.6, lwd = 0.6) + 
    geom_point(size = 0.9) +
    theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    labs(y = "Sum abundance/16S", 
    x = "") + guides(color = FALSE, alpha = FALSE) + labs(title = "ARGs") + scale_y_log10()
ARG.sum.co_outl
```
# Check signifigances
```{r}
# By location
lo_data<-data.frame(SUM=round(mean(sample_data(resfinder_PHY_clean)$SSU_counts)*(sample_sums(resfinder_PHY_clean))), location=as.factor(sample_data(resfinder_PHY_clean)$location))

lo_data[which(lo_data$SUM %in% outliers),]
lo_data <- lo_data[-which(lo_data$SUM %in% outliers),]

fit <- glm.nb(SUM ~ location, data = lo_data, link = log)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(location = "Tukey"))
print(summary(glht(glht.mod)))

# By country
co_data<-data.frame(SUM=round(mean(sample_data(resfinder_PHY_clean)$SSU_counts)*(sample_sums(resfinder_PHY_clean))), country=as.factor(sample_data(resfinder_PHY_clean)$country))

# Exclude outliers
co_data[which(co_data$SUM %in% outliers),]
co_data <- co_data[-which(co_data$SUM %in% outliers),]

fit <- glm.nb(SUM ~ country, data = co_data, link = log)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(country = "Tukey"))
summary(glht(glht.mod))
```
# ARG Classes, sum relative abundance
```{r}
resfinder_PHY_Class <- tax_glom(resfinder_PHY_clean, taxrank = "Class")

# All classes
resfinder_PHY_Class_abund <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_Class), TRUE)), 
    resfinder_PHY_Class)

# Normalize by number of samples in each country
otu_table(resfinder_PHY_Class_abund) <-   (otu_table(resfinder_PHY_Class_abund)[,]/as.matrix(table(sample_data(resfinder_PHY_Class_abund)$location))[, 1])

p17<-plot_bar(resfinder_PHY_Class_abund, x="location", fill="Class")
resfinder.p17 <- p17 + facet_grid(~ country, scales = "free") + 
  ggtitle("Sum of relative abundances of ARGs by classes/16S") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle=55, hjust=1, size = 10),
        axis.title.x = element_blank()) +
  guides(fill = guide_legend(label.theme = element_text(size = 7.5), keywidth = 1.5, keyheight = 1.5))
# longer object length is not a multiple of shorter object length

# Remove OTU separators
resfinder.p17 + geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack", size = 0.5)
```
# Top 12 ARG gene classes, abundance
```{r}
resfinder_PHY_Gene <- tax_glom(resfinder_PHY_clean, taxrank = "Gene")

resfinder_PHY_Gene_abund <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_Gene), TRUE)[1:12]), 
    resfinder_PHY_Gene)

# Normalize by number of samples in each country
otu_table(resfinder_PHY_Gene_abund) <-   (otu_table(resfinder_PHY_Gene_abund)[,]/as.matrix(table(sample_data(resfinder_PHY_Gene_abund)$location))[, 1])

p18<-plot_bar(resfinder_PHY_Gene_abund, x="location", fill="Gene")
resfinder.p18 <- p18 + facet_grid(~ country, scales = "free") + 
  ggtitle("Sum of relative abundances of ARGs by genes/16S") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle=55, hjust=1, size = 10),
        axis.title.x = element_blank()) +
  guides(fill = guide_legend(label.theme = element_text(size = 7.5), keywidth = 1.5, keyheight = 1.5))

# Remove OTU separators
resfinder.p18 + geom_bar(aes(color=Gene, fill=Gene), stat="identity", position="stack", size = 0.5)
```
# Betalactams
```{r}
resfinder_PHY_betalactams = subset_taxa(resfinder_PHY_clean, Class=="Betalactam")
any(taxa_sums(resfinder_PHY_betalactams) == 0)
resfinder_PHY_betalactams <- prune_taxa(taxa_sums(resfinder_PHY_betalactams) > 0, resfinder_PHY_betalactams)
any(taxa_sums(resfinder_PHY_betalactams) == 0)

resfinder_PHY_betalactams_Gene <- tax_glom(resfinder_PHY_betalactams, "Gene")

# Top12
resfinder_PHY_betalactams_Gene_abund <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_betalactams_Gene), TRUE)[1:12]), 
    resfinder_PHY_betalactams_Gene)

# Normalize by sample number
otu_table(resfinder_PHY_betalactams_Gene_abund) <- (otu_table(resfinder_PHY_betalactams_Gene_abund)[, ]/as.matrix(table(sample_data(resfinder_PHY_betalactams_Gene_abund)$location))[, 
    1])

p19<-plot_bar(resfinder_PHY_betalactams_Gene_abund, x="location", fill="Gene")
resfinder.p19 <- p19 + facet_grid(~ country, scales = "free") + 
  ggtitle("Sum of relative abundances of top 12 betalactams/16S") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle=55, hjust=1, size = 10),
        axis.title.x = element_blank()) +
  guides(fill = guide_legend(label.theme = element_text(size = 7.5), keywidth = 1.5, keyheight = 1.5))

# Remove OTU separators
resfinder.p19 + geom_bar(aes(color=Gene, fill=Gene), stat="identity", position="stack", size = 0.5)
```
# Metronidazole
```{r}
resfinder_PHY_metronidazole = subset_taxa(resfinder_PHY_clean, Class=="Metronidazole")
any(taxa_sums(resfinder_PHY_metronidazole) == 0)
resfinder_PHY_metronidazole <- prune_taxa(taxa_sums(resfinder_PHY_metronidazole) > 0, resfinder_PHY_metronidazole)
any(taxa_sums(resfinder_PHY_metronidazole) == 0)

resfinder_PHY_metronidazole_Gene <- tax_glom(resfinder_PHY_metronidazole, "Gene")

# Top12
resfinder_PHY_metronidazole_Gene_abund <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_metronidazole_Gene), TRUE)[1:12]), 
    resfinder_PHY_metronidazole_Gene)

# Normalize by sample number
otu_table(resfinder_PHY_metronidazole_Gene_abund) <- (otu_table(resfinder_PHY_metronidazole_Gene_abund)[, ]/as.matrix(table(sample_data(resfinder_PHY_metronidazole_Gene_abund)$location))[, 
    1])

p20<-plot_bar(resfinder_PHY_metronidazole_Gene_abund, x="location", fill="Gene")
resfinder.p20 <- p20 + facet_grid(~ country, scales = "free") + 
  ggtitle("Sum of relative abundances of top metronidazole ARGs/16S") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle=55, hjust=1, size = 10),
        axis.title.x = element_blank()) +
  guides(fill = guide_legend(label.theme = element_text(size = 7.5), keywidth = 1.5, keyheight = 1.5))

# Remove OTU separators
resfinder.p20 + geom_bar(aes(color=Gene, fill=Gene), stat="identity", position="stack", size = 0.5)
```


# DESEQ?
```{r}
ARGd <- ARG_genemat_norm[, ] * 10^5 + 1
as.integer(as.matrix(ARGd))

ARG_DESEQ <- phyloseq(otu_table(resfinder_PHY, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_resfinder)))
ARG_DESEQ_LU <- subset_samples(ARG_DESEQ, (country == "Benin"))
dds_arg_lu = phyloseq_to_deseq2(ARG_DESEQ_LU, ~location)

dds_arg_lu$location <- relevel(dds_arg_lu$location, "Savalou area")
dds_arg_lu = DESeq(dds_arg_lu, fitType = "mean", test = "Wald", betaPrior = FALSE)
# Error in estimateSizeFactorsForMatrix(counts(object), locfunc = locfunc, : every gene contains at least one zero, cannot compute log geometric means
```
# Beta diversity?
```{r}
# Add tree to the phyloseq object
random_tree = rtree(ntaxa(metaxa_PHY), rooted=TRUE, tip.label=taxa_names(metaxa_PHY))
physeq1 <- merge_phyloseq(metaxa_PHY, sample_data(metadata), random_tree)

physeq2 <- phyloseq(otu_table(OTU_metaxa, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaxa)), sample_data(metadata), random_tree)
identical(physeq1, physeq2)
```
# Rarefication?
```{r}
```


```{bash}

```


