---
title: "R_scripts_final"
#output: html_document
output: rmarkdown::github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set working directory
```{r}
setwd("~/Desktop/Git/AMRIWA/RFiles")
```
## Load required libraries
```{r, warning=FALSE, error=FALSE, message=FALSE, results='hide'}
library(tidyverse)
library(phyloseq)
library(viridis)
library(stringr)
library(vegan)
library(RColorBrewer)
library(BiocManager)
#BiocManager::install("microbiome")
library(microbiome)
#library(microbiomeutilities)
library(ggplot2)
library(knitr)
library(ggpubr)
library(pheatmap)
library(MASS)
library(gplots)
library(grid)
library(cowplot)
library(ggpubr)
library(DESeq2)
library(plyr)
library(multcomp)
library(randomForest)
library(corrplot)
library(ggrepel)
library(Hmisc)
source("HighstatLibV13.R")
library(ggcorrplot)
library(plyr)
library(lattice)
library(VennDiagram)
library(ggsn)
library(sf)
library(psych)
library(usefun)
library(patchwork)
library(sf)
library(rnaturalearth)
library(rnaturalearthdata)
library(ggspatial)
```
## Load metadata
```{r, warning=F}
metadata <-read.table("metadata.txt", sep="\t", header = T, row.names = 1, fill = 1, dec = ".", na.strings = "NA")

metadata$DNA_ng_µl <- as.numeric(gsub(",", ".", gsub("\\.", "", metadata$DNA_ng_µl)))
metadata$A260_280 <- as.numeric(gsub(",", ".", gsub("\\.", "", metadata$A260_280)))
metadata$M_Seqs_trimmed <- as.numeric(gsub(",", ".", gsub("\\.", "", metadata$M_Seqs_trimmed)))
metadata$lat <- as.numeric(gsub(",", ".", gsub("\\.", "", metadata$lat)))
metadata$long <- as.numeric(gsub(",", ".", gsub("\\.", "", metadata$long)))
#str(metadata)
```
## Load Metaxa2 results and create a phyloseq object
```{r}
metaxa_genus <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaxa_genus.txt")

# Create OTU table
OTU_metaxa <- metaxa_genus[,-1]
# Match sample ID order with metadata file
match <- match(rownames(metadata), colnames(OTU_metaxa))
OTU_metaxa <- OTU_metaxa[,match]
all(colnames(OTU_metaxa) == rownames(metadata))
# Create tax table
tax_table_metaxa <- data.frame(str_split_fixed(data.frame(metaxa_genus) [,1], ";", 6))
colnames(tax_table_metaxa) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus")
# Check if samples are in order
identical(rownames(metadata), colnames(OTU_metaxa))
# Combine into phyloseq object
metaxa_PHY <- phyloseq(otu_table(OTU_metaxa, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaxa)), sample_data(metadata))

# Exclude taxa "Unknown", "Unclassified", "Eukaryota", "Mitochondria", "Archaea", "Chloroplast"
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown"))
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Unclassified"))
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Eukaryota"))
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Mitochondria"))
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Archaea"))
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Chloroplast"))

# Add SSU counts to metadata
metadata$SSU_counts <- sample_sums(metaxa_PHY)

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaxa_PHY = subset_samples(metaxa_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaxa_PHY <- subset_samples(metaxa_PHY, country != "UK")

## Exclude biological / technical replicates
metaxa_PHY <- subset_samples(metaxa_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
metaxa_PHY_stat <- subset_samples(metaxa_PHY, category == "WA hospital effluent" | category == "North Eu hospital effluent")


# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
#BH <- data.frame(c("BH01",	"BH02",	"BH03",	"BH05",	"BH06",	"BH07",	"BH09",	"BH27",	"BH29",	"BH30",	"BH34A",	"BH35",	"BH36",	#"BH37",	"BH38",	"BH39",	"BH44",	"BH46",	"BH47",	"BH48",	"BH49",	"BH50",	"BH58",	"BH60",	"BH61"))
#colnames(BH) <- c("sample")
#random_BH <- sample_n(BH, 8)

#BFH <- data.frame(c("BFH1",	"BFH10",	"BFH11",	"BFH12",	"BFH13",	"BFH14",	"BFH15",	"BFH16",	"BFH17",	"BFH18",	"BFH19",	#"BFH20",	"BFH21",	"BFH22",	"BFH23",	"BFH25",	"BFH28",	"BFH29",	"BFH30",	"BFH31",	"BFH32",	"BFH33",	"BFH34",	"BFH35",	#"BFH36",	"BFH37",	"BFH38A",	"BFH39",	"BFH4",	"BFH40",	"BFH41",	"BFH6",	"BFH8",	"BFH9"))
#colnames(BFH) <- c("sample")
#random_BFH <- sample_n(BFH, 8)

# Sample set 1
metaxa_PHY_stat_equal <- subset_samples(metaxa_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 2
#metaxa_PHY_stat_equal <- subset_samples(metaxa_PHY, alias == "BFH34" | 	alias == "BFH35" | 	alias == "BFH18" | 	alias == "BFH32" | alias == "BFH14" | 	alias == "BFH41" | 	alias == "BFH22" |  	alias == "BFH23" |  alias == "BH48" | 	alias == "BH61" | 	alias == "BH30" |  	alias == "BH47" | alias == "BH46" | 	alias == "BH60" | 	alias == "BH39" | 	alias == "BH44" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" |  alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 3
#metaxa_PHY_stat_equal <- subset_samples(metaxa_PHY, alias == "BFH20" | 	alias == "BFH8" | 	alias == "BFH30" | 	alias == "BFH16" | alias == "BFH33" | 	alias == "BFH25" | 	alias == "BFH10" |  	alias == "BFH37" |  alias == "BH49" | 	alias == "BH60" | 	alias == "BH36" |  	alias == "BH46" |  alias == "BH02" | 	alias == "BH30" | 	alias == "BH61" | 	alias == "BH37" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Create phyloseq object with only West African samples
metaxa_PHY_WA_stat <- subset_samples(metaxa_PHY, continent == "West Africa")

# Create phyloseq object with only West African samples
metaxa_PHY_WA_WW_stat <- subset_samples(metaxa_PHY_WA_stat, category == "WA hospital effluent")
```
## Load data for rpoB
```{r}
HMM_RESULT_TABLE <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/HMM_RESULT_TABLE.txt", row.names=1)
HMM_RESULT_TABLE$SUM = rowSums(HMM_RESULT_TABLE[,c(2,3)])

# Reorder samples to match metadata and add to metadata
# SUM
match <- match(rownames(metadata), rownames(HMM_RESULT_TABLE))
rpoB_counts <- HMM_RESULT_TABLE[match,]
metadata$rpoB_counts <- rpoB_counts$SUM

# R1
# Reorder samples to match metadata
match <- match(rownames(metadata), rownames(HMM_RESULT_TABLE))
R1_rpoB_counts <- HMM_RESULT_TABLE[match,]
metadata$R1_rpoB_counts <- rpoB_counts$R1
```
## Load Metaphlan3 results and create a phyloseq object
```{r}
OTU_metaphlan <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_merged_abundance_table_species.txt", header=T)

# Make sure tax tabe is in order
#tax_table_metaphlan <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", quote="\"", comment.char="")
#identical(tax_table_metaphlan$V1, OTU_metaphlan$clade_name)

tax_table_metaphlan <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", header=FALSE, sep=";")
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Remove "__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

match <- match(rownames(metadata), colnames(OTU_metaphlan))
OTU_metaphlan  <- OTU_metaphlan[,match]
all(rownames(metadata) == colnames(OTU_metaphlan))

# Combine into phyloseq object
metaphlan_PHY <- phyloseq(otu_table(OTU_metaphlan, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaphlan)), sample_data(metadata))
# Check that sums are ~100
#sample_sums(metaphlan_PHY)

# Exclude Viruses, Eukaryota & Archaea
metaphlan_PHY <- subset_taxa(metaphlan_PHY, Kingdom != "Viruses" & Kingdom != "Eukaryota" & Kingdom != "Archaea")

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_PHY = subset_samples(metaphlan_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaphlan_PHY <- subset_samples(metaphlan_PHY, country != "UK")
## Exclude biological / technical replicates
metaphlan_PHY <- subset_samples(metaphlan_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
metaphlan_PHY_stat <- subset_samples(metaphlan_PHY, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
# Sample set 1
metaphlan_PHY_stat_equal <- subset_samples(metaphlan_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 2
#metaphlan_PHY_stat_equal <- subset_samples(metaphlan_PHY, alias == "BFH34" | 	alias == "BFH35" | 	alias == "BFH18" | 	alias == "BFH32" | alias == "BFH14" | 	alias == "BFH41" | 	alias == "BFH22" |  	alias == "BFH23" |  alias == "BH48" | 	alias == "BH61" | 	alias == "BH30" |  	alias == "BH47" | alias == "BH46" | 	alias == "BH60" | 	alias == "BH39" | 	alias == "BH44" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" |  alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 3
#metaphlan_PHY_stat_equal <- subset_samples(metaphlan_PHY, alias == "BFH20" | 	alias == "BFH8" | 	alias == "BFH30" | 	alias == "BFH16" | alias == "BFH33" | 	alias == "BFH25" | 	alias == "BFH10" |  	alias == "BFH37" |  alias == "BH49" | 	alias == "BH60" | 	alias == "BH36" |  	alias == "BH46" |  alias == "BH02" | 	alias == "BH30" | 	alias == "BH61" | 	alias == "BH37" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Create phyloseq object with only West African samples
metaphlan_PHY_WA_stat <- subset_samples(metaphlan_PHY, continent == "West Africa")

# Create phyloseq object with only West African wastewaters
metaphlan_PHY_WA_WW_stat <- subset_samples(metaphlan_PHY_WA_stat, category == "WA hospital effluent")
```
## Load ResFinder results and create a phyloseq object
```{r}
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Divide by ARG gene lengths
## Get the lengths in terminal
# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
OTU_resfinder_length_SSU_norm <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(OTU_resfinder_length_SSU_norm))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], OTU_resfinder_length_SSU_norm[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Reload to get new row numbers
write.table(OTU_resfinder_length_SSU_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_length_SSU_norm.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_resfinder_length_SSU_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_length_SSU_norm.txt", row.names=NULL)
OTU_resfinder_length_SSU_norm$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

# Combine to phyloseq object
resfinder_PHY <- phyloseq(otu_table(OTU_resfinder_length_SSU_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY = subset_samples(resfinder_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
resfinder_PHY <- subset_samples(resfinder_PHY, country != "UK")
## Exclude biological / technical replicates
resfinder_PHY <- subset_samples(resfinder_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")
# Create phyloseq object with only hospital WW samples sequenced here
resfinder_PHY_stat <- subset_samples(resfinder_PHY, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
# Sample set 1
resfinder_PHY_stat_equal <- subset_samples(resfinder_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 2
#resfinder_PHY_stat_equal <- subset_samples(resfinder_PHY, alias == "BFH34" | 	alias == "BFH35" | 	alias == "BFH18" | 	alias == "BFH32" | alias == "BFH14" | 	alias == "BFH41" | 	alias == "BFH22" |  	alias == "BFH23" |  alias == "BH48" | 	alias == "BH61" | 	alias == "BH30" |  	alias == "BH47" | alias == "BH46" | 	alias == "BH60" | 	alias == "BH39" | 	alias == "BH44" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" |  alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 3
#resfinder_PHY_stat_equal <- subset_samples(resfinder_PHY, alias == "BFH20" | 	alias == "BFH8" | 	alias == "BFH30" | 	alias == "BFH16" | alias == "BFH33" | 	alias == "BFH25" | 	alias == "BFH10" |  	alias == "BFH37" |  alias == "BH49" | 	alias == "BH60" | 	alias == "BH36" |  	alias == "BH46" |  alias == "BH02" | 	alias == "BH30" | 	alias == "BH61" | 	alias == "BH37" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Create phyloseq object with only West African samples
resfinder_PHY_WA_stat <- subset_samples(resfinder_PHY, continent == "West Africa")

# Create phyloseq object with only West African wastewaters
resfinder_PHY_WA_WW_stat <- subset_samples(resfinder_PHY_WA_stat, category == "WA hospital effluent")
```
## Normalization with rpoB
```{r}
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Normalization with rpoB counts
OTU_resfinder_rpoB_norm <- t(t(OTU_resfinder)/metadata$rpoB_counts)
all(rownames(metadata) == colnames(OTU_resfinder_rpoB_norm))
all(rownames(OTU_resfinder_rpoB_norm) == clusters_tax_table_resfinder$Gene)

# Reload to get new row numbers
write.table(OTU_resfinder_rpoB_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_rpoB_norm.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_resfinder_rpoB_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_rpoB_norm.txt", row.names=NULL)
OTU_resfinder_rpoB_norm$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

# Combine to phyloseq object
resfinder_rpob_PHY <- phyloseq(otu_table(OTU_resfinder_rpoB_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_rpob_PHY = subset_samples(resfinder_rpob_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
resfinder_rpob_PHY <- subset_samples(resfinder_rpob_PHY, country != "UK")
## Exclude biological / technical replicates
resfinder_rpob_PHY <- subset_samples(resfinder_rpob_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
resfinder_rpob_PHY_stat <- subset_samples(resfinder_rpob_PHY, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
# Sample set 3
resfinder_rpob_PHY_stat_equal <- subset_samples(resfinder_rpob_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 2
#resfinder_rpob_PHY_stat_equal <- subset_samples(resfinder_rpob_PHY, alias == "BFH34" | 	alias == "BFH35" | 	alias == "BFH18" | 	alias == "BFH32" | alias == "BFH14" | 	alias == "BFH41" | 	alias == "BFH22" |  	alias == "BFH23" |  alias == "BH48" | 	alias == "BH61" | 	alias == "BH30" |  	alias == "BH47" | alias == "BH46" | 	alias == "BH60" | 	alias == "BH39" | 	alias == "BH44" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" |  alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 3
#resfinder_rpob_PHY_stat_equal <- subset_samples(resfinder_rpob_PHY, alias == "BFH20" | 	alias == "BFH8" | 	alias == "BFH30" | 	alias == "BFH16" | alias == "BFH33" | 	alias == "BFH25" | 	alias == "BFH10" |  	alias == "BFH37" |  alias == "BH49" | 	alias == "BH60" | 	alias == "BH36" |  	alias == "BH46" |  alias == "BH02" | 	alias == "BH30" | 	alias == "BH61" | 	alias == "BH37" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Create phyloseq object with only West African samples
resfinder_rpob_PHY_WA_stat <- subset_samples(resfinder_rpob_PHY, continent == "West Africa")

# Create phyloseq object with only West African wastewaters
resfinder_rpob_PHY_WA_WW_stat <- subset_samples(resfinder_rpob_PHY_WA_stat, category == "WA hospital effluent")
```
## Load CARD results for the samples sequenced here
```{r}
CARD_metadata <- data.frame(resfinder_PHY_stat@sam_data)

# Headers simplified & ";" changed into tab in excel
OTU_CARD <-as.matrix(read.table("mod_COUNT_TABLE.txt", header= T, check.names = F, row.names = 1))
match <- match(rownames(CARD_metadata), colnames(OTU_CARD))
OTU_CARD <- OTU_CARD[,match]

# Normalization with Metaxa2 SSU counts
all(colnames(OTU_CARD) == rownames(CARD_metadata))
OTU_CARD_norm <- t(t(OTU_CARD)/CARD_metadata$SSU_counts)
all(rownames(CARD_metadata) == colnames(OTU_CARD_norm))
identical(OTU_CARD[106, 4]/CARD_metadata$SSU_counts[4], OTU_CARD_norm[106, 4])

# Normalization with HMM rpoB counts
#OTU_CARD_norm_rpoB <- t(t(OTU_CARD)/CARD_metadata$rpoB_counts)
#all(rownames(CARD_metadata) == colnames(OTU_CARD_norm_rpoB))

# Tax table (Modified in excel)
tax_table_CARD <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_CARD.txt", header=FALSE, sep="\t")
colnames(tax_table_CARD) <- c("Gene", "Class")
all(tax_table_CARD$Gene == rownames(OTU_CARD_norm))

# Modify 
tax_table_CARD_renamed = tax_table_CARD
tax_table_CARD_renamed$Gene <- as.matrix(gsub(pattern = "_[A-Z].*", replacement = "", tax_table_CARD$Gene))
# Have a look at the hits with duplicate names
dups <-tax_table_CARD_renamed[duplicated(tax_table_CARD_renamed$Gene)|duplicated(tax_table_CARD_renamed$Gene),]
# They are not those ones of interest here so we will not spend time on renaming them

# Reload to get new row numbers
write.table(OTU_CARD_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_CARD_norm.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_CARD_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_CARD_norm.txt", row.names=NULL)
OTU_CARD_norm$row.names<-NULL

CARD_PHY_stat <- phyloseq(otu_table(OTU_CARD_norm, taxa_are_rows = TRUE), sample_data(CARD_metadata), 
    tax_table(as.matrix(tax_table_CARD_renamed)))
```
## Load MGE results
```{r}
OTU_MGE <-as.matrix(read.table("cp_MGE_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_MGE))
OTU_MGE <- OTU_MGE[,match]
all(colnames(OTU_MGE) == rownames(metadata))

# Tax table
MGE_tax_table_trim <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", header=FALSE)
colnames(MGE_tax_table_trim) <- c("Gene", "Element", "Class")

# Reorder tax_table to match
match <- match(rownames(OTU_MGE), MGE_tax_table_trim$Gene)
MGE_tax_table_trim <- MGE_tax_table_trim[match,]
all(rownames(OTU_MGE) == MGE_tax_table_trim$Gene)

MGE_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_lengths.txt", header=FALSE, comment.char="#", check.names = F)
match <- match(rownames(OTU_MGE), MGE_lengths$V1)
MGE_lengths <- MGE_lengths[match,]
all(rownames(MGE_tax_table_trim$Gene) == MGE_lengths$V1)
OTU_MGE_length_norm <- OTU_MGE/MGE_lengths[, 2]

# Normalization with Metaxa2 SSU counts
OTU_MGE_length_SSU_norm <- t(t(OTU_MGE_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(OTU_MGE_length_SSU_norm))
all(rownames(OTU_MGE_length_SSU_norm) == MGE_tax_table_trim$Gene)

# Reload to get new row numbers
write.table(OTU_MGE_length_SSU_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_MGE_length_SSU_norm", 
            row.names=T, sep = "\t", col.names = T)
OTU_MGE_length_SSU_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_MGE_length_SSU_norm", row.names=NULL)
OTU_MGE_length_SSU_norm$row.names<-NULL

# Reload to get new row numbers
write.table(MGE_tax_table_trim, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", row.names=F, sep = "\t", col.names = T)
MGE_tax_table_trim <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", row.names=NULL)

# Combine to phyloseq object
MGE_PHY <- phyloseq(otu_table(OTU_MGE_length_SSU_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(MGE_tax_table_trim)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
MGE_PHY = subset_samples(MGE_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
MGE_PHY <- subset_samples(MGE_PHY, country != "UK")
## Exclude biological / technical replicates
MGE_PHY <- subset_samples(MGE_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")
# Create phyloseq object with only hospital WW samples sequenced here
MGE_PHY_stat <- subset_samples(MGE_PHY, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
# Sample set 1
MGE_PHY_stat_equal <- subset_samples(MGE_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 2
#MGE_PHY_stat_equal <- subset_samples(MGE_PHY, alias == "BFH34" | 	alias == "BFH35" | 	alias == "BFH18" | 	alias == "BFH32" | alias == "BFH14" | 	alias == "BFH41" | 	alias == "BFH22" |  	alias == "BFH23" |  alias == "BH48" | 	alias == "BH61" | 	alias == "BH30" |  	alias == "BH47" | alias == "BH46" | 	alias == "BH60" | 	alias == "BH39" | 	alias == "BH44" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" |  alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 3
#MGE_PHY_stat_equal <- subset_samples(MGE_PHY, alias == "BFH20" | 	alias == "BFH8" | 	alias == "BFH30" | 	alias == "BFH16" | alias == "BFH33" | 	alias == "BFH25" | 	alias == "BFH10" |  	alias == "BFH37" |  alias == "BH49" | 	alias == "BH60" | 	alias == "BH36" |  	alias == "BH46" |  alias == "BH02" | 	alias == "BH30" | 	alias == "BH61" | 	alias == "BH37" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Create phyloseq object with only West African samples
MGE_PHY_WA_stat <- subset_samples(MGE_PHY, continent == "West Africa")

# Create phyloseq object with only West African wastewaters
MGE_PHY_WA_WW_stat <- subset_samples(MGE_PHY_WA_stat, category == "WA hospital effluent")

# Get class 1 integrons
MGE_PHY_int <- tax_glom(MGE_PHY, taxrank = "Class")
MGE_PHY_int <- subset_taxa(MGE_PHY_int, Class == "intI1")

MGE_PHY_int_stat <- tax_glom(MGE_PHY_stat, taxrank = "Class")
MGE_PHY_int_stat <- subset_taxa(MGE_PHY_int_stat, Class == "intI1")

MGE_PHY_int_stat_equal <- tax_glom(MGE_PHY_stat_equal, taxrank = "Class")
MGE_PHY_int_stat_equal <- subset_taxa(MGE_PHY_int_stat_equal, Class == "intI1")

MGE_PHY_qac <- tax_glom(MGE_PHY_stat, taxrank = "Class")
MGE_PHY_qac <- subset_taxa(MGE_PHY_qac, Class == "qacEdelta")
```
## Load Virulence genes
```{r}
OTU_VFDB <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/VF_COUNT_TABLE", sep=";")

## Modify headers in terminal
#sed -n 's/\(;\).*/\1/p' VF_COUNT_TABLE > output_headers.txt
#sed -i 's/;//g' output_headers.txt
#sed -i '/^$/d' output_headers.txt

output_headers <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/output_headers.txt", quote="\"", comment.char="")
colnames(output_headers) <- c("Accession")
# Check
all(output_headers$V1 == OTU_VFDB$X)

# Reorder according to metadata
match <- match(rownames(metadata), colnames(OTU_VFDB))
OTU_VFDB <- OTU_VFDB[,match]
all(colnames(OTU_VFDB) == rownames(metadata))

# Normalization with Metaxa2 SSU counts
OTU_VFDB_SSU_norm <- t(t(OTU_VFDB)/metadata$SSU_counts)
all(rownames(metadata) == colnames(OTU_VFDB_SSU_norm))
#all(rownames(OTU_VFDB_SSU_norm) == tax_table_VFDB$V1)

# Reload to get new row numbers
write.table(OTU_VFDB_SSU_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_VFDB_SSU_norm.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_VFDB_SSU_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_VFDB_SSU_norm.txt", row.names=NULL)
OTU_VFDB_SSU_norm$row.names<-NULL

## Split columns so that there are more columns
tax_table_VFDB <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/VF.txt", header=FALSE, sep = " ")
new_columns <- str_split_fixed(tax_table_VFDB$V2, ";", 4)
colnames(new_columns) <- c("Gene", "Description", "Class", "Reference")

tax_table_VFDB <- tax_table_VFDB[-2]
colnames(tax_table_VFDB) <- c("Accession")
tax_table_VFDB <- cbind(tax_table_VFDB, new_columns)

# order as in the genemat
match <- match(output_headers$Accession, tax_table_VFDB$Accession)
tax_table_VFDB <- tax_table_VFDB[match,]
identical(tax_table_VFDB$Accession, output_headers$Accession)

# Reload to get new row numbers
write.table(tax_table_VFDB, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_VFDB.txt", row.names=F, sep = "\t", col.names = T)
tax_table_VFDB <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_VFDB.txt", row.names=NULL)
identical(tax_table_VFDB$Accession, output_headers$Accession)

VFDB_PHY <- phyloseq(otu_table(OTU_VFDB_SSU_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_VFDB)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
VFDB_PHY = subset_samples(VFDB_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
VFDB_PHY <- subset_samples(VFDB_PHY, country != "UK")
## Exclude biological / technical replicates
VFDB_PHY <- subset_samples(VFDB_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")
# Create phyloseq object with only hospital WW samples sequenced here
VFDB_PHY_stat <- subset_samples(VFDB_PHY, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
# Sample set 1
VFDB_PHY_stat_equal <- subset_samples(VFDB_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 2
#VFDB_PHY_stat_equal <- subset_samples(VFDB_PHY, alias == "BFH34" | 	alias == "BFH35" | 	alias == "BFH18" | 	alias == "BFH32" | alias == "BFH14" | 	alias == "BFH41" | 	alias == "BFH22" |  	alias == "BFH23" |  alias == "BH48" | 	alias == "BH61" | 	alias == "BH30" |  	alias == "BH47" | alias == "BH46" | 	alias == "BH60" | 	alias == "BH39" | 	alias == "BH44" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" |  alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 3
#VFDB_PHY_stat_equal <- subset_samples(VFDB_PHY, alias == "BFH20" | 	alias == "BFH8" | 	alias == "BFH30" | 	alias == "BFH16" | alias == "BFH33" | 	alias == "BFH25" | 	alias == "BFH10" |  	alias == "BFH37" |  alias == "BH49" | 	alias == "BH60" | 	alias == "BH36" |  	alias == "BH46" |  alias == "BH02" | 	alias == "BH30" | 	alias == "BH61" | 	alias == "BH37" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Create phyloseq object with only West African samples
VFDB_PHY_WA_stat <- subset_samples(VFDB_PHY, continent == "West Africa")

# Create phyloseq object with only West African wastewaters
VFDB_PHY_WA_WW_stat <- subset_samples(VFDB_PHY_WA_stat, category == "WA hospital effluent")
```
## Load crAssphage results
```{r}
crassphage_table <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table.txt", header = T, row.names = 1)

# Match the order according to metadata
match <- match(rownames(metadata), rownames(crassphage_table))
crassphage_table <- crassphage_table[match,]
identical(rownames(metadata), rownames(crassphage_table))
# Drop columns
crassphage_table = subset(crassphage_table, select = -c(coverage, alias) )
identical(rownames(metadata), rownames(crassphage_table))
crassphage_table <- t(crassphage_table)

crassphage_table_norm <- t(t(crassphage_table)/metadata$SSU_counts)
identical(crassphage_table[1, 12]/metadata$SSU_counts[12], crassphage_table_norm[1, 12])

write.table(crassphage_table_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt", row.names=T, sep = "\t", col.names = T)
crassphage_table_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt")

# Reload to get new row numbers
all(rownames(metadata) == colnames(crassphage_table_norm))
write.table(crassphage_table_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt", 
            row.names=T, sep = "\t", col.names = T)
crassphage_table_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt", row.names=NULL)
crassphage_table_norm$row.names<-NULL

# Create phyloseq object
crass_PHY <- phyloseq(otu_table(crassphage_table_norm, taxa_are_rows = T), sample_data(metadata))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
crass_PHY = subset_samples(crass_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
crass_PHY <- subset_samples(crass_PHY, country != "UK")
## Exclude biological / technical replicates
crass_PHY <- subset_samples(crass_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")
# Create phyloseq object with only hospital WW samples sequenced here
crass_PHY_stat <- subset_samples(crass_PHY, category == "WA hospital effluent" | category == "North Eu hospital effluent")


# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
# Sample set 1
#crass_PHY_stat_equal <- subset_samples(crass_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 2
#crass_PHY_stat_equal <- subset_samples(crass_PHY, alias == "BFH34" | 	alias == "BFH35" | 	alias == "BFH18" | 	alias == "BFH32" | alias == "BFH14" | 	alias == "BFH41" | 	alias == "BFH22" |  	alias == "BFH23" |  alias == "BH48" | 	alias == "BH61" | 	alias == "BH30" |  	alias == "BH47" | alias == "BH46" | 	alias == "BH60" | 	alias == "BH39" | 	alias == "BH44" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" |  alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Sample set 3
#crass_PHY_stat_equal <- subset_samples(crass_PHY, alias == "BFH20" | 	alias == "BFH8" | 	alias == "BFH30" | 	alias == "BFH16" | alias == "BFH33" | 	alias == "BFH25" | 	alias == "BFH10" |  	alias == "BFH37" |  alias == "BH49" | 	alias == "BH60" | 	alias == "BH36" |  	alias == "BH46" |  alias == "BH02" | 	alias == "BH30" | 	alias == "BH61" | 	alias == "BH37" | alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

# Create phyloseq object with only West African samples
crass_PHY_WA_stat <- subset_samples(crass_PHY, continent == "West Africa")

# Create phyloseq object with only West African wastewaters
crass_PHY_WA_WW_stat <- subset_samples(crass_PHY_WA_stat, category == "WA hospital effluent")
```
##### Modelling ARG abundance #####
## Data exploration
```{r, warning=FALSE, error=FALSE, message=FALSE, results='hide'}
df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_stat),
               intI1_SUM=sample_sums(MGE_PHY_int_stat),
               MGE_SUM=sample_sums(MGE_PHY_stat),
               hospital_section=as.factor(sample_data(resfinder_PHY_stat)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_stat)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_stat)$hospital),
               country=as.factor(sample_data(resfinder_PHY_stat)$country),
               R1_rpoB_counts=as.factor(sample_data(resfinder_PHY_stat)$R1_rpoB_counts),
               no_of_beds=as.factor(sample_data(resfinder_PHY_stat)$no_of_beds),
               long=as.factor(sample_data(resfinder_PHY_stat)$long),
               lat=as.factor(sample_data(resfinder_PHY_stat)$lat),
               A260_280=as.numeric(sample_data(resfinder_PHY_stat)$A260_280),
               DNA_ng_µl=as.numeric(sample_data(resfinder_PHY_stat)$DNA_ng_µl),
               M_Seqs_trimmed=as.numeric(sample_data(resfinder_PHY_stat)$M_Seqs_trimmed))
              
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
df$R1_rpoB_counts <- as.character(df$R1_rpoB_counts)
df$R1_rpoB_counts <- as.numeric(df$R1_rpoB_counts)
df$no_of_beds <- as.character(df$no_of_beds)
df$no_of_beds <- as.numeric(df$no_of_beds)
```
# Spatial positions
```{r}
#theme_bw()
world <- ne_countries(scale = "medium", returnclass = "sf")
class(world)

gps <- metadata[!duplicated(metadata[,c('lat','long')]),]
gps <- gps[ , c("country", "lat", "long", "hospital")] 
gps <- subset(gps, country=="Benin" | country == "Burkina Faso")
gps0 <- data.frame("Burkina Faso", "12.500000", "-1.666670", "H")
rownames(gps0) <- "BFH13_S131"
colnames(gps0) <- c("country", "lat", "long", "hospital")
gps <- rbind(gps, gps0)

# Add important cities
gps_labels <- data.frame(country = c("country_name", "Benin", "Benin", "country_name", 
                                     "Burkina Faso", "Burkina Faso", "ocean"),
                               lat = c("10.544904033009432", "6.3676953", "9.3400159", "13.740788326149952",  
                                       "12.3681873", "11.1757783", "4.944956754100344"),
                               long = c("2.3165032566686428", "2.4252507", "2.6278258", "-1.0794179365270806", 
                                        "-1.5270944", "-4.2957591", "2.376996878456601"),
                             hospital = c("nd", "nd", "nd", "nd", "nd", "nd", "nd"))
rownames(gps_labels) <- c("Benin", "Cotonou", "Parakou", "Burkina Faso", 
                          "Ouagadougou", "Bobo Dioulasso", "Gulf of Guinea")

gps_data <- rbind(gps, gps_labels)
gps_data$Label <- c("nd", "nd", "nd", "nd", "nd", "nd", "nd", "nd", "nd", "nd",
                    "Benin", "Cotonou", "Parakou", "Burkina Faso", 
                    "Ouagadougou", "Bobo Dioulasso", "Gulf of Guinea")
gps_data$lat <- as.numeric(gps_data$lat)
gps_data$long <- as.numeric(gps_data$long)

p_map1 <- ggplot(data = world) +
    geom_sf() +
    borders("world", colour="black", fill="wheat1")  +
    theme(panel.background = element_rect(fill = "azure1", colour = "azure1")) +
    geom_point(data = subset(gps_data, Label == "nd"), 
               aes(x = long, y = lat), size = 4, shape = 16, color = "#B2182B") +
    geom_text_repel(data = subset(gps_data, Label == "nd"), 
                    mapping = aes(x = long, y = lat, label = hospital, family = "Times"), size = 7.5, 
                    #hjust = -1.3, vjust = -0.1,
                    point.padding = 1e-06) +
  coord_sf(ylim = c(4.5, 14.75), xlim = c(-6, 3.95), expand = T) + 
  theme(axis.text = element_text(family = "Times", size = 16),
        axis.title = element_blank()) +
annotation_scale(location = "bl", width_hint = 0.2, height = unit(0.3, "cm"))

# Add cities
p_map2 <- p_map1 + 
  geom_point(data = subset(gps_data, Label == "Porto Novo" | Label == "Cotonou" | Label == "Parakou" | 
                             Label == "Ouagadougou" | Label == "Bobo Dioulasso"),
               aes(x = long, y = lat), size = 5, shape = 9, color = "black") +
  geom_label_repel(data=subset(gps_data, Label == "Porto Novo" | Label == "Cotonou" | Label == "Parakou" | 
                             Label == "Ouagadougou" | Label == "Bobo Dioulasso"),  
                      aes(x = long, y = lat, label = Label), color = "black", size = 8, family = "Times", 
                   box.padding = 1.75)
p_map2

# Add countries
p_map3 <- p_map2 + 
  geom_point(data = subset(gps_data, Label == "Benin" | Label == "Burkina Faso" | Label == "Gulf of Guinea"),
               aes(x = long, y = lat), size = 0, shape = 16, color = "black") +
  geom_text_repel(data = subset(gps_data, Label == "Benin" | Label == "Burkina Faso" | Label == "Gulf of Guinea"),
                                                 aes(x = long, y = lat, label = Label), 
                  color = "#4C4B49", size = 13, family = "Times")
p_map3

ggsave(filename = "p_map.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Fin
world <- ne_countries(scale = "medium", returnclass = "sf")
class(world)

gps <- metadata[!duplicated(metadata[,c('lat','long')]),]
gps <- gps[ , c("country", "lat", "long")] 
gps_Fin <- subset(gps, country=="Finland")

Fin_map <- ggplot(data = world) +
    geom_sf() +
    borders("world", colour="black", fill="wheat1")  +
    theme(panel.background = element_rect(fill = "azure1", colour = "azure1")) +
    geom_point(data = subset(gps_Fin), 
               aes(x = long, y = lat), size = 4, shape = 16, color = "#B2182B") +
  coord_sf(ylim = c(60, 67), xlim = c(22, 36), expand = T) + 
  theme(axis.text = element_text(family = "Times", size = 16),
#        plot.title = element_text(family = "Times", face = "bold", size = 28, hjust = 0.5, vjust = 0.5, colour = "#4C4B49"),
        axis.title = element_blank()) +
#  ggtitle("Burkina Faso") +
annotation_scale(location = "bl", width_hint = 0.1)

Fin_map <- Fin_map + theme(plot.margin = ggplot2::margin(0, 0, 0, 0, "cm"))
```
```{r}
# Outliers
MyVar <- c("country", "hospital", "hospital_section", "SSU_counts", "R1_rpoB_counts", "no_of_beds", "ARG_SUM", "intI1_SUM", "A260_280", "DNA_ng_µl", "M_Seqs_trimmed")
Mydotplot(df[,MyVar])

# Identify the outlier
#par(mfrow = c(1, 1))
#plot(x = df$country, 
# 	   y = df$SSU_counts)
#identify(x = df$country,
#	      y = df$SSU_counts)

# Remove it?
#df <- df[-60, ]
#dim(df)
# No, we shoud not remove it since it is most likely  a
# a real observation, and there are not enough observations to be sure.

# Collinearity
MySel <- c("ARG_SUM", "intI1_SUM", "country", "hospital", "hospital_section", "SSU_counts", "R1_rpoB_counts", "continent", "A260_280", "DNA_ng_µl", "M_Seqs_trimmed")
Mypairs(df[,MySel])
# Obviously, there is collinearity between continent and country.
# We can continue with onlu country. Similarly we can apply only SSU_counts 
# as they are colloinear with R1_rpoB_counts.

#par(mfrow = c(1, 2))
boxplot(ARG_SUM ~ country, data = df, main = "ARGs")
boxplot(intI1_SUM ~ country, data = df, main = "intI1")

# Numer of zeros in the response variable
#100 * sum(df$ARG_SUM == 0) / nrow(df)

#number of observations per level of a categorical covariate
table(df$country)
table(df$hospital)
# Unequal number of hospitals. Let's not include that as a covariate.

# Conditional boxplots to study the collinearity of the covariates
par(mfrow = c(2, 2))
boxplot(SSU_counts ~ country, 
        xlab = "country",
        data = df, 
        main = "SSU_counts")
boxplot(R1_rpoB_counts ~ country, 
        xlab = "country",
        data = df, 
        main = "rpoB counts")
boxplot(M_Seqs_trimmed ~ country, 
        xlab = "country",
        data = df, 
        main = "M_Seqs_trimmed")
boxplot(DNA_ng_µl ~ country, 
        xlab = "country",
        data = df, 
        main = "DNA_ng_µl")

# There seems to be variation in the SSU_counts
# between hospital WW collected in different countries.
# Although the SSU and rpoB counts seemed to be collinear, 
# it seems that the SSU_counts have a greater variance.
# Let's pick that among these two.

# Interactions
# SSU counts and ARG abundances by country
p <- ggplot(data = df,
            aes(x = SSU_counts, 
                y = ARG_SUM))
p <- p + geom_point()
p <- p + xlab("SSU_counts") + ylab("ARGs")
p <- p + theme(text = element_text(size=15))
p <- p + geom_smooth(method = "glm")
p <- p + facet_grid(~country)
p 
# There seems to be some kind of interaction defined by country, 
# since the smoothers are poistive/negative/with different slopes
# among different countries.

p <- ggplot(data = df,
            aes(x = SSU_counts, 
                y = intI1_SUM))
p <- p + geom_point()
p <- p + xlab("SSU_counts") + ylab("IntI1")
p <- p + theme(text = element_text(size=15))
p <- p + geom_smooth(method = "glm") #You can also do glm
p <- p + facet_grid(~country)
p
# There seems to be some kind of interaction defined by country.
# However, the quality of the data (e.g. only few observations in Finland)
# is too low to be able to include this interaction to the model.
# Maybe it is good not to include this interaction to the model due to the low quality data.

# The relationship of class 1 integron to the ARG sum
p <- ggplot()
p <- p + geom_point(data = df, 
                    aes(y = ARG_SUM, 
                        x = intI1_SUM),
                    shape = 1, 
                    size = 2)
p <- p + xlab("intI") + ylab("args")
p <- p + theme(text = element_text(size=15)) 
p <- p + geom_smooth(data = df, 
                     method = glm,
                     aes(x = intI1_SUM, 
                         y = ARG_SUM))
p <- p + facet_grid(. ~ country) 
p
# In Finland the steeper correlation relies solely on one datapoint
# Thus it is difficult to say whether this is real. 
# Due to the low quality of this data, this interactions should not be included in the model.
# We can however build a model and test whether it is significant.

# Does the hospital section have a role?
par(mfrow = c(1, 2))
boxplot(ARG_SUM ~ hospital_section, 
        xlab = "country",
        data = df, 
        main = "hospital_section")
boxplot(intI1_SUM ~ hospital_section, 
        xlab = "country",
        data = df, 
        main = "hospital_section")
# No.

# ANALYSIS
# The model:
#Call:  glm(formula = ARG_SUM ~ country + SSU_counts + intI1_SUM + country:intI1_SUM, 
#    family = Gamma(link = "log"), data = df)
# would have been ideal, but was discarded due to the fact that the interactions were not 
# based on high quality data and relied on single influental data points.

# MODEL SELECTION
# Let's fit a model with Gamma distribution with a log link.
Mall <- glm(ARG_SUM ~ country + SSU_counts, + intI1_SUM,
           data = df, family="Gamma"(link="log"))
summary(Mall)

M1 <- glm(ARG_SUM ~ country,
           data = df, family="Gamma"(link="log"))
summary(M1)
step(M1, direction = "both", scope=formula(Mall))

# We are left with:
#Call:  glm(formula = ARG_SUM ~ country + intI1_SUM + SSU_counts, family = Gamma(link = "log"), 
#    data = df)
# Let us build the new selected model
M1 <- glm(ARG_SUM ~ country + intI1_SUM + SSU_counts,
           data = df, family="Gamma"(link="log"))
summary(M1)
#                      Estimate Std. Error t value Pr(>|t|)    
#(Intercept)         -7.639e-01  1.502e-01  -5.085 3.65e-06 ***
#countryBurkina Faso -3.405e-01  1.021e-01  -3.335 0.001443 ** 
#countryFinland      -6.439e-01  1.637e-01  -3.935 0.000213 ***
#intI1_SUM            1.878e+00  2.570e-01   7.308 6.28e-10 ***
#SSU_counts           7.411e-06  2.196e-06   3.375 0.001278 ** 

# Generalized R^2:
(21.8860 - 7.9953) / 21.8860
# R^2 = 0.6346843

# Let's save also the simpliest model
M0 <- glm(ARG_SUM ~ country + SSU_counts, data = df, family= "Gamma"(link="log"))
summary(M0)

# Generalized R^2:
(21.886 - 14.580) / 21.886
# R^2 = 0.3338207

# MODEL VALIDATION

# Homogeneity
# Plot residuals vs fitted values
F1 <- fitted(M1)
E1 <- resid(M1, type = "pearson")
par(mfrow = c(1,1), cex.lab = 1.5, mar = c(5,5,2,2))
plot(x = F1, 
     y = E1,
     xlab = "Fitted values",
     ylab = "Pearson residuals")
abline(h = 0, lty = 2)
# No patterns, we are good.

# Influential observations
par(mfrow = c(1, 1))
plot(cooks.distance(M1), type = "h", ylim = c(0, 1))
abline(h = 1)
# There are no influental observations
#cooksd <- data.frame(cooks.distance(M1))
#rownames(cooksd) <- rownames(M1[["data"]])
#cooksd[cooksd$cooks.distance.M1. > 1,]

# Normality
par(cex.lab = 1.5, mar = c(5,5,2,2))
E1 <- resid(M1)
hist(E1, breaks = 15, xlab = "Residuals", main = "")

# Independence due to model misfit
df$E1 <- E1   #Put E1 inside the df
MySel <- c("SSU_counts", "intI1_SUM", "country", "M_Seqs_trimmed")
MyMultipanel.ggp2(Z = df, 
                  varx = MySel, 
                  vary = "E1", 
                  ylab = "Residuals",
                  addSmoother = TRUE,
                  addRegressionLine = FALSE,
                  addHorizontalLine = TRUE) 
# Some / No clear non-linear patterns in these graphs.

boxplot(E1 ~ country, 
        data = df,
        ylab = "Residuals")
abline(h = 0)
# Looks good.

# Check for spatial dependency
# Spatial patterns in the residuals?
MyCex <- 3 * abs(E1) / max(E1)
MyCol <- ifelse(E1 > 0, "red", "blue")

xyplot(long ~ lat, data = df, cex = MyCex, col = MyCol)

xyplot(long ~ lat, data = df_Benin, cex = MyCex, col = MyCol)

xyplot(long ~ lat, data = df_BF, cex = MyCex, col = MyCol)

xyplot(long ~ lat, data = df_Finland, cex = MyCex, col = MyCol)
# In general, some spatial dependency can be detected. 
# However, the number of samples in different locations is quite low.

# MODEL INTERPRETATION
M1 <- glm(ARG_SUM ~ country + SSU_counts + intI1_SUM,
           data = df, family="Gamma"(link="log"))
summary(M1)

M0 <- glm(ARG_SUM ~ country,
           data = df, family="Gamma"(link="log"))
summary(M0)

# Fit model
# Predictions
glht.M1 <- glht(M1, mcp(country = "Tukey"))
summary(glht(glht.M1))

pvalues <- tibble::tribble(
  ~group1, ~group2, ~p,
      "Burkina Faso",     "Finland", 0.07249,
      "Benin",     "Burkina Faso", 0.00214,
      "Benin",     "Finland", 0.001,
  )
pvalues

cols <- get_palette(c("#B2182B", "#44AA99", "#2166AC"), 3)

# Plot
dfA <- cbind(df, Mean = predict(M1, newdata = df, type = "response"), SE = predict(M1, 
    newdata = df, type = "response", se.fit = T)$se.fit)

resfinder_M1 <- ggplot(dfA, aes(x = country, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = country, y = ARG_SUM, color = country), size = 12, alpha = 1, width = 0.3) + 
#  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.5, lwd = 0.75) + geom_point(size = 0.9) + 
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 0, size = 32, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 36, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 40, family = "Times")) +
  labs(y = "ARGs normalized to 16S rRNA", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of ARGs (ResFinder)")
resfinder_M1 + stat_pvalue_manual(pvalues, label = "p", y.position = 4, step.increase = 0.05, tip.length = 0.01, size = 9)

#ggsave(filename = "resfinder_sum_M1_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# OR
MyData <- ddply(df, 
                .(country), 
                summarise,
                intI1_SUM = seq(min(intI1_SUM), 
                                    max(intI1_SUM), 
                                    length = 25),
                SSU_counts = seq(min(SSU_counts), 
                                   max(SSU_counts), 
                                   length = 25))
head(MyData)

P1 <- predict(M1, newdata = MyData, se = TRUE, type = "response")

MyData$Pred <- P1$fit                    #Predicted values
MyData$selo <- P1$fit - 1.96 * P1$se.fit #lower bound
MyData$seup <- P1$fit + 1.96 * P1$se.fit #lower bound
head(MyData)

# Plot
#cols <- get_palette(c("#B2182B", "#44AA99", "#2166AC"), 3)

#p <- ggplot()
#p <- p + geom_point(data = df, aes(y = ARG_SUM, x = intI1_SUM, col = country), shape = 16, size = 3)
#p <- p + xlab("intI1_SUM") + ylab("ARG_SUM")
#p <- p + theme(text = element_text(size=15)) 
#p <- p + geom_line(data = MyData, aes(x = intI1_SUM, y = Pred, col = country, group = country))
#p <- p + geom_ribbon(data = MyData, aes(x = intI1_SUM, ymax = seup, ymin = selo, group = country,col = country, fill = country),
#                     alpha = 0.5) + scale_color_manual(values=cols) + theme_minimal()
#ggsave(filename = "predict_M1_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Simple model
```{r}
cols <- get_palette(c("#B2182B", "#44AA99", "#2585E7"), 3)

glht.M0 <- glht(M0, mcp(country = "Tukey"))
summary(glht(glht.M0))

pvalues <- tibble::tribble(
  ~group1, ~group2, ~p,
    "Benin",     "Burkina Faso", 0.001,
    "Benin",     "Finland", 0.001,
      "Burkina Faso",     "Finland", 0.0105
  )
pvalues

dfA <- cbind(df, Mean = predict(M0, newdata = df, type = "response"), SE = predict(M0, 
    newdata = df, type = "response", se.fit = T)$se.fit)

resfinder_M0 <- ggplot(dfA, aes(x = country, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = country, y = ARG_SUM, color = country), size = 7.5, alpha = 1, width = 0.3) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.5, lwd = 0.75) + geom_point(size = 0.9) + 
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 0, size = 18, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 16, family = "Times"),
        axis.title.y = element_text(size = 16, family = "Times"),
        legend.position = "none",
        plot.title = element_text(size = 18, family = "Times", face = "bold")) +
  labs(y = "Normalized to 16S rRNA", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Relative sum abundance of ARGs")
ARG_sum <- resfinder_M0 + stat_pvalue_manual(pvalues, label = "p", y.position = 2.3, step.increase = 0.05, tip.length = 0.01, size = 5)
ARG_sum

ggsave(filename = "resfinder_sum_M0_new.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Hospitals / hospital sections
```{r}
# Benin
resfinder_PHY_stat_Ben <- subset_samples(resfinder_PHY_stat, country == "Benin")
df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_stat_Ben),
               hospital_section=as.factor(sample_data(resfinder_PHY_stat_Ben)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_stat_Ben)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_stat_Ben)$hospital))
              
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
#str(df)

# Simple model
M0 <- glm(ARG_SUM ~ hospital,
           data = df, family="Gamma"(link="log"))
summary(M0)

glht.M0 <- glht(M0, mcp(hospital = "Tukey"))
summary(glht(glht.M0))

# BF
resfinder_PHY_stat_BF <- subset_samples(resfinder_PHY_stat, country == "Burkina Faso")
df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_stat_BF),
               hospital_section=as.factor(sample_data(resfinder_PHY_stat_BF)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_stat_BF)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_stat_BF)$hospital))
              
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
#str(df)

# Simple model
M0 <- glm(ARG_SUM ~ hospital,
           data = df, family="Gamma"(link="log"))
summary(M0)

glht.M0 <- glht(M0, mcp(hospital = "Tukey"))
summary(glht(glht.M0))

# Finland
resfinder_PHY_stat_Fin <- subset_samples(resfinder_PHY_stat, country == "Finland")
df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_stat_Fin),
               hospital_section=as.factor(sample_data(resfinder_PHY_stat_Fin)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_stat_Fin)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_stat_Fin)$hospital))
              
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
#str(df)

# Simple model
M0 <- glm(ARG_SUM ~ hospital,
           data = df, family="Gamma"(link="log"))
summary(M0)

glht.M0 <- glht(M0, mcp(hospital = "Tukey"))
summary(glht(glht.M0))


# Hospital section
df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_stat),
               hospital_section=as.factor(sample_data(resfinder_PHY_stat)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_stat)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_stat)$hospital))
              
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
#str(df)

# Simple model
M0 <- glm(ARG_SUM ~ hospital_section,
           data = df, family="Gamma"(link="log"))
summary(M0)

glht.M0 <- glht(M0, mcp(hospital_section = "Tukey"))
summary(glht(glht.M0))
```
##### Modelling MGE abundance #####
```{r}
## Data exploration
df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_stat),
               intI1_SUM=sample_sums(MGE_PHY_int_stat),
               MGE_SUM=sample_sums(MGE_PHY_stat),
               hospital_section=as.factor(sample_data(resfinder_PHY_stat)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_stat)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_stat)$hospital),
               country=as.factor(sample_data(resfinder_PHY_stat)$country),
               R1_rpoB_counts=as.factor(sample_data(resfinder_PHY_stat)$R1_rpoB_counts),
               no_of_beds=as.factor(sample_data(resfinder_PHY_stat)$no_of_beds),
               long=as.factor(sample_data(resfinder_PHY_stat)$long),
               lat=as.factor(sample_data(resfinder_PHY_stat)$lat),
               A260_280=as.numeric(sample_data(resfinder_PHY_stat)$A260_280),
               DNA_ng_µl=as.numeric(sample_data(resfinder_PHY_stat)$DNA_ng_µl),
               M_Seqs_trimmed=as.numeric(sample_data(resfinder_PHY_stat)$M_Seqs_trimmed))
              
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
df$R1_rpoB_counts <- as.character(df$R1_rpoB_counts)
df$R1_rpoB_counts <- as.numeric(df$R1_rpoB_counts)
df$no_of_beds <- as.character(df$no_of_beds)
df$no_of_beds <- as.numeric(df$no_of_beds)

df$long <- as.numeric(gsub(",", ".", gsub("\\.", "", df$long)))
df$lat <- as.numeric(gsub(",", ".", gsub("\\.", "", df$lat)))                   
#str(df)

# Simple model
M0 <- glm(MGE_SUM ~ country,
           data = df, family="Gamma"(link="log"))
summary(M0)

glht.M0 <- glht(M0, mcp(country = "Tukey"))
summary(glht(glht.M0))

pvalues <- tibble::tribble(
  ~group1, ~group2, ~p,
    "Benin",     "Burkina Faso", 0.991,
    "Benin",     "Finland", 0.879,
      "Burkina Faso",     "Finland", 0.911
  )
pvalues

dfA <- cbind(df, Mean = predict(M0, newdata = df, type = "response"), SE = predict(M0, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#B2182B", "#44AA99", "#2585E7"), 3)

MGE_M0 <- ggplot(dfA, aes(x = country, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = country, y = MGE_SUM, color = country), size = 7.5, alpha = 1, width = 0.3) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.5, lwd = 0.75) + geom_point(size = 0.9) + 
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 0, size = 18, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 16, family = "Times"),
        axis.title.y = element_blank(),
        legend.position = "none",
        plot.title = element_text(size = 18, family = "Times", face = "bold")) +
  labs(y = "Normalized to 16S rRNA", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Relative sum abundance of MGEs")
MGE_sum <- MGE_M0 + stat_pvalue_manual(pvalues, label = "p", y.position = 7, step.increase = 0.05, tip.length = 0.01, size = 5)
MGE_sum

ggsave(filename = "MGE_sum_M0_new.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
##### Modelling intI1 abundance #####
```{r}
## Data exploration
df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_stat),
               intI1_SUM=sample_sums(MGE_PHY_int_stat),
               MGE_SUM=sample_sums(MGE_PHY_stat),
               hospital_section=as.factor(sample_data(resfinder_PHY_stat)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_stat)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_stat)$hospital),
               country=as.factor(sample_data(resfinder_PHY_stat)$country),
               R1_rpoB_counts=as.factor(sample_data(resfinder_PHY_stat)$R1_rpoB_counts),
               no_of_beds=as.factor(sample_data(resfinder_PHY_stat)$no_of_beds),
               long=as.factor(sample_data(resfinder_PHY_stat)$long),
               lat=as.factor(sample_data(resfinder_PHY_stat)$lat),
               A260_280=as.numeric(sample_data(resfinder_PHY_stat)$A260_280),
               DNA_ng_µl=as.numeric(sample_data(resfinder_PHY_stat)$DNA_ng_µl),
               M_Seqs_trimmed=as.numeric(sample_data(resfinder_PHY_stat)$M_Seqs_trimmed))
              
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
df$R1_rpoB_counts <- as.character(df$R1_rpoB_counts)
df$R1_rpoB_counts <- as.numeric(df$R1_rpoB_counts)
df$no_of_beds <- as.character(df$no_of_beds)
df$no_of_beds <- as.numeric(df$no_of_beds)

df$long <- as.numeric(gsub(",", ".", gsub("\\.", "", df$long)))
df$lat <- as.numeric(gsub(",", ".", gsub("\\.", "", df$lat)))                   
#str(df)

# Simple model
M0 <- glm(intI1_SUM ~ country,
           data = df, family="Gamma"(link="log"))
summary(M0)

glht.M0 <- glht(M0, mcp(country = "Tukey"))
summary(glht(glht.M0))

pvalues <- tibble::tribble(
  ~group1, ~group2, ~p,
    "Benin",     "Burkina Faso", 0.013,
    "Benin",     "Finland", 0.001,
      "Burkina Faso",     "Finland", 0.001
  )
pvalues

dfA <- cbind(df, Mean = predict(M0, newdata = df, type = "response"), SE = predict(M0, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#B2182B", "#44AA99", "#2585E7"), 3)

intI1_M0 <- ggplot(dfA, aes(x = country, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = country, y = intI1_SUM, color = country), size = 7.5, alpha = 1, width = 0.3) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.5, lwd = 0.75) + geom_point(size = 0.9) + 
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 0, size = 18, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 16, family = "Times"),
        axis.title.y = element_blank(),
        legend.position = "none",
        plot.title = element_text(size = 18, family = "Times", face = "bold")) +
  labs(y = "Normalized to 16S rRNA", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Relative sum abundance of intI1")
intI1_sum <- intI1_M0 + stat_pvalue_manual(pvalues, label = "p", y.position = 1.05, step.increase = 0.05, tip.length = 0.01, size = 5)
intI1_sum

ggsave(filename = "intI1_sum_M0_new.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
# Figures in grids
```{r}
design <-"
###
ABC
###
"
ARG_sum + MGE_sum + intI1_sum + plot_layout(design = design) + plot_annotation(tag_levels = c("A", "B", "C")) &
  theme(plot.tag = element_text(size = 24, family = "Times"))

ggsave(filename = "sums_grid.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
# Relative sums of ARGs with all samples based on category
```{r}
# Subset samples which belong to category among >2 other samples
resfinder_PHY_cat_stat <- subset_samples(resfinder_PHY, category != "WA street gutter" &
                                        category != "WA street gutter, sediment" &
                                        category != "WA soil inbetween tanks" &
                                        category != "WA empty hospital septic tank" &
                                        category != "WA treated, receiving municipality channel" &
                                        category != "WA treated, drinking water" &
                                        category != "WA treated, receiving river" &
                                        category != "WA treated, hand washing" &
                                        category != "WA in-patient feces")

MGE_PHY_int_cat_stat <- subset_samples(MGE_PHY_int, category != "WA street gutter" &
                                        category != "WA street gutter, sediment" &
                                        category != "WA soil inbetween tanks" &
                                        category != "WA empty hospital septic tank" &
                                        category != "WA treated, receiving municipality channel" &
                                        category != "WA treated, drinking water" &
                                        category != "WA treated, receiving river" &
                                        category != "WA treated, hand washing" &
                                        category != "WA in-patient feces")

df<-data.frame(ARG_SUM=sample_sums(resfinder_PHY_cat_stat),
               intI1_SUM=sample_sums(MGE_PHY_int_cat_stat),
               hospital_section=as.factor(sample_data(resfinder_PHY_cat_stat)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_cat_stat)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_cat_stat)$hospital),
               country=as.factor(sample_data(resfinder_PHY_cat_stat)$country),
               R1_rpoB_counts=as.factor(sample_data(resfinder_PHY_cat_stat)$R1_rpoB_counts),
               no_of_beds=as.factor(sample_data(resfinder_PHY_cat_stat)$no_of_beds),
               sample_material=as.factor(sample_data(resfinder_PHY_cat_stat)$sample_material))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
df$R1_rpoB_counts <- as.character(df$R1_rpoB_counts)
df$R1_rpoB_counts <- as.numeric(df$R1_rpoB_counts)
df$no_of_beds <- as.character(df$no_of_beds)
df$no_of_beds <- as.numeric(df$no_of_beds)
#str(df)

# Simple model
M0 <- glm(ARG_SUM ~ category,
           data = df, family="Gamma"(link="log"))
summary(M0)

# Predictions
glht.M0 <- glht(M0, mcp(category = "Tukey"))
summary(glht(glht.M0))

# Plot
dfA <- cbind(df, Mean = predict(M0, newdata = df, type = "response"), SE = predict(M0, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#B2182B", "#44AA99", "#2166AC"), 7)

cat_M0 <- ggplot(dfA, aes(x = category, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = category, y = ARG_SUM, color = category), size = 12, alpha = 1, width = 0.3) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.5, lwd = 0.75) + geom_point(size = 0.9) + 
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 45, size = 12, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 36, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 40, family = "Times")) +
  labs(y = "IntI1 normalized to 16S rRNA", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of ARGs (resfinder)")
cat_M0

# Reorder samples
sam_order <- c("WA hospital effluent", 
               "Eu other hospital effluent",
               "North Eu hospital effluent",
               "Eu other wwtp influent", 
               "North Eu wwtp influent", 
               "WA river, drinking water", 
               "North Eu treated")
cat_M0$data$category <- factor(cat_M0$data$category, levels = sam_order)
cat_M0

#ggsave(filename = "resfinder_sum_categories_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Diversity
# ResFinder
```{r}
# Create phyloseq object with the count data
OTU_resfinder_counts <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

match <- match(rownames(metadata), colnames(OTU_resfinder_counts))
OTU_resfinder_counts <- OTU_resfinder_counts[,match]
all(colnames(OTU_resfinder_counts) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder_counts), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder_counts) == clusters_tax_table_resfinder$Gene)

# Reload to get new row numbers
write.table(OTU_resfinder_counts, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_counts.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_resfinder_counts <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_counts.txt", row.names=NULL)
OTU_resfinder_counts$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

resfinder_PHY_counts <- phyloseq(otu_table(OTU_resfinder_counts, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY_counts = subset_samples(resfinder_PHY_counts, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
resfinder_PHY_counts <- subset_samples(resfinder_PHY_counts, country != "UK")
## Exclude biological / technical replicates
resfinder_PHY_counts_stat <- subset_samples(resfinder_PHY_counts, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")
# Create phyloseq object with only hospital WW samples sequenced here
resfinder_PHY_counts_stat <- subset_samples(resfinder_PHY_counts_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")

PHY <- prune_taxa(taxa_sums(resfinder_PHY_counts_stat) > 0, resfinder_PHY_counts_stat)

# Rarefy
#  90% of the minimum sample depth
rarecurve(t(otu_table(PHY)), step=50, cex=0.5)
PHY_rarefied = rarefy_even_depth(PHY, rngseed=1)
#567OTUs were removed because they are no longer 
#present in any sample after random subsampling
rarecurve(t(otu_table(PHY_rarefied)), step=50, cex=0.5)

tab <- microbiome::alpha(PHY_rarefied, index = "all")
#kable(head(tab))

PHY.meta <- meta(PHY_rarefied)
#kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$country)
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- get_palette(c("#B2182B", "#44AA99", "#2166AC"), 3)

shannon <- ggboxplot(PHY.meta, x = "country", y = "Shannon", fill = "country", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.75, size = 36, hjust = 0.75, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 36, family = "Times"),
        legend.position = "none",
        legend.title = element_blank(),
        axis.title.y = element_text(size = 36, family = "Times"),
        plot.title = element_text(size = 40, family = "Times", face = "bold")) +
  labs(title = "Alpha diversity, Shannon index (ResFinder)")

a <- shannon + stat_compare_means(comparisons = vars.pairs, 
                             label = "p", 
                             hide.ns = F, ref.group = ".all.",
                             size = 10, fontfamily = "Times")
a
#ggsave(filename = "resfinder_diversities_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Diversity
# Metaphlan3
```{r}
# transform into counts
PHY <- transform_sample_counts(metaphlan_PHY_stat, function(x) 1E6 * x/sum(x))
otu_table(PHY) <- round(otu_table(PHY), digits = 0)

PHY <- prune_taxa(taxa_sums(PHY) > 0, PHY)

# Sequences should be rarefied prior running Metaphlan3!

tab <- microbiome::alpha(PHY, index = "all")
#kable(head(tab))

PHY.meta <- meta(PHY)
#kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$country)
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- get_palette(c("#B2182B", "#44AA99", "#2166AC"), 3)

shannon <- ggboxplot(PHY.meta, x = "country", y = "Shannon", fill = "country", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.75, size = 36, hjust = 0.75, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 36, family = "Times"),
        legend.position = "none",
        legend.title = element_blank(),
        axis.title.y = element_text(size = 36, family = "Times"),
        plot.title = element_text(size = 40, family = "Times", face = "bold")) +
  labs(title = "Alpha diversity, Shannon index (Metaphlan3)")

a <- shannon + stat_compare_means(comparisons = vars.pairs, 
                             label = "p", 
                             hide.ns = F, ref.group = ".all.",
                             size = 10, fontfamily = "Times")
a
#ggsave(filename = "metaphlan_diversities_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Diversity
# Mobilome (MGE database)
```{r}
# Create phyloseq object with the count data
OTU_MGE_counts <-as.matrix(read.table("cp_MGE_genemat.txt", header= T, check.names = F, row.names = 1))

match <- match(rownames(metadata), colnames(OTU_MGE_counts))
OTU_MGE_counts <- OTU_MGE_counts[,match]
all(colnames(OTU_MGE_counts) == rownames(metadata))

# Tax table
MGE_tax_table_trim <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", header=FALSE)
colnames(MGE_tax_table_trim) <- c("Gene", "Element", "Class")

# Reorder tax_table to match OTU_MGE
match <- match(rownames(OTU_MGE_counts), MGE_tax_table_trim$Gene)
MGE_tax_table_trim <- MGE_tax_table_trim[match,]
all(rownames(OTU_MGE_counts) == MGE_tax_table_trim$Gene)

# Reload to get new row numbers
write.table(OTU_MGE_counts, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_MGE_counts", 
            row.names=T, sep = "\t", col.names = T)
OTU_MGE_counts <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_MGE_counts", row.names=NULL)
OTU_MGE_counts$row.names<-NULL

# Reload to get new row numbers
write.table(MGE_tax_table_trim, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", row.names=F, sep = "\t", col.names = T)
MGE_tax_table_trim <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", row.names=NULL)

MGE_PHY_counts <- phyloseq(otu_table(OTU_MGE_counts, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(MGE_tax_table_trim)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
MGE_PHY_counts = subset_samples(MGE_PHY_counts, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
MGE_PHY_counts <- subset_samples(MGE_PHY_counts, country != "UK")
## Exclude biological / technical replicates
MGE_PHY_counts_stat <- subset_samples(MGE_PHY_counts, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")
# Create phyloseq object with only hospital WW samples sequenced here
MGE_PHY_counts_stat <- subset_samples(MGE_PHY_counts_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")

PHY <- prune_taxa(taxa_sums(MGE_PHY_counts_stat) > 0, MGE_PHY_counts_stat)

# Rarefy
#  90% of the minimum sample depth
rarecurve(t(otu_table(PHY)), step=50, cex=0.5)
PHY_rarefied = rarefy_even_depth(PHY, rngseed=1)
# 489OTUs were removed because they are no longer 
# present in any sample after random subsampling

tab <- microbiome::alpha(PHY_rarefied, index = "all")
#kable(head(tab))

PHY.meta <- meta(PHY_rarefied)
#kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$country)
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- get_palette(c("#B2182B", "#44AA99", "#2166AC"), 3)

#"#81A88D", "#B2182B", "#85D4E3", "#F4A582", "#4393C3", "#2166AC", "#D6604D"
shannon <- ggboxplot(PHY.meta, x = "country", y = "Shannon", fill = "country", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.75, size = 36, hjust = 0.75, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 36, family = "Times"),
        legend.position = "none",
        legend.title = element_blank(),
        axis.title.y = element_text(size = 36, family = "Times"),
        plot.title = element_text(size = 40, family = "Times", face = "bold")) +
  labs(title = "Alpha diversity, Shannon index (MGE)")

a <- shannon + stat_compare_means(comparisons = vars.pairs, 
                             label = "p", 
                             hide.ns = F, ref.group = ".all.",
                             size = 10, fontfamily = "Times")
a
#ggsave(filename = "mge_diversities_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Ordination, ARGs (ResFinder)
```{r}
resfinder_PHY_ord <- ordinate(resfinder_PHY_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_stat, resfinder_PHY_ord, color = "country")
resfinder.p_ord <- p_ord + 
 # scale_color_manual(values = c("#B2182B", "#44AA99", "#2166AC")) +
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 3.5) + 
  stat_ellipse(level = 0.90, linetype = 1) +
#    geom_text_repel(mapping = aes(label = hospital), size = 7, family = "Times", hjust = 1.2) +
    theme_minimal() + labs(title= "Resistome", subtitle = "Hospital WWs in Benin, Burkina Faso and Finland") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 20, family = "Times"),
          legend.text = element_text(size = 36, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 36, family = "Times"),
          axis.text = element_text(size = 18, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
resfinder.p_ord

# Save
#ggsave(filename = "ord_resfinder_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Test significance using pair-wise adonis
#resfinder_temp <- subset_samples(resfinder_PHY_stat_equal, (country == "Benin" | country == "Finland"))
#resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
#adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

#resfinder_temp <- subset_samples(resfinder_PHY_stat_equal, (country == "Benin" | country == "Burkina Faso"))
#resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
#adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

#resfinder_temp <- subset_samples(resfinder_PHY_stat_equal, (country == "Burkina Faso" | country == "Finland"))
#resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
#adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))
```
## Ordination, taxa (Metaphlan3)
```{r}
PHY = transform_sample_counts(metaphlan_PHY_stat, function(x) 1E6 * x/sum(x))

metaphlan_PHY_ord <- ordinate(PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(PHY, metaphlan_PHY_ord, color = "country")
metaphlan.p_ord <- p_ord + 
  #scale_color_manual(values = c("#B2182B", "#44AA99", "#2166AC")) +
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 3.5) + 
  stat_ellipse(level = 0.90, linetype = 1) +
#    geom_text_repel(mapping = aes(label = hospital), size = 7, family = "Times", hjust = 1.2) +
    theme_minimal() + labs(title= "Taxonomical composition", subtitle = "Hospital WWs in Benin, Burkina Faso and Finland") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 20, family = "Times"),
          legend.text = element_text(size = 50, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 36, family = "Times"),
          axis.text = element_text(size = 18, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaphlan.p_ord

# Save
#ggsave(filename = "ord_metaphlan_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Test significance using pair-wise adonis
#metaphlan_temp <- subset_samples(metaphlan_PHY_stat_equal, (country == "Benin" | country == "Finland"))
#metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
#adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

#metaphlan_temp <- subset_samples(metaphlan_PHY_stat_equal, (country == "Benin" | country == "Burkina Faso"))
#metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
#adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

#metaphlan_temp <- subset_samples(metaphlan_PHY_stat_equal, (country == "Burkina Faso" | country == "Finland"))
#metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
#adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))
```
## Ordination, MGEs (Mobilome)
```{r}
MGE_PHY_ord <- ordinate(MGE_PHY_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_stat, MGE_PHY_ord, color = "country")
MGE.p_ord <- p_ord + 
  #scale_color_manual(values = c("#B2182B", "#44AA99", "#2166AC")) +
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 3.5) + 
  stat_ellipse(level = 0.90, linetype = 1) +
  #geom_text_repel(mapping = aes(label = hospital), size = 7, family = "Times", hjust = 1.2) +
   theme_minimal() + labs(title= "Mobilome", subtitle = "Hospital WWs in Benin, Burkina Faso and Finland") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 20, family = "Times"),
          legend.text = element_text(size = 50, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 36, family = "Times"),
          axis.text = element_text(size = 18, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

# Save
#ggsave(filename = "ord_mge_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Test significance using pair-wise adonis
#MGE_temp <- subset_samples(MGE_PHY_stat_equal, (country == "Benin" | country == "Finland"))
#MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
#adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))

#MGE_temp <- subset_samples(MGE_PHY_stat_equal, (country == "Benin" | country == "Burkina Faso"))
#MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
#adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))

#MGE_temp <- subset_samples(MGE_PHY_stat_equal, (country == "Burkina Faso" | country == "Finland"))
#MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
#adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))
```
# Figure panel for ordination plots, with text labels
```{r}
resfinder_PHY_ord <- ordinate(resfinder_PHY_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_stat, resfinder_PHY_ord, color = "country")
arg <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 2) +
  stat_ellipse(level = 0.90, linetype = 1) +
    geom_text_repel(mapping = aes(label = hospital), size = 4, family = "Times", hjust = 1.2, hjust = 0.5) +
    theme_minimal() + labs(title = "Resistome") +
    theme(plot.title = element_text(size = 20, family = "Times", face = "bold"),
    #      legend.text = element_text(size = 18, family = "Times"),
    #      legend.title = element_blank(),
          legend.position = "none",
          axis.title = element_text(size = 18, family = "Times"),
          axis.text = element_text(size = 18, family = "Times")) +
  theme(plot.margin = unit(c(0.1, 0.1, 0.1, 1), "cm")) +
    coord_fixed() +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
arg

#leg_ord <- get_legend(arg)

# Convert to a ggplot and print
#as_ggplot(leg_ord)

PHY = transform_sample_counts(metaphlan_PHY_stat, function(x) 1E6 * x/sum(x))

metaphlan_PHY_ord <- ordinate(PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(PHY, metaphlan_PHY_ord, color = "country")
mp <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 2) +
  stat_ellipse(level = 0.90, linetype = 1) +
    geom_text_repel(mapping = aes(label = hospital), size = 4, family = "Times", hjust = 1.2) +
    theme_minimal() + labs(title= "Taxonomical composition") + 
    theme(plot.title = element_text(size = 20, family = "Times", face = "bold", hjust = 0.5),
          legend.position = "none",
          axis.title = element_text(size = 18, family = "Times"),
          axis.text = element_text(size = 18, family = "Times")) +
  theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
    coord_fixed() +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
mp

MGE_PHY_ord <- ordinate(MGE_PHY_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_stat, MGE_PHY_ord, color = "country")
mge <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 2) +
  stat_ellipse(level = 0.90, linetype = 1) +
  geom_text_repel(mapping = aes(label = hospital), size = 4, family = "Times", hjust = 1.2) +
   theme_minimal() + labs(title= "Mobilome") +
    theme(plot.title = element_text(size = 20, family = "Times", face = "bold", hjust = 0.5),
          legend.position = "none",
          axis.title = element_text(size = 18, family = "Times"),
          axis.text = element_text(size = 18, family = "Times")) +
  theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
  coord_fixed() +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
mge

p <- arg + mp + mge +
  plot_layout(nrow = 2) + plot_annotation(tag_levels = list(c("A", "B", "C"))) &
  theme(plot.tag = element_text(size = 24, family = "Times"), plot.tag.position = c(0, 1))

p_leg <- p + inset_element(leg_ord, left = 1, bottom = 1, right = 1.7, top = 0)

# Save
ggsave(filename = "ord_patch_supp.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
# Figure panel for ordination plots, without text labels
```{r}
resfinder_PHY_ord <- ordinate(resfinder_PHY_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_stat, resfinder_PHY_ord, color = "country")
arg <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 5) +
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title = "Resistome") +
    theme(plot.title = element_text(size = 26, family = "Times", face = "bold"),
          legend.position = "bottom",
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 18, family = "Times"),
          axis.text = element_text(size = 18, family = "Times")) +
  theme(plot.margin = unit(c(0.1, 0.1, 0.1, 1), "cm")) +
    coord_fixed() +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
arg

PHY = transform_sample_counts(metaphlan_PHY_stat, function(x) 1E6 * x/sum(x))

metaphlan_PHY_ord <- ordinate(PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(PHY, metaphlan_PHY_ord, color = "country")
mp <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 3.5) +
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Taxonomical composition") + 
    theme(plot.title = element_text(size = 20, family = "Times", face = "bold", hjust = 0.5),
          legend.position = "none",
          axis.title = element_text(size = 12, family = "Times"),
          axis.text = element_text(size = 12, family = "Times")) +
  theme(plot.margin = unit(c(0.1, 0.1, 0.1, 1), "cm")) +
    coord_fixed() +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=4)))
mp

MGE_PHY_ord <- ordinate(MGE_PHY_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_stat, MGE_PHY_ord, color = "country")
mge <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 3.5) +
  stat_ellipse(level = 0.90, linetype = 1) +
   theme_minimal() + labs(title= "Mobilome") +
    theme(plot.title = element_text(size = 20, family = "Times", face = "bold", hjust = 0.5),
          legend.position = "none",
          axis.title = element_text(size = 12, family = "Times"),
          axis.text = element_text(size = 12, family = "Times")) +
  theme(plot.margin = unit(c(0.1, 0.1, 0.1, 1), "cm")) +
  coord_fixed() +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=4)))
mge

layout <- "
AAAABBB
AAAACCC
"
arg + mp + mge +
  plot_layout(design = layout) + plot_annotation(tag_levels = list(c("A", "B", "C"))) &
  theme(plot.tag = element_text(size = 24, family = "Times"), plot.tag.position = c(0, 1))

# Save
ggsave(filename = "ord_patch_new.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Ordination, intI1 (Mobilome)
```{r}
intI1_PHY_ord <- ordinate(MGE_PHY_int_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_int_stat, intI1_PHY_ord, color = "country")
intI1.p_ord <- p_ord + 
  #scale_color_manual(values = c("#B2182B", "#44AA99", "#2166AC")) +
  scale_color_manual(values = c("#B2182B", "#72D39D", "#2585E7")) +
  geom_point(size = 5.5) + 
  stat_ellipse(level = 0.90, linetype = 1) +
    geom_text_repel(mapping = aes(label = hospital), size = 3, family = "Times", hjust = 1.2) +
    theme_minimal() + labs(title= "Mobilome, IntI1 (MGE Database)", subtitle = "Hospital WWs in Benin, Burkina Faso and Finland") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
intI1.p_ord

# Save
#ggsave(filename = "ord_intI1_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Test significance using pair-wise adonis
#MGE_temp <- subset_samples(MGE_PHY_int_stat_equal, (country == "Benin" | country == "Finland"))
#MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
#adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))

#MGE_temp <- subset_samples(MGE_PHY_int_stat_equal, (country == "Benin" | country == "Burkina Faso"))
#MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
#adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))

#MGE_temp <- subset_samples(MGE_PHY_int_stat_equal, (country == "Burkina Faso" | country == "Finland"))
#MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
#adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))
```
## Ordination, VFDBs (Virulence genes)
```{r}
VFDB_PHY_ord <- ordinate(VFDB_PHY_stat, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(VFDB_PHY_stat, VFDB_PHY_ord, color = "country")
VFDB.p_ord <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#44AA99", "#2166AC")) +
  geom_point(size = 5.5) + 
  stat_ellipse(level = 0.90, linetype = 1) +
 #   geom_text_repel(mapping = aes(label = hospital), size = 3, family = "Times", hjust = 1.2) +
    theme_minimal() + labs(title= "Virulence Genes (VFDB)", subtitle = "Hospital WWs in Benin (25), BF (34) and Finland (8)") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
VFDB.p_ord

# Save
#ggsave(filename = "ord_vfdb_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Test significance using pair-wise adonis
#VFDB_temp <- subset_samples(VFDB_PHY_stat_equal, (country == "Benin" | country == "Finland"))
#VFDB_dist <- vegdist(t(otu_table(VFDB_temp)), dist = "horn")
#adonis(VFDB_dist ~ country, data = data.frame(sample_data(VFDB_temp), permutations = 9999))

#VFDB_temp <- subset_samples(VFDB_PHY_stat_equal, (country == "Benin" | country == "Burkina Faso"))
#VFDB_dist <- vegdist(t(otu_table(VFDB_temp)), dist = "horn")
#adonis(VFDB_dist ~ country, data = data.frame(sample_data(VFDB_temp), permutations = 9999))

#VFDB_temp <- subset_samples(VFDB_PHY_stat_equal, (country == "Burkina Faso" | country == "Finland"))
#VFDB_dist <- vegdist(t(otu_table(VFDB_temp)), dist = "horn")
#adonis(VFDB_dist ~ country, data = data.frame(sample_data(VFDB_temp), permutations = 9999))
```
## Compare hospitals within each country
# Ordination, ARGs (ResFinder)
```{r}
resfinder_PHY_WA_WW_stat_Ben <- subset_samples(resfinder_PHY_WA_WW_stat, country == "Benin")

cols <- get_palette(c("#332288", "#44AA99", "#88CCEE", "#B94ED9", "#F22D3D"), 5)

resfinder_PHY_ord <- ordinate(resfinder_PHY_WA_WW_stat_Ben, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_WA_WW_stat_Ben, resfinder_PHY_ord, color = "hospital")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  geom_point(size = 7) + 
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Resistome (ResFinder)", subtitle = "Hospital WWs in Benin") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
resfinder.p_ord

# Save
#ggsave(filename = "ord_res_Ben_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

resfinder_PHY_WA_WW_stat_BF <- subset_samples(resfinder_PHY_WA_WW_stat, country == "Burkina Faso")

resfinder_PHY_ord <- ordinate(resfinder_PHY_WA_WW_stat_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_WA_WW_stat_BF, resfinder_PHY_ord, color = "hospital")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  scale_shape_manual(values=c(18, 16)) +
  geom_point(size = 7) + 
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Resistome (ResFinder)", subtitle = "Hospital WWs in Burkina Faso") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
resfinder.p_ord

# Save
#ggsave(filename = "ord_res_BF_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

resfinder_PHY_stat_Fin <- subset_samples(resfinder_PHY_stat, category == "North Eu hospital effluent")

resfinder_PHY_ord <- ordinate(resfinder_PHY_stat_Fin, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_stat_Fin, resfinder_PHY_ord, color = "hospital")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  scale_shape_manual(values=c(18, 16)) +
  geom_point(size = 7) + 
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Resistome (ResFinder)", subtitle = "Hospital WWs in Finland") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
resfinder.p_ord

# Save
#ggsave(filename = "ord_res_Fin_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## WA WW samples
## Ordination, taxa (Metaphlan3)
```{r}
metaphlan_PHY_WA_WW_stat_Ben <- subset_samples(metaphlan_PHY_WA_WW_stat, country == "Benin")

metaphlan_PHY_ord <- ordinate(metaphlan_PHY_WA_WW_stat_Ben, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY_WA_WW_stat_Ben, metaphlan_PHY_ord, color = "hospital")
metaphlan.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  geom_point(size = 7) + 
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Taxonomy (Metaphlan3)", subtitle = "Hospital WWs in Benin") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaphlan.p_ord

#ggsave(filename = "ord_Ben_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

metaphlan_PHY_WA_WW_stat_BF <- subset_samples(metaphlan_PHY_WA_WW_stat, country == "Burkina Faso")

metaphlan_PHY_ord <- ordinate(metaphlan_PHY_WA_WW_stat_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY_WA_WW_stat_BF, metaphlan_PHY_ord, color = "hospital")
metaphlan.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  geom_point(size = 7) + 
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Taxonomy (Metaphlan3)", subtitle = "Hospital WWs in BF") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaphlan.p_ord

#ggsave(filename = "ord_BF_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

metaphlan_PHY_WA_WW_stat_Fin <- subset_samples(metaphlan_PHY, category == "North Eu hospital effluent")

metaphlan_PHY_ord <- ordinate(metaphlan_PHY_WA_WW_stat_Fin, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY_WA_WW_stat_Fin, metaphlan_PHY_ord, color = "hospital")
metaphlan.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  geom_point(size = 7) + 
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Taxonomy (Metaphlan3)", subtitle = "Hospital WWs in Finland") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaphlan.p_ord

#ggsave(filename = "ord_Fin_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## WA WW samples
## Ordination, mobilome (MGE database)
```{r}
MGE_PHY_WA_WW_stat_Ben <- subset_samples(MGE_PHY_WA_WW_stat, country == "Benin")

MGE_PHY_ord <- ordinate(MGE_PHY_WA_WW_stat_Ben, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_WA_WW_stat_Ben, MGE_PHY_ord, color = "hospital")
MGE.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  geom_point(size = 7) + 
   geom_text(mapping = aes(label = hospital), size = 3, hjust = 1.5, vjust = -0.95) +
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Mobilome (MGE database)", subtitle = "Hospital WWs in Benin") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

#ggsave(filename = "ord_mge_Ben_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

MGE_PHY_WA_WW_stat_BF <- subset_samples(MGE_PHY_WA_WW_stat, country == "Burkina Faso")

MGE_PHY_ord <- ordinate(MGE_PHY_WA_WW_stat_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_WA_WW_stat_BF, MGE_PHY_ord, color = "hospital")
MGE.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  geom_point(size = 7) + 
   geom_text(mapping = aes(label = hospital), size = 3, hjust = 1.5, vjust = -0.95) +
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Mobilome (MGE database)", subtitle = "Hospital WWs in BF") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

#ggsave(filename = "ord_mge_BF_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

MGE_PHY_WA_WW_stat_Fin <- subset_samples(MGE_PHY, category == "North Eu hospital effluent")

MGE_PHY_ord <- ordinate(MGE_PHY_WA_WW_stat_Fin, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_WA_WW_stat_Fin, MGE_PHY_ord, color = "hospital")
MGE.p_ord <- p_ord + 
  scale_color_manual(values=cols) + 
  geom_point(size = 7) + 
   geom_text(mapping = aes(label = hospital), size = 3, hjust = 1.5, vjust = -0.95) +
  stat_ellipse(level = 0.90, linetype = 1) +
    theme_minimal() + labs(title= "Mobilome (MGE database)", subtitle = "Hospital WWs in Finland") + 
    theme(plot.title = element_text(size = 36, family = "Times", face = "bold"),
          plot.subtitle = element_text(size = 28, family = "Times"),
          legend.text = element_text(size = 28, family = "Times"),
          legend.title = element_blank(),
          axis.title = element_text(size = 28, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

#ggsave(filename = "ord_mge_Fin_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## CrAssphage
```{r}
# Plot orrelation of crAssphage & ARGs
ARG_relative_sum <- data.frame(sample_sums(resfinder_PHY_stat))
crAssphage_relative_sum <- data.frame(sample_sums(crass_PHY_stat))
all(rownames(ARG_relative_sum) == rownames(crAssphage_relative_sum))

# Join data
crass_res <- cbind(ARG_relative_sum, crAssphage_relative_sum)
colnames(crass_res) <- c("ARGs", "crAssphage")

# Plot
cor <- ggplot(crass_res, aes(x=ARGs, y=crAssphage)) +
  geom_point(size=7, shape=19, color = "#3110D2") +
  geom_smooth(method="lm", se=TRUE, fullrange=FALSE, level=0.95, color = "#FB2A38", fill = "#8A91F8") +
    theme_bw() +
  theme(axis.title = element_text(size = 30, family = "Times"),
        axis.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
 xlab("ARGs") + ylab("crAssphage") +
  ggtitle("Correlation of relative sums of ARGs and crAssphage")
cor2 <- cor + stat_cor(method = "pearson", label.x = 0.1, label.y = 0)
cor2
```
## Correlation between SSU & rpoB counts
```{r}
## Use either *_PHY / *_PHY_stat_equal / *_PHY_WA_WW_stat
SSU_counts <- data.frame(sample_data(resfinder_PHY_stat)$SSU_counts)
R1_rpoB_counts <- data.frame(sample_data(resfinder_PHY_stat)$R1_rpoB_counts)
bacterial_counts <- cbind(SSU_counts, R1_rpoB_counts)
colnames(bacterial_counts) <- c("SSU_counts", "R1_rpoB_counts")

p <- ggplot(bacterial_counts, aes(x=SSU_counts, y=R1_rpoB_counts)) +
  geom_point(size=7, shape=19, color = "#3110D2") +
  geom_smooth(method="lm", se=TRUE, fullrange=FALSE, level=0.95, color = "#FB2A38", fill = "#8A91F8") +
    theme_bw() +
  theme(axis.title = element_text(size = 30, family = "Times"),
        axis.text = element_text(size = 32, family = "Times"),
        plot.title = element_text(size = 36, family = "Times"),
        plot.subtitle = element_text(size = 28, family = "Times")) +
 xlab("16s rRNA counts") + ylab("R1 rpoB counts") +
  labs(title= "Correlation of 16s rRNA and rpoB counts", subtitle = "Hospital WWs in Benin (25), BF (34) and Finland (8)")
cor <- p + stat_cor(method = "pearson", label.x = 100000, label.y = 1000, )
cor

correl<-corr.test(SSU_counts, R1_rpoB_counts, use="pairwise", method="pearson",
                            adjust="fdr",alpha=.05,ci=TRUE)
r <- data.frame(correl$r)
p <- data.frame(correl$p)
p.ad <- data.frame(correl$p.adj)

#ggsave(filename = "SSU_rpoB_cor_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Correlation of Intl1 & all ARGs
```{r}
ARG_relative_sum <- data.frame(sample_sums(resfinder_PHY_stat))
MGE_relative_sum <- data.frame(sample_sums(MGE_PHY_stat))
all(rownames(ARG_relative_sum) == rownames(MGE_relative_sum))

# Join data
mge_res <- cbind(ARG_relative_sum, MGE_relative_sum)
colnames(mge_res) <- c("ARGs", "MGEs")

# Plot
cor <- ggplot(mge_res, aes(x=ARGs, y=MGEs)) +
  geom_point(size=7, shape=19, color = "#3110D2") +
  geom_smooth(method="lm", se=TRUE, fullrange=FALSE, level=0.95, color = "#FB2A38", fill = "#8A91F8") +
    theme_bw() +
  theme(axis.title = element_text(size = 30, family = "Times"),
        axis.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times"),
        plot.subtitle = element_text(size = 28, family = "Times")) +
 xlab("ARG") + ylab("MGEs") +
  labs(title= "Correlation of relative sums of ARGs and MGEs", subtitle = "Hospital WWs in Benin (25), BF (34) and Finland (8)")
cor2 <- cor + stat_cor(method = "pearson", label.x = 2, label.y = 1.5)
cor2

#ggsave(filename = "ARG_MGE_cor_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

## Intl1
ARG_relative_sum <- data.frame(sample_sums(resfinder_PHY_stat))
intI1_relative_sum <- data.frame(sample_sums(MGE_PHY_int_stat))
all(rownames(ARG_relative_sum) == rownames(intI1_relative_sum))

# Join data
intl_res <- cbind(ARG_relative_sum, intI1_relative_sum)
colnames(intl_res) <- c("ARGs", "intI1")

# Plot
cor <- ggplot(intl_res, aes(x=ARGs, y=intI1)) +
  geom_point(size=7, shape=19, color = "#3110D2") +
  geom_smooth(method="lm", se=TRUE, fullrange=FALSE, level=0.95, color = "#FB2A38", fill = "#8A91F8") +
    theme_bw() +
  theme(axis.title = element_text(size = 30, family = "Times"),
        axis.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times"),
        plot.subtitle = element_text(size = 28, family = "Times")) +
 xlab("ARG") + ylab("intI1") +
  labs(title= "Correlation of relative sums of ARGs and Int1", subtitle = "Hospital WWs in Benin, Burkina Faso and Finland")
cor2 <- cor + stat_cor(method = "pearson", label.x = 1, label.y = 1.5)
cor2

correl<-corr.test(ARG_relative_sum, intI1_relative_sum, use="pairwise", method="pearson",
                            adjust="fdr",alpha=.05,ci=TRUE)

r <- data.frame(correl$r)
p <- data.frame(correl$p)
p.ad <- data.frame(correl$p.adj)

# Correlation coefficients for each country:
# Benin = 0.79, p = 2.5 x 10-6; BF=R2=0.54, p = 0.00086; Finland = 0.83, p = 0.01 

#ggsave(filename = "ARG_intl1_cor_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## DESeq2
# ResFinder
# Fin-Ben
```{r}
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Divide by ARG gene lengths
## Get the lengths in terminal
# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
deseq_OTU_resfinder <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(deseq_OTU_resfinder))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], deseq_OTU_resfinder[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Deseq
deseq_OTU <- deseq_OTU_resfinder[, ] * 10^5 + 1

# Reload to get new row numbers
write.table(deseq_OTU, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU.txt", 
            row.names=T, sep = "\t", col.names = T)
deseq_OTU <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU.txt", row.names=NULL)
deseq_OTU$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

resfinder_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_deseq = subset_samples(resfinder_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
resfinder_deseq <- subset_samples(resfinder_deseq, country != "UK")

## Exclude biological / technical replicates
resfinder_deseq <- subset_samples(resfinder_deseq, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
resfinder_deseq_stat <- subset_samples(resfinder_deseq, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Take pair wise comparisons
deseq_PHY = subset_samples(resfinder_deseq_stat, country == "Benin" | country == "Finland")

hist(log10(apply(otu_table(deseq_PHY), 1, var)), xlab = "log10(variance)")

# Let's set a threashold for the variance
varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_resfinder = res[which(res$padj < alpha), ]
sigtab_resfinder = cbind(as(sigtab_resfinder, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_resfinder), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_resfinder = merge(sigtab_resfinder, as.data.frame(n), by = 0)
#kable(sigtab_resfinder, caption = "Taxonomy")

# Plot
# Class
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Class = factor(as.character(sigtab_resfinder$Class), levels = names(x))

# Gene cluster
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Cluster_name, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Cluster_name = factor(as.character(sigtab_resfinder$Cluster_name), levels = names(x))

# Gene 
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Gene, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Gene = factor(as.character(sigtab_resfinder$Gene), levels = names(x))

sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]
#write.table(sorted_sigtab, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/DESeq2_Ben_Fin.txt", 
#            row.names=T, sep = "\t", col.names = T)

Fin_Ben <- subset(sorted_sigtab,log2FoldChange>=0)
Ben_Fin <- subset(sorted_sigtab,log2FoldChange<=0)
Ben_Fin <- Ben_Fin[order(Ben_Fin$log2FoldChange), ]
head(Fin_Ben)
head(Ben_Fin)
```
# Plot
# Benin-Finland
```{r}
# Take top 25 from each and combine back to one table
#Fin25 <- Fin_Ben[1:25,]
#Ben25 <- Ben_Fin[1:25,]
#Ben_Fin_top_sigtab <- rbind(Fin25, Ben25)

# Shorten gene names and order again
#Ben_Fin_top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", Ben_Fin_top_sigtab$Gene)

#cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 13)
#p <- ggplot(data = Ben_Fin_top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
#  geom_bar(stat = "identity",
#  aes(fill = Class), position = position_dodge(preserve = "single")) + 
#  ylab("Log2 Fold Change\nBenin                     Finland") + 
#  scale_fill_manual(values = cols) +
#  ggtitle("Enriched ARGs in Beninise / Finnish samples (25 with highest log 2 fold change values of each country)") +
#  coord_flip() +
#  theme_minimal() +
#  theme(axis.text.x = element_text(family = "Times", size = 10),
#        axis.text.y = element_text(family = "Times", size = 12, face = "bold.italic"),
#        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
#        axis.title.y = element_blank(),
#        legend.text = element_text(family = "Times", size = 20),
#        legend.title = element_blank(),
#        plot.title = element_text(family = "Times", size = 20, face = "bold"))

#ggsave(filename = "BEN_FIN_deseq50_resfinder_genes_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## DESeq2
# ResFinder
# BF-Fin
```{r}
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Divide by ARG gene lengths
## Get the lengths in terminal
# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
deseq_OTU_resfinder <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(deseq_OTU_resfinder))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], deseq_OTU_resfinder[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Deseq
deseq_OTU <- deseq_OTU_resfinder[, ] * 10^5 + 1

# Reload to get new row numbers
write.table(deseq_OTU, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU.txt", 
            row.names=T, sep = "\t", col.names = T)
deseq_OTU <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU.txt", row.names=NULL)
deseq_OTU$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

resfinder_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_deseq = subset_samples(resfinder_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
resfinder_deseq <- subset_samples(resfinder_deseq, country != "UK")

## Exclude biological / technical replicates
resfinder_deseq <- subset_samples(resfinder_deseq, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
resfinder_deseq_stat <- subset_samples(resfinder_deseq, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Take pair wise comparisons
deseq_PHY = subset_samples(resfinder_deseq_stat, country == "Burkina Faso" | country == "Finland")

hist(log10(apply(otu_table(deseq_PHY), 1, var)), xlab = "log10(variance)")

# Let's set a threashold for the variance
varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_resfinder = res[which(res$padj < alpha), ]

sigtab_resfinder = cbind(as(sigtab_resfinder, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_resfinder), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_resfinder = merge(sigtab_resfinder, as.data.frame(n), by = 0)

#kable(sigtab_resfinder, caption = "Taxonomy")

# Plot
# Class
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Class = factor(as.character(sigtab_resfinder$Class), levels = names(x))

# Gene cluster
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Cluster_name, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Cluster_name = factor(as.character(sigtab_resfinder$Cluster_name), levels = names(x))

# Gene 
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Gene, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Gene = factor(as.character(sigtab_resfinder$Gene), levels = names(x))

sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]

# Save BF
#write.table(sorted_sigtab, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/DESeq2_BF_Fin.txt", 
#            row.names=T, sep = "\t", col.names = T)

Fin_BF <- subset(sorted_sigtab,log2FoldChange>=0)
BF_Fin <- subset(sorted_sigtab,log2FoldChange<=0)
BF_Fin <- BF_Fin[order(BF_Fin$log2FoldChange), ]
head(Fin_BF)
head(BF_Fin)
```
## Plot
# BF-Finland
```{r}
# Take top 10 from each and combine back to one table
#Fin25 <- Fin_BF[1:25,]
#BF25 <- BF_Fin[1:25,]
#BF_Fin_top_sigtab <- rbind(Fin25, BF25)

# Shorten gene names and order again
#BF_Fin_top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", BF_Fin_top_sigtab$Gene)

#cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 10)
#p <- ggplot(data = BF_Fin_top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
#  geom_bar(stat = "identity",
#    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("                            Log2 Fold Change\n              Burkina Faso                     Finland") + 
#  scale_fill_manual(values = cols) +
#  ggtitle("Enriched ARGs in Burkinabe / Finnish samples (25 with highest log 2 fold change values of each country)") +
#  coord_flip() +
#  theme_minimal() +
#  theme(axis.text.x = element_text(family = "Times", size = 10),
#        axis.text.y = element_text(family = "Times", size = 12, face = "bold.italic"),
#        axis.title.x = element_text(family = "Times",  size = 20, face = "bold", hjust = -0.01),
#        axis.title.y = element_blank(),
#        legend.text = element_text(family = "Times", size = 20),
#        legend.title = element_blank(),
#        plot.title = element_text(family = "Times", size = 20, face = "bold"))

#ggsave(filename = "BF_FIN_deseq50_resfinder_genes_new.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## DESeq2
# ResFinder with clusters
```{r}
#OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
#match <- match(rownames(metadata), colnames(OTU_resfinder))
#OTU_resfinder <- OTU_resfinder[,match]
#all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
#clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
#colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
#col_order <- c("Class", "Cluster_name", "Gene")
#clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
#match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
#clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
#all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Divide by ARG gene lengths
## Get the lengths in terminal
# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
#resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
#all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
#OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
#deseq_OTU_resfinder <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
#all(rownames(metadata) == colnames(deseq_OTU_resfinder))
#identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], deseq_OTU_resfinder[2025, 5])
#all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Deseq
#deseq_OTU <- deseq_OTU_resfinder[, ] * 10^5 + 1

# Merge ARGs so that cluster names instead of gene names can be used
#clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt")
#all(clusters_tax_table_resfinder$Gene == rownames(deseq_OTU))

#Class <- as.data.frame(clusters_tax_table_resfinder$Class)
#Cluster_name <- as.data.frame(clusters_tax_table_resfinder$Cluster_name)
#Gene <- as.data.frame(clusters_tax_table_resfinder$Gene)

#OTU_DESeq <- cbind(deseq_OTU, Class, Cluster_name, Gene)

#names(OTU_DESeq)[names(OTU_DESeq) == "clusters_tax_table_resfinder$Class"] <- "Class"
#names(OTU_DESeq)[names(OTU_DESeq) == "clusters_tax_table_resfinder$Cluster_name"] <- "Cluster_name"
#names(OTU_DESeq)[names(OTU_DESeq) == "clusters_tax_table_resfinder$Gene"] <- "Gene"

#all(OTU_DESeq$Gene == clusters_tax_table_resfinder$Gene)

# Merge
#merged_deseq <- ddply(OTU_DESeq,"Cluster_name",numcolwise(sum))
# Remove duplicates
#uniq_clust <- clusters_tax_table_resfinder[!duplicated(clusters_tax_table_resfinder[ , c("Cluster_name")]),]
# Reorder to match OTU
#match <- match(uniq_clust$Cluster_name, merged_deseq$Cluster_name)
#merged_deseq <- merged_deseq[match,]
#all(merged_deseq$Cluster_name == uniq_clust$Cluster_name)

# Tax table
#match <- match(merged_deseq$Cluster_name, clusters_tax_table_resfinder$Cluster_name)
#heat_clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
#identical(merged_deseq$Cluster_name, heat_clusters_tax_table_resfinder$Cluster_name)

# Reload to get new row numbers
#write.table(merged_deseq, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/merged_deseq.txt", 
#            row.names=T, sep = "\t", col.names = T)
#merged_deseq <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/merged_deseq.txt", row.names=NULL)
#merged_deseq$row.names<-NULL
#merged_deseq$Cluster_name <- NULL

# Reload to get new row numbers
#write.table(heat_clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
#heat_clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=NULL)

# Create phyloseq
#merged_deseq_resfinder_PHY <- phyloseq(otu_table(merged_deseq, taxa_are_rows = TRUE), sample_data(metadata), 
#    tax_table(as.matrix(heat_clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
#merged_deseq_resfinder_PHY = subset_samples(merged_deseq_resfinder_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
#merged_deseq_resfinder_PHY = subset_samples(merged_deseq_resfinder_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
#merged_deseq_resfinder_PHY <- subset_samples(merged_deseq_resfinder_PHY, country != "UK")

## Exclude biological / technical replicates
#merged_deseq_resfinder_PHY_stat <- subset_samples(merged_deseq_resfinder_PHY, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
#merged_deseq_resfinder_PHY_stat <- subset_samples(merged_deseq_resfinder_PHY_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")

# Take pair wise comparisons
#deseq_PHY = subset_samples(merged_deseq_resfinder_PHY_stat, country == "Benin" | country == "Finland")
#deseq_PHY = subset_samples(merged_deseq_resfinder_PHY_stat, country == "Burkina Faso" | country == "Finland")
#deseq_PHY = subset_samples(merged_deseq_resfinder_PHY_stat, country == "Benin" | country == "Burkina Faso")

#hist(log10(apply(otu_table(deseq_PHY), 1, var)), xlab = "log10(variance)")

#varianceThreshold = 50
#keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
#deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
#deseq_PHY

#dds = phyloseq_to_deseq2(deseq_PHY, ~country)

#dds$category <- relevel(dds$country, "Benin", "Finland")
#dds$category <- relevel(dds$country, "Burkina Faso", "Finland")
#dds$category <- relevel(dds$country, "Benin", "Burkina Faso")

#dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

#res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
#resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

#res = res[order(res$padj, na.last=NA), ]

#alpha = 0.05
#sigtab_resfinder = res[which(res$padj < alpha), ]

#sigtab_resfinder = cbind(as(sigtab_resfinder, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_resfinder),     ], "matrix"))

#otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
#otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

#n <- rowSums(otu_table(deseq_PHY))

#sigtab_resfinder = merge(sigtab_resfinder, as.data.frame(n), by = 0)

#kable(sigtab_resfinder, caption = "Taxonomy")

# Plot
# Class
#x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Class, function(x) max(x))
#x = sort(x, TRUE)
#sigtab_resfinder$Class = factor(as.character(sigtab_resfinder$Class), levels = names(x))

# Gene cluster
#x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Cluster_name, function(x) max(x))
#x = sort(x, TRUE)
#sigtab_resfinder$Cluster_name = factor(as.character(sigtab_resfinder$Cluster_name), levels = names(x))

# Gene 
#x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Gene, function(x) max(x))
#x = sort(x, TRUE)
#sigtab_resfinder$Gene = factor(as.character(sigtab_resfinder$Gene), levels = names(x))

#sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]
```
## DESeq2
# Taxonomy (Metaxa2), Genus
```{r}
metaxa_deseq_PHY <- prune_taxa(taxa_sums(metaxa_PHY_stat) > 0, metaxa_PHY_stat)

metaxa_deseq_PHY <- tax_glom(metaxa_deseq_PHY, taxrank = "Genus")

# Take pair wise comparisons
deseq_PHY = subset_samples(metaxa_deseq_PHY, country == "Benin" | country == "Finland")

# Let's set a threashold for the variance
varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaxa = res[which(res$padj < alpha), ]

sigtab_metaxa = cbind(as(sigtab_metaxa, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaxa), 
    ], "matrix"))

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaxa = merge(sigtab_metaxa, as.data.frame(n), by = 0)

kable(sigtab_metaxa, caption = "Taxonomy")

# Phylum
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Phylum = factor(as.character(sigtab_metaxa$Phylum), levels = names(x))

# Class
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Class = factor(as.character(sigtab_metaxa$Class), levels = names(x))

# Genus
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Genus = factor(as.character(sigtab_metaxa$Genus), levels = names(x))

sorted_sigtab <- sigtab_metaxa[order(-sigtab_metaxa$log2FoldChange), ]

# Save Ben
#write.table(sorted_sigtab, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaxa2_DESeq2_Ben_Fin.txt", 
#            row.names=T, sep = "\t", col.names = T)

metaxa_Fin <- subset(sorted_sigtab,log2FoldChange>=0)
metaxa_Ben <- subset(sorted_sigtab,log2FoldChange<=0)
metaxa_Ben <- metaxa_Ben[order(metaxa_Ben$log2FoldChange), ]
head(metaxa_Fin)
head(metaxa_Ben)
```
## Plot
# Benin-Finland
```{r}
# Genus
#metaxa_Fin20 <- metaxa_Fin[1:20,]
#metaxa_Ben20 <- metaxa_Ben[1:20,]
#metaxa_top_sigtab <- rbind(metaxa_Fin20, metaxa_Ben20)

#cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 15)

#p <- ggplot(data = metaxa_top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
#  geom_bar(stat = "identity",
#   aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                 Finland") + 
#  scale_fill_manual(values = cols) +
#  ggtitle("Enriched taxa in Finnish / Beninise samples (20 taxa with highest log-2-fold change for each country)") +
#  coord_flip() +
#  theme_minimal() +
#  theme(axis.text.x = element_text(family = "Times", size = 14),
#        axis.text.y = element_text(family = "Times", size = 16, face = "bold.italic"),
#        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
#        axis.title.y = element_blank(),
#        legend.text = element_text(family = "Times", size = 20, face = "italic"),
#        legend.title = element_blank(),
#        plot.title = element_text(family = "Times", size = 20, face = "bold"))
#p

#ggsave(filename = "FIN_BEN_deseq40_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# ESKAPE etc
#metaxa_selected_sigtable <- subset(sigtab_metaxa, Genus =="Staphylococcus" | Genus == "Escherichia-Shigella" | Genus == "Enterococcus" | Genus == "Acinetobacter" | Genus == "Enterobacter" | Genus == "Pseudomonas" | Genus == "Clostridium" | Genus == "Citrobacter" | Genus == "Streptococcus" | Genus == "Aeromonas" | Genus == "Stenotrophomonas" | Genus == "Campylobacter" | Genus == "Corynebacterium" | Genus == "Mycobacterium" | Genus == "Neisseria" | Genus == "Salmonella" | Genus == "Vibrio" | Genus == "Klebsiella" | Genus =="Unclassified Staphylococcaceae" | Genus == "Unclassified Enterobacteriaceae" | Genus == "Unclassified Enterococcaceae" | Genus == "Unclassified Pseudomonadaceae" | Genus =="Unclassified Clostridiaceae"| Genus == "Unclassified Streptococcaceae" | Genus == "Unclassified Aeromonadaceae"| Genus == "Unclassified Campylobacteraceae" | Genus == "Unclassified Corynebacteriaceae" | Genus == "Unclassified Mycobacteriaceae" | Genus == "Unclassified Moraxellaceae" | Genus == "Unclassified Vibrionaceae"| Genus == "Unclassified Xanthomonadaceae" | Genus == "Unclassified Neisseriaceae")

#sorted_metaxa_selected_sigtable <- metaxa_selected_sigtable[order(-metaxa_selected_sigtable$log2FoldChange), ]

#selected_metaxa_Fin <- subset(sorted_metaxa_selected_sigtable,log2FoldChange>=0)
#selected_metaxa_Ben <- subset(sorted_metaxa_selected_sigtable,log2FoldChange<=0)
#selected_metaxa_Ben <- selected_metaxa_Ben[order(selected_metaxa_Ben$log2FoldChange), ]

#Fin11 <- selected_metaxa_Fin[1:11,]
#Ben9 <- selected_metaxa_Ben[1:9,]
#top_sigtab <- rbind(Fin11, Ben9)

#cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 8)
#p <- ggplot(data = top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
#  geom_bar(stat = "identity",
#    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                  Finland") + 
#  scale_fill_manual(values = cols) +
#  ggtitle("Differentially abundant clinically relevant taxa in\n Finnish / Beninise hospital WW") +
#  coord_flip() +
#  theme_minimal() +
#  theme(axis.text.x = element_text(family = "Times", size = 14),
#        axis.text.y = element_text(family = "Times", size = 24, face = "bold.italic"),
#        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
#        axis.title.y = element_blank(),
#        legend.text = element_text(family = "Times", size = 26, face = "italic"),
#        legend.title = element_blank(),
#        plot.title = element_text(family = "Times", size = 28, face = "bold"))
#p

#ggsave(filename = "FIN_BEN_deseq_selected_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## DESeq2
# Taxonomy (Metaxa2), Genus
```{r}
#metaxa_deseq_PHY <- prune_taxa(taxa_sums(metaxa_PHY_stat) > 0, metaxa_PHY_stat)

#metaxa_deseq_PHY <- tax_glom(metaxa_deseq_PHY, taxrank = "Genus")

# Take pair wise comparisons
#deseq_PHY = subset_samples(metaxa_deseq_PHY, country == "Burkina Faso" | country == "Finland")

#varianceThreshold = 50
#keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
#deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
#deseq_PHY

#dds = phyloseq_to_deseq2(deseq_PHY, ~country)

#ds$category <- relevel(dds$country, "Burkina Faso", "Finland")

#dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

#res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
#resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

#res = res[order(res$padj, na.last=NA), ]

#alpha = 0.05
#sigtab_metaxa = res[which(res$padj < alpha), ]

#sigtab_metaxa = cbind(as(sigtab_metaxa, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaxa),     ], "matrix"))

#n <- rowSums(otu_table(deseq_PHY))

#sigtab_metaxa = merge(sigtab_metaxa, as.data.frame(n), by = 0)

#kable(sigtab_metaxa, caption = "Taxonomy")

# Phylum
#x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Phylum, function(x) max(x))
#x = sort(x, TRUE)
#sigtab_metaxa$Phylum = factor(as.character(sigtab_metaxa$Phylum), levels = names(x))

# Class
#x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Class, function(x) max(x))
#x = sort(x, TRUE)
#sigtab_metaxa$Class = factor(as.character(sigtab_metaxa$Class), levels = names(x))

# Genus
#x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Genus, function(x) max(x))
#x = sort(x, TRUE)
#sigtab_metaxa$Genus = factor(as.character(sigtab_metaxa$Genus), levels = names(x))

#sorted_sigtab <- sigtab_metaxa[order(-sigtab_metaxa$log2FoldChange), ]

#write.table(sorted_sigtab, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaxa2_DESeq2_BF_Fin.txt", 
#            row.names=T, sep = "\t", col.names = T)
```
## Plot
# BF-Finland
```{r}
#metaxa_Fin <- subset(sorted_sigtab,log2FoldChange>=0)
#metaxa_BF <- subset(sorted_sigtab,log2FoldChange<=0)
#metaxa_BF <- metaxa_BF[order(metaxa_BF$log2FoldChange), ]

# Genus
#metaxa_Fin20 <- metaxa_Fin[1:20,]
#metaxa_BF20 <- metaxa_BF[1:20,]
#metaxa_top_sigtab <- rbind(metaxa_Fin20, metaxa_BF20)

#cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 15)

#p <- ggplot(data = metaxa_top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
#  geom_bar(stat = "identity",
#    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                 Finland") + 
#  scale_fill_manual(values = cols) +
#  ggtitle("Enriched taxa in Finnish / Burkinabe samples (20 taxa with highest log-2-fold change for each country)") +
#  coord_flip() +
#  theme_minimal() +
#  theme(axis.text.x = element_text(family = "Times", size = 14),
#        axis.text.y = element_text(family = "Times", size = 16, face = "bold.italic"),
#        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
#        axis.title.y = element_blank(),
#        legend.text = element_text(family = "Times", size = 20, face = "italic"),
#        legend.title = element_blank(),
#        plot.title = element_text(family = "Times", size = 20, face = "bold"))
#p

#ggsave(filename = "FIN_BF_deseq40_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# ESKAPE etc
#metaxa_selected_sigtable <- subset(sigtab_metaxa, Genus =="Staphylococcus" | Genus == "Escherichia-Shigella" | Genus == "Enterococcus" | Genus == "Acinetobacter" | Genus == "Enterobacter" | Genus == "Pseudomonas" | Genus == "Clostridium" | Genus == "Citrobacter" | Genus == "Streptococcus" | Genus == "Aeromonas" | Genus == "Stenotrophomonas" | Genus == "Campylobacter" | Genus == "Corynebacterium" | Genus == "Mycobacterium" | Genus == "Neisseria" | Genus == "Salmonella" | Genus == "Vibrio" | Genus == "Klebsiella" | Genus =="Unclassified Staphylococcaceae" | Genus == "Unclassified Enterobacteriaceae" | Genus == "Unclassified Enterococcaceae" | Genus == "Unclassified Pseudomonadaceae" | Genus =="Unclassified Clostridiaceae"| Genus == "Unclassified Streptococcaceae" | Genus == "Unclassified Aeromonadaceae"| Genus == "Unclassified Campylobacteraceae" | Genus == "Unclassified Corynebacteriaceae" | Genus == "Unclassified Mycobacteriaceae" | Genus == "Unclassified Moraxellaceae" | Genus == "Unclassified Vibrionaceae"| Genus == "Unclassified Xanthomonadaceae" | Genus == "Unclassified Neisseriaceae")

#sorted_metaxa_selected_sigtable <- metaxa_selected_sigtable[order(-metaxa_selected_sigtable$log2FoldChange), ]

#metaxa_Fin <- subset(sorted_metaxa_selected_sigtable,log2FoldChange>=0)
#metaxa_BF <- subset(sorted_metaxa_selected_sigtable,log2FoldChange<=0)
#metaxa_BF <- metaxa_BF[order(metaxa_BF$log2FoldChange), ]

#Fin5 <- metaxa_Fin[1:5,]
#BF9 <- metaxa_BF[1:9,]
#top_sigtab <- rbind(Fin5, BF9)

#cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 8)
#p <- ggplot(data = top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
#  geom_bar(stat = "identity",
#    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                  Finland") + 
#  scale_fill_manual(values = cols) +
#  ggtitle("Differentially abundant clinically relevant taxa in \nFinnish / Burkinabe hospital WW") +
#  coord_flip() +
#  theme_minimal() +
#  theme(axis.text.x = element_text(family = "Times", size = 14),
#        axis.text.y = element_text(family = "Times", size = 24, face = "bold.italic"),
#        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
#        axis.title.y = element_blank(),
#        legend.text = element_text(family = "Times", size = 26, face = "italic"),
#        legend.title = element_blank(),
#        plot.title = element_text(family = "Times", size = 28, face = "bold"))
#p

#ggsave(filename = "FIN_BF_deseq_selected_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## DESeq2
# Taxonomy (Metaphlan3), Genus/Species
# Benin-Finland
```{r}
# Load data
OTU_metaphlan <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_merged_abundance_table_species.txt", header=T)

# Make sure tax tabe is in order
#tax_table_metaphlan <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", quote="\"", comment.char="")
#identical(tax_table_metaphlan$V1, OTU_metaphlan$clade_name)

tax_table_metaphlan <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", header=FALSE, sep=";")
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Remove "__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

match <- match(rownames(metadata), colnames(OTU_metaphlan))
OTU_metaphlan  <- OTU_metaphlan[,match]
all(rownames(metadata) == colnames(OTU_metaphlan))

OTU_metaphlan_deseq = OTU_metaphlan

vec <- as.vector(metadata$SSU_counts)
deseq_OTU <- mapply(FUN = `*`, as.data.frame(OTU_metaphlan_deseq), vec)

metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_metaphlan)))

# OLD WAY
#deseq_OTU <- OTU_metaphlan[, ] * 10^5 + 1

#metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
#    tax_table(as.matrix(tax_table_metaphlan)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_deseq = subset_samples(metaphlan_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaphlan_deseq <- subset_samples(metaphlan_deseq, country != "UK")

## Exclude biological / technical replicates
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")

metaphlan_deseq_stat <- prune_taxa(taxa_sums(metaphlan_deseq_stat) > 0, metaphlan_deseq_stat)

# Take pair wise comparisons
deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Benin" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaphlan = res[which(res$padj < alpha), ]

sigtab_metaphlan = cbind(as(sigtab_metaphlan, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaphlan), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaphlan = merge(sigtab_metaphlan, as.data.frame(n), by = 0)

#kable(sigtab_metaphlan, caption = "Taxonomy")

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Class = factor(as.character(sigtab_metaphlan$Class), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Phylum = factor(as.character(sigtab_metaphlan$Phylum), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Genus = factor(as.character(sigtab_metaphlan$Genus), levels = names(x))
```
## Save (Species)
# Benin-Finland
```{r}
sorted_sigtab <- sigtab_metaphlan[order(-sigtab_metaphlan$log2FoldChange), ]

#write.table(sorted_sigtab, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaphlan3_DESeq2_Ben_Fin_s.txt", 
#            row.names=T, sep = "\t", col.names = T)
head(sorted_sigtab)
```
## DESeq2
# Taxonomy (Metaphlan3), Genus/Species
# Benin-Finland
```{r}
# Load data
OTU_metaphlan <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_merged_abundance_table_species.txt", header=T)

# Make sure tax tabe is in order
#tax_table_metaphlan <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", quote="\"", comment.char="")
#identical(tax_table_metaphlan$V1, OTU_metaphlan$clade_name)

tax_table_metaphlan <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", header=FALSE, sep=";")
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Remove "__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

match <- match(rownames(metadata), colnames(OTU_metaphlan))
OTU_metaphlan  <- OTU_metaphlan[,match]
all(rownames(metadata) == colnames(OTU_metaphlan))

OTU_metaphlan_deseq = OTU_metaphlan

vec <- as.vector(metadata$SSU_counts)
deseq_OTU <- mapply(FUN = `*`, as.data.frame(OTU_metaphlan_deseq), vec)

metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_metaphlan)))

# OLD WAY
#deseq_OTU <- OTU_metaphlan[, ] * 10^5 + 1

#metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
#    tax_table(as.matrix(tax_table_metaphlan)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_deseq = subset_samples(metaphlan_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaphlan_deseq <- subset_samples(metaphlan_deseq, country != "UK")

## Exclude biological / technical replicates
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")

metaphlan_deseq_stat <- prune_taxa(taxa_sums(metaphlan_deseq_stat) > 0, metaphlan_deseq_stat)

# Take pair wise comparisons
deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Benin" | country == "Finland")

# Get genus
deseq_PHY <- tax_glom(deseq_PHY, taxrank = "Genus")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaphlan = res[which(res$padj < alpha), ]

sigtab_metaphlan = cbind(as(sigtab_metaphlan, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaphlan), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaphlan = merge(sigtab_metaphlan, as.data.frame(n), by = 0)

#kable(sigtab_metaphlan, caption = "Taxonomy")

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Class = factor(as.character(sigtab_metaphlan$Class), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Phylum = factor(as.character(sigtab_metaphlan$Phylum), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Genus = factor(as.character(sigtab_metaphlan$Genus), levels = names(x))
```
## Save and plot (Genus)
# Fin_Benin
```{r}
sorted_metaphlan_sigtable <- sigtab_metaphlan[order(-sigtab_metaphlan$log2FoldChange), ]

#write.table(sorted_metaphlan_sigtable, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaphlan3_DESeq2_Ben_Fin_g.txt", 
#            row.names=T, sep = "\t", col.names = T)

head(sorted_metaphlan_sigtable)
```
## DESeq2
# Taxonomy (Metaphlan3), Genus/Species
# BF-Finland
```{r}
# Load data
OTU_metaphlan <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_merged_abundance_table_species.txt", header=T)

# Make sure tax tabe is in order
#tax_table_metaphlan <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", quote="\"", comment.char="")
#identical(tax_table_metaphlan$V1, OTU_metaphlan$clade_name)

tax_table_metaphlan <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", header=FALSE, sep=";")
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Remove "__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

match <- match(rownames(metadata), colnames(OTU_metaphlan))
OTU_metaphlan  <- OTU_metaphlan[,match]
all(rownames(metadata) == colnames(OTU_metaphlan))

OTU_metaphlan_deseq = OTU_metaphlan

vec <- as.vector(metadata$SSU_counts)
deseq_OTU <- mapply(FUN = `*`, as.data.frame(OTU_metaphlan_deseq), vec)

metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_metaphlan)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_deseq = subset_samples(metaphlan_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaphlan_deseq <- subset_samples(metaphlan_deseq, country != "UK")

## Exclude biological / technical replicates
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")

metaphlan_deseq_stat <- prune_taxa(taxa_sums(metaphlan_deseq_stat) > 0, metaphlan_deseq_stat)

# Take pair wise comparisons
deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaphlan = res[which(res$padj < alpha), ]

sigtab_metaphlan = cbind(as(sigtab_metaphlan, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaphlan), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaphlan = merge(sigtab_metaphlan, as.data.frame(n), by = 0)

#kable(sigtab_metaphlan, caption = "Taxonomy")

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Class = factor(as.character(sigtab_metaphlan$Class), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Phylum = factor(as.character(sigtab_metaphlan$Phylum), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Genus = factor(as.character(sigtab_metaphlan$Genus), levels = names(x))
```
## Save (Species)
# BF-Finland
```{r}
sorted_sigtab <- sigtab_metaphlan[order(-sigtab_metaphlan$log2FoldChange), ]

#write.table(sorted_sigtab, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaphlan3_DESeq2_BF_Fin_s.txt", 
#            row.names=T, sep = "\t", col.names = T)

head(sorted_sigtab)
```
## DESeq2
# Taxonomy (Metaphlan3), Genus/Species
# BF-Finland
```{r}
# Load data
OTU_metaphlan <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_merged_abundance_table_species.txt", header=T)

# Make sure tax tabe is in order
#tax_table_metaphlan <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", quote="\"", comment.char="")
#identical(tax_table_metaphlan$V1, OTU_metaphlan$clade_name)

tax_table_metaphlan <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", header=FALSE, sep=";")
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Remove "__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

match <- match(rownames(metadata), colnames(OTU_metaphlan))
OTU_metaphlan  <- OTU_metaphlan[,match]
all(rownames(metadata) == colnames(OTU_metaphlan))

OTU_metaphlan_deseq = OTU_metaphlan

vec <- as.vector(metadata$SSU_counts)
deseq_OTU <- mapply(FUN = `*`, as.data.frame(OTU_metaphlan_deseq), vec)

metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_metaphlan)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_deseq = subset_samples(metaphlan_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaphlan_deseq <- subset_samples(metaphlan_deseq, country != "UK")

## Exclude biological / technical replicates
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")

metaphlan_deseq_stat <- prune_taxa(taxa_sums(metaphlan_deseq_stat) > 0, metaphlan_deseq_stat)

# Take pair wise comparisons
deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Burkina Faso" | country == "Finland")

# Get genus
deseq_PHY <- tax_glom(deseq_PHY, taxrank = "Genus")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaphlan = res[which(res$padj < alpha), ]

sigtab_metaphlan = cbind(as(sigtab_metaphlan, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaphlan), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaphlan = merge(sigtab_metaphlan, as.data.frame(n), by = 0)

#kable(sigtab_metaphlan, caption = "Taxonomy")

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Class = factor(as.character(sigtab_metaphlan$Class), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Phylum = factor(as.character(sigtab_metaphlan$Phylum), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Genus = factor(as.character(sigtab_metaphlan$Genus), levels = names(x))
```
## Save and plot (Genus)
## Fin_BF
```{r}
sorted_metaphlan_sigtable <- sigtab_metaphlan[order(-sigtab_metaphlan$log2FoldChange), ]

#write.table(sorted_metaphlan_sigtable, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaphlan3_Genus_DESeq2_BF_Fin_g.txt", 
#            row.names=T, sep = "\t", col.names = T)
head(sorted_metaphlan_sigtable)
```
## DESeq2
# MGE (Mobilome)
# Benin-Finland / BF-Finland
```{r}
# Deseq
deseq_OTU <- OTU_MGE_length_SSU_norm[, ] * 10^5 + 1

MGE_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(MGE_tax_table_trim)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
MGE_deseq = subset_samples(MGE_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
MGE_deseq <- subset_samples(MGE_deseq, country != "UK")

## Exclude biological / technical replicates
MGE_deseq_stat <- subset_samples(MGE_deseq, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH62")

# Create phyloseq object with only hospital WW samples sequenced here
MGE_deseq_stat <- subset_samples(MGE_deseq_stat, category == "WA hospital effluent" | category == "North Eu hospital effluent")


MGE_deseq_stat <- prune_taxa(taxa_sums(MGE_deseq_stat) > 0, MGE_deseq_stat)

# Take pair wise comparisons
deseq_PHY = subset_samples(MGE_deseq_stat, country == "Benin" | country == "Finland")
#deseq_PHY = subset_samples(MGE_deseq_stat, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")
#dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
#plotDispEsts(dds)
#head(res)
#summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_MGE = res[which(res$padj < alpha), ]

sigtab_MGE = cbind(as(sigtab_MGE, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_MGE), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_MGE = merge(sigtab_MGE, as.data.frame(n), by = 0)

#kable(sigtab_MGE, caption = "Taxonomy")

x = tapply(sigtab_MGE$log2FoldChange, sigtab_MGE$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_MGE$Class = factor(as.character(sigtab_MGE$Class), levels = names(x))

x = tapply(sigtab_MGE$log2FoldChange, sigtab_MGE$Element, function(x) max(x))
x = sort(x, TRUE)
sigtab_MGE$Element = factor(as.character(sigtab_MGE$Element), levels = names(x))

# Sort
sorted_sigtab <- sigtab_MGE[order(-sigtab_MGE$log2FoldChange), ]
```
## Heatmap, ResFinder
```{r}
## Load data as above
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Divide by ARG gene lengths
## Get the lengths in terminal
# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
OTU_resfinder_length_SSU_norm <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(OTU_resfinder_length_SSU_norm))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], OTU_resfinder_length_SSU_norm[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Merge ARGs so that cluster names instead of gene names can be used
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt")
all(clusters_tax_table_resfinder$Gene == rownames(OTU_resfinder_length_SSU_norm))

Class <- as.data.frame(clusters_tax_table_resfinder$Class)
Cluster_name <- as.data.frame(clusters_tax_table_resfinder$Cluster_name)
Gene <- as.data.frame(clusters_tax_table_resfinder$Gene)

heat_length_SSU_norm <- cbind(OTU_resfinder_length_SSU_norm, Class, Cluster_name, Gene)

names(heat_length_SSU_norm)[names(heat_length_SSU_norm) == "clusters_tax_table_resfinder$Class"] <- "Class"
names(heat_length_SSU_norm)[names(heat_length_SSU_norm) == "clusters_tax_table_resfinder$Cluster_name"] <- "Cluster_name"
names(heat_length_SSU_norm)[names(heat_length_SSU_norm) == "clusters_tax_table_resfinder$Gene"] <- "Gene"

all(heat_length_SSU_norm$Gene == clusters_tax_table_resfinder$Gene)

# Merge
merged <- ddply(heat_length_SSU_norm,"Cluster_name",numcolwise(sum))
# Remove duplicates
uniq_clust <- clusters_tax_table_resfinder[!duplicated(clusters_tax_table_resfinder[ , c("Cluster_name")]),]
# Reorder to match OTU
match <- match(uniq_clust$Cluster_name, merged$Cluster_name)
merged <- merged[match,]
all(merged$Cluster_name == uniq_clust$Cluster_name)

# Tax table
match <- match(merged$Cluster_name, clusters_tax_table_resfinder$Cluster_name)
heat_clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
identical(merged$Cluster_name, heat_clusters_tax_table_resfinder$Cluster_name)

# Reload to get new row numbers
write.table(merged, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_OTU_resfinder.txt", 
            row.names=T, sep = "\t", col.names = T)
heat_OTU_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_OTU_resfinder.txt", row.names=NULL)
heat_OTU_resfinder$row.names<-NULL
heat_OTU_resfinder$Cluster_name <- NULL

# Reload to get new row numbers
write.table(heat_clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
heat_clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=NULL)

# Create phyloseq
heat_resfinder_PHY <- phyloseq(otu_table(heat_OTU_resfinder, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(heat_clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
heat_resfinder_PHY = subset_samples(heat_resfinder_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")

# Subset samples
heat_resfinder_PHY <- subset_samples(heat_resfinder_PHY, category == "WA hospital effluent" | 
                                       category ==  "WA treated, receiving municipality channel" | 
                                       category == "WA treated, receiving river" |
                                       category == "WA empty hospital septic tank" | category == "WA treated, drinking water" | 
                                       category == "WA street gutter" | category == "WA street gutter, sediment" | 
                                       category == "WA in-patient feces" | category == "WA treated, hand washing" | 
                                       category == "WA treated, hand washing" | category == "WA river, drinking water" | 
                                       category == "North Eu hospital effluent")
```
## Plot Heatmap
```{r}
# Exctract ARGs
heat_resfinder_PHY_Clusters <- tax_glom(heat_resfinder_PHY, taxrank = "Cluster_name")

selected <- subset_taxa(heat_resfinder_PHY_Clusters, 
                        Cluster_name == "blaOXA-370_1_clust" | Cluster_name == "blaIMP-58_1_clust" |
                        Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaCTX-M-211_1_clust" | 
                        Cluster_name == "blaCTX-M-4_1_clust" | Cluster_name == "blaCTX-M-110_1_clust" |
                        Cluster_name == "blaKPC-34_1_clust" | Cluster_name == "catB3_2_clust" |     
                        Cluster_name == "blaGES-1_1_clust" | Cluster_name == "blaVEB-1_1_clust" |
                        Cluster_name == "cfxA_1_clust" | Cluster_name == "blaCMY-150_2_clust" |       
                        Cluster_name == "aph(6)-Id_1_clust" | Cluster_name == "aadA2_1_clust" |
                        Cluster_name == "cmlA1_1_clust" | Cluster_name == "mcr-3.1_1_clust" | 
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "erm(B)_18_clust" |
                        Cluster_name == "mcr-1.11_1_clust" | Cluster_name == "qnrA1_1_clust" |
                        Cluster_name == "nimA_1" | Cluster_name == "nimE_1_clust" |
                        Cluster_name == "sul1_11_clust" | Cluster_name == "mcr-5.1_1_clust" |
                        Cluster_name == "tet(A)_6_clust" | 
                        Cluster_name == "dfrA1_1_clust" | Cluster_name == "erm(33)_1" |
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "blaOXA-368_1_clust" | 
                        Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaOXA-392_1_clust")  

# OTU matrix
heat_OTU = as(otu_table(selected), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Tax table matrix
heat_tax = as(tax_table(selected), "matrix")

# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# Into dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
match <- match(rownames(heat.df), rownames(heat_tax.df))
temp <- heat_tax.df[match,]
rownames(heat.df) <- temp$V3
#metadata_row <- data.frame(temp$V2, row.names=rownames(heat.df))

# Col annotation
category <- as.matrix(sample_data(selected)[["category"]])
category <- as.factor(category)
category <- data.frame(category)
colnames(category) <- c("category")
rownames(category) <- as.matrix(colnames(otu_table(selected)))

ann_colors <- list(category = c("WA hospital effluent" = "#F04744", 
                                    "WA treated, receiving municipality channel" = "#AF32D9", 
                                    "WA treated, receiving river" = "black",
                                    "WA empty hospital septic tank" = "#B1B6FC",
                                    "WA treated, drinking water" = "#FCF0C0", 
                                    "WA street gutter" = "#ffe433", 
                                    "WA street gutter, sediment" = "#A03DBD",
                                    "WA in-patient feces" = "#588c7e", 
                                    "WA treated, hand washing" = "#f4a688",
                                    "WA river, drinking water" = "#00ced1", 
                                    "North Eu hospital effluent" = "#A1BD14"))

rownames(heat.df) <- c("aadA2_1_clust__(aadA)", "aph(6)-Id_1_clust__(aph(6)-Id)", "blaNDM-18_1_clust__(blaNDM-1)", 
                     "blaOXA-368_1_clust__(blaOXA-10)", "blaOXA-370_1_clust__(blaOXA-48)", "blaOXA-392_1_clust__(blaOXA-1)",
                     "blaIMP-58_1_clust__(blaIMP)", "blaCMY-150_2_clust__(blaCMY-2)", "blaCTX-M-211_1_clust__(Group I blaCTX-M)",
                     "blaCTX-M-4_1_clust__(Group II blaCTX-M)", "blaCTX-M-110_1_clust__(Group III blaCTX-M)", 
                     "blaKPC-34_1_clust__(blaKPC-2)", "blaVEB-1_1_clust__(blaVEB-1)", "cfxA_1_clust__(cfxA)", 
                     "blaVIM-48_1_clust__(blaVIM-7)", "blaGES-1_1_clust__(blaGES-5)", "mcr-1.11_1_clust__(mcr-1)", 
                     "mcr-3.1_1_clust__(mcr-3)", "mcr-5.1_1_clust__(mcr-5)", "erm(B)_18_clust__(ermB)", "mph(E)_1_clust__(mphE)", 
                     "erm(33)_1__(ermA)", "nimA_1__(nimA)", "nimE_1_clust__(nimE)", 
                     "catB3_2_clust__(catB3)", "cmlA1_1_clust__(cmlA)", "qnrA1_1_clust__(qnrA)", 
                     "sul1_11_clust__(sul1)", "tet(A)_6_clust__(tetA)", "dfrA1_1_clust__(dfrA)")

# Row annotation
metadata_row <- data.frame(temp$V2, row.names=rownames(heat.df))
colnames(metadata_row) <- c("AB class")

## Plot log
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

heat <- pheatmap(log10(heat.df + 0.000001), cluster_rows = F, cluster_cols = T, 
                        cellwidth = 9, cellheight = 17, 
                        border_color = "white",
                        fontsize=10,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "", 
                        angle_col = 90, legend = TRUE, fontsize_row = 12, fontsize_col = 9, 
                        labels_row = as.expression(newnames),
                       # filename = "selected_all_heat.png",
                        annotation_row = metadata_row,
                        annotation_col = category,
                        annotation_colors = ann_colors,
                        clustering_distance_cols = "euclidean",
                        treeheight_col = 18,
                        cutree_cols = 4, gaps_row = c(2, 16, 19, 22, 24, 26, 27, 28, 29)
                        )
heat
```
## Heatmap for taxa
````{r}
metaphlan_PHY_Species <- tax_glom(metaphlan_PHY_stat, taxrank = "Species")
# ESKAPE and other relevant taxa
selected <- subset_taxa(metaphlan_PHY_Species,  Species == "Acinetobacter_baumannii" | Species == "Acinetobacter_nosocomialis" 
                        | Species == "Acinetobacter_bouvetii"
                        | Species == "Acinetobacter_johnsonii" | Species == "Acinetobacter_radioresistens" | Species == "Acinetobacter_lwoffii" | Species == "Acinetobacter_calcoaceticus" | Species == "Acinetobacter_haemolyticus" | Species == "Acinetobacter_bereziniae" | Species == "Acinetobacter_venetianus" | Species == "Acinetobacter_calcoaceticus" | Species == "Acinetobacter_pittii" | Species == "Acinetobacter_guillouiae" | Species == "Acinetobacter_schindleri" | Species == "Acinetobacter_bereziniae" | Species == "Acinetobacter_kyonggiensis" |  Species == "Enterobacter_cloacae_complex" | Species == "Enterococcus_faecium" | Species == "Klebsiella_pneumoniae" | Species == "Staphylococcus_aureus" | Species == "Pseudomonas_aeruginosa_group" | Species == "Escherichia_coli"
                        #| Species == "Stenotrophomonas_maltophilia"
                        )

# Filter out low abundance taxa
selected <- subset_taxa(selected, taxa_sums(selected) != 0)

# OTU matrix
heat_OTU = as(otu_table(selected), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Tax table matrix
heat_tax = as(tax_table(selected), "matrix")

# Swap colnames
match <- match(rownames(heat.df), rownames(heat_tax))
temp <- heat_tax[match,]
all(rownames(temp) == rownames(heat_OTU))
all(rownames(temp) == rownames(heat_tax))
rownames(heat.df) <- temp[, 7]

new_df <- heat.df[ order(row.names(heat.df)), ]
new_tax = heat_tax
rownames(new_tax) <- paste(selected@tax_table[,7])
new_tax[ order(row.names(new_tax)), ]

# Col annotation
country <- as.matrix(sample_data(selected)[["country"]])
country <- as.factor(country)
country <- data.frame(country)
colnames(country) <- c("country")
rownames(country) <- as.matrix(colnames(otu_table(selected)))
country$country <- gsub(" ", "_", country$country)

ann_colors <- list(country = c("Benin" = "#B2182B", 
                                    "Burkina_Faso" = "#44AA99", 
                                    "Finland" = "#2166AC"))

colnames(new_df) <- gsub(pattern = "_[A-Z].*", replacement = "_", colnames(new_df))

## Plot log
newnames <- lapply(
  rownames(new_df),
  function(x) bquote(italic(.(x))))

heat <- pheatmap(sqrt(new_df), cluster_rows = F, cluster_cols = T, 
                        border_color = "grey",
                        colorRampPalette(brewer.pal(9, "Blues"))(100), 
                        main = "Relative abundance of clinically relevant species,\n (Metaphlan3, square root transormed)", 
                        angle_col = 90, legend = TRUE, fontsize_row = 11,
                        labels_row = as.expression(newnames),
                        filename = "eskape_heat.png",
                        annotation_col = country,
                        clustering_distance_cols = "euclidean",
                        show_colnames = T,
                 cellwidth = 13,
                 cellheight = 26,
                 gaps_row = rep(c(12)),
                 annotation_colors = ann_colors
                        )
heat
```
## 15 most abundant ARGs / country
```{r}
# Benin
resfinder_PHY_stat_Ben <- subset_samples(resfinder_PHY_stat, country == "Benin")

resfinder_PHY_stat_Ben_abun <- tax_glom(resfinder_PHY_stat_Ben, taxrank = "Gene")

# Take 15 most abundant
resfinder_PHY_stat_Ben_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_stat_Ben_abun), TRUE)[1:15]), 
    resfinder_PHY_stat_Ben_abun)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 18)
rfc <- plot_bar(resfinder_PHY_stat_Ben_abun, fill = "Gene") 
rfc_plot_Ben <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of ARGs (ResFinder)"))) +
  ggtitle("ARG Abundance normalized to 16s rRNA") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, family = "Times", face = "bold", angle = 45, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 24, family = "Times", hjust = 0.5, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))

# BF
resfinder_PHY_stat_BF <- subset_samples(resfinder_PHY_stat, country == "Burkina Faso")

resfinder_PHY_stat_BF_abun <- tax_glom(resfinder_PHY_stat_BF, taxrank = "Gene")

# Take 15 most abundant
resfinder_PHY_stat_BF_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_stat_BF_abun), TRUE)[1:15]), 
    resfinder_PHY_stat_BF_abun)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 18)
rfc <- plot_bar(resfinder_PHY_stat_BF_abun, fill = "Gene") 
rfc_plot_BF <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of ARGs (ResFinder)"))) +
  ggtitle("ARG Abundance normalized to 16s rRNA") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, family = "Times", face = "bold", angle = 45, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 24, family = "Times", hjust = 0.5, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))

# Finland
resfinder_PHY_stat_Fin <- subset_samples(resfinder_PHY_stat, country == "Finland")

resfinder_PHY_stat_Fin_abun <- tax_glom(resfinder_PHY_stat_Fin, taxrank = "Gene")

# Take 15 most abundant
resfinder_PHY_stat_Fin_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_stat_Fin_abun), TRUE)[1:15]), 
    resfinder_PHY_stat_Fin_abun)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 18)
rfc <- plot_bar(resfinder_PHY_stat_Fin_abun, fill = "Gene") 
rfc_plot_Fin <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of ARGs (ResFinder)"))) +
  ggtitle("ARG Abundance normalized to 16s rRNA") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, family = "Times", face = "bold", angle = 45, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 24, family = "Times", hjust = 0.5, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))

# Create dataframe
Benin <- data.frame(resfinder_PHY_stat_Ben_abun@tax_table)$Gene
BF <- data.frame(resfinder_PHY_stat_BF_abun@tax_table)$Gene
Finland <- data.frame(resfinder_PHY_stat_Fin_abun@tax_table)$Gene

top_ARGs <- data.frame(Benin, BF, Finland)
```
## Interesting ARGs
# MCR
```{r}
cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#F22D3D", "#FDA4B3", "#882255"), 8)
#cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#F22D3D", "#FDA4B3", "#882255"), 11)

resfinder_PHY_mcr <- subset_taxa(resfinder_PHY_stat, Class == "Polymyxin")

# save sums
#resfinder_PHY_mcr_pruned <- tax_glom(resfinder_PHY_mcr, taxrank = "Cluster_name")
#resfinder_PHY_mcr_pruned <- subset_taxa(resfinder_PHY_mcr_pruned, Cluster_name == "mcr-3.1_1_clust")
#df <- data.frame(sample_sums(resfinder_PHY_mcr_pruned))

# plot
resfinder_PHY_mcr <- subset_samples(resfinder_PHY_mcr, sample_sums(resfinder_PHY_mcr) >= 0.0005)
resfinder_PHY_mcr <- subset_taxa(resfinder_PHY_mcr, taxa_sums(resfinder_PHY_mcr) != 0)

hospital <- factor(c("I", "I", "I", "J", "J", "J", "J", "J", "J", "J", "G", "A", "A", "B", "B", "B", "B", "B", "B", "C", "C", "C", "K", "M", "M", "L"))

rfc <- plot_bar(resfinder_PHY_mcr, fill = "Cluster_name")
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols, labels = c("mcr-1", "mcr-10", "mcr-3", "mcr-3.17", "mcr-4", "mcr-5", "mcr-7", "mcr-9")) +
  labs(y = expression(atop(bold("ARGs to 16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_mcr)))), labels=hospital, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, family = "Times", angle = 0, face = "bold"),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 0),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 28, family = "Times", face = "italic"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(5, "char"),
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 30, family = "Times", hjust = 0, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))
rfc_plot

ggsave(filename = "mcr_bar_plot.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Visualization using heatmap
#resfinder_PHY_mcr <- subset_taxa(resfinder_PHY_selected, Cluster_name == "mcr-5.1_1_clust" | Cluster_name == "mcr-1.11_1_clust" | Cluster_name == "mcr-3.17_1" | Cluster_name == "mcr-4.1_1_clust" | Cluster_name == "mcr-7.1_1" | Cluster_name == "mcr-9_1" | Cluster_name == "mcr-10_1" | Cluster_name == "mcr-8_1" | Cluster_name == "mcr-6.1_1" | Cluster_name == "mcr-2.1_1_clust")

#otu_table(resfinder_PHY_mcr)[otu_table(resfinder_PHY_mcr) < 0] <- 0
#otu_table(resfinder_PHY_mcr)[otu_table(resfinder_PHY_mcr) > 0] <- 1

#heat_df_mcr = otu_table(resfinder_PHY_mcr)

#rownames(heat_df_mcr) <- c("mcr-1.1",	"mcr-1.2",	"mcr-1.3",	"mcr-1.4",	"mcr-1.5",	"mcr-1.6",	"mcr-1.7",	"mcr-1.8",	"mcr-1.9",	"mcr-1.10",	"mcr-1.11",	"mcr-1.12",	"mcr-1.13",	"mcr-1.14",	"mcr-2.1",	"mcr-2.2",	"mcr-3.17",	"mcr-4.1",	"mcr-4.2",	"mcr-4.3",	"mcr-4.6",	"mcr-4.4",	"mcr-4.5",	"mcr-5.1",	"mcr-5.2",	"mcr-6.1",	"mcr-7.1",	"mcr-8",	"mcr-9",	"mcr-10")

#colnames(heat_df_mcr) <- sample_data(resfinder_PHY_mcr)$alias

#pheatmap(heat_df_mcr, cluster_cols = F, cluster_rows = T)
```
## Interesting ARGs
# MRSA & VRE & tigecycline ARGs
```{r}
# mec variants
resfinder_PHY_Cluster <- subset_taxa(resfinder_PHY_stat, Cluster_name == "mecA_1_clust" | Cluster_name == "mecB_1" | Cluster_name == "mecC2_1_clust" | Cluster_name == "mecD_1")

# Metronidazole
resfinder_PHY_Cluster <- subset_taxa(resfinder_PHY_stat, Class == "Metronidazole")
resfinder_PHY_Cluster <- subset_taxa(resfinder_PHY, Class == "Metronidazole")

# van variants
resfinder_PHY_Cluster <- subset_taxa(resfinder_PHY_stat, Cluster_name == "VanHAX_1_clust" | Cluster_name == "VanHAX_PA" | Cluster_name == "VanHBX_1_clust" | Cluster_name == "VanC1XY_1_clust"  | Cluster_name == "VanC3XY_1_clust" | Cluster_name == "VanHDX_6" | Cluster_name == "VanHDX_7_clust" | Cluster_name == "VanHDX_3_clust" | Cluster_name == "VanHFX_1" | Cluster_name == "VanE_1_clust" | Cluster_name == "VanGXY_1" | Cluster_name == "VanG2XY_1" | Cluster_name == "VanLXY_1" | Cluster_name == "VanHMX_1" | Cluster_name == "VanNXY_1" | Cluster_name == "VanHOX_1" | Cluster_name == "vanXmurFvanWI_1" | Cluster_name == "vanXmurFvanKWI_1" | Cluster_name == "vanXmurFvanKWI_2")

# tigecycline
resfinder_PHY_Cluster <- subset_taxa(resfinder_PHY_stat, Cluster_name == "tet(X)_1_clust" | Cluster_name == "tet(X3)_1" |
                                           Cluster_name == "tet(X)_3_clust" | Cluster_name == "tet(37)_1" |
                                           Cluster_name == "tet(47)_1" | Cluster_name == "tet(48)_1" |
                                           Cluster_name == "tet(49)_1" | Cluster_name == "tet(50)_1" |
                                           Cluster_name == "tet(51)_1" | Cluster_name == "tet(52)_1" |
                                           Cluster_name == "tet(53)_1" | Cluster_name == "tet(54)_1" |
                                           Cluster_name == "tet(55)_1")

resfinder_PHY_Cluster <- subset_samples(resfinder_PHY_Cluster, sample_sums(resfinder_PHY_Cluster) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 43)
rfc <- plot_bar(resfinder_PHY_Cluster, fill = "Gene") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARGs to 16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 9, family = "Times", angle = 45, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~hospital, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 20, family = "Times", hjust = 0, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white"))

# Check out also with the CARD data
CARD_PHY_Cluster <- subset_taxa(CARD_PHY_stat, Class == "glycylcycline")
CARD_PHY_Cluster <- subset_taxa(CARD_PHY_Cluster, Gene == "tetX" |
                                      Gene == "Tet(X4)" | Gene == "Tet(X3)" |
                                      Gene == "Tet(47)" | Gene == "tet(48)" |
                                      Gene == "tet(49)" | Gene == "tet(51)" |
                                      Gene == "tet(56)")

CARD_PHY_Cluster <- subset_samples(CARD_PHY_Cluster, sample_sums(CARD_PHY_Cluster) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 24)
card <- plot_bar(CARD_PHY_Cluster, fill = "Gene") 
card_plot <- card +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARGs to 16s rRNA"), atop("CARD")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 9, family = "Times", angle = 45, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 20, family = "Times", hjust = 0, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white"))
```
## Interesting ARGs
# Carbapenemases
```{r}
resfinder_PHY_Cluster_1 <- subset_taxa(resfinder_PHY_stat, Cluster_name == "blaKPC-34_1_clust"| Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaIMP-1_1_clust" | Cluster_name == "blaOXA-397_1_clust")

resfinder_PHY_Cluster_2 <- subset_taxa(resfinder_PHY_stat, Gene == "blaGES-2_1_AF326355" | Gene == "blaGES-4_1_AB116723" | Gene == "blaGES-5_1_DQ236171" 
                                       | Gene == "blaGES-6_1_AY494718" | Gene == "blaGES-14_1_GU207844" | Gene == "blaGES-16_1_HM173356" 
                                       | Gene == "blaGES-18_1_JQ028729" | Gene == "blaGES-20_1_JN596280" # carbapenemase blaGES
                                       
                                       | Gene == "blaOXA-48_1_AY236073" | Gene == "blaOXA-162_1_GU197550" | Gene == "blaOXA-181_1_CM004561" 
                                       | Gene == "blaOXA-199_1_JN704570" | Gene == "blaOXA-204_1_KP027885" | Gene == "blaOXA-232_1_JX423831" 
                                       | Gene == "blaOXA-244_1_KP659189" | Gene == "blaOXA-245_1_JX438001" | Gene == "blaOXA-247_1_JX893517" 
                                       | Gene == "blaOXA-247_1_JX893517" | Gene == "blaOXA-514_1_KU866382" | Gene == "blaOXA-515_1_KU866383" 
                                       | Gene == "blaOXA-517_1_KU878974") # blaOXA-48-like

resfinder_PHY_Cluster <- merge_phyloseq(resfinder_PHY_Cluster_1, resfinder_PHY_Cluster_2)

#resfinder_PHY_Cluster <- subset_samples(resfinder_PHY_Cluster, sample_sums(resfinder_PHY_Cluster) != 0)

cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#FDA4B3", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
alias <- factor(c("BFH1",	"BFH10",	"BFH11",	"BFH12",	"BFH13",	"BFH14",	"BFH15",	"BFH16",	"BFH17",	"BFH18",	"BFH19",	"BFH20",	"BFH21",	"BFH22",	"BFH23",	"BFH25",	"BFH28",	"BFH29",	"BFH30",	"BFH31",	"BFH32",	"BFH33",	"BFH34",	"BFH35",	"BFH36",	"BFH37",	"BFH38A",	"BFH39",	"BFH4",	"BFH40",	"BFH41",	"BFH6",	"BFH8",	"BFH9",	"BH01",	"BH02",	"BH03",	"BH05",	"BH06",	"BH07",	"BH09",	"BH27",	"BH29",	"BH30",	"BH34A",	"BH35",	"BH36",	"BH37",	"BH38",	"BH39",	"BH44",	"BH46",	"BH47",	"BH48",	"BH49",	"BH50",	"BH58",	"BH60",	"BH61",	"FH1",	"FH2",	"FH3",	"FH4",	"FH5",	"FH6",	"FH7",	"FH9"))

hospital <- factor(c("F",	"G",	"G",	"G",	"H",	"H",	"H",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"F",	"J",	"J",	"G",	"G",	"G",	"A",	"A",	"A",	"A",	"A",	"A",	"A",	"B",	"B",	"B",	"B",	"B",	"B",	"B",	"B",	"B",	"C",	"C",	"C",	"C",	"C",	"C",	"D",	"D",	"D",	"K",	"K",	"K",	"M",	"M",	"M",	"N",	"L"))

rfc <- plot_bar(resfinder_PHY_Cluster, fill = "Cluster_name") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  labs(y = expression(atop(bold("ARGs/16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs in hospital WW samples") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_Cluster)))), labels=hospital, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, family = "Times", angle = 0, hjust = 1, vjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 0),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 28, family = "Times", face = "italic"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(5, "char"),
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free") +
  theme(strip.text.x= element_text(size = 28, family = "Times", hjust = 0, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))
rfc_plot

ggsave(filename = "carbapenemases.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Check out also with the CARD data
```{r}
CARD_PHY_Cluster <- subset_taxa(CARD_PHY_stat, Class == "carbapenem")
CARD_PHY_Cluster <- subset_taxa(CARD_PHY_stat, Gene == "KPC-4" | Gene == "KPC-8" | Gene == "KPC-10" | Gene == "KPC-12" | Gene == "NDM-10" | Gene == "NDM-6" | Gene == "NDM-24" | Gene == "NDM-13" | Gene == "GES-14" | Gene == "GES-15" | Gene == "GES-18" | Gene == "GES-20" | Gene == "GES-21" | Gene == "VIM-5" | Gene == "VIM-6" | Gene == "VIM-7" | Gene == "VIM-8" | Gene == "VIM-12" | Gene == "VIM-17" | Gene == "VIM-30" | Gene == "VIM-35" | Gene == "VIM-37" | Gene == "VIM-38" | Gene == "IMP-1" | Gene == "IMP-2" | Gene == "IMP-4" | Gene == "IMP-11" | Gene == "IMP-14" | Gene == "IMP-15" | Gene == "IMP-28" | Gene == "IMP-29"| Gene == "IMP-31" | Gene == "IMP-33" | Gene == "IMP-26" | Gene == "IMP-42" | Gene == "IMP-48" | Gene == "IMP-51" | Gene == "OXA-2")

CARD_PHY_Cluster <- subset_samples(CARD_PHY_Cluster, sample_sums(CARD_PHY_Cluster) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 38)
card <- plot_bar(CARD_PHY_Cluster, fill = "Gene") 
card_plot <- card +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARGs to 16s rRNA"), atop("CARD")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 9, family = "Times", angle = 45, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 0),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 20, family = "Times", hjust = 0, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white"))
```
## Carbapenemases in other than hospital WW
```{r}
resfinder_PHY_Cluster_1 <- subset_taxa(resfinder_PHY, Cluster_name == "blaKPC-34_1_clust"| Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaIMP-1_1_clust" | Cluster_name == "blaOXA-397_1_clust")

resfinder_PHY_Cluster_2 <- subset_taxa(resfinder_PHY, Gene == "blaGES-2_1_AF326355" | Gene == "blaGES-4_1_AB116723" | Gene == "blaGES-5_1_DQ236171" 
                                       | Gene == "blaGES-6_1_AY494718" | Gene == "blaGES-14_1_GU207844" | Gene == "blaGES-16_1_HM173356" 
                                       | Gene == "blaGES-18_1_JQ028729" | Gene == "blaGES-20_1_JN596280" # carbapenemase blaGES
                                       
                                       | Gene == "blaOXA-48_1_AY236073" | Gene == "blaOXA-162_1_GU197550" | Gene == "blaOXA-181_1_CM004561" 
                                       | Gene == "blaOXA-199_1_JN704570" | Gene == "blaOXA-204_1_KP027885" | Gene == "blaOXA-232_1_JX423831" 
                                       | Gene == "blaOXA-244_1_KP659189" | Gene == "blaOXA-245_1_JX438001" | Gene == "blaOXA-247_1_JX893517" 
                                       | Gene == "blaOXA-247_1_JX893517" | Gene == "blaOXA-514_1_KU866382" | Gene == "blaOXA-515_1_KU866383" 
                                       | Gene == "blaOXA-517_1_KU878974") # blaOXA-48-like

resfinder_PHY_Cluster <- merge_phyloseq(resfinder_PHY_Cluster_1, resfinder_PHY_Cluster_2)

## Benin
# Feces
resfinder_PHY_feces <- subset_samples(resfinder_PHY_Cluster, alias == "BH20" | alias == "BH22" | alias == "BH24" | alias == "BH25")
df <- sample_sums(resfinder_PHY_feces)

cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#FDA4B3", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
names <- paste(resfinder_PHY_feces@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_feces, fill = "Cluster_name") 
rfc_plot1 <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  ggtitle("Benin") +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  scale_y_continuous(limits = c(0,0.01)) +
  labs(y = expression(atop(bold("ARGs/16s rRNA"), atop("ResFinder")))) +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_feces)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 30, family = "Times", angle = 0, face = "bold"),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 0),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 40, family = "Times", face = "bold")) +
  facet_grid(~plot_name, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 36, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))
rfc_plot1

ggsave(filename = "carbapenemases_feces_ben.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

# Drinking
resfinder_PHY_ben_drink <- subset_samples(resfinder_PHY_Cluster, alias == "BSE100" | alias == "BSE74" | alias == "BSE79"
                                    | alias == "BSE93"| alias ==  "BH11")
df <- sample_sums(resfinder_PHY_ben_drink)

cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#FDA4B3", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
names <- paste(resfinder_PHY_ben_drink@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_ben_drink, fill = "Cluster_name") 
rfc_plot2 <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  scale_y_continuous(limits = c(0,0.01)) +
  labs(y = expression(atop(bold("ARGs/16s rRNA"), atop("ResFinder")))) +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_ben_drink)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 30, family = "Times", angle = 0, face = "bold"),
      # axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 0),
      # axis.title.y = element_text(size = 30, family = "Times"),
      axis.title.y = element_blank(),
      axis.text.y = element_blank(),
        axis.title.x = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~plot_name, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 36, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))
rfc_plot2

ggsave(filename = "carbapenemases_drinking_ben.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

# other
resfinder_PHY_ben_other <- subset_samples(resfinder_PHY_Cluster, alias ==  "BH13" | alias == "BH14"| alias == "BH32" | alias == "BH52")
df <- sample_sums(resfinder_PHY_ben_other)

cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#FDA4B3", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
names <- paste(resfinder_PHY_ben_other@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_ben_other, fill = "Cluster_name") 
rfc_plot3 <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  scale_y_continuous(limits = c(0,0.01)) +
  labs(y = expression(atop(bold("ARGs/16s rRNA"), atop("ResFinder")))) +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_ben_other)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 30, family = "Times", angle = 0, face = "bold"),
       # axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 0),
       # axis.title.y = element_text(size = 30, family = "Times"),
       axis.title.y = element_blank(),
       axis.text.y = element_blank(),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times",  face = "italic"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~plot_name, scales = "free", space = "free", 
            labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 36, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))
rfc_plot3

ggsave(filename = "carbapenemases_other_ben.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

## Burkina Faso
# other
resfinder_PHY_BF_other <- subset_samples(resfinder_PHY_Cluster, alias == "BFH27" | alias ==  "BFH42" | alias == "BFH26")
df <- sample_sums(resfinder_PHY_BF_other)

cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#FDA4B3", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)

names <- paste(resfinder_PHY_BF_other@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_BF_other, fill = "Cluster_name") 
rfc_plot4 <- rfc +
  geom_bar(stat="identity", size = 0, color = NA) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  labs(y = expression(atop(bold("ARGs/16s rRNA"), atop("ResFinder")))) +
  ggtitle("Burkina Faso") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_BF_other)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 30, family = "Times", angle = 0, face = "bold"),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 0),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~plot_name, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 36, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))
rfc_plot4

ggsave(filename = "carbapenemases_other_bf.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 


###################
# Sample sums
# feces
resfinder_PHY_feces <- subset_samples(resfinder_PHY, alias == "BH20" | alias == "BH22" | alias == "BH24" | alias == "BH25")
resfinder_PHY_feces <- tax_glom(resfinder_PHY_feces, taxrank = "Gene")

# Take 15 most abundant
resfinder_PHY_feces_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_feces), TRUE)[1:15]), 
    resfinder_PHY_feces)

# Specific ARGs
resfinder_PHY_subset <- subset_samples(resfinder_PHY, alias == "BH11")
resfinder_PHY_subset <- tax_glom(resfinder_PHY_subset, taxrank = "Cluster_name")

resfinder_PHY_subset <- subset_taxa(resfinder_PHY_feces, Cluster_name == "nimE_1_clust")
resfinder_PHY_subset <- subset_taxa(resfinder_PHY_subset, taxa_sums(resfinder_PHY_subset) != 0)

resfinder_PHY_subset <- subset_taxa(resfinder_PHY_feces, Cluster_name == "nimA_1")
resfinder_PHY_subset <- subset_taxa(resfinder_PHY_subset, taxa_sums(resfinder_PHY_subset) != 0)

resfinder_PHY_subset <- subset_taxa(resfinder_PHY_subset, Cluster_name == "blaNDM-18_1_clust")
resfinder_PHY_subset <- subset_taxa(resfinder_PHY_subset, taxa_sums(resfinder_PHY_subset) != 0)
df <- sample_sums(resfinder_PHY_subset)

# drinking
resfinder_PHY_ben_drink <- subset_samples(resfinder_PHY, alias == "BSE100" | alias == "BSE74" | alias == "BSE79"
                                    | alias == "BSE93"| alias ==  "BH11")
resfinder_PHY_ben_drink <- subset_taxa(resfinder_PHY_ben_drink, taxa_sums(resfinder_PHY_ben_drink) != 0)
df <- data.frame(tax_table(resfinder_PHY_ben_drink))

resfinder_PHY_ben_drink <- tax_glom(resfinder_PHY_ben_drink, taxrank = "Gene")

# Take 15 most abundant
resfinder_PHY_ben_drink_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_ben_drink), TRUE)[1:15]), 
    resfinder_PHY_ben_drink)

# other, Benin
resfinder_PHY_ben_other <- subset_samples(resfinder_PHY, alias ==  "BH13" | alias == "BH14"| alias == "BH32" | alias == "BH52")
resfinder_PHY_ben_other <- subset_taxa(resfinder_PHY_ben_other, taxa_sums(resfinder_PHY_ben_other) != 0)
df <- data.frame(tax_table(resfinder_PHY_ben_other))

resfinder_PHY_ben_other <- tax_glom(resfinder_PHY_ben_other, taxrank = "Gene")

# Take 15 most abundant
resfinder_PHY_ben_other_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_ben_other), TRUE)[1:15]), 
    resfinder_PHY_ben_other)

# other, BF
resfinder_PHY_BF_other <- subset_samples(resfinder_PHY, alias == "BFH27" | alias ==  "BFH42" | alias == "BFH26")
resfinder_PHY_BF_other <- subset_taxa(resfinder_PHY_BF_other, taxa_sums(resfinder_PHY_BF_other) != 0)
df <- data.frame(tax_table(resfinder_PHY_BF_other))

resfinder_PHY_BF_other <- tax_glom(resfinder_PHY_BF_other, taxrank = "Gene")

# Take 15 most abundant
resfinder_PHY_BF_other_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_BF_other), TRUE)[1:15]), 
    resfinder_PHY_BF_other)
```
# Figure panel for ordination plots
```{r}
resfinder_PHY_Cluster_1 <- subset_taxa(resfinder_PHY_stat, Cluster_name == "blaKPC-34_1_clust"| Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaIMP-1_1_clust" | Cluster_name == "blaOXA-397_1_clust")

resfinder_PHY_Cluster_2 <- subset_taxa(resfinder_PHY_stat, Gene == "blaGES-2_1_AF326355" | Gene == "blaGES-4_1_AB116723" | Gene == "blaGES-5_1_DQ236171" 
                                       | Gene == "blaGES-6_1_AY494718" | Gene == "blaGES-14_1_GU207844" | Gene == "blaGES-16_1_HM173356" 
                                       | Gene == "blaGES-18_1_JQ028729" | Gene == "blaGES-20_1_JN596280" # carbapenemase blaGES
                                       
                                       | Gene == "blaOXA-48_1_AY236073" | Gene == "blaOXA-162_1_GU197550" | Gene == "blaOXA-181_1_CM004561" 
                                       | Gene == "blaOXA-199_1_JN704570" | Gene == "blaOXA-204_1_KP027885" | Gene == "blaOXA-232_1_JX423831" 
                                       | Gene == "blaOXA-244_1_KP659189" | Gene == "blaOXA-245_1_JX438001" | Gene == "blaOXA-247_1_JX893517" 
                                       | Gene == "blaOXA-247_1_JX893517" | Gene == "blaOXA-514_1_KU866382" | Gene == "blaOXA-515_1_KU866383" 
                                       | Gene == "blaOXA-517_1_KU878974") # blaOXA-48-like

resfinder_PHY_Cluster <- merge_phyloseq(resfinder_PHY_Cluster_1, resfinder_PHY_Cluster_2)

cols <- get_palette(c("#332288", "#117733", "#52BFAD", "#88CCEE", "#DDCC77", "#FDA4B3", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)

hospital <- factor(c("F",	"G",	"G",	"G",	"H",	"H",	"H",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"I",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"J",	"F",	"J",	"J",	"G",	"G",	"G",	"A",	"A",	"A",	"A",	"A",	"A",	"A",	"B",	"B",	"B",	"B",	"B",	"B",	"B",	"B",	"B",	"C",	"C",	"C",	"C",	"C",	"C",	"D",	"D",	"D",	"K",	"K",	"K",	"M",	"M",	"M",	"N",	"L"))

rfc <- plot_bar(resfinder_PHY_Cluster, fill = "Cluster_name") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  labs(y = expression(atop(bold("ARGs/16s rRNA")))) +
  ggtitle("Hospital wastewaters") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_Cluster)))), labels=hospital, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 18, family = "Times", angle = 0, hjust = 1, vjust = 1),
        axis.text.y = element_text(size = 16, family = "Times", angle = 0),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
  #    legend.text = element_text(size = 14, family = "Times",  face = "italic"),
  #    legend.title = element_blank(),
  #     legend.key = element_rect(size = 1, color = "white"),
  #     legend.key.size = unit(0.5, "cm"),
  #     legend.spacing.y = unit(2, "char"),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 26, family = "Times", face = "bold")) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.01), breaks=seq(0, 0.1, 0.05)) +
  facet_grid(~country, scales = "free", space = "free") +
  theme(strip.text.x= element_text(size = 16, family = "Times", hjust = 0, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))

# Save legend
#leg <- get_legend(rfc_plot)

# Convert to a ggplot and print
#as_ggplot(leg)

# Other than HWW
resfinder_PHY_Cluster_1 <- subset_taxa(resfinder_PHY, Cluster_name == "blaKPC-34_1_clust"| Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaIMP-1_1_clust" | Cluster_name == "blaOXA-397_1_clust")

resfinder_PHY_Cluster_2 <- subset_taxa(resfinder_PHY, Gene == "blaGES-2_1_AF326355" | Gene == "blaGES-4_1_AB116723" | Gene == "blaGES-5_1_DQ236171" 
                                       | Gene == "blaGES-6_1_AY494718" | Gene == "blaGES-14_1_GU207844" | Gene == "blaGES-16_1_HM173356" 
                                       | Gene == "blaGES-18_1_JQ028729" | Gene == "blaGES-20_1_JN596280" # carbapenemase blaGES
                                       
                                       | Gene == "blaOXA-48_1_AY236073" | Gene == "blaOXA-162_1_GU197550" | Gene == "blaOXA-181_1_CM004561" 
                                       | Gene == "blaOXA-199_1_JN704570" | Gene == "blaOXA-204_1_KP027885" | Gene == "blaOXA-232_1_JX423831" 
                                       | Gene == "blaOXA-244_1_KP659189" | Gene == "blaOXA-245_1_JX438001" | Gene == "blaOXA-247_1_JX893517" 
                                       | Gene == "blaOXA-247_1_JX893517" | Gene == "blaOXA-514_1_KU866382" | Gene == "blaOXA-515_1_KU866383" 
                                       | Gene == "blaOXA-517_1_KU878974") # blaOXA-48-like

resfinder_PHY_Cluster <- merge_phyloseq(resfinder_PHY_Cluster_1, resfinder_PHY_Cluster_2)
## Benin
# Feces
resfinder_PHY_feces <- subset_samples(resfinder_PHY_Cluster, alias == "BH20" | alias == "BH22" | alias == "BH24" | alias == "BH25")
df <- sample_sums(resfinder_PHY_feces)

names <- paste(resfinder_PHY_feces@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_feces, fill = "Cluster_name") 
rfc_plot1 <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  ggtitle("Benin") +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  labs(y = expression(atop(bold("ARGs/16s rRNA")))) +
      ggtitle("Benin") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_feces)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 20, family = "Times", angle = 0, face = "bold"),
        axis.text.y = element_text(size = 16, family = "Times", angle = 0),
        axis.title.y = element_text(size = 24, family = "Times"),
        axis.title.x = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 16, family = "Times", face = "bold")) +
  scale_y_continuous(limits = c(0, 0.002), labels = scales::number_format(accuracy = 0.001), breaks = seq(0, 0.002, by = 0.001)) +
  facet_grid(~plot_name, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 14, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))

# Drinking
resfinder_PHY_ben_drink <- subset_samples(resfinder_PHY_Cluster, alias == "BSE100" | alias == "BSE74" | alias == "BSE79"
                                    | alias == "BSE93"| alias ==  "BH11")
df <- sample_sums(resfinder_PHY_ben_drink)

names <- paste(resfinder_PHY_ben_drink@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_ben_drink, fill = "Cluster_name") 
rfc_plot2 <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  labs(y = expression(atop(bold("ARGs/16s rRNA")))) +
      ggtitle("Benin") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_ben_drink)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 20, family = "Times", angle = 0, face = "bold"),
        axis.text.y = element_text(size = 16, family = "Times", angle = 0),
        axis.title.x = element_blank(), 
        axis.title.y = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 16, family = "Times", face = "bold")) +
  scale_y_continuous(limits = c(0, 0.002), labels = scales::number_format(accuracy = 0.001), breaks = seq(0, 0.002, by = 0.001)) +
  facet_grid(~plot_name, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 14, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))

# other
resfinder_PHY_ben_other <- subset_samples(resfinder_PHY_Cluster, alias ==  "BH13" | alias == "BH14"| alias == "BH32" | alias == "BH52")
df <- sample_sums(resfinder_PHY_ben_other)

names <- paste(resfinder_PHY_ben_other@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_ben_other, fill = "Cluster_name") 
rfc_plot3 <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  labs(y = expression(atop(bold("ARGs/16s rRNA")))) +
    ggtitle("Benin") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_ben_other)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 20, family = "Times", angle = 0, face = "bold"),
        axis.text.y = element_text(size = 16, family = "Times", angle = 0),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 16, family = "Times", face = "bold")) +
 scale_y_continuous(limits = c(0, 0.01), labels = scales::number_format(accuracy = 0.001), breaks = seq(0, 0.01, by = 0.005)) +
  facet_grid(~plot_name, scales = "free", space = "free", 
            labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 14, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))

## Burkina Faso
# other
resfinder_PHY_BF_other <- subset_samples(resfinder_PHY_Cluster, alias == "BFH27" | alias ==  "BFH42" | alias == "BFH26")
df <- sample_sums(resfinder_PHY_BF_other)

names <- paste(resfinder_PHY_BF_other@sam_data$alias)

rfc <- plot_bar(resfinder_PHY_BF_other, fill = "Cluster_name") 
rfc_plot4 <- rfc +
  geom_bar(stat="identity", size = 0, color = NA) +
  scale_fill_manual(values = cols, labels = c("blaGES", "blaIMP", "blaKPC", "blaNDM", "blaOXA-48", "blaOXA-58", "blaVIM")) +
  labs(y = expression(atop(bold("ARGs/16s rRNA")))) +
  ggtitle("Burkina Faso") +
  scale_x_discrete(breaks=levels(factor(rownames(sample_data(resfinder_PHY_BF_other)))), labels=names, expression(bar("x"))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 20, family = "Times", angle = 0, face = "bold"),
        axis.text.y = element_text(size = 16, family = "Times", angle = 0),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 16, family = "Times", face = "bold")) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.01), breaks=seq(0, 0.02, 0.01)) +
  facet_grid(~plot_name, scales = "free", space = "free", labeller = label_wrap_gen(width = 25, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 11, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=1))

# OR
layout <- "
AAAAAA
AAAAAA
AAAAAA
BBBCCC
DDDEE#
"
p <- rfc_plot + rfc_plot1 + rfc_plot2 + rfc_plot3 + rfc_plot4 + 
  plot_layout(design = layout) + plot_annotation(tag_levels = list(c("A", "B", "C", "D", "E"))) &
  theme(plot.tag = element_text(size = 24, family = "Times"))

p_leg <- p + inset_element(leg, left = 1.65, bottom = 1, right = 1, top = 0)
p_leg

ggsave(filename = "carbapenemases_grid.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
# Taxonomy
```{r}
# 15 most abundant taxa in hospital WW in each country
metaphlan_PHY_Ben <- subset_samples(metaphlan_PHY_stat, country == "Benin")
metaphlan_PHY_BF <- subset_samples(metaphlan_PHY_stat, country == "Burkina Faso")
metaphlan_PHY_Fin <- subset_samples(metaphlan_PHY_stat, country == "Finland")

# At genus level
metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY_Ben, taxrank = "Genus")
metaphlan_PHY_Genus_abund <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_Genus), TRUE)[1:15]), metaphlan_PHY_Genus)
#tax_table(metaphlan_PHY_Genus_abund)
# At species level
metaphlan_PHY_Species <- tax_glom(metaphlan_PHY_Ben, taxrank = "Species")
metaphlan_PHY_Species_abund <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_Species), TRUE)[1:15]), metaphlan_PHY_Species)
#tax_table(metaphlan_PHY_Species_abund)

# Check out the prevalence for some clinically relevant taxa
# Stenotrophomonas_maltophilia & Acinetobacter spp. & Enterococcus & Staphylococcus
metaphlan_PHY_Species <- tax_glom(metaphlan_PHY_stat, taxrank = "Species")
metaphlan_PHY_selected <- subset_taxa(metaphlan_PHY_Species, Species == "Stenotrophomonas_maltophilia")
metaphlan_PHY_selected <- subset_samples(metaphlan_PHY_selected, sample_sums(metaphlan_PHY_selected) != 0)

metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY_stat, taxrank = "Genus")
metaphlan_PHY_selected <- subset_taxa(metaphlan_PHY_Species, Genus == "Enterococcus")
metaphlan_PHY_selected <- subset_samples(metaphlan_PHY_selected, sample_sums(metaphlan_PHY_selected) != 0)

metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY_stat, taxrank = "Genus")
metaphlan_PHY_selected <- subset_taxa(metaphlan_PHY_Genus, Genus == "Staphylococcus")
metaphlan_PHY_selected <- subset_samples(metaphlan_PHY_selected, sample_sums(metaphlan_PHY_selected) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 15)
mp <- plot_bar(metaphlan_PHY_selected, fill = "Genus") 
mp_plot <- mp +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of genera"))) +
  ggtitle("Species possibly showing intrinsic resistance to carbapenems (Metaphlan3)") +
  theme_minimal() +
  theme(
    #axis.text.x = element_blank(),
    axis.text.x = element_text(size = 9, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 19, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFDF9"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 26, family = "Times", hjust = 0, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white")) + guides(fill=guide_legend(ncol=2))
```
# Global sewage, mcr (Hendriksen et al.)
```{r}
#GlobalSewage_ResFinder <- readRDS("GlobalSewage_ResFinder.rds")
#tax_table_GlobalSewage <- data.frame(tax_table(GlobalSewage_ResFinder))
#OTU_table_GlobalSewage <- data.frame(t(otu_table(GlobalSewage_ResFinder)))
#GlobalSewage_metadata <- data.frame(GlobalSewage_ResFinder@sam_data)

# Add country as column to metadata (GlobalSewage_metadata)
#GlobalSewage_metadata$country <- rownames(GlobalSewage_metadata)
#rownames(GlobalSewage_metadata)[rownames(GlobalSewage_metadata) == "Viet Nam"] <- "Viet.Nam"
#rownames(GlobalSewage_metadata)[rownames(GlobalSewage_metadata) == "Sri Lanka"] <- "Sri.Lanka"
#rownames(GlobalSewage_metadata)[rownames(GlobalSewage_metadata) == "South Africa"] <- "South.Africa"
#rownames(GlobalSewage_metadata)[rownames(GlobalSewage_metadata) == "New Zealand"] <- "New.Zealand"
#rownames(GlobalSewage_metadata)[rownames(GlobalSewage_metadata) == "Czech Republic"] <- "Czech.Republic"
#rownames(GlobalSewage_metadata)[rownames(GlobalSewage_metadata) == "Cote d'Ivoire"] <- "Cote.d.Ivoire"

#GlobalSewage_metadata$continent <- c("Europe", "Oceania", "Europe", "Africa", "South America", "Europe", "Africa", "North America", "Africa", "Asia", "South America", "Africa", "Europe", "Europe", "Europe", "South America", "Africa", "Europe", "Africa", "Europe", "Europe", "Africa", "Europe", "Europe", "Asia", "Middle East", "Europe", "Middle East", "Europe", "Asia", "Africa", "Europe", "Europe", "Europe", "Europe", "Asia", "Europe", "Europe", "Asia", "Europe", "Oceania", "Africa", "Europe", "Asia", "South America", "Europe", "Africa", "Europe", "Asia", "Europe", "Europe", "Africa", "Europe", "Asia", "Europe", "Europe", "Africa", "Africa", "Middle East", "North America", "Asia", "Africa")

# Check
#identical(rownames(OTU_table_GlobalSewage), rownames(tax_table_GlobalSewage))
#all(colnames(OTU_table_GlobalSewage) == rownames(GlobalSewage_metadata))

#GlobalSewage_ResFinder <- phyloseq(otu_table(OTU_table_GlobalSewage, taxa_are_rows=TRUE), 
#                       tax_table(as.matrix(tax_table_GlobalSewage)), sample_data(GlobalSewage_metadata))

# All
#GlobalSewage_ResFinder_mcr <- subset_taxa(GlobalSewage_ResFinder, Gene == "mcr-3.20" | Gene == "mcr-3.4" | Gene == "mcr-1.3" | Gene == "mcr-3.8" | Gene == "mcr-1.14" | Gene == "mcr-4.3" | Gene == "mcr-3.24" | Gene == "mcr-1.13" | Gene == "mcr-3.19" | Gene == "mcr-3.15" | Gene == "mcr-8" | Gene == "mcr-3.17" | Gene == "mcr-3.11" | Gene == "mcr-3.16" | Gene == "mcr-1.2" | Gene == "mcr-1.1" | Gene == "mcr-3.6" | Gene == "mcr-4.6" | Gene == "mcr-3.21" | Gene == "mcr-3.9" | Gene == "mcr-3.25" | Gene == "mcr-2.2" | Gene == "mcr-3.2" | Gene == "mcr-3.1" | Gene == "mcr-4.4" | Gene == "mcr-3.13" | Gene == "mcr-4.2" | Gene == "mcr-1.5" | Gene == "mcr-3.12" | Gene == "mcr-3.10" | Gene == "mcr-3.23" | Gene == "mcr-4.5" | Gene == "mcr-3.3" | Gene == "mcr-3.18" | Gene == "mcr-3.5" | Gene == "mcr-4.1" | Gene == "mcr-7.1" | Gene == "mcr-3.7" | Gene == "mcr-3.14" | Gene == "mcr-5.2" | Gene == "mcr-5.1")

# Variants
#GlobalSewage_ResFinder_mcr <- subset_taxa(GlobalSewage_ResFinder, Gene == "mcr-5.1" | Gene == "mcr-5.2")

#cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 41)
#rfc <- plot_bar(GlobalSewage_ResFinder_mcr, fill = "Gene") 
#rfc_plot <- rfc +
#  geom_bar(stat="identity", color = NA, size = 0) +
#  scale_fill_manual(values = cols) +
#  labs(y = expression(bold("Relative abundance (ResFinder)"))) +
#  ggtitle("ARG Abundance normalized to 16s rRNA") +
#  theme_minimal() +
#  theme(axis.text.x = element_text(size = 10, family = "Times", face = "bold", angle = 90),
#        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
#        axis.title.y = element_text(size = 30, family = "Times"),
#        axis.title.x = element_blank(),
#        legend.text = element_text(size = 18, family = "Times"),
#        legend.title = element_blank(),
#        legend.key = element_rect(size = 2, color = "white"),
#        legend.key.size = unit(1, "cm"),
#        legend.spacing.y = unit(2.2, "char"),
#        panel.background = element_rect(fill = "#FFFCF3"),
#        panel.grid.minor = element_blank(),
#        panel.grid.major = element_blank(),
#        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
#  facet_grid(~continent, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
#  theme(strip.text.x= element_text(size = 24, family = "Times", hjust = 0.5, vjust = 0.5, angle = 0),
#        strip.background = element_rect(colour = "white"))
```
## Mantel tests
```{r}
# horn-morisita
ARG_dist <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat))), method = "horn")

MGE_dist <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat))), method = "horn")

intI1_dist <- vegdist(t(as.matrix(otu_table(MGE_PHY_int_stat))), method = "horn")

PHY_bacteria <- subset_taxa(metaphlan_PHY_stat, Kingdom %in% c("Bacteria"))
PHY_bacteria <- subset_samples(PHY_bacteria, sample_sums(PHY_bacteria) != 0)

BACT_dist <- vegdist(t(as.matrix(otu_table(PHY_bacteria))), method = "horn")

mantel(ARG_dist, MGE_dist, method = "spearman", permutations = 9999, na.rm = TRUE)
#Mantel statistic r: 0.575 
#      Significance: 1e-04 
#Upper quantiles of permutations (null model):
#   90%    95%  97.5%    99% 
#0.0834 0.1095 0.1315 0.1585 
#Permutation: free
#Number of permutations: 9999

mantel(ARG_dist, BACT_dist, method = "spearman", permutations = 9999, na.rm = TRUE)
#Mantel statistic r: 0.4672 
#      Significance: 1e-04 
#Upper quantiles of permutations (null model):
#   90%    95%  97.5%    99% 
#0.0927 0.1193 0.1440 0.1758 
#Permutation: free
#Number of permutations: 9999

mantel(MGE_dist, BACT_dist, method = "spearman", permutations = 9999, na.rm = TRUE)
#Mantel statistic r: 0.2811 
#      Significance: 1e-04 
#Upper quantiles of permutations (null model):
#   90%    95%  97.5%    99% 
#0.0816 0.1048 0.1256 0.1483 
#Permutation: free
#Number of permutations: 9999
```
## Mantel tests
# For each country
```{r, results=F}
# For each country
resfinder_PHY_stat_Ben <- subset_samples(resfinder_PHY_stat, country == "Benin")
resfinder_PHY_stat_BF <- subset_samples(resfinder_PHY_stat, country == "Burkina Faso")
resfinder_PHY_stat_Fin <- subset_samples(resfinder_PHY_stat, country == "Finland")

MGE_PHY_stat_Ben <- subset_samples(MGE_PHY_stat, country == "Benin")
MGE_PHY_stat_BF <- subset_samples(MGE_PHY_stat, country == "Burkina Faso")
MGE_PHY_stat_Fin <- subset_samples(MGE_PHY_stat, country == "Finland")

metaphlan_PHY_stat_Ben <- subset_samples(metaphlan_PHY_stat, country == "Benin")
metaphlan_PHY_stat_BF <- subset_samples(metaphlan_PHY_stat, country == "Burkina Faso")
metaphlan_PHY_stat_Fin <- subset_samples(metaphlan_PHY_stat, country == "Finland")

# Distance matrices
ARG_dist_Ben <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat_Ben))), method = "horn")
ARG_dist_BF <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat_BF))), method = "horn")
ARG_dist_Fin <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat_Fin))), method = "horn")

MGE_dist_Ben <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat_Ben))), method = "horn")
MGE_dist_BF <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat_BF))), method = "horn")
MGE_dist_Fin <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat_Fin))), method = "horn")

PHY_bacteria_Ben <- subset_taxa(metaphlan_PHY_stat_Ben, Kingdom %in% c("Bacteria"))
PHY_bacteria_Ben <- subset_samples(PHY_bacteria_Ben, sample_sums(PHY_bacteria_Ben) != 0)

PHY_bacteria_BF <- subset_taxa(metaphlan_PHY_stat_BF, Kingdom %in% c("Bacteria"))
PHY_bacteria_BF <- subset_samples(PHY_bacteria_BF, sample_sums(PHY_bacteria_BF) != 0)

PHY_bacteria_Fin <- subset_taxa(metaphlan_PHY_stat_Fin, Kingdom %in% c("Bacteria"))
PHY_bacteria_Fin <- subset_samples(PHY_bacteria_Fin, sample_sums(PHY_bacteria_Fin) != 0)

BACT_dist_Ben <- vegdist(t(as.matrix(otu_table(PHY_bacteria_Ben))), method = "horn")
BACT_dist_BF <- vegdist(t(as.matrix(otu_table(PHY_bacteria_BF))), method = "horn")
BACT_dist_Fin <- vegdist(t(as.matrix(otu_table(PHY_bacteria_Fin))), method = "horn")

# ARG-MGE
mantel(ARG_dist_Ben, MGE_dist_Ben, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_BF, MGE_dist_BF, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_Fin, MGE_dist_Fin, method = "spearman", permutations = 9999, na.rm = TRUE)

# ARG-bacteria
mantel(ARG_dist_Ben, BACT_dist_Ben, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_BF, BACT_dist_BF, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_Fin, BACT_dist_Fin, method = "spearman", permutations = 9999, na.rm = TRUE)

# MGE-bacteria
mantel(MGE_dist_Ben, BACT_dist_Ben, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(MGE_dist_BF, BACT_dist_BF, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(MGE_dist_Fin, BACT_dist_Fin, method = "spearman", permutations = 9999, na.rm = TRUE)
```
## Mantel tests
# For each country, equal sample sizes
```{r, results = F}
ARG_dist <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat_equal))), method = "horn")
MGE_dist <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat_equal))), method = "horn")

PHY_bacteria <- subset_taxa(metaphlan_PHY_stat_equal, Kingdom %in% c("Bacteria"))
PHY_bacteria <- subset_samples(PHY_bacteria, sample_sums(PHY_bacteria) != 0)

BACT_dist <- vegdist(t(as.matrix(otu_table(PHY_bacteria))), method = "horn")

mantel(ARG_dist, MGE_dist, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist, BACT_dist, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(MGE_dist, BACT_dist, method = "spearman", permutations = 9999, na.rm = TRUE)

# For each country
resfinder_PHY_stat_Ben <- subset_samples(resfinder_PHY_stat_equal, country == "Benin")
resfinder_PHY_stat_BF <- subset_samples(resfinder_PHY_stat_equal, country == "Burkina Faso")
resfinder_PHY_stat_Fin <- subset_samples(resfinder_PHY_stat_equal, country == "Finland")

MGE_PHY_stat_Ben <- subset_samples(MGE_PHY_stat_equal, country == "Benin")
MGE_PHY_stat_BF <- subset_samples(MGE_PHY_stat_equal, country == "Burkina Faso")
MGE_PHY_stat_Fin <- subset_samples(MGE_PHY_stat_equal, country == "Finland")

MGE_PHY_int_stat_Ben <- subset_samples(MGE_PHY_int_stat_equal, country == "Benin")
MGE_PHY_int_stat_BF <- subset_samples(MGE_PHY_int_stat_equal, country == "Burkina Faso")
MGE_PHY_int_stat_Fin <- subset_samples(MGE_PHY_int_stat_equal, country == "Finland")

metaphlan_PHY_stat_Ben <- subset_samples(metaphlan_PHY_stat_equal, country == "Benin")
metaphlan_PHY_stat_BF <- subset_samples(metaphlan_PHY_stat_equal, country == "Burkina Faso")
metaphlan_PHY_stat_Fin <- subset_samples(metaphlan_PHY_stat_equal, country == "Finland")

# Distance matrices
ARG_dist_Ben <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat_Ben))), method = "horn")
ARG_dist_BF <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat_BF))), method = "horn")
ARG_dist_Fin <- vegdist(t(as.matrix(otu_table(resfinder_PHY_stat_Fin))), method = "horn")

MGE_dist_Ben <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat_Ben))), method = "horn")
MGE_dist_BF <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat_BF))), method = "horn")
MGE_dist_Fin <- vegdist(t(as.matrix(otu_table(MGE_PHY_stat_Fin))), method = "horn")

PHY_bacteria_Ben <- subset_taxa(metaphlan_PHY_stat_Ben, Kingdom %in% c("Bacteria"))
PHY_bacteria_Ben <- subset_samples(PHY_bacteria_Ben, sample_sums(PHY_bacteria_Ben) != 0)

PHY_bacteria_BF <- subset_taxa(metaphlan_PHY_stat_BF, Kingdom %in% c("Bacteria"))
PHY_bacteria_BF <- subset_samples(PHY_bacteria_BF, sample_sums(PHY_bacteria_BF) != 0)

PHY_bacteria_Fin <- subset_taxa(metaphlan_PHY_stat_Fin, Kingdom %in% c("Bacteria"))
PHY_bacteria_Fin <- subset_samples(PHY_bacteria_Fin, sample_sums(PHY_bacteria_Fin) != 0)

BACT_dist_Ben <- vegdist(t(as.matrix(otu_table(PHY_bacteria_Ben))), method = "horn")
BACT_dist_BF <- vegdist(t(as.matrix(otu_table(PHY_bacteria_BF))), method = "horn")
BACT_dist_Fin <- vegdist(t(as.matrix(otu_table(PHY_bacteria_Fin))), method = "horn")

# ARG-MGE
mantel(ARG_dist_Ben, MGE_dist_Ben, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_BF, MGE_dist_BF, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_Fin, MGE_dist_Fin, method = "spearman", permutations = 9999, na.rm = TRUE)

# ARG-bacteria
mantel(ARG_dist_Ben, BACT_dist_Ben, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_BF, BACT_dist_BF, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(ARG_dist_Fin, BACT_dist_Fin, method = "spearman", permutations = 9999, na.rm = TRUE)

# MGE-bacteria
mantel(MGE_dist_Ben, BACT_dist_Ben, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(MGE_dist_BF, BACT_dist_BF, method = "spearman", permutations = 9999, na.rm = TRUE)
mantel(MGE_dist_Fin, BACT_dist_Fin, method = "spearman", permutations = 9999, na.rm = TRUE)
```
## Correlation of intI and all ARGs?
```{r}
## Benin
# ARG names
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
rownames(tax) <- paste(tax$sp, tax$n, sep="")
tax <- tax[c(-4, -5)]

args <- subset_samples(resfinder_PHY_stat, country %in% c("Benin"))
int <- subset_samples(MGE_PHY_int_stat, country %in% c("Benin"))

arg_matrix <- otu_table(args)
arg_matrix = arg_matrix[ rowSums(arg_matrix)!=0, ]

match <- match(rownames(arg_matrix), rownames(tax))
arg_tax <- tax[match,]

rownames(arg_matrix) <- arg_tax$Gene

int_matrix <- data.frame(sample_sums(otu_table(int)))
arg_matrix <- t(arg_matrix)

Benin_correl<-corr.test(arg_matrix, int_matrix, use="pairwise", method="pearson",
                            adjust="fdr",alpha=.05,ci=TRUE)

r <- data.frame(Benin_correl$r)
p <- data.frame(Benin_correl$p)
p.ad <- data.frame(Benin_correl$p.adj)

Benin_cor_data <- data.frame(r, p, p.ad)
Benin_cor_data$Gene <- rownames(Benin_cor_data)
colnames(Benin_cor_data) <- c("r", "p", "p.ad", "Gene")

Benin_cor_data_filt <- Benin_cor_data[which(Benin_cor_data$p.ad < 0.05), ]

Benin_pos <- Benin_cor_data_filt[which(Benin_cor_data_filt$r > 0), ]
Benin_neg <- Benin_cor_data_filt[which(Benin_cor_data_filt$r < 0), ]

## Burkina Faso
# ARG names
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
rownames(tax) <- paste(tax$sp, tax$n, sep="")
tax <- tax[c(-4, -5)]

args <- subset_samples(resfinder_PHY_stat, country %in% c("Burkina Faso"))
int <- subset_samples(MGE_PHY_int_stat, country %in% c("Burkina Faso"))

arg_matrix <- otu_table(args)
arg_matrix = arg_matrix[ rowSums(arg_matrix)!=0, ]

match <- match(rownames(arg_matrix), rownames(tax))
arg_tax <- tax[match,]

rownames(arg_matrix) <- arg_tax$Gene

int_matrix <- data.frame(sample_sums(otu_table(int)))
arg_matrix <- t(arg_matrix)

BF_correl<-corr.test(arg_matrix, int_matrix, use="pairwise", method="pearson",
                            adjust="fdr",alpha=.05,ci=TRUE)

r <- data.frame(BF_correl$r)
p <- data.frame(BF_correl$p)
p.ad <- data.frame(BF_correl$p.adj)

BF_cor_data <- data.frame(r, p, p.ad)
BF_cor_data$Gene <- rownames(BF_cor_data)
colnames(BF_cor_data) <- c("r", "p", "p.ad", "Gene")

BF_cor_data_filt <- BF_cor_data[which(BF_cor_data$p.ad < 0.05), ]

BF_pos <- BF_cor_data_filt[which(BF_cor_data_filt$r > 0), ]
BF_neg <- BF_cor_data_filt[which(BF_cor_data_filt$r < 0), ]

## Finland
# ARG names
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
rownames(tax) <- paste(tax$sp, tax$n, sep="")
tax <- tax[c(-4, -5)]

args <- subset_samples(resfinder_PHY_stat, country %in% c("Finland"))
int <- subset_samples(MGE_PHY_int_stat, country %in% c("Finland"))

arg_matrix <- otu_table(args)
arg_matrix = arg_matrix[ rowSums(arg_matrix)!=0, ]

match <- match(rownames(arg_matrix), rownames(tax))
arg_tax <- tax[match,]

rownames(arg_matrix) <- arg_tax$Gene

int_matrix <- data.frame(sample_sums(otu_table(int)))
arg_matrix <- t(arg_matrix)

Finland_correl<-corr.test(arg_matrix, int_matrix, use="pairwise", method="pearson",
                            adjust="fdr",alpha=.05,ci=TRUE)

r <- data.frame(Finland_correl$r)
p <- data.frame(Finland_correl$p)
p.ad <- data.frame(Finland_correl$p.adj)

Finland_cor_data <- data.frame(r, p, p.ad)
Finland_cor_data$Gene <- rownames(Finland_cor_data)
colnames(Finland_cor_data) <- c("r", "p", "p.ad", "Gene")

Finland_cor_data_filt <- Finland_cor_data[which(Finland_cor_data$p.ad < 0.05), ]

Finland_pos <- Finland_cor_data_filt[which(Finland_cor_data_filt$r > 0), ]
Finland_neg <- Finland_cor_data_filt[which(Finland_cor_data_filt$r < 0), ]

## All
# intI1
# ARG names
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
rownames(tax) <- paste(tax$sp, tax$n, sep="")
tax <- tax[c(-4, -5)]

args = resfinder_PHY_stat
int = MGE_PHY_int_stat

arg_matrix <- otu_table(args)
arg_matrix = arg_matrix[ rowSums(arg_matrix)!=0, ]

match <- match(rownames(arg_matrix), rownames(tax))
arg_tax <- tax[match,]

rownames(arg_matrix) <- arg_tax$Gene

int_matrix <- data.frame(sample_sums(otu_table(int)))
arg_matrix <- t(arg_matrix)

correl<-corr.test(arg_matrix, int_matrix, use="pairwise", method="pearson",
                            adjust="fdr",alpha=.05,ci=TRUE)
r <- data.frame(correl$r)
p <- data.frame(correl$p)
p.ad <- data.frame(correl$p.adj)

cor_data <- data.frame(r, p, p.ad)
cor_data$Gene <- rownames(cor_data)
colnames(cor_data) <- c("r", "p", "p.ad", "Gene")

cor_data_filt <- cor_data[which(cor_data$p.ad < 0.05), ]

pos <- cor_data_filt[which(cor_data_filt$r > 0), ]
neg <- cor_data_filt[which(cor_data_filt$r < 0), ]

#write.table(neg, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/neg.txt", 
#            row.names=F, sep = "\t", col.names = T)

## All
# qacEdelta
# ARG names
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
rownames(tax) <- paste(tax$sp, tax$n, sep="")
tax <- tax[c(-4, -5)]

args = resfinder_PHY_stat
qacEdelta <- tax_glom(MGE_PHY_stat, taxrank = "Class")
qacEdelta <- subset_taxa(qacEdelta, Class == "qacEdelta")

arg_matrix <- otu_table(args)
arg_matrix = arg_matrix[ rowSums(arg_matrix)!=0, ]


match <- match(rownames(arg_matrix), rownames(tax))
arg_tax <- tax[match,]

rownames(arg_matrix) <- arg_tax$Gene

qacEdelta_matrix <- data.frame(sample_sums(otu_table(qacEdelta)))
arg_matrix <- t(arg_matrix)

correl<-corr.test(arg_matrix, qacEdelta_matrix, use="pairwise", method="pearson",
                            adjust="fdr",alpha=.05,ci=TRUE)
r <- data.frame(correl$r)
p <- data.frame(correl$p)
p.ad <- data.frame(correl$p.adj)

cor_data <- data.frame(r, p, p.ad)
cor_data$Gene <- rownames(cor_data)
colnames(cor_data) <- c("r", "p", "p.ad", "Gene")

cor_data_filt <- cor_data[which(cor_data$p.ad < 0.05), ]

pos <- cor_data_filt[which(cor_data_filt$r > 0), ]
neg <- cor_data_filt[which(cor_data_filt$r < 0), ]

#write.table(pos, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/pos.txt", 
#            row.names=F, sep = "\t", col.names = T)
```
## Which taxa correlate with ARG sum (draft, takes forever to run)
```{r}
# Benin
#args <- subset_samples(resfinder_PHY_stat, country %in% c("Benin"))
#mp <- subset_samples(metaphlan_PHY_stat, country %in% c("Benin"))
#mp_genus <- tax_glom(mp, taxrank = "Genus")
#mp <- prune_taxa(names(sort(taxa_sums(mp_genus), TRUE)[1:12]), mp_genus)
#mp_genus
#arg_matrix <- otu_table(args)
#arg_matrix = arg_matrix[ rowSums(arg_matrix)!=0, ]

# ARG names
#tax <- data.frame(clusters_tax_table_resfinder)
#tax$sp <- rep("sp", times = 3104)
#tax$n <- rep(1:3104, each=1)
#colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
#rownames(tax) <- paste(tax$sp, tax$n, sep="")
#tax <- tax[c(-4, -5)]

#match <- match(rownames(arg_matrix), rownames(tax))
#arg_tax <- tax[match,]
#rownames(arg_matrix) <- arg_tax$Gene
#arg_matrix <- t(arg_matrix)

#metaphlan_PHY_stat_Genus_Ben <- tax_glom(mp, taxrank = "Genus")
#otu_matrix <- as.matrix(otu_table(mp_genus))
#otu_matrix = otu_matrix[ rowSums(otu_matrix)!=0, ]
#otu_matrix <- t(otu_matrix)

#Benin_correl_mp<-corr.test(arg_matrix, otu_matrix, use="pairwise", method="spearman", adjust="fdr",alpha=.05,ci=F)

#r <- data.frame(Benin_correl$r)
#p <- data.frame(Benin_correl$p)
#p.ad <- data.frame(Benin_correl$p.adj)

#Benin_cor_data <- data.frame(r, p, p.ad)
#Benin_cor_data$Gene <- rownames(Benin_cor_data)
#colnames(Benin_cor_data) <- c("r", "p", "p.ad", "Gene")

#Benin_cor_data_filt <- Benin_cor_data[which(Benin_cor_data$p.ad < 0.05), ]

#Benin_pos <- Benin_cor_data_filt[which(Benin_cor_data_filt$r > 0), ]
#Benin_neg <- Benin_cor_data_filt[which(Benin_cor_data_filt$r < 0), ]
```
## Figures for correlation matrix for ARGs & intI1
```{r}
intI1 <- data.frame(sample_sums(MGE_PHY_int_stat))
colnames(intI1) <- c("intI1")
qacEdelta <- data.frame(sample_sums(MGE_PHY_qac))
colnames(qacEdelta) <- c("qacEdelta")

# DESeq2: Fin-Ben
# Benin
BenFin20 <- Ben_Fin[1:20,]

pattern_Ben_Fin <- as.matrix(BenFin20$Row.names)

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_Ben_Fin, ]
all(rownames(arg_data) == BenFin20$Row.names)

rownames(arg_data) <- BenFin20$Gene
# shorten gene names
rownames(arg_data) <- gsub(pattern = "_[A-Z].*", replacement = "", rownames(arg_data))
arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

par(family="Times New Roman", cex=1.5)
cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M1 <- M[ , -c(1:20)]
p_mat1 <- p_mat[ , -c(1:20)]

# Finland
FinBen20 <- Fin_Ben[1:20,]
pattern_Fin_Ben <- as.matrix(FinBen20$Row.names)

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_Fin_Ben, ]
all(rownames(arg_data) == FinBen20$Row.names)

rownames(arg_data) <- FinBen20$Gene
# shorten gene names
rownames(arg_data) <- gsub(pattern = "_[A-Z].*", replacement = "", rownames(arg_data))
arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

par(family="Times New Roman", cex=1.5)
cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M2 <- M[ , -c(1:20)]
p_mat2 <- p_mat[ , -c(1:20)]

# DESeq2: Fin-BF
# BF
BFFin20 <- BF_Fin[1:20,]

pattern_BF_Fin <- as.matrix(BFFin20$Row.names)

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_BF_Fin, ]
all(rownames(arg_data) == BFFin20$Row.names)

rownames(arg_data) <- BFFin20$Gene
# shorten gene names
rownames(arg_data) <- gsub(pattern = "_[A-Z].*", replacement = "", rownames(arg_data))
arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M3 <- M[ , -c(1:20)]
p_mat3 <- p_mat[ , -c(1:20)]

# Finland
FinBF20 <- Fin_BF[1:20,]
pattern_Fin_BF <- as.matrix(FinBF20$Row.names)

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_Fin_BF, ]
all(rownames(arg_data) == FinBF20$Row.names)

rownames(arg_data) <- FinBF20$Gene
# shorten gene names
rownames(arg_data) <- gsub(pattern = "_[A-Z].*", replacement = "", rownames(arg_data))
arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M4<- M[ , -c(1:20)]
p_mat4 <- p_mat[ , -c(1:20)]

# Plot with ggcorrplot
# For the legend
p_mat1[is.na(p_mat1)] = 0
p_mat2[is.na(p_mat2)] = 0
p_mat3[is.na(p_mat3)] = 0
p_mat4[is.na(p_mat4)] = 0

m0 <- ggcorrplot(M1, p.mat = p_mat1, type = "full", insig = "blank", method = "square",
                         ggtheme = ggplot2::theme_classic() +
                         theme(axis.text = element_text(face = "italic", family = "Times", size = 9, angle = 20),
                         plot.title = element_text(size=9, face="bold", family = "Times"),
                         legend.title = element_blank(),
                         legend.text = element_text(family = "Times", size = 20),
                         legend.key.size = unit(1.4, "cm")))

title1 <- ggdraw() + draw_label("Differentially abundant ARGs in HWWs in Benin vs. Finland",
    fontface = 'bold', x = 0.32, hjust = 0.1, y = 0.35, fontfamily = "Times", size = 26)
title2 <- ggdraw() + draw_label("Differentially abundant ARGs in HWWs in Burkina Faso vs. Finland",
    fontface = 'bold', x = 0.35, hjust = 0.1, y = 0.35, fontfamily = "Times", size = 26)

m1 <- ggcorrplot(M1, p.mat = p_mat1, type = "full", insig = "blank", method = "square",
                 ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                         legend.position = "none", plot.margin = unit(c(0, 0, 0, 0), "cm"),
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 20),
                         axis.text.x.bottom = element_text(size = 16, angle = 35, face = "italic"),
                         plot.title = element_text(size=26, face="bold", family = "Times"))) + ggtitle("Benin")

m2 <- ggcorrplot(M2, p.mat = p_mat2, type = "full", insig = "blank", method = "square",
                 ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                         legend.position = "none", plot.margin = unit(c(0, 0, 0, 0), "cm"),
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 20),
                         axis.text.x.bottom = element_text(size = 16, angle = 35, face = "italic"),
                         plot.title = element_text(size=26, face="bold", family = "Times"))) + ggtitle("Finland")

m3 <- ggcorrplot(M3, p.mat = p_mat3, type = "full", insig = "blank", method = "square",
                 ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                         legend.position = "none", plot.margin = unit(c(0, 0, 0, 0), "cm"),
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 20),
                         axis.text.x.bottom = element_text(size = 16, angle = 35, face = "italic"),
                         plot.title = element_text(size=26, face="bold", family = "Times"))) + ggtitle("Burkina Faso")

m4 <- ggcorrplot(M4, p.mat = p_mat4, type = "full", insig = "blank", method = "square",
                 ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                          legend.position = "none", plot.margin = unit(c(0, 0, 0, 0), "cm"),
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 20),
                         axis.text.x.bottom = element_text(size = 16, angle = 35, face = "italic"),
                         plot.title = element_text(size=26, face="bold", family = "Times"))) + ggtitle("Finland")

# Extract the legend from one of the plots
legend <- get_legend(m0)

# Some inception with cowplot...
A <- plot_grid(title1, m1,m2, NULL, ncol = 1, rel_heights = c(0.5, 1, 1, 0.1))
B <- plot_grid(NULL, title2, m3,m4, ncol = 1,  rel_heights = c(0.1, 0.5, 1, 1))
AB <- plot_grid(A, B, ncol = 1)

#plot_grid(AB, legend, ncols = 2, rel_widths = c(2, 1))
AB

ggsave(filename = "ARGs_corr_deseq.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Same but for 15 most abundant ARGs
```{r}
#top_ARGs
MGE_PHY_qac <- tax_glom(MGE_PHY_stat, taxrank = "Class")
MGE_PHY_qac <- subset_taxa(MGE_PHY_qac, Class == "qacEdelta")

intI1 <- data.frame(sample_sums(MGE_PHY_int_stat))
colnames(intI1) <- c("intI1")
qacEdelta <- data.frame(sample_sums(MGE_PHY_qac))
colnames(qacEdelta) <- c("qacEdelta")

# Benin
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
tax$Row.names <- paste(tax$sp, tax$n, sep="")

pattern_Ben <- data.frame(dplyr::filter(tax,
                                  grepl("aph\\(6\\)-Id_1_M28829|aph\\(6\\)-Id_4_CP000971|aph\\(3''\\)-Ib_3_AF321550|ant\\(3''\\)-Ia_1_X02340|aph\\(3''\\)-Ib_2_AF024602|aph\\(3''\\)-Ib_5_AF321551|aph\\(6\\)-Id_2_AF024602|blaOXA-347_1_ACWG01000053|blaOXA-129_1_FJWZ01000025|blaOXA-256_1_HE616889|cmlA1_1_M64556|sul1_5_EU780013|tet\\(A\\)_6_AF534183|tet\\(C\\)_2_AY046276|tet\\(C\\)_3_AF055345", 
                                        Gene)))
pattern_Benin <- as.matrix(pattern_Ben[,6])

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_Benin, ]
all(rownames(arg_data) == pattern_Benin)

rownames(arg_data) <- pattern_Ben$Gene
rownames(arg_data)
# shorten gene names
#rownames(arg_data) <- gsub(pattern = "_[A-Z].*", replacement = "", rownames(arg_data))
rownames(arg_data) = c("aph(6)-Id_1", "aph(6)-Id_4", "aph(3'')-Ib_3", "ant(3'')-Ia_1", "aph(3'')-Ib_2", "aph(3'')-Ib_5", "aph(6)-Id_2", "blaOXA-347", "blaOXA-129", "blaOXA-256", "cmlA1", "sul1", "tet(A)_6", "tet(C)_2", "tet(C)_3")

arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M1 <- M[ , -c(1:15)]
p_mat1 <- p_mat[ , -c(1:15)]

# BF
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
tax$Row.names <- paste(tax$sp, tax$n, sep="")

pattern_BF <- data.frame(dplyr::filter(tax,
                                  grepl("aph\\(6\\)-Id_1_M28829|ant\\(3''\\)-Ia_1_X02340|aph\\(3''\\)-Ib_2_AF024602|aph\\(3''\\)-Ib_5_AF321551|blaOXA-101_1_AM412777|blaCMY-4_1_LNHZ01000079|msr\\(E\\)_1_FR751518|nimA_1_X71444|cmlA1_1_M64556|cmlA1_2_AB212941|qnrS2_1_DQ485530|sul1_5_EU780013|tet\\(39\\)_1_KT346360|tet\\(C\\)_2_AY046276|tet\\(C\\)_3_AF055345", 
                                        Gene)))
pattern_BurkinaFaso <- as.matrix(pattern_BF[,6])

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_BurkinaFaso, ]
all(rownames(arg_data) == pattern_BurkinaFaso)

rownames(arg_data) <- pattern_BF$Gene
rownames(arg_data)

# shorten gene names
#rownames(arg_data) <- gsub(pattern = "_[A-Z].*", replacement = "", rownames(arg_data))
rownames(arg_data) = c("aph(6)-Id_1", "ant(3'')-Ia_1", "aph(3'')-Ib_2", "aph(3'')-Ib_5", "blaOXA-101", "blaCMY-4", "msr(E)", "nimA", "cmlA1_1", "cmlA1_2", "qnrS2", "sul1", "tet(39)", "tet(C)_2", "tet(C)_3")

arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M2 <- M[ , -c(1:15)]
p_mat2 <- p_mat[ , -c(1:15)]

# Finland
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
tax$Row.names <- paste(tax$sp, tax$n, sep="")

pattern_Fin <- data.frame(dplyr::filter(tax,
                                  grepl("aac\\(6'\\)-Ib-Hangzhou_1_FJ503047|aph\\(6\\)-Id_1_M28829|aac\\(6'\\)-IIa_1_M29695|ant\\(3''\\)-Ia_1_X02340|aadA1b_1_M95287|blaOXA-427_1_KX827604|blaMOX-6_1_GQ152601|cfxA6_1_GQ342996|blaOXA-296_1_APOH01000009|erm\\(B\\)_18_X66468|msr\\(E\\)_1_FR751518|mph\\(E\\)_1_DQ839391|mph\\(E\\)_2_JF769133|tet\\(39\\)_1_KT346360|tet\\(Q\\)_1_L33696", 
                                        Gene)))
pattern_Finland <- as.matrix(pattern_Fin[,6])

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_Finland, ]
all(rownames(arg_data) == pattern_Finland)

rownames(arg_data) <- pattern_Fin$Gene
rownames(arg_data)
# shorten gene names
#rownames(arg_data) <- gsub(pattern = "_[A-Z].*", replacement = "", rownames(arg_data))
rownames(arg_data) = c("aac(6')-Ib-Hangzhou", "aph(6)-Id_1", "aac(6')-IIa_1", "ant(3'')-Ia_1", "aadA1b", "blaOXA-427", "blaMOX-6", "cfxA6", "blaOXA-296", "erm(B)", "msr(E)", "mph(E)_1", "mph(E)_2", "tet(39)", "tet(Q)")

arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M3 <- M[ , -c(1:15)]
p_mat3 <- p_mat[ , -c(1:15)]

# Plot with ggcorrplot
# For the legend
p_mat1[is.na(p_mat1)] = 0
p_mat2[is.na(p_mat2)] = 0
p_mat3[is.na(p_mat3)] = 0

m0 <- ggcorrplot(M1, p.mat = p_mat1, type = "lower", insig = "blank", method = "square",
                         ggtheme = ggplot2::theme_classic() +
                         theme(axis.text = element_text(face = "italic", family = "Times", size = 30, angle = 20),
                         plot.title = element_text(size=14, face="bold", family = "Times"),
                         legend.title = element_blank(),
                         legend.text = element_text(family = "Times", size = 20),
                         legend.key.size = unit(1.4, "cm")))

m1 <- ggcorrplot(M1, p.mat = p_mat1, type = "full", insig = "blank", method = "square",
                 ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                         legend.position = "none",
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 18),
                         axis.text.x.bottom = element_text(size = 18, angle = 35, face = "italic"),
                         plot.title = element_text(size=22, face="bold", family = "Times"))) +
  ggtitle("15 most abundant ARGs in Benin")
m2 <- ggcorrplot(M2, p.mat = p_mat2, type = "full", insig = "blank", method = "square",
                         ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                         legend.position = "none",
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 18),
                         axis.text.x.bottom = element_text(size = 18, angle = 35, face = "italic"),
                         plot.title = element_text(size=22, face="bold", family = "Times"))) +
  ggtitle("15 most abundant ARGs in Burkina Faso")
m3 <- ggcorrplot(M3, p.mat = p_mat3, type = "full", insig = "blank", method = "square",
                 ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                         legend.position = "none",
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 18),
                         axis.text.x.bottom = element_text(size = 18, angle = 35, face = "italic"),
                         plot.title = element_text(size=22, face="bold", family = "Times"))) +
  ggtitle("15 most abundant ARGs in Finland")

# Extract the legend from one of the plots
legend <- get_legend(m0)

# Plot
A <- plot_grid(m1,m2, m3, ncol = 1)
plot_grid(A, legend, ncol = 2, rel_widths = c(1, 0.25))

#ggsave(filename = "top15_ARGs_corr_deseq.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Same but for non-wastewater samples in West Africa
```{r}
intI1 <- data.frame(sample_sums(MGE_PHY_int_stat))
colnames(intI1) <- c("intI1")
qacEdelta <- data.frame(sample_sums(MGE_PHY_qac))
colnames(qacEdelta) <- c("qacEdelta")

# Pristine
intI1 <- data.frame(sample_sums(MGE_PHY_int_stat))
colnames(intI1) <- c("intI1")
qacEdelta <- data.frame(sample_sums(MGE_PHY_qac))
colnames(qacEdelta) <- c("qacEdelta")

tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
tax$Row.names <- paste(tax$sp, tax$n, sep="")

pattern_prist <- data.frame(dplyr::filter(tax,
                                  grepl("aadA2_1_NC_010870|aph\\(6\\)-Id_1_M28829|ant\\(2''\\)-Ia_6_AJ871915|ant\\(2''\\)-Ia_9_HM367610|ant\\(2''\\)-Ia_10_HM367617|ant\\(3''\\)-Ia_1_X02340|blaOXA-129_1_FJWZ01000025|blaOXA-256_1_HE616889|blaADC-25_1_EF016355|blaPAU-1_1_MH053445|msr\\(E\\)_1_FR751518|mph\\(E\\)_1_DQ839391|mph\\(E\\)_2_JF769133|sul1_5_EU780013|tet\\(39\\)_1_KT346360", 
                                        Gene)))
pattern_pristine <- as.matrix(pattern_prist[,6])

args <- data.frame(otu_table(resfinder_PHY_stat))
arg_data <- args[pattern_pristine, ]
all(rownames(arg_data) == pattern_pristine)

rownames(arg_data) <- pattern_prist$Gene
rownames(arg_data)

arg_data = t(arg_data)

df <- data.frame(arg_data, intI1, qacEdelta)

cor <- rcorr(as.matrix(df))
M <- cor$r
p_mat <- cor$P
M5 <- M[ , -c(1:15)]
p_mat5 <- p_mat[ , -c(1:15)]

p_mat5[is.na(p_mat5)] = 0

m5 <- ggcorrplot(M5, p.mat = p_mat5, type = "full", insig = "blank", method = "square",
                 ggtheme = ggplot2::theme_classic() +
                   theme(axis.text = element_text(face = "italic", family = "Times"),
                         legend.position = "none",
                         axis.text.y.left = element_text(angle = 0, face = "bold.italic", size = 18),
                         axis.text.x.bottom = element_text(size = 18, angle = 35, face = "italic"),
                         plot.title = element_text(size=22, face="bold", family = "Times"))) +
  ggtitle("15 most abundant ARGs in pristine samples")
```
# "Core" resistome and unique ARGs
```{r}
Ben_temp <- otu_table(subset_samples(resfinder_PHY_stat, country %in%  c("Benin")))[rowSums(otu_table(subset_samples(resfinder_PHY_stat, country %in% c("Benin")))) > 0]
nrow(Ben_temp) # 1738

BF_temp <- otu_table(subset_samples(resfinder_PHY_stat, country %in%  c("Burkina Faso")))[rowSums(otu_table(subset_samples(resfinder_PHY_stat, country %in% c("Burkina Faso")))) > 0]
nrow(BF_temp) # 2131

Fin_temp <- otu_table(subset_samples(resfinder_PHY_stat, country %in%  c("Finland")))[rowSums(otu_table(subset_samples(resfinder_PHY_stat, country %in% c("Finland")))) > 0]
nrow(Fin_temp) # 1555

length(intersect(row.names(Ben_temp), (row.names(BF_temp)))) # 1664
length(intersect(row.names(BF_temp), (row.names(Fin_temp)))) # 1414
length(intersect(row.names(Ben_temp), (row.names(Fin_temp)))) # 1295

grid.newpage()
ven.p <- draw.triple.venn(area1 = nrow(Ben_temp), area2 = nrow(BF_temp), area3 = nrow(Fin_temp), 
                 n12 = length(intersect(row.names(Ben_temp), (row.names(BF_temp)))), 
                 n23 = length(intersect(row.names(BF_temp), (row.names(Fin_temp)))), 
                 n13 = length(intersect(row.names(Ben_temp), (row.names(Fin_temp)))), 
                 n123 = length(intersect(intersect(row.names(Ben_temp), (row.names(BF_temp))), row.names(Fin_temp))), 
    fontfamily = "Times", category = c("Benin", "Burkina Faso", "Finland"),
    lty = "blank", fill = c("#B2182B", "#44AA99", "#2166AC"),
    alpha = 0.75,  cex = 4.5, cat.cex = 6, rotation.degree = 0, label.col = "white", cat.dist = 0.05,
    filename = "Venn_diagram.png", output=TRUE, imagetype="png", margin = 0.08)
grid.draw(ven.p)

# Pairwise
grid.newpage()
p1 <- draw.pairwise.venn(area1 = nrow(Ben_temp), area2 = nrow(BF_temp), 
                            cross.area = length(intersect(row.names(Ben_temp), (row.names(BF_temp)))),
    fontfamily = "Times", category = c("Benin", "Burkina Faso"),
    lty = "blank", fill = c("#B2182B", "#44AA99"),
    alpha = 0.75,  cex = 5, cat.cex = 6, rotation.degree = 0, label.col = "white", 
    cat.pos = c(300, 105), cat.dist = c(-0.07, -0.06),
                                ext.pos = 10,
                                ext.dist = -0.05,
                                ext.length = 0.75,
                                ext.line.lwd = 2,
                                ext.line.lty = "dashed")
grid.draw(p1)

grid.newpage()
p2 <- draw.pairwise.venn(area1 = nrow(Fin_temp), area2 = nrow(BF_temp), 
                            cross.area = length(intersect(row.names(Fin_temp), (row.names(BF_temp)))),
    fontfamily = "Times", category = c("Finland", "Burkina Faso"),
    lty = "blank", fill = c("#2166AC", "#44AA99"),
    alpha = 0.65,  cex = 4, cat.cex = 4, rotation.degree = 0, label.col = "white", cat.dist = 0.02,
    filename = "Venn_diagram.png", output=TRUE, imagetype="png", margin = 0.08)
grid.draw(p2)

grid.newpage()
p3 <- draw.pairwise.venn(area1 = nrow(Ben_temp), area2 = nrow(Fin_temp), 
                            cross.area = length(intersect(row.names(Ben_temp), (row.names(Fin_temp)))),
    fontfamily = "Times", category = c("Benin", "Finland"),
    lty = "blank", fill = c("#B2182B", "#2166AC"),
    alpha = 0.75,  cex = 6, cat.cex = 4, rotation.degree = 0, label.col = "white", cat.dist = 0.03,
    filename = "Venn_diagram.png", output=TRUE, imagetype="png", margin = 0.08, euler.d = TRUE, scaled = TRUE)
grid.draw(p3)

# And which ARGs are those?
tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
rownames(tax) <- paste(tax$sp, tax$n, sep="")
tax <- tax[c(-4, -5)]

match <- match(rownames(Ben_temp), rownames(tax))
Ben_names <- tax[match,]

match <- match(rownames(BF_temp), rownames(tax))
BF_names <- tax[match,]

match <- match(rownames(Fin_temp), rownames(tax))
Fin_names <- tax[match,]

#write.table(Ben_names, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/counts_Ben.txt", row.names=F, sep = "\t", col.names = T)

#write.table(BF_names, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/counts_BF.txt", row.names=F, sep = "\t", col.names = T)

#write.table(Fin_names, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/counts_Fin.txt", row.names=F, sep = "\t", col.names = T)

# What about the unique ARGs?
# Core
counts <- data.frame(otu_table(resfinder_PHY_stat))
counts[counts > 0] <- 1
core <- counts[rowSums(counts)==67,]

tax <- data.frame(clusters_tax_table_resfinder)
tax$sp <- rep("sp", times = 3104)
tax$n <- rep(1:3104, each=1)
colnames(tax) <- c("Class", "Cluster_name", "Gene", "sp", "n")
rownames(tax) <- paste(tax$sp, tax$n, sep="")
tax <- tax[c(-4, -5)]

match <- match(rownames(core), rownames(tax))
core_names <- tax[match,]

#write.table(core_names, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/core_names.txt", row.names=F, sep = "\t", col.names = T)

# Unique for Benin
temp1 <- intersect(row.names(Ben_temp), row.names(Fin_temp))
temp2 <- intersect(row.names(Ben_temp), row.names(BF_temp))
temp <- c(temp1, temp2)
temp <- data.frame(temp)
temp <- data.frame(unique(temp))
rownames(temp) <- temp$temp

unique_Ben <- data.frame(names = outersect(rownames(temp), rownames(Ben_temp)))
rownames(unique_Ben) <- unique_Ben$names

match <- match(rownames(unique_Ben), rownames(tax))
unique_Ben <- tax[match,]

#write.table(unique_Ben, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/unique_Ben.txt",  row.names=F, sep = "\t", col.names = T)

# BF
temp1 <- intersect(row.names(BF_temp), row.names(Fin_temp))
temp2 <- intersect(row.names(BF_temp), row.names(Ben_temp))
temp <- c(temp1, temp2)
temp <- data.frame(temp)
temp <- data.frame(unique(temp))
rownames(temp) <- temp$temp

unique_BF <- data.frame(names = outersect(rownames(temp), rownames(BF_temp)))
rownames(unique_BF) <- unique_BF$names

match <- match(rownames(unique_BF), rownames(tax))
unique_BF <- tax[match,]

#write.table(unique_BF, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/unique_BF.txt", row.names=F, sep = "\t", col.names = T)

# Finland
temp1 <- intersect(row.names(Fin_temp), row.names(BF_temp))
temp2 <- intersect(row.names(Fin_temp), row.names(Ben_temp))
temp <- c(temp1, temp2)
temp <- data.frame(temp)
temp <- data.frame(unique(temp))
rownames(temp) <- temp$temp

unique_Fin <- data.frame(names = outersect(rownames(temp), rownames(Fin_temp)))
rownames(unique_Fin) <- unique_Fin$names

match <- match(rownames(unique_Fin), rownames(tax))
unique_Fin <- tax[match,]

#write.table(unique_Fin, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/unique_Fin.txt", row.names=F, sep = "\t", col.names = T)
```





