---
title: "R_scripts"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Set working directory
```{r}
setwd("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles")
```
## Load libraries
```{r, warning=FALSE, error=FALSE, message=FALSE}
library(phyloseq)
library(dplyr)
library(viridis)
library(stringr)
library(vegan)
library(RColorBrewer)
library(BiocManager)
BiocManager::install("microbiome")
library(microbiome)
library(microbiomeutilities)
library(ggplot2)
library(knitr)
library(ggpubr)
library(pheatmap)
```
## Load metadata
```{r}
metadata <-read.table("metadata.txt", sep="\t", header = T, row.names = 1, fill = 1)
```
# Metaxa2
```{r}
# Load data
metaxa_genus <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaxa_genus.txt")

# Create OTU table
OTU_metaxa <- metaxa_genus[,-1]

# Create tax table
tax_table_metaxa <- data.frame(str_split_fixed(data.frame(metaxa_genus) [,1], ";", 6))
colnames(tax_table_metaxa) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus")

# Combine into phyloseq object
metaxa_PHY <- phyloseq(otu_table(OTU_metaxa, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaxa)), sample_data(metadata))
```
## Add SSU counts into metadata
```{r}
metadata$SSU_counts <- sample_sums(metaxa_PHY)
```
## Filter samples
```{r}
# With low quality (BFH24_S142, BH63_S118, FH10_S171)
metaxa_PHY = subset_samples(metaxa_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Feces samples (BH20_S88, BH22_S89, BH24_S90, BH25_S91)
metaxa_PHY_ww = subset_samples(metaxa_PHY, name!="BH20_S88" & name!="BH22_S89" & name!="BH24_S90" & name!="BH25_S91")

# Clear taxa that is not present in any of the remaining samples
any(taxa_sums(metaxa_PHY_ww) == 0)
metaxa_PHY <- prune_taxa(taxa_sums(metaxa_PHY_ww) > 0, metaxa_PHY_ww)
any(taxa_sums(metaxa_PHY) == 0)
```
## Ordinate Metaxa2 result to study wheather the taxonomy can be explained by the variable "country"
```{r}
# Change into relative data
metaxa_rel_PHY <- transform_sample_counts(metaxa_PHY, function(x) x/sum(x))

# Ordinate and plot data
metaxa_rel_PHY_ord <- ordinate(metaxa_rel_PHY, method = "PCoA", distance = "horn")
p1 <- plot_ordination(metaxa_rel_PHY, metaxa_rel_PHY_ord, color = "country", label = "name")
metaxa.p1 <- p1 + scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) + geom_point(colour = "black", 
    pch = 21, size = 2, alpha = 0.5) + stat_ellipse(level = 0.9, linetype = 1) +
    theme_minimal() + labs(title = "Metaxa2")
metaxa.p1

# Test significance using pair-wise adonis
metaxa_temp <- subset_samples(metaxa_PHY, (country == "Benin" | country == "Finland"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

metaxa_temp <- subset_samples(metaxa_PHY, (country == "Burkina Faso" | country == "Finland"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

metaxa_temp <- subset_samples(metaxa_PHY, (country == "Benin" | country == "Burkina Faso"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

# The explanatory variable "country" explains some (~15.7 %) of the difference in the grouping of taxonomy in the comparison between of Benin and Finland.
```
# Metaphlan3
```{r}
# Modify data in command-line to match sample names in metadata file
## tr '|' ';' <merged_abundance_table.txt > mod_merged_abundance_table.txt
## sed -i 's/_profile//g' mod_merged_abundance_table.txt
## sed -i 's/BFH38-A_S156/BFH38.A_S156/g' mod_merged_abundance_table.txt
## sed -i 's/BFH38-B_S157/BFH38.B_S157/g' mod_merged_abundance_table.txt
## sed -i 's/BH34-A_S98/BH34.A_S98/g' mod_merged_abundance_table.txt
## sed -i 's/BH34-B_S99/BH34.B_S99/g' mod_merged_abundance_table.txt

# Load data
metaphlan <- as.matrix(read.table("mod_merged_abundance_table.txt", fill = 1, header = T, check.names = F))

# Exclude the first column "NCBI_tax_id"
metaphlan <- subset(metaphlan, select = -c(NCBI_tax_id))

# Create OTU table
OTU_metaphlan <- metaphlan[,-1]

# Change values as numeric
class(OTU_metaphlan) <- "numeric"

# Create tax_table
tax_table_metaphlan <- data.frame(str_split_fixed(data.frame(metaphlan) [,1], ";", 7))
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# Clean "*__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

# Combine into phyloseq object
metaphlan_PHY <- phyloseq(otu_table(OTU_metaphlan, taxa_are_rows = TRUE), tax_table(as.matrix(tax_table_metaphlan)), sample_data(metadata))
```
## Filter smaples
```{r}
# Virus
metaphlan_PHY <- subset_taxa(metaphlan_PHY, Kingdom != "Virus")

# With low quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_PHY = subset_samples(metaxa_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Feces samples (BH20_S88, BH22_S89, BH24_S90, BH25_S91)
metaphlan_PHY_ww = subset_samples(metaphlan_PHY, name!="BH20_S88" & name!="BH22_S89" & name!="BH24_S90" & name!="BH25_S91")

# Clear taxa that is not present in any of the remaining samples
any(taxa_sums(metaphlan_PHY_ww) == 0)
metaxa_PHY <- prune_taxa(taxa_sums(metaphlan_PHY_ww) > 0, metaphlan_PHY_ww)
any(taxa_sums(metaphlan_PHY) == 0)
```
## Ordinate Metaphlan3 results to study wheather the taxonomy can be explained by the variable "country"
```{r}
# Ordinate and plot data
metaphlan_PHY_ord <- ordinate(metaphlan_PHY, method = "PCoA", distance = "horn")
p2 <- plot_ordination(metaphlan_PHY, metaphlan_PHY_ord, color = "country", label = "name")
metaphlan.p2 <- p2 + scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) + 
    geom_point(colour = "black", pch = 21, size = 2, alpha = 0.5) + stat_ellipse(level = 0.9, linetype = 1) +
    theme_minimal() + labs(title = "Metaphlan3")
metaphlan.p2

# Test significance using pair-wise adonis
metaphlan_temp <- subset_samples(metaphlan_PHY, (country == "Benin" | country == "Finland"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY, (country == "Burkina Faso" | country == "Finland"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY, (country == "Burkina Faso" | country == "Benin"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

# The results are almost identical to the ones from Metaxa2 analysis: The explanatory variable "country" explains some (~15.7 %) of the difference in the grouping of taxonomy in the comparison between of Benin and Finland.
```
# ResFinder
## Load data
```{r}
# Modify mapping output file "ARG_genemat.txt" in command line to match sample names in metadata file
## sed 's/BFH38-A_S156/BFH38.A_S156/g' ARG_genemat.txt > mod_ARG_genemat.txt
## sed -i 's/BFH38-B_S157/BFH38.B_S157/g' mod_ARG_genemat.txt
## sed -i 's/BH34-A_S98/BH34.A_S98/g' mod_ARG_genemat.txt
## sed -i 's/BH34-B_S99/BH34.B_S99/g' mod_ARG_genemat.txt

ARG_genemat <-as.matrix(read.table("mod_ARG_genemat.txt", header= T, check.names = F, row.names = 1))
```
## Load ResFinder gene lengths
```{r}
# Modify "resfinder.fasta" file so that only hits remain
## seqkit grep -f ARG_genes.txt resfinder.fasta > filtered_resfinder.fasta
# Check if there are correct number of lines
## grep ">" filtered_resfinder.fasta | wc -l
# Print out the gene lengths of these genes into file "lengths_resfinder.txt"
## cat filtered_resfinder.fasta | awk '$0 ~ ">" {if (NR > 1) {print c;} c=0;printf substr($0,2,100) "\t"; } $0 !~ ">" {c+=length($0);} END { print c; }' > lengths_resfinder.txt
# Reorder in excel to match with file "ARG_genemat_norm"
lengths_resfinder <-as.matrix(read.table("lengths_resfinder.txt", header= F, check.names = T, row.names = 1))
colnames(lengths_resfinder) <- c("Length")
```
## Normalization with gene lengths
```{r}
# Divide by ResFinder hit gene lengths
ARG_length_norm <- ARG_genemat/lengths_resfinder[, 1]

# Divide by SSU counts and normalize to bacterial 16S rRNA length (1550)
ARG_genemat_norm <- t(t(ARG_length_norm)/metadata$SSU_counts) * 1550

# Check if correct:
identical(ARG_genemat[2020, 4]/metadata$SSU_counts[4], ARG_genemat_norm[2020, 4])

# Save and load again to exclude row.names
write.table(ARG_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)

ARG_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=NULL)
ARG_genemat_norm$row.names<-NULL
```
## Normalize ARG counts to SSU
### Do not run if the above is run
```{r}
ARG_genemat_norm <- t(t(ARG_genemat)/metadata$SSU_counts)

# Check if correct:
identical(ARG_genemat[2020, 4]/metadata$SSU_counts[10], ARG_genemat_norm[2020, 4])

# Save and load again to exclude row.names
write.table(ARG_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)

ARG_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=NULL)
ARG_genemat_norm$row.names<-NULL
```
## Create phyloseq object
```{r}
# Create tax table
tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_resfinder.txt", header=FALSE, sep=";")
colnames(tax_table_resfinder) <- c("Gene", "Class")

# Combine to phyloseq object
resfinder_PHY <- phyloseq(otu_table(ARG_genemat_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_resfinder)))
```
## Filter samples
```{r}
# With low quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY = subset_samples(resfinder_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Feces samples (BH20_S88, BH22_S89, BH24_S90, BH25_S91)
resfinder_PHY_ww = subset_samples(resfinder_PHY, name!="BH20_S88" & name!="BH22_S89" & name!="BH24_S90" & name!="BH25_S91")

# Check if there are taxa that are not present in any of the remaining samples
any(taxa_sums(resfinder_PHY_ww) == 0)
resfinder_PHY <- prune_taxa(taxa_sums(resfinder_PHY_ww) > 0, resfinder_PHY_ww)
any(taxa_sums(resfinder_PHY) == 0)
```
## Ordinate ResFinder mapping results to study wheather the taxonomy can be explained by the variable "country"
```{r}
# Ordinate and plot data
resfinder_PHY_ord <- ordinate(resfinder_PHY, method = "PCoA", distance = "horn")
p3 <- plot_ordination(resfinder_PHY, resfinder_PHY_ord, color = "country", label = "name")
resfinder.p3 <- p3 + scale_color_manual(values=c("#FF333F", "#35E0F5", "#531592")) + 
    geom_point(colour = "black", pch = 21, size = 2, alpha = 0.5) + stat_ellipse(level = 0.9, linetype = 1) +
    theme_minimal() + labs(title = "ResFinder")
resfinder.p3

# Test significance between all pairs
resfinder_temp <- subset_samples(resfinder_PHY, (country == "Finland" | country == "Benin"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

resfinder_temp <- subset_samples(resfinder_PHY, (country == "Burkina Faso" | country == "Benin"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

resfinder_temp <- subset_samples(resfinder_PHY, (country == "Finland" | country == "Burkina Faso"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

# The explanatory variable "country" explains some (~10.1 % [Normalized by gene lengths: ~10.2 %]) of the difference in the grouping of taxonomy in the comparison between of Benin and Finland and some (~12.5 % [Normalized by gene lengths: ~12.9 %]) between Burkina Faso and Finland.
```
# Diversity indexes by Metaxa2
```{r}
# Take a look at the indexes (now not the relative data, we'll see...)
metaxa_tab <-microbiome::alpha(metaxa_PHY, index = "all")
kable(head(metaxa_tab))

# Create data table
metaxa.meta <- meta(metaxa_PHY)
kable(head(metaxa.meta))

# Add indexes to data table
## Shannon
metaxa.meta$diversity_shannon <- metaxa_tab$diversity_shannon
## Gini Simpson
metaxa.meta$diversity_gini_simpson <- metaxa_tab$diversity_gini_simpson
## Inverse Simpson
metaxa.meta$diversity_inverse_simpson <- metaxa_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(metaxa.meta$diversity_shannon)
shapiro.test(metaxa.meta$diversity_shannon) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaxa.meta$diversity_shannon)

hist(metaxa.meta$diversity_gini_simpson)
shapiro.test(metaxa.meta$diversity_gini_simpson) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaxa.meta$diversity_gini_simpson)

hist(metaxa.meta$diversity_inverse_simpson)
shapiro.test(metaxa.meta$diversity_inverse_simpson)
qqnorm(metaxa.meta$diversity_inverse_simpson)

# Create a list of pairwise comparisons
div.country <- levels(metaxa.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])
print(country.pairs)

# Plot
## Shannon
p4 <- ggviolin(metaxa.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p4 <- p4 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p4)

## Gini Simpson
p5 <- ggviolin(metaxa.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p5 <- p5 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p5)

# The probability that two individuals randomly selected from a sample will belong to same species is higher in Finland than in Benin.

## Inverse Simpson
p6 <- ggviolin(metaxa.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p6 <- p6 + stat_compare_means(comparisons = country.pairs, method = "t.test") 
print(p6)

# The alpha diversity is the highest in Benin compared to Burkina Faso and Finland. Of Burkina Faso and Finland the diversity is higher in Burkina Faso. 
```
# Diversity indexes by Metaphlan3
```{r}
# Take a look at the indexes
metaphlan_tab <-microbiome::alpha(metaphlan_PHY, index = "all")
kable(head(metaphlan_tab))

# Create data table
metaphlan.meta <- meta(metaphlan_PHY)
kable(head(metaxa.meta))

# Add indexes to data table
## Shannon
metaphlan.meta$diversity_shannon <- metaphlan_tab$diversity_shannon
## Gini Simpson
metaphlan.meta$diversity_gini_simpson <- metaphlan_tab$diversity_gini_simpson
## Inverse Simpson
metaphlan.meta$diversity_inverse_simpson <- metaphlan_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(metaphlan.meta$diversity_shannon)
shapiro.test(metaphlan.meta$diversity_shannon) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaphlan.meta$diversity_shannon)

hist(metaphlan.meta$diversity_gini_simpson)
shapiro.test(metaphlan.meta$diversity_gini_simpson) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(metaphlan.meta$diversity_gini_simpson)

hist(metaphlan.meta$diversity_inverse_simpson)
shapiro.test(metaphlan.meta$diversity_inverse_simpson) 
qqnorm(metaxa.meta$diversity_inverse_simpson)

# Plot
## Shannon
p7 <- ggviolin(metaphlan.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p7 <- p7 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p7)

## Gini Simpson
p8 <- ggviolin(metaphlan.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p8 <- p8 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p8)

# The outlier samole in here is FH5_S166

## Inverse Simpson
p9 <- ggviolin(metaphlan.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p9 <- p9 + stat_compare_means(comparisons = country.pairs, method = "t.test") 
print(p9)

# The results are identical to those from Metaxa2.
```
# Diversity indexes by ResFinder
```{r}
# Shannon
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY, index = "diversity_shannon")
kable(head(resfinder_tab))

## Create data table
resfinder.meta <- meta(resfinder_PHY)
kable(head(resfinder.meta))

## Add indexes to data table
resfinder.meta$diversity_shannon <- resfinder_tab$diversity_shannon

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_shannon)
shapiro.test(resfinder.meta$diversity_shannon) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(resfinder.meta$diversity_shannon)

# Plot
p10 <- ggviolin(resfinder.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p10 <- p10 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p10)

# Gini Simpson
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY, index = "diversity_gini_simpson")
kable(head(resfinder_tab))

## Add indexes to data table
resfinder.meta$diversity_gini_simpson <- resfinder_tab$diversity_gini_simpson

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_gini_simpson)
shapiro.test(resfinder.meta$diversity_gini_simpson) # Does not fit the assumption of normal distribution > wilcox test.
qqnorm(resfinder.meta$diversity_gini_simpson)

# Plot
p11 <- ggviolin(resfinder.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p11 <- p11 + stat_compare_means(comparisons = country.pairs, method = "wilcox.test") 
print(p11)

# The outlier sample here is FH6-S167

# Inverse Simpson
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY, index = "diversity_inverse_simpson")
kable(head(resfinder_tab))

## Add indexes to data table
resfinder.meta$diversity_inverse_simpson <- resfinder_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(metaphlan.meta$diversity_inverse_simpson)
shapiro.test(metaphlan.meta$diversity_inverse_simpson) 
qqnorm(metaxa.meta$diversity_inverse_simpson)

# Plot
p12 <- ggviolin(resfinder.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#FF333F", "#35E0F5", "#531592")) 
p12 <- p12 + stat_compare_means(comparisons = country.pairs, method = "t.test") 
print(p12)

# The alpha diversity of resistance genes detected by ResFinder seems slightly higher in Burkina Faso than in Benin and in Finland.
```
# Taxonomic composition by Metaxa2 using microbiome package
```{r}
# Load data
metaxa.com <- metaxa_PHY
taxic_metaxa <- as.data.frame(metaxa.com@tax_table) 

# Add OTU ids
taxic_metaxa$OTU <- rownames(taxic_metaxa) 
colnames(taxic_metaxa)

# Convert into a matrix.
taxmat_metaxa <- as.matrix(taxic_metaxa)

# Convert into phyloseq compatible file.
new.tax_metaxa <- tax_table(taxmat_metaxa)  

# Incroporate into new phyloseq object
tax_table(metaxa.com) <- new.tax_metaxa

# Class level
metaxa_top6class <- aggregate_top_taxa(metaxa.com, top = 6, "Class")

metaxa_top6class.rel <- microbiome::transform(metaxa_top6class, "compositional")

metaxa_top6class.p13 <- plot_composition(metaxa_top6class.rel, group_by = "country") + 
  theme(legend.position = "bottom") + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, size = 8), axis.title = element_blank()) +
    ggtitle("Relative abundance of top 6 classes, Metaxa2")

metaxa_top6class.p13 + 
    scale_fill_manual(values = c("#1B9E77","#D95F02","#7570B3","#E7298A","#66A61E","#E6AB02", "#FFFFFF"))
```
# Top 12 genera merged by location, Metaxa2
```{r}
metaxa_PHY_genus <- tax_glom(metaxa_PHY, "Genus")

metaxa_PHY_genus_abun <- prune_taxa(names(sort(taxa_sums(metaxa_PHY_genus), TRUE)[1:12]), 
    metaxa_PHY_genus)

# Transform_1
metaxa_PHY_genus_abun_trans <- transform_sample_counts(metaxa_PHY_genus_abun, function(x) x / sum(x))

# Merge by location
mrg_loc <- merge_samples(metaxa_PHY_genus_abun_trans, "location")

# Transform_2
mrg_loc_trans <- transform_sample_counts(mrg_loc, function(x) x / sum(x))

# Normalize by sample number / location
otu_table(mrg_loc_trans) <- otu_table(mrg_loc_trans)[, ]/as.matrix(table(sample_data(metaxa_PHY)$location))[, 
    1]

# Set colors
nb.cols <- 12
mycolors <- colorRampPalette(brewer.pal(8, "Dark2"))(nb.cols)
# Facet labs
country.labs <- c("1" = "Benin", "2" = "Burkina Faso", "3" = "Finland")

# Plot
metaxa.p14 <- plot_bar(mrg_loc_trans, fill = "Genus", title = "12 most abundant genera") + 
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country = country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=75, hjust=1, size = 10),
        axis.title.x = element_blank()) +
        ylab("Relative abundance (%)") + xlab("") +
  scale_fill_manual(values=mycolors) +
  guides(fill = guide_legend(label.theme = element_text(size = 8), keywidth = 1.8, keyheight = 1.8, title = "Genus"))
metaxa.p14
```
# Taxonomic composition by Metaphlan3 using microbiome package
```{r}
# Load data
metaphlan.com <- metaphlan_PHY
taxic_metaphlan <- as.data.frame(metaphlan.com@tax_table) 

# Add OTU ids
taxic_metaphlan$OTU <- rownames(taxic_metaphlan) 
colnames(taxic_metaphlan)

# Convert into a matrix.
taxmat_metaphlan <- as.matrix(taxic_metaphlan)

# Convert into phyloseq compatible file.
new.tax_metaphlan <- tax_table(taxmat_metaphlan)  

# Incroporate into new phyloseq object
tax_table(metaphlan.com) <- new.tax_metaphlan

# Class level
metaphlan_top6class <- aggregate_top_taxa(metaphlan.com, top = 6, "Class")

metaphlan_top6class.rel <- microbiome::transform(metaphlan_top6class, "compositional")

metaphlan_top6class.p15 <- plot_composition(metaphlan_top6class.rel, group_by = "country") + 
  theme(legend.position = "bottom") + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, size = 8), axis.title = element_blank()) +
    ggtitle("Relative abundance of top 6 classes, Metaphlan3")

metaphlan_top6class.p15 + 
    scale_fill_manual(values = c("#1B9E77","#D95F02","#7570B3","#E7298A","#66A61E","#E6AB02", "#FFFFFF"))

# The relative abundance of top 6 classes seems identical between the two methods Metaxa2 and Metaphlan.
# Deltaproteobacteria seems more common in Beninise samples compared to BF and Finland. In which instead the proportion of Gammaproteobacteria is higher. Betaproteobacteria are present in lower proportion in Finnish samples.
```
# Top 12 genera merged by location, Metaphlan3
```{r}
metaphlan_PHY_genus <- tax_glom(metaphlan_PHY, "Genus")

metaphlan_PHY_genus_abun <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_genus), TRUE)[1:12]), 
    metaphlan_PHY_genus)

# Transform_1
metaphlan_PHY_genus_abun_trans <- transform_sample_counts(metaphlan_PHY_genus_abun, function(x) x / sum(x))

# Merge by location
mrg_loc <- merge_samples(metaxa_PHY_genus_abun_trans, "location")

# Transform_2
mrg_loc_trans <- transform_sample_counts(mrg_loc, function(x) x / sum(x))

# Normalize by sample number / location
otu_table(mrg_loc_trans) <- otu_table(mrg_loc_trans)[, ]/as.matrix(table(sample_data(metaxa_PHY)$location))[, 
    1]

# Set colors
nb.cols <- 12
mycolors <- colorRampPalette(brewer.pal(8, "Dark2"))(nb.cols)
# Facet labs
country.labs <- c("1" = "Benin", "2" = "Burkina Faso", "3" = "Finland")

# Plot
metaphlan.p16 <- plot_bar(mrg_loc_trans, fill = "Genus", title = "12 most abundant genera") + 
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country = country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=75, hjust=1, size = 10),
        axis.title.x = element_blank()) +
        ylab("Relative abundance (%)") + xlab("") +
  scale_fill_manual(values=mycolors) +
  guides(fill = guide_legend(label.theme = element_text(size = 8), keywidth = 1.8, keyheight = 1.8, title = "Genus"))
metaphlan.p16
```
# DRAFT Heatmap of Metaxa2 results
```{r}
heat.sample <- plot_taxa_heatmap(metaxa_PHY,
                                 subset.top = 50,
                                 taxonomic.level = "Genus",
                                 VariableA = "country",
                                 heatcolors = rev(brewer.pal(12, "GnBu")),
                                 transformation = "log10",
                                 cluster_cols = F)
heat.sample
```
# DRAFT Boxplot of relative abundance of top 6 class
```{r}
pn <- plot_taxa_boxplot(metaxa_PHY,
                        taxonomic.level = "Class",
                        top.otu = 6, VariableA = "country",
                        title = "Relative abudance plot", color = "Dark2")
print(pn)
```





