---
title: "R_scripts_new_samples"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Set working directory
```{r}
setwd("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles")
```
## Load required libraries
```{r, warning=FALSE, error=FALSE, message=FALSE, results='hide'}
library(tidyverse)
library(phyloseq)
library(viridis)
library(stringr)
library(vegan)
library(RColorBrewer)
library(BiocManager)
#BiocManager::install("microbiome")
library(microbiome)
#library(microbiomeutilities)
library(ggplot2)
library(knitr)
library(ggpubr)
library(pheatmap)
library(MASS)
library(gplots)
library(grid)
library(cowplot)
library(ggpubr)
library(DESeq2)
library(plyr)
library(multcomp)
library(randomForest)
```
## Load metadata
```{r}
metadata <-read.table("metadata.txt", sep="\t", header = T, row.names = 1, fill = 1)
```
## Load Metaxa2 results and create phyloseq object
```{r}
metaxa_genus <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/metaxa_genus.txt")

# Create OTU table
OTU_metaxa <- metaxa_genus[,-1]

# Match sample ID order with metadata file
match <- match(rownames(metadata), colnames(OTU_metaxa))
OTU_metaxa <- OTU_metaxa[,match]
all(colnames(OTU_metaxa) == rownames(metadata))

# Create tax table
tax_table_metaxa <- data.frame(str_split_fixed(data.frame(metaxa_genus) [,1], ";", 6))
colnames(tax_table_metaxa) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus")

# Check if samples are in order
identical(rownames(metadata), colnames(OTU_metaxa))

# Combine into phyloseq object
metaxa_PHY <- phyloseq(otu_table(OTU_metaxa, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaxa)), sample_data(metadata))
# Exclude taxa "Unknown"
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown"))
# Exclude taxa "Unclassified"
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Unclassified"))
# Exclude taxa "Eukaryota"
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Eukaryota"))

## Sums for SSU count
# Exclude taxa "Unknown"
metaxa_PHY_SSU <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown"))
# Exclude taxa "Mitochondria"
metaxa_PHY_SSU <- subset_taxa(metaxa_PHY_SSU, !Domain %in% c("Mitochondria"))
# Exclude taxa "Chloroplast"
metaxa_PHY_SSU <- subset_taxa(metaxa_PHY_SSU, !Domain %in% c("Chloroplast"))
# Exclude taxa "Unclassified"
metaxa_PHY_SSU <- subset_taxa(metaxa_PHY_SSU, !Domain %in% c("Unclassified"))
# Exclude taxa "Eukaryota"
metaxa_PHY_SSU <- subset_taxa(metaxa_PHY_SSU, !Domain %in% c("Eukaryota"))
# Exclude taxa "Archaea"
metaxa_PHY_SSU <- subset_taxa(metaxa_PHY_SSU, !Domain %in% c("Archaea"))

## Add SSU counts to metadata
metadata$SSU_counts <- sample_sums(metaxa_PHY_SSU)

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaxa_PHY = subset_samples(metaxa_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
metaxa_PHY <- subset_samples(metaxa_PHY, country != "UK")
```
## Load Metaphlan3
```{r}
# OTU table
#sed -i 's/_profile//g' mod_merged_abundance_table_species.txt
#sed -n '2p'  merged_abundance_table.txt > merged_abundance_table_species.txt
#sed -i 's/_profile//g' merged_abundance_table_species.txt
#grep -E "s__" merged_abundance_table.txt >> merged_abundance_table_species.txt
#tr '|' ';' <merged_abundance_table_species.txt > mod_merged_abundance_table_species.txt

# tax table
#awk '{print $1}' mod_merged_abundance_table_species.txt > tax_table_metaphlan
#sed '1d' -i tax_table_metaphlan

# Load data
OTU_metaphlan <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_merged_abundance_table_species.txt", header=T)

# Make sure tax tabe is in order
#tax_table_metaphlan <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", quote="\"", comment.char="")
#identical(tax_table_metaphlan$V1, OTU_metaphlan$clade_name)

tax_table_metaphlan <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan", header=FALSE, sep=";")
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Remove "__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))

match <- match(rownames(metadata), colnames(OTU_metaphlan))
OTU_metaphlan  <- OTU_metaphlan[,match]
all(rownames(metadata) == colnames(OTU_metaphlan))

# Combine into phyloseq object
metaphlan_PHY <- phyloseq(otu_table(OTU_metaphlan, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaphlan)), sample_data(metadata))

# Exclude viruses
metaphlan_PHY <- subset_taxa(metaphlan_PHY, Kingdom != "Viruses" & Kingdom != "Eukaryota")
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_PHY = subset_samples(metaphlan_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
metaphlan_PHY <- subset_samples(metaphlan_PHY, country != "UK")
```
## Load ResFinder results
```{r}
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names added manually in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
# Divide by ARG gene lengths
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
OTU_resfinder_length_SSU_norm <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(OTU_resfinder_length_SSU_norm))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], OTU_resfinder_length_SSU_norm[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Reload to get new row numbers
write.table(OTU_resfinder_length_SSU_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_length_SSU_norm.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_resfinder_length_SSU_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_length_SSU_norm.txt", row.names=NULL)
OTU_resfinder_length_SSU_norm$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

# Combine to phyloseq object
resfinder_PHY <- phyloseq(otu_table(OTU_resfinder_length_SSU_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Save relative sums of ARGs
metadata$ARG_relative_sum <- sample_sums(resfinder_PHY)

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY = subset_samples(resfinder_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
resfinder_PHY <- subset_samples(resfinder_PHY, country != "UK")
```
## Load MGE results
```{r}
# cp MGE_genemat.txt cp_MGE_genemat.txt
# sed -i -e 's/\[//g' cp_MGE_genemat.txt
# sed -i -e 's/\]//g' cp_MGE_genemat.txt
# sed -i -e 's/\"//g' cp_MGE_genemat.txt
OTU_MGE <-as.matrix(read.table("cp_MGE_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_MGE))
OTU_MGE <- OTU_MGE[,match]
all(colnames(OTU_MGE) == rownames(metadata))

# Tax table (MGE_tax_table_trim.txt from Github)
# sed -i -e 's/\[//g' cp_MGE.fasta
# sed -i -e 's/\]//g' cp_MGE.fasta
# sed -i -e 's/\"//g' cp_MGE.fasta
# seqkit fx2tab --length --name --header-line cp_MGE.fasta > MGE_lengths.txt
MGE_tax_table_trim <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", header=FALSE)
colnames(MGE_tax_table_trim) <- c("Gene", "Element", "Class")

match <- match(rownames(OTU_MGE), MGE_tax_table_trim$Gene)
MGE_tax_table_trim <- MGE_tax_table_trim[match,]

# seqkit fx2tab --length --name --header-line MGE.fasta > MGE_lengths.txt
# Divide by ARG gene lengths
MGE_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_lengths.txt", header=FALSE, comment.char="#", check.names = F)
match <- match(rownames(OTU_MGE), MGE_lengths$V1)
MGE_lengths <- MGE_lengths[match,]
all(rownames(MGE_tax_table_trim$Gene) == MGE_lengths$V1)

OTU_MGE_length_norm <- OTU_MGE/MGE_lengths[, 2]

# Normalization with Metaxa2 SSU counts
OTU_MGE_length_SSU_norm <- t(t(OTU_MGE_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(OTU_MGE_length_SSU_norm))
all(rownames(OTU_MGE_length_SSU_norm) == MGE_tax_table_trim$Gene)

# Reload to get new row numbers
write.table(OTU_MGE_length_SSU_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_MGE_length_SSU_norm", 
            row.names=T, sep = "\t", col.names = T)
OTU_MGE_length_SSU_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_MGE_length_SSU_norm", row.names=NULL)
OTU_MGE_length_SSU_norm$row.names<-NULL

# Reload to get new row numbers
write.table(MGE_tax_table_trim, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", row.names=F, sep = "\t", col.names = T)
MGE_tax_table_trim <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_tax_table_trim.txt", row.names=NULL)

# Combine to phyloseq object
MGE_PHY <- phyloseq(otu_table(OTU_MGE_length_SSU_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(MGE_tax_table_trim)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
MGE_PHY = subset_samples(MGE_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
MGE_PHY <- subset_samples(MGE_PHY, country != "UK")
```
## Load Virulence genes
```{r}
OTU_VFDB <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/VF_COUNT_TABLE", sep=";")
# Hits
#sed -n 's/\(;\).*/\1/p' VF_COUNT_TABLE > output_headers.txt
#sed -i 's/;//g' output_headers.txt
#sed -i '/^$/d' output_headers.txt
output_headers <- read.table("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/output_headers.txt", quote="\"", comment.char="")
colnames(output_headers) <- c("Accession")
# Check
all(output_headers$V1 == OTU_VFDB$X)

match <- match(rownames(metadata), colnames(OTU_VFDB))
OTU_VFDB <- OTU_VFDB[,match]

# Normalization with Metaxa2 SSU counts
all(colnames(OTU_VFDB) == rownames(metadata))
OTU_VFDB_SSU_norm <- t(t(OTU_VFDB)/metadata$SSU_counts)
all(rownames(metadata) == colnames(OTU_VFDB_SSU_norm))
#identical(OTU_VFDB[200, 4]/metadata$SSU_counts[4], OTU_VFDB_SSU_norm[200, 4])

# Reload to get new row numbers
write.table(OTU_VFDB_SSU_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_VFDB_SSU_norm.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_VFDB_SSU_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_VFDB_SSU_norm.txt", row.names=NULL)
OTU_VFDB_SSU_norm$row.names<-NULL

# Tax table (Modified in excel)
# DB accessions
#grep ">" VFDB_setB_pro.fas > db_headers.txt
#sed -i 's/>//g' db_headers.txt

# In excel separate columns with ";" and replace spaces in other than first column with "_" and get it back to Puhti as "mod_db_headers.txt"

#grep -Fw -f output_headers.txt mod_db_headers.txt > VF.txt

# Change gene name back to tab delimited (not sure if needed)
#sed -i 's/_;/\t/g' VF.txt
#sed -i 's/;/\t/g' VF.txt

#sed 's/ /\t/1g' VF.txt

tax_table_VFDB <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/VF.txt", header=FALSE, sep = " ")
new_columns <- str_split_fixed(tax_table_VFDB$V2, ";", 4)
colnames(new_columns) <- c("Gene", "Description", "Class", "Reference")

tax_table_VFDB <- tax_table_VFDB[-2]
colnames(tax_table_VFDB) <- c("Accession")
tax_table_VFDB <- cbind(tax_table_VFDB, new_columns)

# order as in the genemat
match <- match(output_headers$Accession, tax_table_VFDB$Accession)
tax_table_VFDB <- tax_table_VFDB[match,]
identical(tax_table_VFDB$Accession, output_headers$Accession)
#all(output_headers$Accession == OTU_VFDB$X) # check before rownames are removed

# Reload to get new row numbers
write.table(tax_table_VFDB, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_VFDB.txt", row.names=F, sep = "\t", col.names = T)
tax_table_VFDB <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_VFDB.txt", row.names=NULL)
#identical(tax_table_VFDB$BFH31_S149, output_headers$BFH31_S149) # check before rownames are removed
#identical(tax_table_VFDB$Accession, output_headers$Accession) # check before rownames are removed

VFDB_PHY <- phyloseq(otu_table(OTU_VFDB_SSU_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_VFDB)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
VFDB_PHY = subset_samples(VFDB_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
VFDB_PHY <- subset_samples(VFDB_PHY, country != "UK")
```
## Load crAssphage results
```{r}
# Load data
crassphage_table <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table.txt", header = T, row.names = 1)
identical(rownames(metadata), rownames(crassphage_table))

match <- match(rownames(metadata), rownames(crassphage_table))
crassphage_table <- crassphage_table[match,]
identical(rownames(metadata), rownames(crassphage_table))

crassphage_table <- t(crassphage_table)

#crassphage_table_norm <- t(t(crassphage_table)/metadata$M_Seqs_trimmed)
crassphage_table_norm <- t(t(crassphage_table)/metadata$SSU_counts)

write.table(crassphage_table_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt", row.names=T, sep = "\t", col.names = T)
crassphage_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt")
crassphage_norm <- t(crassphage_norm)

# Create phyloseq object
all(rownames(crassphage_norm) == rownames(metadata))
crass_PHY <- phyloseq(otu_table(crassphage_norm, taxa_are_rows = F), sample_data(metadata))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
crass_PHY = subset_samples(crass_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
crass_PHY <- subset_samples(crass_PHY, country != "UK")


# log transform
crassphage_norm_log <- log10(crassphage_norm)

crassphage_norm_log <- t(crassphage_norm_log)
alias <- as.data.frame(rownames(crassphage_norm_log))
crassphage_norm_log <- cbind(crassphage_norm_log, alias = alias) 

colnames(crassphage_norm_log) <- c("count", "coverage", "alias")
p<-ggplot(data=crassphage_norm_log, aes(x=alias, y=coverage)) +
  geom_bar(stat="identity") +
  theme(axis.text.x = element_text(angle = 90))
```
## Ordination, Metaxa2
```{r}
# Create phyloseq object for relative data
metaxa_rel_PHY <- transform_sample_counts(metaxa_PHY, function(x) x/sum(x))

metaxa_rel_PHY_ord <- ordinate(metaxa_rel_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaxa_rel_PHY, metaxa_rel_PHY_ord, color = "country", shape = "sample_type")
metaxa.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0", "#7EA52A", "#24C0E1", "#661D38", "#8DAEA7", "#939139")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6, 11, 0, 1, 2, 20, 23)) +
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Taxonomy (Metaxa2) All samples") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaxa.p_ord

# Test significance using pair-wise adonis
metaxa_temp <- subset_samples(metaxa_PHY_ww, (continent_level_1 == "Europe" | continent_level_1 == "West Africa"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ continent_level_1, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

metaxa_temp <- subset_samples(metaxa_PHY_WA_hosp, (hospital_section_1 == "surgery" | hospital_section_1 == "mixed"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ hospital_section_1, data = data.frame(sample_data(metaxa_temp), permutations = 9999))
```
## Ordination, Metaphlan3
```{r}
# Subset samples
metaphlan_PHY_ord <- ordinate(metaphlan_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY, metaphlan_PHY_ord, color = "country", shape = "sample_type")
metaphlan.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0", "#7EA52A", "#24C0E1", "#661D38", "#8DAEA7", "#939139")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6, 11, 0, 1, 2, 20, 23)) +
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Taxonomy (Metaphlan3) All samples") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10, family = "Times")) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaphlan.p_ord

# Test significance using pair-wise adonis
metaphlan_temp <- subset_samples(metaphlan_PHY, (continent_level_1 == "Europe" | continent_level_1 == "West Africa"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ continent_level_1, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY, (category_for_model == "Eu other wwtp influent" | category_for_model == "North Eu wwtp influent"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ category_for_model, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))
```
## Ordinations, ResFinder
```{r}
resfinder_PHY_ord <- ordinate(resfinder_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY, resfinder_PHY_ord, color = "country", shape = "sample_type")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0", "#7EA52A", "#24C0E1", "#661D38", "#8DAEA7", "#939139")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6, 11, 0, 1, 2, 20, 23))+
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Resistome (ResFinder) All samples") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10, family = "Times")) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
resfinder.p_ord

# Test significance using pair-wise adonis
resfinder_temp <- subset_samples(resfinder_PHY, (category_for_model == "North Eu wwtp influent" | category_for_model == "Eu other wwtp influent"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ category_for_model, data = data.frame(sample_data(resfinder_temp), permutations = 9999))


# By hospital
resfinder_PHY_BF <- subset_samples(resfinder_PHY, country == "Burkina Faso")
resfinder_PHY_BF <- subset_samples(resfinder_PHY_BF, sample_type == "hospital effluent")

resfinder_PHY_ord <- ordinate(resfinder_PHY_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_BF, resfinder_PHY_ord, color = "hospital", shape = "hospital_section")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0", "#7EA52A", "#24C0E1")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6, 11, 0, 1, 2, 20, 23))+
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Resistome (ResFinder) Burkina Faso, hospital effluents") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 20, family = "Times")) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
resfinder.p_ord
```
## Ordinations, MGE
```{r}
MGE_PHY_ord <- ordinate(MGE_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY, MGE_PHY_ord, color = "country", shape = "category")
MGE.p_ord <- p_ord + scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0", "#7EA52A", "#24C0E1", "#661D38", "#8DAEA7", "#331225")) + 
    geom_point(size = 5) + 
  #  stat_ellipse(level = 0.95, linetype = 1) +
    geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Mobilome (MGE database) Burkina Faso") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

# Test significance using pair-wise adonis
MGE_temp <- subset_samples(MGE_PHY, (continent_level_1 == "Europe" | continent_level_1 == "West Africa"))
MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
adonis(MGE_dist ~ continent_level_1, data = data.frame(sample_data(MGE_temp), permutations = 9999))

MGE_temp <- subset_samples(MGE_PHY, (country == "Benin" | country == "Burkina Faso"))
MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))
```
## Ordinations, VFDB
```{r}
VFDB_PHY_ord <- ordinate(VFDB_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(VFDB_PHY, VFDB_PHY_ord, color = "country", shape = "sample_type")
VFDB.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0", "#7EA52A", "#24C0E1", "#661D38", "#8DAEA7", "#939139")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6, 11, 0, 1, 2, 20, 23))+
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Resistome (ResFinder) All samples") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10, family = "Times")) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
VFDB.p_ord

# Test significance using pair-wise adonis
VFDB_temp <- subset_samples(VFDB_PHY, (continent_level_1 == "Europe" | continent_level_1 == "West Africa"))
VFDB_dist <- vegdist(t(otu_table(VFDB_temp)), dist = "horn")
adonis(VFDB_dist ~ continent_level_1, data = data.frame(sample_data(VFDB_temp), permutations = 9999))

VFDB_temp <- subset_samples(VFDB_PHY, (category == "hospital effluent" | category == "wwtp influent"))
VFDB_dist <- vegdist(t(otu_table(VFDB_temp)), dist = "horn")
adonis(VFDB_dist ~ category, data = data.frame(sample_data(VFDB_temp), permutations = 9999))
```
## Following subset of samples in Benin & Burkina Faso
```{r}
## Koudougou flow of samples towards treatment
# Subset samples
metaxa_PHY_sub <- subset_samples(metaxa_PHY, connected_samples != "nd")
metaxa_PHY_sub_BF <- subset_samples(metaxa_PHY_sub, hospital == "Koudougou Hospital")

metaphlan_PHY_sub <- subset_samples(metaphlan_PHY, connected_samples != "nd")
metaphlan_PHY_sub_BF <- subset_samples(metaphlan_PHY_sub, hospital == "Koudougou Hospital")

resfinder_PHY_sub <- subset_samples(resfinder_PHY, connected_samples != "nd")
resfinder_PHY_sub_BF <- subset_samples(resfinder_PHY_sub, hospital == "Koudougou Hospital")

MGE_PHY_sub <- subset_samples(MGE_PHY, connected_samples != "nd")
MGE_PHY_sub_BF <- subset_samples(MGE_PHY_sub, hospital == "Koudougou Hospital")

VFDB_PHY_sub <- subset_samples(VFDB_PHY, connected_samples != "nd")
VFDB_PHY_sub_BF <- subset_samples(VFDB_PHY_sub, hospital == "Koudougou Hospital")

#ps2 <- tax_glom(metaphlan_PHY_sub_BF, "Genus")
#ps2f <- format_to_besthit(ps2)
#metaphlan_PHY_ord <- ordinate(ps2f, method = "CCA", distance = "bray")
#p <- plot_ordination_utils(ps2f, metaphlan_PHY_ord,
#  color = "connection_level", plot.arrow = TRUE,
#  scale.arrow = 1.3, top.taxa = 5)

#metaphlan.p_ord <- p + 
#  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC")) + 
#  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6)) +
#  geom_point(size = 10) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
#    theme_minimal() + ggtitle("Taxonomy (Metaphlan3) Connected samples, \nKoudougou Hospital") + 
#    theme(plot.title = element_text(size = 25, family = "Times")) +
#    theme(legend.text = element_text(size = 10)) +
#    theme(legend.title = element_blank()) +
#    theme(axis.title = element_text(size = 20)) +
#    guides(fill = guide_legend(override.aes = list(linetype = 0)),
#         color = guide_legend(override.aes = list(linetype = 0, size=5)))
#metaphlan.p_ord

metaphlan_PHY_ord <- ordinate(metaphlan_PHY_sub_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY_sub_BF, metaphlan_PHY_ord, color = "connection_level", shape = "connection_group")
metaphlan.p_ord <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6)) +
  geom_point(size = 8) + 
  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = 0.9, vjust = -1.7) +
    theme_minimal() + ggtitle("Taxonomy (Metaphlan3) Connected samples, \nKoudougou Hospital") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0, size = 10)),
         color = guide_legend(override.aes = list(linetype = 0, size=10)),
         shape = guide_legend(override.aes = list(linetype = 0, size=10)))
metaphlan.p_ord

resfinder_PHY_ord <- ordinate(resfinder_PHY_sub_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_sub_BF, resfinder_PHY_ord, color = "connection_level", shape = "connection_group")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC")) +  
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6))+
  geom_point(size = 8) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = 0.9, vjust = -1.7) +
    theme_minimal() + ggtitle("Resistome (ResFinder) Connected samples, \nKoudougou Hospital") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0, size = 10)),
         color = guide_legend(override.aes = list(linetype = 0, size=10)),
         shape = guide_legend(override.aes = list(linetype = 0, size=10)))
resfinder.p_ord

resfinder_temp <- subset_samples(resfinder_PHY_sub_Ben, (connection_group == "A" | connection_group == "D"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ connection_group, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

MGE_PHY_ord <- ordinate(MGE_PHY_sub_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_sub_BF, MGE_PHY_ord, color = "connection_level", shape = "connection_group")
MGE.p_ord <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6))+
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = -0.1, vjust = -1.7) +
    theme_minimal() + ggtitle("Mobilome (MGE) Connected samples, \n Koudougou Hospital") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

VFDB_ord <- ordinate(VFDB_PHY_sub_BF, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(VFDB_PHY_sub_BF, VFDB_ord, color = "connection_level", shape = "connection_group")
VFDB.p_ord <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6))+
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = -0.1, vjust = -1.7) +
    theme_minimal() + ggtitle("Virulence genes (VFDB) Koudougou Hospital") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
VFDB.p_ord


# Plot bar
cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 18)
rfc <- plot_bar(resfinder_PHY_sub_BF, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance (ResFinder)"))) +
  ggtitle("ARG Abundance normalized to 16s rRNA") +
  theme_minimal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~connection_level, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 24, family = "Times", hjust = 0.5, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))
rfc_plot

# Sum abundance
df<-data.frame(SUM=sample_sums(resfinder_PHY_sub_BF),
               category=as.factor(sample_data(resfinder_PHY_sub_BF)$category_for_model),
               hospital_section=as.factor(sample_data(resfinder_PHY_sub_BF)$hospital_section),
               hospital=as.factor(sample_data(resfinder_PHY_sub_BF)$hospital),
               SSU_counts=as.factor(sample_data(resfinder_PHY_sub_BF)$SSU_counts),
               area=as.factor(sample_data(resfinder_PHY_sub_BF)$area_or_city),
               connection_level=as.factor(sample_data(resfinder_PHY_sub_BF)$connection_level),
               country=as.factor(sample_data(resfinder_PHY_sub_BF)$country))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)

plot(df$SUM~df$connection_level, cex.axis = 0.5, las = 2)

# Fit model with all variables  
#fit <- glm(SUM~connection_level, data=df, family="Gamma"(link="log"))
#summary(fit)

# Tukey's post hoc test
#glht.mod <- glht(fit, mcp(hospital = "Tukey"))
#summary(glht(glht.mod))

#df<-data.frame(SUM=sample_sums(BF),
#               hospital=as.factor(sample_data(BF)$hospital))

cols <- palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"))
resfinder_sum <- ggplot(df, aes(x = connection_level, y = SUM, color = connection_level)) + 
  geom_point(data = df, aes(x = connection_level, y = SUM, color = connection_level), size = 12, alpha = 1) +
  scale_color_manual(values=cols) +
  theme_linedraw() + 
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 0, vjust = 0.8, size = 30, hjust = 0.8, family = "Times", face = "bold"), 
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(y = "Abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Relative abundance of ARGs (ResFinder), \nKoudougou")
resfinder_sum

# Sum abundance of VF
df<-data.frame(SUM=sample_sums(VFDB_PHY_sub_BF),
               category=as.factor(sample_data(VFDB_PHY_sub_BF)$category_for_model),
               hospital_section=as.factor(sample_data(VFDB_PHY_sub_BF)$hospital_section),
               hospital=as.factor(sample_data(VFDB_PHY_sub_BF)$hospital),
               SSU_counts=as.factor(sample_data(VFDB_PHY_sub_BF)$SSU_counts),
               area=as.factor(sample_data(VFDB_PHY_sub_BF)$area_or_city),
               connection_level=as.factor(sample_data(VFDB_PHY_sub_BF)$connection_level),
               country=as.factor(sample_data(VFDB_PHY_sub_BF)$country))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)

plot(df$SUM~df$connection_level, cex.axis = 0.5, las = 2)

cols <- palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"))
VF_sum <- ggplot(df, aes(x = connection_level, y = SUM, color = connection_level)) + 
  geom_point(data = df, aes(x = connection_level, y = SUM, color = connection_level), size = 12, alpha = 1) +
  scale_color_manual(values=cols) +
  theme_linedraw() + 
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 0, vjust = 0.8, size = 30, hjust = 0.8, family = "Times", face = "bold"), 
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(y = "Abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Relative abundance of virulence factors (VFDB), \nKoudougou")
VF_sum

## Diversities
# Metaxa2
PHY <- prune_taxa(taxa_sums(metaxa_PHY_sub_BF) > 0, metaxa_PHY_sub_BF)

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$connection_level) # get the variables
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"))
d <- ggplot(PHY.meta, aes(connection_level, Shannon))
shannon <- d + 
  geom_point(aes(fill = connection_level, colour = connection_level), alpha = 0.9, size = 12) +
  scale_color_manual(values = cols) +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.8, size = 30, hjust = 0.8, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(title = "Alpha diversity, Shannon (Metaxa2)")
print(shannon)

e <- ggplot(PHY.meta, aes(connection_level, Observed))
observed <- e + 
  geom_point(aes(fill = connection_level, colour = connection_level), alpha = 0.9, size = 12) +
  scale_color_manual(values = cols) +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.8, size = 30, hjust = 0.8, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(title = "Alpha diversity, Observed (Metaxa2)")
print(observed)

# ResFinder
# Combine to phyloseq object
resfinder_PHY_counts <- phyloseq(otu_table(OTU_resfinder_counts, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY_counts = subset_samples(resfinder_PHY_counts, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
resfinder_PHY_counts <- subset_samples(resfinder_PHY_counts, country != "UK")

resfinder_PHY_counts_sub <- subset_samples(resfinder_PHY_counts, connected_samples != "nd")
resfinder_PHY_counts_sub_BF <- subset_samples(resfinder_PHY_counts_sub, hospital == "Koudougou Hospital")

PHY <- prune_taxa(taxa_sums(resfinder_PHY_counts_sub_BF) > 0, resfinder_PHY_counts_sub_BF)

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$connection_level) # get the variables
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"))
d <- ggplot(PHY.meta, aes(connection_level, Shannon))
shannon <- d + 
  geom_point(aes(fill = connection_level, colour = connection_level), alpha = 0.9, size = 12) +
  scale_color_manual(values = cols) +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.8, size = 30, hjust = 0.8, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(title = "Alpha diversity, Shannon (ResFinder)")
print(shannon)

e <- ggplot(PHY.meta, aes(connection_level, Observed))
observed <- e + 
  geom_point(aes(fill = connection_level, colour = connection_level), alpha = 0.9, size = 12) +
  scale_color_manual(values = cols) +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.8, size = 30, hjust = 0.8, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(title = "Alpha diversity, Observed (ResFinder)")
print(observed)

#####

## Ougadougou
metaxa_PHY_sub <- subset_samples(metaxa_PHY, connected_samples != "nd")
metaxa_PHY_sub_Ben <- subset_samples(metaxa_PHY_sub, hospital == "Yalgado teaching hospital")

metaphlan_PHY_sub <- subset_samples(metaphlan_PHY, connected_samples != "nd")
metaphlan_PHY_sub_Ben <- subset_samples(metaphlan_PHY_sub, hospital == "Yalgado teaching hospital")

resfinder_PHY_sub <- subset_samples(resfinder_PHY, connected_samples != "nd")
resfinder_PHY_sub_Ben <- subset_samples(resfinder_PHY_sub, hospital == "Yalgado teaching hospital")

MGE_PHY_sub <- subset_samples(MGE_PHY, connected_samples != "nd")
MGE_PHY_sub_Ben <- subset_samples(MGE_PHY_sub, hospital == "Yalgado teaching hospital")

VFDB_PHY_sub <- subset_samples(VFDB_PHY, connected_samples != "nd")
VFDB_PHY_sub_Ben <- subset_samples(VFDB_PHY_sub, hospital == "Yalgado teaching hospital")


cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)
metaphlan_PHY_ord <- ordinate(metaphlan_PHY_sub_Ben, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY_sub_Ben, metaphlan_PHY_ord, color = "connection_level", shape = "connection_group")
metaphlan.p_ord <- p_ord + 
  scale_color_manual(values = cols) + 
  scale_shape_manual(values=c(15, 16, 17, 18, 20, 25, 14, 21)) +
  geom_point(size = 10) + 
  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = 0.9, vjust = -1.7) +
    theme_minimal() + ggtitle("Taxonomy (Metaphlan3) Connected samples, \nYalgado Teaching Hospital") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0, size = 10)),
         color = guide_legend(override.aes = list(linetype = 0, size=10)),
         shape = guide_legend(override.aes = list(linetype = 0, size=10)))
metaphlan.p_ord

resfinder_PHY_ord <- ordinate(resfinder_PHY_sub_Ben, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_sub_Ben, resfinder_PHY_ord, color = "connection_level", shape = "connection_group")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values = cols) + 
  scale_shape_manual(values=c(15, 16, 17, 18, 20, 25, 14, 21)) +
  geom_point(size = 10) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = 0.9, vjust = -1.7) +
    theme_minimal() + ggtitle("Resistome (ResFinder) Connected samples, \nYalgado Teaching Hospital") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0, size = 10)),
         color = guide_legend(override.aes = list(linetype = 0, size=10)),
         shape = guide_legend(override.aes = list(linetype = 0, size=10)))
resfinder.p_ord

MGE_PHY_ord <- ordinate(MGE_PHY_sub_Ben, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_sub_Ben, MGE_PHY_ord, color = "connection_level", shape = "connection_group")
MGE.p_ord <- p_ord + 
  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC")) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6))+
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = -0.1, vjust = -1.7) +
    theme_minimal() + ggtitle("Mobilome (MGE) Connected samples, \n Yalgado teaching Hospital") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)
VFDB_ord <- ordinate(VFDB_PHY_sub_Ben, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(VFDB_PHY_sub_Ben, VFDB_ord, color = "connection_level", shape = "connection_group")
VFDB.p_ord <- p_ord + 
  scale_color_manual(values = cols) + 
  scale_shape_manual(values=c(15, 16, 17, 3, 7, 8, 6))+
  geom_point(size = 5) + 
#  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = connection_for_plot), size = 4, hjust = -0.1, vjust = -1.7) +
    theme_minimal() + ggtitle("Virulence genes (VFDB) Yalgado teaching Hospital") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
VFDB.p_ord

# Plot bar
cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 18)
rfc <- plot_bar(resfinder_PHY_sub_Ben, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance (ResFinder)"))) +
  ggtitle("ARG Abundance normalized to 16s rRNA") +
  theme_minimal() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~connection_for_plot, scales = "free", space = "free", labeller = label_wrap_gen(width = 10, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 18, family = "Times", hjust = 0.5, vjust = 0.5, angle = 0, face = "bold"),
        strip.background = element_rect(colour = "white"))
rfc_plot

# Sum abundance
df<-data.frame(SUM=sample_sums(resfinder_PHY_sub_Ben),
               category=as.factor(sample_data(resfinder_PHY_sub_Ben)$category_for_model),
               hospital_section=as.factor(sample_data(resfinder_PHY_sub_Ben)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_sub_Ben)$SSU_counts),
               connection_level=as.factor(sample_data(resfinder_PHY_sub_Ben)$connection_level_combined))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)

plot(df$SUM~df$connection_level, cex.axis = 0.5, las = 2)

# Plot
df<-data.frame(SUM=sample_sums(resfinder_PHY_sub_Ben),
               connection_level=as.factor(sample_data(resfinder_PHY_sub_Ben)$connection_level_combined))

fit <- glm(SUM~connection_level + SSU_counts, data=df, family="Gamma"(link="log"))
summary(fit)

glht.mod <- glht(fit, mcp(connection_level = "Tukey"))
summary(glht(glht.mod))

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)

resfinder_sum <- ggplot(dfA, aes(x = connection_level, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = connection_level, y = SUM, color = connection_level), size = 8, alpha = 1, width = 0.3) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.3, lwd = 0.5) + geom_point(size = 0.9) + 
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 15, hjust = 0.75, vjust = 0.75, size = 26, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of ARGs (ResFinder), \nYalgado Teaching Hospital
       
       
       ")
resfinder_sum

# Sum abundance virulence
df<-data.frame(SUM=sample_sums(VFDB_PHY_sub_Ben),
               category=as.factor(sample_data(VFDB_PHY_sub_Ben)$category_for_model),
               hospital_section=as.factor(sample_data(VFDB_PHY_sub_Ben)$hospital_section),
               SSU_counts=as.factor(sample_data(VFDB_PHY_sub_Ben)$SSU_counts),
               connection_level=as.factor(sample_data(VFDB_PHY_sub_Ben)$connection_level_combined))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)

plot(df$SUM~df$connection_level, cex.axis = 0.5, las = 2)

# Plot
fit <- glm(SUM~connection_level, data=df, family="Gamma"(link="log"))
summary(fit)

glht.mod <- glht(fit, mcp(connection_level = "Tukey"))
summary(glht(glht.mod))

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)
VF_sum <- ggplot(dfA, aes(x = connection_level, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = connection_level, y = SUM, color = connection_level), size = 8, alpha = 1, width = 0.3) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.3, lwd = 0.5) + geom_point(size = 0.9) + 
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 15, hjust = 0.75, vjust = 0.75, size = 26, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of virulence genes (VFDB), \nYalgado Teaching Hospital
       
       
       ")
VF_sum

## Diversities
# Metaxa2
PHY <- prune_taxa(taxa_sums(metaxa_PHY_sub_Ben) > 0, metaxa_PHY_sub_Ben)

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$connection_level_combined) # get the variables
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)
shannon <- ggboxplot(PHY.meta, x = "connection_level_combined", y = "Shannon", fill = "connection_level_combined", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 15, vjust = 0.75, size = 36, hjust = 0.75, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20, family = "Times"),
        legend.position = "none",
        legend.title = element_blank(),
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity, Shannon index (Metaxa2)")
print(shannon)

a <- shannon + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = F, ref.group = ".all.")

cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)
observed <- ggboxplot(PHY.meta, x = "connection_level_combined", y = "Observed", fill = "connection_level_combined", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 15, vjust = 0.75, size = 36, hjust = 0.75, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20, family = "Times"),
        legend.position = "none",
        legend.title = element_blank(),
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity, Observed index (Metaxa2)")
print(observed)

a <- observed + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = F, ref.group = ".all.")

# ResFinder
# Combine to phyloseq object
resfinder_PHY_counts <- phyloseq(otu_table(OTU_resfinder_counts, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY_counts = subset_samples(resfinder_PHY_counts, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
resfinder_PHY_counts <- subset_samples(resfinder_PHY_counts, country != "UK")

resfinder_PHY_counts_sub <- subset_samples(resfinder_PHY_counts, connected_samples != "nd")
resfinder_PHY_counts_sub_Ben <- subset_samples(resfinder_PHY_counts_sub, hospital == "Yalgado teaching hospital")

PHY <- prune_taxa(taxa_sums(resfinder_PHY_counts_sub_Ben) > 0, resfinder_PHY_counts_sub_Ben)

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$connection_level_combined) # get the variables
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)
shannon <- ggboxplot(PHY.meta, x = "connection_level_combined", y = "Shannon", fill = "connection_level_combined", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 15, vjust = 0.75, size = 36, hjust = 0.75, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20, family = "Times"),
        legend.position = "none",
        legend.title = element_blank(),
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity, Shannon index (ResFinder)")
print(shannon)

a <- shannon + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = F, ref.group = ".all.")

cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 8)
observed <- ggboxplot(PHY.meta, x = "connection_level_combined", y = "Observed", fill = "connection_level_combined", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 15, vjust = 0.75, size = 36, hjust = 0.75, family = "Times", face = "bold"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20, family = "Times"),
        legend.position = "none",
        legend.title = element_blank(),
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity, Observed index (ResFinder)")
print(observed)

a <- observed + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = F, ref.group = ".all.")
```
## Studying different sections of hospitals
```{r}
# Metaxa2
metaxa_PHY_hosp_eff <- subset_samples(metaxa_PHY, sample_type == "hospital effluent")
#metaxa_PHY_hosp_eff <- subset_samples(metaxa_PHY_hosp_eff, country == "Benin" | country == "Burkina Faso")

metaxa_rel_PHY <- transform_sample_counts(metaxa_PHY_hosp_eff, function(x) x/sum(x))

metaxa_PHY_ord <- ordinate(metaxa_rel_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaxa_rel_PHY, metaxa_PHY_ord, color = "hospital_section")
metaxa.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#8DAEA7", "#EFAB15", "#24C0E1", "#661D38")) +
  geom_point(size = 5) + 
  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Taxonomy (Metaxa2) Hospital effluents by sections") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaxa.p_ord

metaxa_temp <- subset_samples(metaxa_PHY_hosp_eff, (hospital_section == "surgery" | hospital_section == "maternity"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ hospital_section, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

# Metaphlan3
metaphlan_PHY_hosp_eff <- subset_samples(metaphlan_PHY, sample_type == "hospital effluent")
# Exclude replicates:
metaphlan_PHY_hosp_eff <- subset_samples(metaphlan_PHY_hosp_eff, alias != "BH10_S84" & 
                                           alias != "BH31_S95" &  alias != "BH32_S96" & alias != "BH33_S97" & 
                                           alias != "BH34A_S98" & alias != "BH34B_S99" & alias != "BFH38B_S157" & 
                                           alias != "FH8_S169" & alias != "BH45_S106" & alias != "BH59_S114" &
                                           alias != "BH61_S116" & alias != "BH62_S117")

metaphlan_PHY_ord <- ordinate(metaphlan_PHY_hosp_eff, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY_hosp_eff, metaphlan_PHY_ord, color = "hospital_section")
metaphlan.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#8DAEA7", "#EFAB15", "#661D38")) +
  geom_point(size = 5) + 
  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = area_or_city), size = 4, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Taxonomy (Metaphlan3) \nHospital effluents by hospital sections") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 20, family = "Times")) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
metaphlan.p_ord

metaphlan_temp <- subset_samples(metaphlan_PHY_hosp_eff, (hospital_section == "other" | hospital_section == "maternity"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ hospital_section, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

# What are the most abundant genera?
metaphlan_PHY_hosp_sec <- subset_samples(metaphlan_PHY, hospital_section == "surgery" | hospital_section == "maternity" |
                                           hospital_section == "maternity_surgery" | hospital_section == "pediatry")
metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY_hosp_sec, taxrank = "Genus")
# Take 15 most abundant
metaphlan_PHY_Genus_abund <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_Genus), TRUE)[1:15]), metaphlan_PHY_Genus)

# Normalize by number of samples in each sample type
otu_table(metaphlan_PHY_Genus_abund) <- (otu_table(metaphlan_PHY_Genus_abund)[,]/as.matrix(table(sample_data(metaphlan_PHY_Genus_abund)$hospital_section))[, 1])

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 15)

mp <- plot_bar(metaphlan_PHY_Genus_abund, fill = "Genus") 
mp_plot <- mp +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("Relative abundance of 15 most abundant taxa"), atop("Metaphlan3")))) +
  ggtitle("15 most abundant genera") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 60, hjust = 1), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~hospital_section, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 20, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))
mp_plot

# ResFinder
resfinder_PHY_hosp_eff <- subset_samples(resfinder_PHY, sample_type == "hospital effluent")

# Exclude replicates:
resfinder_PHY_hosp_eff <- subset_samples(resfinder_PHY_hosp_eff, alias != "BH10_S84" & 
                                           alias != "BH31_S95" & alias != "BH32_S96" & alias != "BH33_S97" & 
                                           alias != "BH34A_S98" & alias != "BH34B_S99" & alias != "BFH38B_S157" & 
                                           alias != "FH8_S169" & alias != "BH45_S106" & alias != "BH59_S114" &
                                           alias != "BH61_S116" & alias != "BH62_S117")

resfinder_PHY_ord <- ordinate(resfinder_PHY_hosp_eff, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY_hosp_eff, resfinder_PHY_ord, color = "hospital_section")
resfinder.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#8DAEA7", "#EFAB15", "#661D38")) +
  geom_point(size = 5) + 
  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = area_or_city), size = 4, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Resistome (ResFinder) \nHospital effluents by sections") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 20, family = "Times")) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20, family = "Times")) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
resfinder.p_ord

resfinder_temp <- subset_samples(resfinder_PHY_hosp_eff, (hospital_section == "surgery" | hospital_section == "maternity"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ hospital_section, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

# MGE
MGE_PHY_hosp_eff <- subset_samples(MGE_PHY, sample_type == "hospital effluent")
#MGE_PHY_hosp_eff <- subset_samples(MGE_PHY_hosp_eff, country == "Benin" | country == "Burkina Faso")

MGE_PHY_ord <- ordinate(MGE_PHY_hosp_eff, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY_hosp_eff, MGE_PHY_ord, color = "hospital_section")
MGE.p_ord <- p_ord + 
  scale_color_manual(values=c("#FF333F", "#8DAEA7", "#EFAB15", "#24C0E1", "#661D38")) +
  geom_point(size = 5) + 
  stat_ellipse(level = 0.95, linetype = 1) +
  geom_text(mapping = aes(label = alias), size = 2.5, hjust = 1, vjust = -0.5) +
    theme_minimal() + ggtitle("Mobilome (MGE db) Hospital effluents by sections") + 
    theme(plot.title = element_text(size = 25, family = "Times")) +
    theme(legend.text = element_text(size = 10)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))
MGE.p_ord

MGE_temp <- subset_samples(MGE_PHY_hosp_eff, (hospital_section == "maternity_surgery" | hospital_section == "surgery"))
MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
adonis(MGE_dist ~ hospital_section, data = data.frame(sample_data(MGE_temp), permutations = 9999))

# Surgery / maternity most abundant Clusters / ARGs
resfinder_PHY_hosp_eff <- subset_samples(resfinder_PHY, sample_type == "hospital effluent")

# Exclude replicates:
resfinder_PHY_hosp_eff <- subset_samples(resfinder_PHY_hosp_eff,  alias != "BH10_S84" & 
                                           alias != "BH31_S95" & alias != "BH32_S96" & alias != "BH33_S97" & 
                                           alias != "BH34A_S98" & alias != "BH34B_S99" & alias != "BFH38B_S157" & 
                                           alias != "FH8_S169" & alias != "BH45_S106" & alias != "BH59_S114" &
                                           alias != "BH61_S116" & alias != "BH62_S117")
maternity <- subset_samples(resfinder_PHY_hosp_eff, hospital_section == "maternity")
surgery <- subset_samples(resfinder_PHY_hosp_eff, hospital_section == "surgery")

maternity_Cluster <- tax_glom(maternity, taxrank = "Cluster_name")
# Take 15 most abundant
maternity_Cluster_abund <- prune_taxa(names(sort(taxa_sums(maternity_Cluster), TRUE)[1:15]), 
    maternity_Cluster)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 18)
rfc <- plot_bar(maternity_Cluster_abund, fill = "Cluster_name") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("ARG Abundance normalized to 16s rRNA, ResFinder"))) +
  ggtitle("Relative abundance of ARGs, Other") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, family = "Times", angle = 45, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 24, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~area_or_city, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 20, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))
rfc_plot
```
## Plot genera
```{r}
metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY, taxrank = "Genus")
metaphlan_PHY_Genus_abund <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_Genus), TRUE)[1:15]), metaphlan_PHY_Genus)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 15)
mp <- plot_bar(metaphlan_PHY_Genus_abund, fill = "Genus") 
mp_plot <- mp +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of genera"))) +
  ggtitle("15 Most abundant genera (Metaphlan3)") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8.5, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~category_for_plots, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 9, family = "Times", hjust = 0, vjust = 0.5, angle = 90),
        strip.background = element_rect(colour = "white"))
mp_plot

# Merge by category_for_model
metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY, taxrank = "Genus")
metaphlan_PHY_Genus_abund <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_Genus), TRUE)[1:15]), metaphlan_PHY_Genus)

mrg <- merge_samples(metaphlan_PHY_Genus_abund, "category_for_model")
otu_table(mrg) <- otu_table(mrg)[, ]/as.matrix(table(sample_data(metaphlan_PHY)$category_for_model))[, 
    1]
mp <- plot_bar(mrg, fill = "Genus") 
mp_plot <- mp +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of genera"))) +
  ggtitle("15 Most abundant genera (Metaphlan3)") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 16, family = "Times", angle = 55, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold"))
mp_plot

# Merge by hospital_section
metaphlan_PHY_sub <- subset_samples(metaphlan_PHY, hospital_section == "surgery" | hospital_section == "maternity" |
                                 hospital_section == "maternity_surgery" | hospital_section == "pediatry")

metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY_sub, taxrank = "Genus")
metaphlan_PHY_Genus_abund <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_Genus), TRUE)[1:15]), metaphlan_PHY_Genus)

mrg <- merge_samples(metaphlan_PHY_Genus_abund, "hospital_section")
otu_table(mrg) <- otu_table(mrg)[, ]/as.matrix(table(sample_data(metaphlan_PHY_sub)$hospital_section))[, 
    1]
mp <- plot_bar(mrg, fill = "Genus") 
mp_plot <- mp +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of genera"))) +
  ggtitle("15 Most abundant genera (Metaphlan3)") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 16, family = "Times", angle = 55, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold"))
mp_plot

# Plot interesting samples: waters for drinking and hand washing & samples of treated water in WA and in Sweden
metaphlan_PHY_sub <- subset_samples(metaphlan_PHY,
                                    #   category_for_plots == "WA treated, receiving municipality channel"
                                    #   category_for_plots == "WA treated, receiving river"
                                    #   category_for_plots == "WA treated, drinking water"
                                    #   category_for_plots == "WA treated, hand washing"
                                    #   category_for_plots == "WA river, drinking water"
                                    #   category_for_plots == "North Eu treated"
                                      )
metaphlan_PHY_Genus <- tax_glom(metaphlan_PHY_sub, taxrank = "Genus")
metaphlan_PHY_Genus_abund <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_Genus), TRUE)[1:15]), metaphlan_PHY_Genus)

#metaphlan_trans <- transform_sample_counts(metaphlan_PHY_Genus_abund, function(x) x / sum(x))
otu_table(metaphlan_PHY_Genus_abund) <- (otu_table(metaphlan_PHY_Genus_abund)[, ]/as.matrix(table(sample_data(metaphlan_PHY_Genus_abund)$category_for_plots))[, 
    1])

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 15)
mp <- plot_bar(metaphlan_PHY_Genus_abund, fill = "Genus") 
mp_plot <- mp +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of genera"))) +
  ggtitle("15 Most abundant genera (Metaphlan3)") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8.5, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~clean_description, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 20, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))
mp_plot

# Plot Acineto
metaphlan_PHY_hospital_effluent <- subset_samples(metaphlan_PHY, sample_type == "hospital effluent" | sample_type == "wwtp influent")
metaphlan_PHY_Acineto <- subset_taxa(metaphlan_PHY_hospital_effluent, Genus == "Acinetobacter")

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 1)
mp <- plot_bar(metaphlan_PHY_Acineto, fill = "Genus") 
mp_plot <- mp +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance of genera"))) +
  ggtitle("Abundance of Acinetobacter (Metaphlan3)") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8.5, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~category_for_model, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 12, family = "Times", hjust = 0, vjust = 0.5, angle = 90),
        strip.background = element_rect(colour = "white"))
mp_plot

# Boxplot
metaphlan_PHY_Acineto <- subset_samples(metaphlan_PHY_Acineto, sample_sums(metaphlan_PHY_Acineto) != 0)
df <- data.frame(SUM=sample_sums(metaphlan_PHY_Acineto),
                 country=as.factor(sample_data(metaphlan_PHY_Acineto)$country),
                 category=as.factor(sample_data(metaphlan_PHY_Acineto)$category_for_model),
                 hospital_section=as.factor(sample_data(metaphlan_PHY_Acineto)$hospital_section))

p <- ggplot(df, aes(x=SUM, y=hospital_section, color = hospital_section)) + 
  geom_boxplot() +
  theme_classic() +
  theme(axis.title = element_blank())
  coord_flip()
p

```
## Plot ARGs by classses, SSU normalized, ResFinder
```{r}
# Get only waters
#resfinder_PHY_sub <- subset_samples(resfinder_PHY, category_for_model == "WA treated" | 
#                                      category_for_model == "WA drinking water" |
#                                      category_for_model == "WA drinking water" |
#                                      category_for_model == "WA hospital effluent" | 
#                                      category_for_model == "Eu other hospital effluent" |    
#                                      category_for_model == "North Eu hospital effluent" |   
#                                      category_for_model == "Eu other wwtp influent" | 
#                                      category_for_model == "North EU wwtp influent" | 
#                                      category_for_model == "North Eu treated")

resfinder_PHY_sub <- subset_samples(resfinder_PHY, country == "Burkina Faso" | sample_type == "hospital_effluent")

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 18)
rfc <- plot_bar(resfinder_PHY_sub, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(bold("Relative abundance (ResFinder)"))) +
  ggtitle("ARG Abundance normalized to 16s rRNA") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8.5, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~hospital, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 9.5, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))
rfc_plot
#ggsave(filename = "rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

## For interestingly similar samples: selected WA based on ordination & Swedish treated
resfinder_PHY_sub <- subset_samples(resfinder_PHY, 
                                      #category_for_plots == "North Eu treated" 
                                      #alias == "BFH8_S126" 
                                      #alias == "BFH42_S161"
                                      #alias == "BH46_S107" 
                                      #alias == "BFH27_S145" 
                                      #alias == "BFH30_S148"
                                      #alias == "BH50_S111"  
                                      #alias == "BH11_S85"
                                      )
resfinder_PHY_sub <- subset_samples(resfinder_PHY, 
                                    category_for_plots == "North Eu treated")
resfinder_PHY_sub_Gene <- tax_glom(resfinder_PHY_sub, taxrank = "Gene")
# Take 15 most abundant
resfinder_PHY_sub_Gene_abund <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_sub_Gene), TRUE)[1:15]), resfinder_PHY_sub_Gene)

otu_table(resfinder_PHY_sub_Gene_abund) <- (otu_table(resfinder_PHY_sub_Gene_abund)[, ]/as.matrix(table(sample_data(resfinder_PHY_sub_Gene_abund)$category_for_plots))[, 
    1])

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 18)
rfc <- plot_bar(resfinder_PHY_sub_Gene_abund, fill = "Gene") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARG Abundance normalized to 16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, family = "Times", angle = 0, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~clean_description, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 15, family = "Times", hjust = 0, vjust = 0.5, angle = 0),
        strip.background = element_rect(colour = "white"))
rfc_plot
```
## Plot MGEs, normalized by SSU
```{r}
MGE_PHY_Class <- tax_glom(MGE_PHY, taxrank = "Class")

# Take 12 most abundant
MGE_PHY_Class_abund <- prune_taxa(names(sort(taxa_sums(MGE_PHY_Class), TRUE)[1:17]), 
    MGE_PHY_Class)

MGE_PHY_Class_abund <- subset_samples(MGE_PHY_Class_abund, sample_sums(MGE_PHY_Class_abund) != 0)

mge <- plot_bar(MGE_PHY_Class_abund, fill = "Class") 
cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 17)
mge_plot <- mge +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("MGEs normalized to 16s rRNA"), atop("MGEB")))) +
  ggtitle("Relative abundance of integrase elements") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 60, hjust = 1), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~country, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))
mge_plot
```
## Relative sums, ResFinder, gamma distribution model
```{r}
## Study measures of variance and mean and normality of the data
## Exclude samples that are unique in their category
#resfinder_PHY_model <- subset_samples(resfinder_PHY, sample_type == "hospital effluent")
#resfinder_PHY_model <- subset_samples(resfinder_PHY_model, country == "Benin" | country == "Burkina Faso")
#resfinder_PHY_model <- subset_samples(resfinder_PHY_model, sample_type != "wwtp influent")

# Exclude replicates:
#resfinder_PHY_model <- subset_samples(resfinder_PHY_model,  alias != "BH10_S84" & 
#                                           alias != "BH31_S95" & alias != "BH32_S96" & alias != "BH33_S97" & 
#                                           alias != "BH34A_S98" & alias != "BH34B_S99" & alias != "BFH38B_S157" & 
#                                           alias != "FH8_S169" & alias != "BH45_S106" & alias != "BH59_S114" &
#                                           alias != "BH61_S116" & alias != "BH62_S117")

#resfinder_PHY_model <- subset_samples(resfinder_PHY_model, country == "Benin" | country == "Burkina Faso")

## Create model
df<-data.frame(SUM=sample_sums(resfinder_PHY_model),
               category_for_model=as.factor(sample_data(resfinder_PHY_model)$category_for_model),
               hospital_section=as.factor(sample_data(resfinder_PHY_model)$hospital_section),
               hospital=as.factor(sample_data(resfinder_PHY_model)$hospital),
               hospital_2=as.factor(sample_data(resfinder_PHY_model)$hospital_2),
               continent_level_1=as.factor(sample_data(resfinder_PHY_model)$continent_level_1),
               continent_level_2=as.factor(sample_data(resfinder_PHY_model)$continent_level_2),
               continent_level_3=as.factor(sample_data(resfinder_PHY_model)$continent_level_3),
               country=as.factor(sample_data(resfinder_PHY_model)$country),
               area_or_city=as.factor(sample_data(resfinder_PHY_model)$area_or_city),
               SSU_counts=as.factor(sample_data(resfinder_PHY_model)$SSU_counts),
               sample_type=as.factor(sample_data(resfinder_PHY_model)$sample_type))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
## Explore data
#plot(df$SUM~df$hospital, cex.axis = 0.5, las = 2)

# Fit model with all variables  
fit <- glm(SUM~hospital, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(hospital_2 = "Tukey"))
summary(glht(glht.mod))

# Plot
df<-data.frame(SUM=sample_sums(resfinder_PHY_model),
               hospital_2=as.factor(sample_data(resfinder_PHY_model)$hospital_2),
               alias=as.factor(sample_data(resfinder_PHY_model)$alias),
               hospital_section=as.factor(sample_data(resfinder_PHY_model)$hospital_section),
               SSU_counts=as.factor(sample_data(resfinder_PHY_model)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_model)$hospital))

fit <- glm(SUM~hospital, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 15)

resfinder_sum <- ggplot(dfA, aes(x = hospital, y = Mean, label = alias)) + 
  geom_jitter(data = dfA, aes(x = hospital, y = SUM, color = hospital_2), size = 5, alpha = 0.9, 
              position = position_jitter(seed = 1)) +
  geom_point() + scale_color_manual(values=cols) + 
  geom_text(aes(x = hospital_2, y = SUM, label = alias), data = dfA, size = 2.5, vjust = -1.5, hjust = 0, family = "Times",
            position = position_jitter(seed = 1)) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 16, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 15, family = "Times"), 
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of ARGs (ResFinder), Hospital effluents
       
       ")
resfinder_sum 


# Category
#resfinder_PHY_model <- subset_samples(resfinder_PHY, category_for_model != "WA street gutter" &
#                                        category_for_model != "WA street gutter, sediment" &
#                                        category_for_model != "WA soil inbetween tanks")

resfinder_PHY_model <- subset_samples(resfinder_PHY, sample_type == "hospital effluent")
resfinder_PHY_model <- subset_samples(resfinder_PHY_model, country != "Spain")
                                      
df<-data.frame(SUM=sample_sums(resfinder_PHY_model),
               category_for_model=as.factor(sample_data(resfinder_PHY_model)$category_for_model),
               hospital_section=as.factor(sample_data(resfinder_PHY_model)$hospital_section),
               hospital=as.factor(sample_data(resfinder_PHY_model)$hospital),
               hospital_2=as.factor(sample_data(resfinder_PHY_model)$hospital_2),
               continent_level_1=as.factor(sample_data(resfinder_PHY_model)$continent_level_1),
               continent_level_2=as.factor(sample_data(resfinder_PHY_model)$continent_level_2),
               continent_level_3=as.factor(sample_data(resfinder_PHY_model)$continent_level_3),
               country=as.factor(sample_data(resfinder_PHY_model)$country),
               area_or_city=as.factor(sample_data(resfinder_PHY_model)$area_or_city),
               SSU_counts=as.factor(sample_data(resfinder_PHY_model)$SSU_counts),
               sample_type=as.factor(sample_data(resfinder_PHY_model)$sample_type),
               alias=as.factor(sample_data(resfinder_PHY_model)$alias))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)

fit <- glm(SUM~country + hospital_section + SSU_counts, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(country = "Tukey"))
summary(glht(glht.mod))

# Plot
fit <- glm(SUM~country, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 4)

resfinder_sum <- ggplot(dfA, aes(x = country, y = Mean)) +
  geom_jitter(data = dfA, aes(x = country, y = SUM, color = country), size = 5, alpha = 0.9) +
  geom_point() + scale_color_manual(values=cols) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 16, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 15, family = "Times"), 
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of ARGs (ResFinder), Hospital effluents
       
       ")
resfinder_sum 

# Category
resfinder_PHY_model <- subset_samples(resfinder_PHY, category_for_model != "WA street gutter" &
                                        category_for_model != "WA street gutter, sediment" &
                                        category_for_model != "WA soil inbetween tanks" &
                                        category_for_model != "WA empty hospital septic tank" &
                                        category_for_model != "WA in-patient feces")

df<-data.frame(SUM=sample_sums(resfinder_PHY_model),
               category_for_model=as.factor(sample_data(resfinder_PHY_model)$category_for_model),
               hospital_section=as.factor(sample_data(resfinder_PHY_model)$hospital_section),
               hospital=as.factor(sample_data(resfinder_PHY_model)$hospital),
               hospital_2=as.factor(sample_data(resfinder_PHY_model)$hospital_2),
               continent_level_1=as.factor(sample_data(resfinder_PHY_model)$continent_level_1),
               continent_level_2=as.factor(sample_data(resfinder_PHY_model)$continent_level_2),
               continent_level_3=as.factor(sample_data(resfinder_PHY_model)$continent_level_3),
               country=as.factor(sample_data(resfinder_PHY_model)$country),
               area_or_city=as.factor(sample_data(resfinder_PHY_model)$area_or_city),
               SSU_counts=as.factor(sample_data(resfinder_PHY_model)$SSU_counts),
               sample_type=as.factor(sample_data(resfinder_PHY_model)$sample_type),
               alias=as.factor(sample_data(resfinder_PHY_model)$alias))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)

fit <- glm(SUM~category_for_model + SSU_counts, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(category_for_model = "Tukey"))
summary(glht(glht.mod))

# Plot
fit <- glm(SUM~category_for_model, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 10)

resfinder_sum <- ggplot(dfA, aes(x = category_for_model, y = Mean)) +
  geom_jitter(data = dfA, aes(x = category_for_model, y = SUM, color = category_for_model), size = 5, alpha = 0.9) +
  geom_point() + scale_color_manual(values=cols) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 20, hjust = 0.5, vjust = 0.6, size = 18, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 15, family = "Times"),
        axis.text.y = element_text(size = 18, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of ARGs (ResFinder)
       
       ")
resfinder_sum 

# Hospital section
resfinder_PHY_model <- subset_samples(resfinder_PHY, sample_type == "hospital effluent")
resfinder_PHY_model <- subset_samples(resfinder_PHY_model, continent_level_1 == "West Africa")

df<-data.frame(SUM=sample_sums(resfinder_PHY_model),
               category_for_model=as.factor(sample_data(resfinder_PHY_model)$category_for_model),
               hospital_section=as.factor(sample_data(resfinder_PHY_model)$hospital_section),
               hospital=as.factor(sample_data(resfinder_PHY_model)$hospital),
               hospital_2=as.factor(sample_data(resfinder_PHY_model)$hospital_2),
               continent_level_1=as.factor(sample_data(resfinder_PHY_model)$continent_level_1),
               continent_level_2=as.factor(sample_data(resfinder_PHY_model)$continent_level_2),
               continent_level_3=as.factor(sample_data(resfinder_PHY_model)$continent_level_3),
               country=as.factor(sample_data(resfinder_PHY_model)$country),
               area_or_city=as.factor(sample_data(resfinder_PHY_model)$area_or_city),
               SSU_counts=as.factor(sample_data(resfinder_PHY_model)$SSU_counts),
               sample_type=as.factor(sample_data(resfinder_PHY_model)$sample_type),
               alias=as.factor(sample_data(resfinder_PHY_model)$alias))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)

fit <- glm(SUM~hospital + hospital_section + SSU_counts, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(hospital_section = "Tukey"))
summary(glht(glht.mod))

# Plot
fit <- glm(SUM~hospital_section, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 10)

resfinder_sum <- ggplot(dfA, aes(x = hospital_section, y = Mean)) +
  geom_jitter(data = dfA, aes(x = hospital_section, y = SUM, color = hospital_section), size = 5, alpha = 0.9) +
  geom_point() + scale_color_manual(values=cols) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 20, hjust = 0.5, vjust = 0.6, size = 18, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 15, family = "Times"),
        axis.text.y = element_text(size = 18, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of ARGs (ResFinder), Hospital effluent
       
       ")
resfinder_sum 

## include sample_type + country + SSU

#scale_y_log10(breaks = c(10^-2, 10^-1, 10^0, 10^1, 10^2), labels = trans_format("log10", math_format(10^.x)), limits=c(10^-2.5, 100))
```
## Gamma VF
```{r}
VFDB_PHY_model <- subset_samples(VFDB_PHY, sample_type == "hospital effluent")
# Exclude Spain
VFDB_PHY_model <- subset_samples(VFDB_PHY_model, country != "Spain")

df<-data.frame(SUM=sample_sums(VFDB_PHY_model),
               category_for_model=as.factor(sample_data(VFDB_PHY_model)$category_for_model),
               hospital_section=as.factor(sample_data(VFDB_PHY_model)$hospital_section),
               hospital=as.factor(sample_data(VFDB_PHY_model)$hospital),
               hospital_2=as.factor(sample_data(VFDB_PHY_model)$hospital_2),
               continent_level_1=as.factor(sample_data(VFDB_PHY_model)$continent_level_1),
               continent_level_2=as.factor(sample_data(VFDB_PHY_model)$continent_level_2),
               continent_level_3=as.factor(sample_data(VFDB_PHY_model)$continent_level_3),
               country=as.factor(sample_data(VFDB_PHY_model)$country),
               area_or_city=as.factor(sample_data(VFDB_PHY_model)$area_or_city),
               SSU_counts=as.factor(sample_data(VFDB_PHY_model)$SSU_counts),
               sample_type=as.factor(sample_data(VFDB_PHY_model)$sample_type),
               alias=as.factor(sample_data(VFDB_PHY_model)$alias))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
## Explore data
#plot(df$SUM~df$hospital, cex.axis = 0.5, las = 2)

# Fit model with all variables  
fit <- glm(SUM~continent_level_3 + SSU_counts, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(continent_level_3 = "Tukey"))
summary(glht(glht.mod))

# Plot
fit <- glm(SUM~continent_level_3, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 15)

VF_sum <- ggplot(dfA, aes(x = continent_level_3, y = Mean, label = alias)) + 
  geom_jitter(data = dfA, aes(x = continent_level_3, y = SUM, color = continent_level_3), size = 5, alpha = 0.9, 
              position = position_jitter(seed = 1)) +
  geom_point() + scale_color_manual(values=cols) + 
    geom_text(aes(x = continent_level_3, y = SUM, label = alias), data = dfA, size = 4, vjust = -1.5, hjust = 0, family = "Times",
            position = position_jitter(seed = 1)) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 16, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 15, family = "Times"), 
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Sum relative abundance of virulence genes (VFDB)
       
       ")
VF_sum 
```
## Model for crassphage
```{r}
crass_PHY_model <- subset_samples(crass_PHY, category_for_model != "WA street gutter" & 
                                        category_for_model != "WA street gutter, sediment" & 
                                        category_for_model != "WA soil inbetween tanks" & 
                                        category_for_model != "WA empty hospital septic tank")

df<-data.frame(SUM=sample_sums(crass_PHY_model),
               category=as.factor(sample_data(crass_PHY_model)$category_for_model),
               hospital_section=as.factor(sample_data(crass_PHY_model)$hospital_section),
               hospital=as.factor(sample_data(crass_PHY_model)$hospital),
               continent_level_1=as.factor(sample_data(crass_PHY_model)$continent_level_1),
               continent_level_2=as.factor(sample_data(crass_PHY_model)$continent_level_2),
               country=as.factor(sample_data(crass_PHY_model)$country),
               SSU_counts=as.factor(sample_data(crass_PHY_model)$SSU_counts),
               area=as.factor(sample_data(crass_PHY_model)$area_or_city))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
#df$crassphage <- as.character(df$crassphage)
#df$crassphage <- as.numeric(df$crassphage)
# Fit model with all variables  
fit <- glm(SUM~country + SSU_counts + hospital_section, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(country = "Tukey"))
summary(glht(glht.mod))

# Plot
df<-data.frame(SUM=sample_sums(crass_PHY),
               country=as.factor(sample_data(crass_PHY)$country))

fit <- glm(SUM~country, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 15)

crass_sum <- ggplot(dfA, aes(x = country, y = Mean)) + scale_color_manual(values=cols) + 
  geom_line() + 
  geom_jitter(data = dfA, aes(x = country, y = SUM, color = country), size = 5, alpha = 0.5, width = 0.3) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.3, lwd = 0.3) + geom_point(size = 0.9) + 
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 15, family = "Times"), 
        axis.title.y = element_text(size = 15, family = "Times"), 
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Sum abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  scale_y_log10(breaks = c(10^-3, 10^-2, 10^-1, 10^0, 10^1), labels = trans_format("log10", math_format(10^.x)), limits=c(10^-3, 10^1.25)) +
  labs(title = "Sum relative abundance of crassphage (normalized to SSU), Country
       
       ")
crass_sum
```
## Diversity metrics
```{r}
# Metaxa2
PHY <- prune_taxa(taxa_sums(metaxa_PHY) > 0, metaxa_PHY)
## Almost all
# Exclude singletons
#PHY <- subset_samples(PHY, category_for_model != "WA empty hospital septic tank" &
#                        category_for_model != "WA street gutter" &
#                        category_for_model != "WA street gutter, sediment" &
#                        category_for_model != "WA soil inbetween tanks")

## By hopspital
#PHY <- subset_samples(PHY, sample_type == "hospital effluent")

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$category_for_model) # get the variables
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

## By hospital
#vars <- levels(PHY.meta$hospital) # get the variables
#vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
#print(vars.pairs)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 15)
shannon <- ggboxplot(PHY.meta, x = "category_for_model", y = "Shannon", fill = "category_for_model", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 40, vjust = 1, size = 16, hjust = 0.95, family = "Times"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20),
        legend.position = "None",
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity, Shannon index (Metaxa2)")
print(shannon)

a <- shannon + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = F, ref.group = ".all.")

observed <- ggboxplot(PHY.meta, x = "category_for_model", y = "Observed", fill = "category_for_model", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 40, vjust = 1, size = 16, hjust = 0.95, family = "Times"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20),
        legend.position = "None",
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity, hospitals, Observed (Metaxa2)
       
       ")
print(observed)

b <- observed + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = TRUE, ref.group = ".all.")


# ResFinder
OTU_resfinder_counts <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder_counts))
OTU_resfinder_counts <- OTU_resfinder_counts[,match]
all(colnames(OTU_resfinder_counts) == rownames(metadata))

# Tax_table (Cluster names added manually in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder_counts), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder_counts) == clusters_tax_table_resfinder$Gene)

# Reload to get new row numbers
write.table(OTU_resfinder_counts, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_counts.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_resfinder_counts <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_counts.txt", row.names=NULL)
OTU_resfinder_counts$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

# Combine to phyloseq object
resfinder_PHY_counts <- phyloseq(otu_table(OTU_resfinder_counts, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY_counts = subset_samples(resfinder_PHY_counts, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
resfinder_PHY_counts <- subset_samples(resfinder_PHY_counts, country != "UK")


PHY <- prune_taxa(taxa_sums(resfinder_PHY_counts) > 0, resfinder_PHY_counts)
## Almost all
# Exclude singletons
#PHY <- subset_samples(PHY, category_for_model != "WA empty hospital septic tank" &
#                        category_for_model != "WA street gutter" &
#                        category_for_model != "WA street gutter, sediment" &
#                        category_for_model != "WA soil inbetween tanks")

## By hopspital
PHY <- subset_samples(PHY, sample_type == "hospital effluent")
PHY <- subset_samples(PHY, country == "Benin" | country == "Burkina Faso" | country == "Finland")

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

# make a pairwise list that we want to compare.
vars <- levels(PHY.meta$hospital) # get the variables
vars.pairs <- combn(seq_along(vars), 2, simplify = FALSE, FUN = function(i)vars[i])
print(vars.pairs)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 15)
shannon <- ggboxplot(PHY.meta, x = "hospital", y = "Shannon", fill = "hospital", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 40, vjust = 0.8, size = 16, hjust = 0.8, family = "Times"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20),
        legend.position = "None",
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity of ARGs, hospitals, Shannon index (ResFinder)
       
       ")
print(shannon)

a <- shannon + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = TRUE, ref.group = ".all.")

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98"), 13)
observed <- ggboxplot(PHY.meta, x = "hospital", y = "Observed", fill = "hospital", add = "jitter") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 40, vjust = 0.8, size = 16, hjust = 0.8, family = "Times"), 
        axis.title.x = element_blank(),
        axis.text.y = element_text(size = 20),
        legend.position = "None",
        axis.title.y = element_text(size = 22, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) +
  labs(title = "Alpha diversity of ARGs, hospitals, Observed (ResFinder)
       
       ")
print(observed)

b <- observed + stat_compare_means(comparisons = vars.pairs, 
                             label = "p.signif", 
                             hide.ns = TRUE, ref.group = ".all.")

```
## Gamma
```{r}
PHY <- prune_taxa(taxa_sums(metaxa_PHY) > 0, metaxa_PHY)
PHY <- subset_samples(PHY, category_for_model != "WA empty hospital septic tank" &
                        category_for_model != "WA street gutter" &
                        category_for_model != "WA street gutter, sediment" &
                        category_for_model != "WA soil inbetween tanks")
#PHY <- subset_samples(PHY, sample_type == "hospital effluent")
#PHY <- subset_samples(PHY, country == "Benin" | country == "Burkina Faso")

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed

df<-data.frame(SUM=PHY.meta$Shannon,
               category_for_model=as.factor(PHY.meta$category_for_model),
               SSU_counts=as.factor(PHY.meta$SSU_counts),
               sample_type=as.factor(PHY.meta$country),
               hospital=as.factor(PHY.meta$hospital),
               country=as.factor(PHY.meta$country),
               hospital_section=as.factor(PHY.meta$hospital_section))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
## Explore data
#plot(df$SUM~df$hospital, cex.axis = 0.5, las = 2)

# Fit model with all variables  
fit <- glm(SUM~SSU_counts + category_for_model, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(category_for_model = "Tukey"))
summary(glht(glht.mod))

# Plot
fit <- glm(SUM~category_for_model, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 15)

metaxa_sum <- ggplot(dfA, aes(x = category_for_model, y = Mean)) + 
  geom_jitter(data = dfA, aes(x = category_for_model, y = SUM, color = category_for_model), size = 5, alpha = 0.9, 
              position = position_jitter(seed = 1)) +
  geom_point() + scale_color_manual(values=cols) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 0.8, size = 16, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 24, family = "Times"),
        axis.text.y = element_text(size = 20, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Shannon", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Alpha diversity (Metaxa2), Shannon index
       
       
       ")
metaxa_sum


# ResFinder
OTU_resfinder_counts <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder_counts))
OTU_resfinder_counts <- OTU_resfinder_counts[,match]
all(colnames(OTU_resfinder_counts) == rownames(metadata))

# Tax_table (Cluster names added manually in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder_counts), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder_counts) == clusters_tax_table_resfinder$Gene)

# Reload to get new row numbers
write.table(OTU_resfinder_counts, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_counts.txt", 
            row.names=T, sep = "\t", col.names = T)
OTU_resfinder_counts <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/OTU_resfinder_counts.txt", row.names=NULL)
OTU_resfinder_counts$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

# Combine to phyloseq object
resfinder_PHY_counts <- phyloseq(otu_table(OTU_resfinder_counts, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY_counts = subset_samples(resfinder_PHY_counts, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
resfinder_PHY_counts <- subset_samples(resfinder_PHY_counts, country != "UK")


PHY <- prune_taxa(taxa_sums(resfinder_PHY_counts) > 0, resfinder_PHY_counts)
## Almost all
# Exclude singletons
#PHY <- subset_samples(PHY, category_for_model != "WA empty hospital septic tank" &
#                        category_for_model != "WA street gutter" &
#                        category_for_model != "WA street gutter, sediment" &
#                        category_for_model != "WA soil inbetween tanks")
PHY <- subset_samples(PHY, sample_type == "hospital effluent")
PHY <- subset_samples(PHY, country == "Benin" | country == "Burkina Faso")

tab <- microbiome::alpha(PHY, index = "all")
kable(head(tab))

PHY.meta <- meta(PHY)
kable(head(PHY.meta))

PHY.meta$Shannon <- tab$diversity_shannon 
PHY.meta$Observed <- tab$observed


df<-data.frame(SUM=PHY.meta$Shannon,
               category_for_model=as.factor(PHY.meta$category_for_model),
               SSU_counts=as.factor(PHY.meta$SSU_counts),
               sample_type=as.factor(PHY.meta$country),
               hospital=as.factor(PHY.meta$hospital),
               country=as.factor(PHY.meta$country),
               hospital_section=as.factor(PHY.meta$hospital_section))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
## Explore data
#plot(df$SUM~df$hospital, cex.axis = 0.5, las = 2)

# Fit model with all variables  
fit <- glm(SUM~SSU_counts + hospital + hospital_section, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(hospital = "Tukey"))
summary(glht(glht.mod))

# Plot
fit <- glm(SUM~hospital, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 15)

resfinder_sum <- ggplot(dfA, aes(x = hospital, y = Mean)) + 
  geom_jitter(data = dfA, aes(x = hospital, y = SUM, color = hospital), size = 5, alpha = 0.9, 
              position = position_jitter(seed = 1)) +
  geom_point() + scale_color_manual(values=cols) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 0.8, size = 16, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 24, family = "Times"),
        axis.text.y = element_text(size = 20, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "Shannon", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Alpha diversity (ResFinder), Shannon index, Hospital effluents in WA
       
       
       ")
resfinder_sum
```
## Heatmap, ResFinder
```{r}
df<- OTU_resfinder
#df1 <- df + 1
df1 <-df

resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
df1_length_norm <- df1/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
df1_length_SSU_norm <- t(t(df1_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(df1_length_SSU_norm))

clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt")
all(clusters_tax_table_resfinder$Gene == rownames(df1_length_SSU_norm))

# Reload to get new row numbers
write.table(df1_length_SSU_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_OTU_resfinder.txt", 
            row.names=T, sep = "\t", col.names = T)
heat_OTU_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_OTU_resfinder.txt", row.names=NULL)
heat_OTU_resfinder$row.names<-NULL
heat_OTU_resfinder$Cluster_name <- NULL

# Reload to get new row numbers
write.table(heat_clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
heat_clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=NULL)

# Create phyloseq
heat_resfinder_PHY <- phyloseq(otu_table(heat_OTU_resfinder, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(heat_clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
heat_resfinder_PHY = subset_samples(heat_resfinder_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
heat_resfinder_PHY <- subset_samples(heat_resfinder_PHY, country != "UK")
```
# Merge genes with same cluster name
```{r}
# normalization with lengths
df<- OTU_resfinder
#df1 <- df + 1
df1 <-df

resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
df1_length_norm <- df1/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
df1_length_SSU_norm <- t(t(df1_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(df1_length_SSU_norm))


clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt")
all(clusters_tax_table_resfinder$Gene == rownames(df1_length_SSU_norm))

Class <- as.data.frame(clusters_tax_table_resfinder$Class)
Cluster_name <- as.data.frame(clusters_tax_table_resfinder$Cluster_name)
Gene <- as.data.frame(clusters_tax_table_resfinder$Gene)

df1_length_SSU_norm <- cbind(df1_length_SSU_norm, Class)
df1_length_SSU_norm <- cbind(df1_length_SSU_norm, Cluster_name)
df1_length_SSU_norm <- cbind(df1_length_SSU_norm, Gene)

names(df1_length_SSU_norm)[names(df1_length_SSU_norm) == "clusters_tax_table_resfinder$Class"] <- "Class"
names(df1_length_SSU_norm)[names(df1_length_SSU_norm) == "clusters_tax_table_resfinder$Cluster_name"] <- "Cluster_name"
names(df1_length_SSU_norm)[names(df1_length_SSU_norm) == "clusters_tax_table_resfinder$Gene"] <- "Gene"

all(df1_length_SSU_norm$Gene == clusters_tax_table_resfinder$Gene)

# Merge
merged_df <- ddply(df1_length_SSU_norm,"Cluster_name",numcolwise(sum))
# Remove duplicates
uniq_clust <- clusters_tax_table_resfinder[!duplicated(clusters_tax_table_resfinder[ , c("Cluster_name")]),]
# Reorder to match OTU
match <- match(uniq_clust$Cluster_name, merged_df$Cluster_name)
merged_df <- merged_df[match,]
all(merged_df$Cluster_name == uniq_clust$Cluster_name)

# Tax table
match <- match(merged_df$Cluster_name, clusters_tax_table_resfinder$Cluster_name)
heat_clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
identical(merged_df$Cluster_name, heat_clusters_tax_table_resfinder$Cluster_name)

# Reload to get new row numbers
write.table(merged_df, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_OTU_resfinder.txt", 
            row.names=T, sep = "\t", col.names = T)
heat_OTU_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_OTU_resfinder.txt", row.names=NULL)
heat_OTU_resfinder$row.names<-NULL
heat_OTU_resfinder$Cluster_name <- NULL

# Reload to get new row numbers
write.table(heat_clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
heat_clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=NULL)

# Create phyloseq
heat_resfinder_PHY <- phyloseq(otu_table(heat_OTU_resfinder, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(heat_clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
heat_resfinder_PHY <- subset_samples(heat_resfinder_PHY, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
heat_resfinder_PHY <- subset_samples(heat_resfinder_PHY, country != "UK")
```
# Plot, selected, all samples
```{r}
# Exctract ARGs
heat_resfinder_PHY_Cluster_name <- tax_glom(heat_resfinder_PHY, taxrank = "Cluster_name")

selected <- subset_taxa(heat_resfinder_PHY_Cluster_name, 
                        Cluster_name == "blaOXA-370_1_clust" | Cluster_name == "blaIMP-58_1_clust" |
                        Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaCTX-M-211_1_clust" | # ctx-m-15
                        Cluster_name == "blaCTX-M-4_1_clust" | Cluster_name == "blaCTX-M-110_1_clust" | # ctx-m-2; ctx-m-14
                        Cluster_name == "blaKPC-34_1_clust" | Cluster_name == "catB3_2_clust" |        # kpc-2
                        Cluster_name == "blaGES-1_1_clust" | Cluster_name == "blaVEB-1_1_clust" |
                        Cluster_name == "cfxA_1_clust" | Cluster_name == "blaCMY-150_2_clust" |        # cmy-2
                        Cluster_name == "aph(6)-Id_1_clust" | Cluster_name == "aadA2_1_clust" |
                        Cluster_name == "cmlA1_1_clust" | Cluster_name == "mcr-3.1_1_clust" | 
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "erm(B)_18_clust" |
                        Cluster_name == "mcr-1.11_1_clust" | Cluster_name == "qnrA1_1_clust" |
                        Cluster_name == "nimA_1" | Cluster_name == "nimE_1_clust" |
                        Cluster_name == "sul1_11_clust" | Cluster_name == "sul2_7_clust" |
                        Cluster_name == "tet(A)_6_clust" | Cluster_name == "tet(S/M)_1_clust" | 
                        Cluster_name == "dfrA1_1_clust" | Cluster_name == "erm(33)_1" |
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "blaOXA-368_1_clust" |     # oxa-10
                        Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaOXA-392_1_clust"|
                        Cluster_name == "mcr-5.1_1_clust")   # oxa-1

# OTU matrix
heat_OTU = as(otu_table(selected), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Tax table matrix
heat_tax = as(tax_table(selected), "matrix")

# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# Into dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
match <- match(rownames(heat.df), rownames(heat_tax.df))
temp <- heat_tax.df[match,]
rownames(heat.df) <- temp$V3

# Annotations
metadata_row <- data.frame(temp$V2, row.names=rownames(heat.df))
colnames(metadata_row) <- c("Antibiotic class")

metadata_col <- as.matrix(sample_data(selected)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(selected)[["alias"]])


ann_colors <- list(metadata_col = c("WA hospital effluent" = "#FD8FD9", 
                                    "WA treated" = "#FB2A38", 
                                    "WA empty hospital septic tank" = "navy",
                                    "WA drinking water" = "#BFD678",
                                    "North Eu treated" = "#921631", 
                                    "WA street gutter" = "#84D2DD", 
                                    "WA street gutter, sediment" = "darkorange",
                                    "WA soil inbetween tanks" = "#568643",
                                    "WA in-patient feces" = "#7B5E49", 
                                    "North Eu wwtp influent" = "#77DDB3", 
                                    "WA river" = "#F79537", 
                                    "Eu other hospital effluent" = "#7504AA", 
                                    "Eu other wwtp influent" = "#DDCC77", 
                                    "North Eu hospital effluent" = "#546C7F"))

rownames(heat.df) <- c("aadA2_1_clust__(aadA)", "aph(6)-Id_1_clust__(aph(6)-Id)", "blaNDM-18_1_clust__(blaNDM-1)", 
                     "blaOXA-368_1_clust__(blaOXA-10)", "blaOXA-370_1_clust__(blaOXA-48)", "blaOXA-392_1_clust__(blaOXA-1)",
                     "blaIMP-58_1_clust__(blaIMP)", "blaCMY-150_2_clust__(blaCMY-2)", "blaCTX-M-211_1_clust__(Group I blaCTX-M)",
                     "blaCTX-M-4_1_clust__(Group II blaCTX-M)", "blaCTX-M-110_1_clust__(Group III blaCTX-M)", 
                     "blaKPC-34_1_clust__(blaKPC-2)", "blaVEB-1_1_clust__(blaVEB)", "cfxA_1_clust__(cfxA)", 
                     "blaVIM-48_1_clust__(blaVIM)", "blaGES-1_1_clust__(blaGES)", "mcr-1.11_1_clust__(mcr1.1)", 
                     "mcr-3.1_1_clust__(mcr3)", "mcr-5.1_1_clust__(mcr5)", "erm(B)_18_clust__(ermB)", "mph(E)_1_clust__(mphE)", 
                     "erm(33)_1__(ermA)", "nimA_1__(nimA)", "nimE_1_clust__(nimE)", 
                     "catB3_2_clust__(catB3)", "cmlA1_1_clust__(cmlA)", "qnrA1_1_clust__(qnrA)", 
                     "sul1_11_clust__(sul1)", "sul2_7_clust__(sul2)", "tet(A)_6_clust__(tetA)", 
                     "tet(S/M)_1_clust__(tetM)", "dfrA1_1_clust__(dfrA)")

## Plot log
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

heat <- pheatmap(log10(heat.df + 0.000001), cluster_rows = F, cluster_cols = T, 
                        cellwidth = 12, cellheight = 20, 
                        border_color = "white",
                        fontsize=5,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "", 
                        angle_col = 90, legend = TRUE, fontsize_row = 12, fontsize_col = 9, 
                        labels_row = as.expression(newnames),
                        #gaps_col = c(1, 9, 17),
                        filename = "selected_all_heat.png",
                    #    annotation_row = metadata_row,
                        annotation_col = metadata_col,
                        annotation_colors = ann_colors,
                        treeheight_row = 0,
                        cutree_cols = 14,
                        treeheight_col = 30,
                        )
heat
```
# Exclude other than sequenced here, selected
```{r}
heat_resfinder_PHY_Cluster_name <- tax_glom(heat_resfinder_PHY, taxrank = "Cluster_name")

heat_resfinder_PHY_Cluster_name <- subset_samples(heat_resfinder_PHY_Cluster_name, 
                                                  country == "Benin" | country == "Burkina Faso" | country == "Finland")
heat_resfinder_PHY_Cluster_name <- subset_samples(heat_resfinder_PHY_Cluster_name, 
                                                  alias != "FIN25" & alias != "FIN25_1" & alias != "FIN25_2")
selected <- subset_taxa(heat_resfinder_PHY_Cluster_name, 
                        Cluster_name == "blaOXA-370_1_clust" | Cluster_name == "blaIMP-58_1_clust" |
                        Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaCTX-M-211_1_clust" | # ctx-m-15
                        Cluster_name == "blaCTX-M-4_1_clust" | Cluster_name == "blaCTX-M-110_1_clust" | # ctx-m-2; ctx-m-14
                        Cluster_name == "blaKPC-34_1_clust" | Cluster_name == "catB3_2_clust" |        # kpc-2
                        Cluster_name == "blaGES-1_1_clust" | Cluster_name == "blaVEB-1_1_clust" |
                        Cluster_name == "cfxA_1_clust" | Cluster_name == "blaCMY-150_2_clust" |        # cmy-2
                        Cluster_name == "aph(6)-Id_1_clust" | Cluster_name == "aadA2_1_clust" |
                        Cluster_name == "cmlA1_1_clust" | Cluster_name == "mcr-3.1_1_clust" | 
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "erm(B)_18_clust" |
                        Cluster_name == "mcr-1.11_1_clust" | Cluster_name == "qnrA1_1_clust" |
                        Cluster_name == "nimA_1" | Cluster_name == "nimE_1_clust" |
                        Cluster_name == "sul1_11_clust" | Cluster_name == "sul2_7_clust" |
                        Cluster_name == "tet(A)_6_clust" | Cluster_name == "tet(S/M)_1_clust" | 
                        Cluster_name == "dfrA1_1_clust" | Cluster_name == "erm(33)_1" |
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "blaOXA-368_1_clust" |     # oxa-10
                        Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaOXA-392_1_clust")   # oxa-1

# OTU matrix
heat_OTU = as(otu_table(selected), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Tax table matrix
heat_tax = as(tax_table(selected), "matrix")

# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# Into dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
match <- match(rownames(heat.df), rownames(heat_tax.df))
temp <- heat_tax.df[match,]
rownames(heat.df) <- temp$V3

# Annotations
metadata_row <- data.frame(temp$V2, row.names=rownames(heat.df))
colnames(metadata_row) <- c("Antibiotic class")

metadata_col <- as.matrix(sample_data(selected)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(selected)[["alias"]])

metadata_col <- as.matrix(sample_data(selected)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(selected)[["alias"]])

ann_colors <- list(metadata_col = c("WA hospital effluent" = "#FD8FD9", 
                                    "WA treated" = "#FB2A38", 
                                    "WA empty hospital septic tank" = "navy",
                                    "WA drinking water" = "#BFD678",
                                    "WA street gutter" = "#84D2DD", 
                                    "WA street gutter, sediment" = "darkorange",
                                    "WA soil inbetween tanks" = "#568643",
                                    "WA in-patient feces" = "#7B5E49", 
                                    "North Eu hospital effluent" = "#77DDB3"))

rownames(heat.df) <- c("aadA2_1_clust__(aadA)", "aph(6)-Id_1_clust__(aph(6)-Id)", "blaNDM-18_1_clust__(blaNDM-1)", 
                     "blaOXA-368_1_clust__(blaOXA-10)", "blaOXA-370_1_clust__(blaOXA-48)", "blaOXA-392_1_clust__(blaOXA-1)",
                     "blaIMP-58_1_clust__(blaIMP)", "blaCMY-150_2_clust__(blaCMY-2)", "blaCTX-M-211_1_clust__(Group I blaCTX-M)",
                     "blaCTX-M-4_1_clust__(Group II blaCTX-M)", "blaCTX-M-110_1_clust__(Group III blaCTX-M)", 
                     "blaKPC-34_1_clust__(blaKPC-2)", "blaVEB-1_1_clust__(blaVEB)", "cfxA_1_clust__(cfxA)", 
                     "blaVIM-48_1_clust__(blaVIM)", "blaGES-1_1_clust__(blaGES)", "mcr-1.11_1_clust__(mcr1.1)", 
                     "mcr-3.1_1_clust__(mcr3)", "erm(B)_18_clust__(ermB)", "mph(E)_1_clust__(mphE)", 
                     "erm(33)_1__(ermA)", "nimA_1__(nimA)", "nimE_1_clust__(nimE)", 
                     "catB3_2_clust__(catB3)", "cmlA1_1_clust__(cmlA)", "qnrA1_1_clust__(qnrA)", 
                     "sul1_11_clust__(sul1)", "sul2_7_clust__(sul2)", "tet(A)_6_clust__(tetA)", 
                     "tet(S/M)_1_clust__(tetM)", "dfrA1_1_clust__(dfrA)")

newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

heat <- pheatmap(log10(heat.df + 0.000001), cluster_rows = F, cluster_cols = T, 
                        cellwidth = 10, cellheight = 18, 
                        border_color = "white",
                        fontsize=5,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "", 
                        angle_col = 90, legend = TRUE, fontsize_row = 12, fontsize_col = 9, 
                        labels_row = as.expression(newnames),
                        gaps_row = c(2, 16, 18, 21, 23, 25, 26, 28, 30, 31),
                        filename = "sub_selected_heat.png",
                        annotation_row = metadata_row,
                        annotation_col = metadata_col,
                        annotation_colors = ann_colors,
                        treeheight_row = 0,
                        cutree_cols = 10,
                        treeheight_col = 30,
                        )
heat
```
## Benin / BF
# Exclude other than sequenced here, selected
```{r}
heat_resfinder_PHY_Cluster_name <- tax_glom(heat_resfinder_PHY, taxrank = "Cluster_name")

heat_resfinder_PHY_Cluster_name <- subset_samples(heat_resfinder_PHY_Cluster_name, 
                                                  country == "Benin")
selected <- subset_taxa(heat_resfinder_PHY_Cluster_name, 
                        Cluster_name == "blaOXA-370_1_clust" | Cluster_name == "blaIMP-58_1_clust" |
                        Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaCTX-M-211_1_clust" | # ctx-m-15
                        Cluster_name == "blaCTX-M-4_1_clust" | Cluster_name == "blaCTX-M-110_1_clust" | # ctx-m-2; ctx-m-14
                        Cluster_name == "blaKPC-34_1_clust" | Cluster_name == "catB3_2_clust" |        # kpc-2
                        Cluster_name == "blaGES-1_1_clust" | Cluster_name == "blaVEB-1_1_clust" |
                        Cluster_name == "cfxA_1_clust" | Cluster_name == "blaCMY-150_2_clust" |        # cmy-2
                        Cluster_name == "aph(6)-Id_1_clust" | Cluster_name == "aadA2_1_clust" |
                        Cluster_name == "cmlA1_1_clust" | Cluster_name == "mcr-3.1_1_clust" | 
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "erm(B)_18_clust" |
                        Cluster_name == "mcr-1.11_1_clust" | Cluster_name == "qnrA1_1_clust" |
                        Cluster_name == "nimA_1" | Cluster_name == "nimE_1_clust" |
                        Cluster_name == "sul1_11_clust" | Cluster_name == "sul2_7_clust" |
                        Cluster_name == "tet(A)_6_clust" | Cluster_name == "tet(S/M)_1_clust" | 
                        Cluster_name == "dfrA1_1_clust" | Cluster_name == "erm(33)_1" |
                        Cluster_name == "mph(E)_1_clust" | Cluster_name == "blaOXA-368_1_clust" |     # oxa-10
                        Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaOXA-392_1_clust")   # oxa-1

# OTU matrix
heat_OTU = as(otu_table(selected), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Tax table matrix
heat_tax = as(tax_table(selected), "matrix")

# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# Into dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
match <- match(rownames(heat.df), rownames(heat_tax.df))
temp <- heat_tax.df[match,]
rownames(heat.df) <- temp$V3

# Annotations
metadata_row <- data.frame(temp$V2, row.names=rownames(heat.df))
colnames(metadata_row) <- c("Antibiotic class")

metadata_col <- as.matrix(sample_data(selected)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(selected)[["alias"]])

metadata_col <- as.matrix(sample_data(selected)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(selected)[["alias"]])

ann_colors <- list(metadata_col = c("WA hospital effluent" = "#FD8FD9", 
                                    "WA treated" = "#FB2A38",
                                    "WA empty hospital septic tank" = "navy",
                                    "WA drinking water" = "#BFD678",
                                    "WA street gutter" = "#84D2DD", 
                                    "WA street gutter, sediment" = "darkorange",
                                    "WA soil inbetween tanks" = "#568643",
                                    "WA in-patient feces" = "#7B5E49"
                                    #"North Eu hospital effluent" = "#77DDB3"
                                    ))

rownames(heat.df) <- c("aadA2_1_clust__(aadA)", "aph(6)-Id_1_clust__(aph(6)-Id)", "blaNDM-18_1_clust__(blaNDM-1)", 
                     "blaOXA-368_1_clust__(blaOXA-10)", "blaOXA-370_1_clust__(blaOXA-48)", "blaOXA-392_1_clust__(blaOXA-1)",
                     "blaIMP-58_1_clust__(blaIMP)", "blaCMY-150_2_clust__(blaCMY-2)", "blaCTX-M-211_1_clust__(Group I blaCTX-M)",
                     "blaCTX-M-4_1_clust__(Group II blaCTX-M)", "blaCTX-M-110_1_clust__(Group III blaCTX-M)", 
                     "blaKPC-34_1_clust__(blaKPC-2)", "blaVEB-1_1_clust__(blaVEB)", "cfxA_1_clust__(cfxA)", 
                     "blaVIM-48_1_clust__(blaVIM)", "blaGES-1_1_clust__(blaGES)", "mcr-1.11_1_clust__(mcr1.1)", 
                     "mcr-3.1_1_clust__(mcr3)", "erm(B)_18_clust__(ermB)", "mph(E)_1_clust__(mphE)", 
                     "erm(33)_1__(ermA)", "nimA_1__(nimA)", "nimE_1_clust__(nimE)", 
                     "catB3_2_clust__(catB3)", "cmlA1_1_clust__(cmlA)", "qnrA1_1_clust__(qnrA)", 
                     "sul1_11_clust__(sul1)", "sul2_7_clust__(sul2)", "tet(A)_6_clust__(tetA)", 
                     "tet(S/M)_1_clust__(tetM)", "dfrA1_1_clust__(dfrA)")

newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

heat <- pheatmap(log10(heat.df + 0.000001), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 15, cellheight = 15, 
                        border_color = "white",
                        fontsize=5,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "", 
                        angle_col = 90, legend = TRUE, fontsize_row = 12, fontsize_col = 9, 
                        labels_row = as.expression(newnames),
                        gaps_row = c(2, 16, 18, 21, 23, 25, 26, 28, 30, 31),
                        filename = "Benin_selected_heat.png",
                        annotation_row = metadata_row,
                        annotation_col = metadata_col,
                        annotation_colors = ann_colors,
                        treeheight_row = 0,
                        cutree_cols = 10,
                        treeheight_col = 30,
                        )
heat
```


# Exclude other than sequenced here, top100
```{r}
# 100 most abundant
heat_resfinder_PHY_Cluster_name <- tax_glom(heat_resfinder_PHY, taxrank = "Cluster_name")

heat_resfinder_PHY_abun <- prune_taxa(names(sort(taxa_sums(heat_resfinder_PHY_Cluster_name), TRUE)[1:100]), 
    heat_resfinder_PHY_Cluster_name)

top100 <- subset_samples(heat_resfinder_PHY_abun, country == "Benin" | country == "Burkina Faso" | country == "Finland")
top100 = subset_samples(top100, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171" & alias != "FIN25" & alias != "FIN25_1" & alias != "FIN25_2")

# OTU matrix
heat_OTU = as(otu_table(top100), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Tax table matrix
heat_tax = as(tax_table(top100), "matrix")

# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# Into dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
match <- match(rownames(heat.df), rownames(heat_tax.df))
temp <- heat_tax.df[match,]
rownames(heat.df) <- temp$V3

# Annotations
metadata_row <- data.frame(temp$V2, row.names=rownames(heat.df))
colnames(metadata_row) <- c("Antibiotic class")

metadata_col <- as.matrix(sample_data(top100)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(top100)[["alias"]])

metadata_col <- as.matrix(sample_data(top100)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(top100)[["alias"]])

ann_colors <- list(metadata_col = c("WA hospital effluent" = "#FD8FD9", 
                                    "WA treated" = "#FB2A38", 
                                    "WA empty hospital septic tank" = "navy",
                                    "WA drinking water" = "#BFD678",
                                    "WA street gutter" = "#84D2DD", 
                                    "WA street gutter, sediment" = "darkorange",
                                    "WA soil inbetween tanks" = "#568643",
                                    "WA in-patient feces" = "#7B5E49",
                                    "North Eu hospital effluent" = "#77DDB3"
                                    ))
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

heat <- pheatmap(log10(heat.df + 0.000001), cluster_rows = F, cluster_cols = T, 
                        cellwidth = 12, cellheight = 20, 
                        border_color = "white",
                        fontsize=5,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "", 
                        angle_col = 90, legend = TRUE, fontsize_row = 12, fontsize_col = 9, 
                        labels_row = as.expression(newnames),
                        #gaps_col = c(1, 9, 17),
                        filename = "top100_heat.png",
                    #    annotation_row = metadata_row,
                        annotation_col = metadata_col,
                        annotation_colors = ann_colors,
                        treeheight_row = 0,
                        cutree_cols = 14,
                        treeheight_col = 30,
                        )
heat
```
## All everything
```{r}
#heat_resfinder_PHY_Cluster_name <- tax_glom(heat_resfinder_PHY, taxrank = "Cluster_name")

all <- subset_samples(heat_resfinder_PHY_Gene, country == "Benin" | country == "Burkina Faso" | country == "Finland")
all = subset_samples(all, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171" & alias != "FIN25" & alias != "FIN25_1" & alias != "FIN25_2")

# OTU matrix
heat_OTU = as(otu_table(all), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Tax table matrix
heat_tax = as(tax_table(all), "matrix")

# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# Into dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
match <- match(rownames(heat.df), rownames(heat_tax.df))
temp <- heat_tax.df[match,]
rownames(heat.df) <- temp$V3

# Annotations
metadata_row <- data.frame(temp$V2, row.names=rownames(heat.df))
colnames(metadata_row) <- c("Antibiotic class")

metadata_col <- as.matrix(sample_data(all)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(all)[["alias"]])

metadata_col <- as.matrix(sample_data(all)[["category_for_model"]])
colnames(metadata_col) <- c("category")
metadata_col <- as.factor(metadata_col)
metadata_col <- data.frame(metadata_col)
rownames(metadata_col) <- as.matrix(sample_data(all)[["alias"]])

ann_colors <- list(metadata_col = c("WA hospital effluent" = "#FD8FD9", 
                                    "WA treated" = "#FB2A38", 
                                    "WA empty hospital septic tank" = "navy",
                                    "WA drinking water" = "#BFD678",
                                    "WA street gutter" = "#84D2DD", 
                                    "WA street gutter, sediment" = "darkorange",
                                    "WA soil inbetween tanks" = "#568643",
                                    "WA in-patient feces" = "#7B5E49",
                                    "North Eu hospital effluent" = "#77DDB3"
                                    ))
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

heat <- pheatmap(log10(heat.df + 0.000001), cluster_rows = F, cluster_cols = T, 
                        cellwidth = 12, cellheight = 20, 
                        border_color = "white",
                        fontsize=5,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "", 
                        angle_col = 90, legend = TRUE, fontsize_row = 12, fontsize_col = 9, 
                        labels_row = as.expression(newnames),
                        #gaps_col = c(1, 9, 17),
                        filename = "all_every_heat.png",
                    #    annotation_row = metadata_row,
                        annotation_col = metadata_col,
                        annotation_colors = ann_colors,
                        treeheight_row = 0,
                        cutree_cols = 14,
                        treeheight_col = 30,
                        )
heat
```
## DESeq2
# Metaxa2
```{r}
metaxa_deseq <- metaxa_PHY
metaxa_deseq <- subset_samples(metaxa_deseq,
                                 hospital_section == "surgery" |
                                 hospital_section == "maternity")

metaxa_deseq = prune_taxa(taxa_sums(metaxa_deseq) > 10, metaxa_deseq)

hist(log10(apply(otu_table(metaxa_deseq), 1, var)), xlab = "log10(variance)")

varianceThreshold = 30
keepOTUs = apply(otu_table(metaxa_deseq), 1, var) > varianceThreshold
metaxa_deseq1 = prune_taxa(keepOTUs, metaxa_deseq)
metaxa_deseq1

dds_metaxa = phyloseq_to_deseq2(metaxa_deseq1, ~hospital_section)

dds_metaxa$country <- relevel(dds_metaxa$hospital_section, "surgery", "maternity")

dds_metaxa = DESeq(dds_metaxa, fitType = "mean", test = "Wald", betaPrior = FALSE)

res_metaxa = results(dds_metaxa, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds_metaxa)
plotDispEsts(dds_metaxa)
head(res_metaxa)
summary(res_metaxa)

res_metaxa = res_metaxa[order(res_metaxa$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaxa = res_metaxa[which(res_metaxa$padj < alpha), ]

sigtab_metaxa = cbind(as(sigtab_metaxa, "data.frame"), as(tax_table(metaxa_PHY)[rownames(sigtab_metaxa), 
    ], "matrix"))

n <- rowSums(otu_table(metaxa_deseq1))

sigtab_metaxa = merge(sigtab_metaxa, as.data.frame(n), by = 0)

kable(sigtab_metaxa, caption = "Taxonomy")

# Plot
# Domain
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Domain, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Domain = factor(as.character(sigtab_metaxa$Domain), levels = names(x))

# Phylum
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Phylum = factor(as.character(sigtab_metaxa$Phylum), levels = names(x))

# Family 
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Family, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Family = factor(as.character(sigtab_metaxa$Family), levels = names(x))

# Class
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Class = factor(as.character(sigtab_metaxa$Class), levels = names(x))

# Genus 
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Genus = factor(as.character(sigtab_metaxa$Genus), levels = names(x))


cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 17)

# Plot
b <- ggplot(sigtab_metaxa, aes(x = Genus, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 0.9) + 
  theme_classic() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 12, face = "italic"), 
        axis.text.y = element_text(size = 22), 
        axis.title.x = element_blank(), axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 10), legend.title = element_text(size = 24), 
        legend.position = c(0.80, 0.80), legend.key.width = unit(0.15, "lines"), 
        legend.key.height = unit(0.5, "lines"),
        legend.background = element_rect(colour = "white"),
        plot.title = element_text(size = 24)) + 
  scale_color_manual(values = cols) + 
    scale_size(range = c(1, 5), breaks = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)) + 
    labs(color = "") + guides(size = FALSE) + ylim(-10, 10) + 
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  surgery         maternity") + 
    ggtitle("DESeq analysis, Metaxa2")
b
```
## DESeq2
# Metaphlan3
```{r}
metaphlan_data <- OTU_metaphlan[, ] * 10^5 + 1

metaphlan_deseq <- phyloseq(otu_table(metaphlan_data, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_metaphlan)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_deseq = subset_samples(metaphlan_deseq, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
metaphlan_deseq <- subset_samples(metaphlan_deseq, country != "UK")

metaphlan_deseq <- subset_samples(metaphlan_deseq,
                                 hospital_section == "surgery" |
                                 hospital_section == "maternity")

metaphlan_deseq = prune_taxa(taxa_sums(metaphlan_deseq) > 10, metaphlan_deseq)

hist(log10(apply(otu_table(metaphlan_deseq), 1, var)), xlab = "log10(variance)")

varianceThreshold = 30
keepOTUs = apply(otu_table(metaphlan_deseq), 1, var) > varianceThreshold
metaphlan_deseq1 = prune_taxa(keepOTUs, metaphlan_deseq)
metaphlan_deseq1

dds_metaphlan = phyloseq_to_deseq2(metaphlan_deseq1, ~hospital_section)

dds_metaphlan$country <- relevel(dds_metaphlan$hospital_section, "surgery", "maternity")

dds_metaphlan = DESeq(dds_metaphlan, fitType = "mean", test = "Wald", betaPrior = FALSE)

res_metaphlan = results(dds_metaphlan, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds_metaphlan)
plotDispEsts(dds_metaphlan)
head(res_metaphlan)
summary(res_metaphlan)

res_metaphlan = res_metaphlan[order(res_metaphlan$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaphlan = res_metaphlan[which(res_metaphlan$padj < alpha), ]

sigtab_metaphlan = cbind(as(sigtab_metaphlan, "data.frame"), as(tax_table(metaphlan_PHY)[rownames(sigtab_metaphlan), 
    ], "matrix"))

n <- rowSums(otu_table(metaphlan_deseq1))

sigtab_metaphlan = merge(sigtab_metaphlan, as.data.frame(n), by = 0)

kable(sigtab_metaphlan, caption = "Taxonomy")

# Plot
# Domain
x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Kingdom, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Domain = factor(as.character(sigtab_metaphlan$Kingdom), levels = names(x))

# Phylum
x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Phylum = factor(as.character(sigtab_metaphlan$Phylum), levels = names(x))

# Family 
x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Family, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Family = factor(as.character(sigtab_metaphlan$Family), levels = names(x))

# Class
x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Class = factor(as.character(sigtab_metaphlan$Class), levels = names(x))

# Genus 
x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Genus = factor(as.character(sigtab_metaphlan$Genus), levels = names(x))


cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 18)

# Plot
b <- ggplot(sigtab_metaphlan, aes(x = Genus, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 0.9) + 
  theme_classic() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 12, face = "italic"), 
        axis.text.y = element_text(size = 22), 
        axis.title.x = element_blank(), axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 10), legend.title = element_text(size = 24), 
        legend.position = c(0.80, 0.80), legend.key.width = unit(0.15, "lines"), 
        legend.key.height = unit(0.5, "lines"),
        legend.background = element_rect(colour = "white"),
        plot.title = element_text(size = 24)) + 
  scale_color_manual(values = cols) + 
    scale_size(range = c(1, 5), breaks = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)) + 
    labs(color = "") + guides(size = FALSE) + ylim(-10, 10) + 
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  surgery         other") + 
    ggtitle("DESeq analysis, Metaphlan3")
b
```
## DESeq2
# ResFinder
```{r}
# OTU table
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names added manually in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
# Divide by ARG gene lengths
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
deseq_OTU_resfinder <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(deseq_OTU_resfinder))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], deseq_OTU_resfinder[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Deseq
arg_data <- deseq_OTU_resfinder[, ] * 10^5 + 1

# Reload to get new row numbers
write.table(arg_data, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/arg_data.txt", 
            row.names=T, sep = "\t", col.names = T)
arg_data <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/arg_data.txt", row.names=NULL)
arg_data$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

resfinder_deseq <- phyloseq(otu_table(arg_data, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_deseq = subset_samples(resfinder_deseq, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
resfinder_deseq <- subset_samples(resfinder_deseq, country != "UK")

# Exclude replicates:
#resfinder_deseq <- subset_samples(resfinder_deseq,  alias != "BH10_S84" & 
#                                           alias != "BH31_S95" & alias != "BH32_S96" & alias != "BH33_S97" & 
#                                           alias != "BH34A_S98" & alias != "BH34B_S99" & alias != "BFH38B_S157" & 
#                                           alias != "FH8_S169" & alias != "BH45_S106" & alias != "BH59_S114" &
#                                           alias != "BH61_S116" & alias != "BH62_S117")

# Take pair wise comparisons
resfinder_deseq = subset_samples(resfinder_deseq, hospital_2 == "Nanoro_SOUKA__SourDeVie" | hospital_2 == "Yalgado teaching hospital")

hist(log10(apply(otu_table(resfinder_deseq), 1, var)), xlab = "log10(variance)")

varianceThreshold = 50
keepOTUs = apply(otu_table(resfinder_deseq), 1, var) > varianceThreshold
resfinder_deseq1 = prune_taxa(keepOTUs, resfinder_deseq)
resfinder_deseq1

dds_resfinder = phyloseq_to_deseq2(resfinder_deseq1, ~hospital_2)

dds_resfinder$category <- relevel(dds_resfinder$hospital_2, "Nanoro_SOUKA__SourDeVie", "Yalgado teaching hospital")

dds_resfinder = DESeq(dds_resfinder, fitType = "mean", test = "Wald", betaPrior = FALSE)

res_resfinder = results(dds_resfinder, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds_resfinder)
plotDispEsts(dds_resfinder)
head(res_resfinder)
summary(res_resfinder)

res_resfinder = res_resfinder[order(res_resfinder$padj, na.last=NA), ]

alpha = 0.05
sigtab_resfinder = res_resfinder[which(res_resfinder$padj < alpha), ]

sigtab_resfinder = cbind(as(sigtab_resfinder, "data.frame"), as(tax_table(resfinder_deseq)[rownames(sigtab_resfinder), 
    ], "matrix"))

n <- rowSums(otu_table(resfinder_deseq1))

sigtab_resfinder = merge(sigtab_resfinder, as.data.frame(n), by = 0)

kable(sigtab_resfinder, caption = "Taxonomy")

# Plot
# Class
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Class = factor(as.character(sigtab_resfinder$Class), levels = names(x))

# Gene cluster
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Cluster_name, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Cluster_name = factor(as.character(sigtab_resfinder$Cluster_name), levels = names(x))

# Gene 
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Gene, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Gene = factor(as.character(sigtab_resfinder$Gene), levels = names(x))

# Plot surgery vs other
# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 11)

b <- ggplot(sigtab_resfinder, aes(x = Cluster_name, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.95, vjust = 1, size = 9, face = "italic"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 24, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 24, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
   # scale_size(range = c(1, 5), breaks = seq(2.5)) + 
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  Yalgado teaching Hospital         Nanoro, SOUKA, SourDeVie") + 
    ggtitle("DESeq2 (ResFinder): ARGs in small hospitals vs. Yalgado teaching hospital in Burkina Faso")
b
```
## Plot deseq2 in general
```{r}
# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225", "#B94ED9"), 13)
b <- ggplot(sigtab_resfinder, aes(x = Cluster_name, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 0.9) + 
 # theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 0.95, vjust = 1, size = 9, face = "italic"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 10), legend.title = element_text(size = 30, family = "Times"), 
        legend.position = c(0.80, 0.80), legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(colour = "white"),
        plot.title = element_text(size = 24, family = "Times")) + 
  scale_color_manual(values = cols) + 
    scale_size(range = c(1, 5), breaks = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)) + 
    labs(color = "") + guides(size = FALSE) + ylim(-20, 20) + 
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  other         surgery") + 
    ggtitle("DESeq analysis, ResFinder")
b
```
## Boxplot, DESeq2 results
```{r}
## Boxplot
dat <- sigtab_resfinder[grepl("blaCMY-*", sigtab_resfinder$Cluster_name), ]


cols <- get_palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"), 2)

p <- ggplot(dat, aes(x=Cluster_name, y=log2FoldChange)) + 
  geom_boxplot() +
  scale_color_manual(values=cols) + 
  geom_jitter(data = dat, aes(x = Cluster_name, y = log2FoldChange, color = Cluster_name), size = 5, alpha = 1, width = 0) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 20, family = "Times", face = "bold"), 
        axis.text.y = element_text(angle = 0, hjust = 1, vjust = 1, size = 20, family = "Times"),
        axis.title.y = element_text(size = 20, family = "Times"), 
        plot.title = element_text(size = 30, family = "Times", face = "italic")) +
  labs(y = "log2 fold change\nNorth Eu hospital effluent        WA hospital effluent", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "blaCMY")
p


#ggsave(filename = "RF_deseq_BF_Ben.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)  
```
## New deseq
```{r}
temp_OTU <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(temp_OTU))
temp_OTU <- temp_OTU[,match]
all(colnames(temp_OTU) == rownames(metadata))

# Reload to get new row numbers
write.table(temp_OTU, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/temp_OTU.txt", 
            row.names=T, sep = "\t", col.names = T)
temp_OTU <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/temp_OTU.txt", row.names=NULL)
temp_OTU$row.names<-NULL

# Load tax table
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt")

Class <- as.data.frame(clusters_tax_table_resfinder$Class)
Cluster_name <- as.data.frame(clusters_tax_table_resfinder$Cluster_name)
Gene <- as.data.frame(clusters_tax_table_resfinder$Gene)

temp_OTU <- cbind(temp_OTU, Class)
temp_OTU <- cbind(temp_OTU, Cluster_name)
temp_OTU <- cbind(temp_OTU, Gene)

names(temp_OTU)[names(temp_OTU) == "clusters_tax_table_resfinder$Class"] <- "Class"
names(temp_OTU)[names(temp_OTU) == "clusters_tax_table_resfinder$Cluster_name"] <- "Cluster_name"
names(temp_OTU)[names(temp_OTU) == "clusters_tax_table_resfinder$Gene"] <- "Gene"

# Merge clusters
merged_temp_OTU <- ddply(temp_OTU,"Cluster_name",numcolwise(sum))
# Remove duplicates
uniq_clust <- clusters_tax_table_resfinder[!duplicated(clusters_tax_table_resfinder[ , c("Cluster_name")]),]
match <- match(uniq_clust$Cluster_name, merged_temp_OTU$Cluster_name)
merged_temp_OTU <- merged_temp_OTU[match,]
all(merged_temp_OTU$Cluster_name == uniq_clust$Cluster_name)
# Reload to get new row numbers
write.table(uniq_clust, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
deseq_clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_clusters_tax_table_resfinder.txt", row.names=NULL)

# Reload to get new row numbers
write.table(merged_temp_OTU, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU_resfinder.txt", 
            row.names=T, sep = "\t", col.names = T)
deseq_OTU_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU_resfinder.txt", row.names=NULL)
deseq_OTU_resfinder$row.names<-NULL
deseq_OTU_resfinder$Cluster_name <- NULL

arg_data <- deseq_OTU_resfinder[, ] * 10^5 + 1

resfinder_deseq <- phyloseq(otu_table(arg_data, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(deseq_clusters_tax_table_resfinder)))

# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_deseq = subset_samples(resfinder_deseq, alias != "BFH24_S142" & alias != "BH63_S118" & alias != "FH10_S171")
resfinder_deseq <- subset_samples(resfinder_deseq, country != "UK")

# Take pair wise comparisons
resfinder_deseq = subset_samples(resfinder_deseq, category == "WA hospital effluent" | category == "North Eu hospital effluent")

hist(log10(apply(otu_table(resfinder_deseq), 1, var)), xlab = "log10(variance)")

varianceThreshold = 50
keepOTUs = apply(otu_table(resfinder_deseq), 1, var) > varianceThreshold
resfinder_deseq1 = prune_taxa(keepOTUs, resfinder_deseq)
resfinder_deseq1

dds_resfinder = phyloseq_to_deseq2(resfinder_deseq1, ~category)

dds_resfinder$category <- relevel(dds_resfinder$category, "WA hospital effluent", "North Eu hospital effluent")

dds_resfinder = DESeq(dds_resfinder, fitType = "mean", test = "Wald", betaPrior = FALSE)

res_resfinder = results(dds_resfinder, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds_resfinder)
plotDispEsts(dds_resfinder)
head(res_resfinder)
summary(res_resfinder)

res_resfinder = res_resfinder[order(res_resfinder$padj, na.last=NA), ]

alpha = 0.05
sigtab_resfinder = res_resfinder[which(res_resfinder$padj < alpha), ]

sigtab_resfinder = cbind(as(sigtab_resfinder, "data.frame"), as(tax_table(resfinder_deseq)[rownames(sigtab_resfinder), 
    ], "matrix"))

n <- rowSums(otu_table(resfinder_deseq1))

sigtab_resfinder = merge(sigtab_resfinder, as.data.frame(n), by = 0)

kable(sigtab_resfinder, caption = "Taxonomy")

# Plot
# Class
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Class = factor(as.character(sigtab_resfinder$Class), levels = names(x))

# Gene cluster
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Cluster_name, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Cluster_name = factor(as.character(sigtab_resfinder$Cluster_name), levels = names(x))

# Gene 
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Gene, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Gene = factor(as.character(sigtab_resfinder$Gene), levels = names(x))

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225", "#B94ED9"), 13)

# Plot
b <- ggplot(sigtab_resfinder, aes(x = Cluster_name, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 0.9) + 
 # theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 0.95, vjust = 1, size = 9, face = "italic"), 
        axis.text.y = element_text(size = 22), 
        axis.title.x = element_blank(), axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 10), legend.title = element_text(size = 24), 
        legend.position = c(0.80, 0.80), legend.key.width = unit(0.15, "lines"), 
        legend.key.height = unit(0.5, "lines"),
        legend.background = element_rect(colour = "white"),
        plot.title = element_text(size = 24)) + 
  scale_color_manual(values = cols) + 
    scale_size(range = c(1, 5), breaks = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)) + 
    labs(color = "") + guides(size = FALSE) + ylim(-20, 20) + 
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  WA hospital effluent         North Eu hospital effluent") + 
    ggtitle("DESeq analysis, ResFinder")
b
#ggsave(filename = "RF_deseq_BF_Ben.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)  
```
## Quinolone / mcr sums
```{r}
# Take 15 most abundant
resfinder_PHY_sub <- subset_samples(resfinder_PHY, country == "Finland" | country == "Benin" | country == "Burkina Faso")
resfinder_PHY_sub <- subset_samples(resfinder_PHY_sub, sample_type != "wwtp influent")

#resfinder_PHY_Class <- tax_glom(resfinder_PHY_sub, taxrank = "Class")

resfinder_PHY_Class_abund <- subset_taxa(resfinder_PHY_sub, Class == "Polymyxin")

resfinder_PHY_mcr20 <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_Class_abund), TRUE)[1:54]), 
    resfinder_PHY_Class_abund)

resfinder_PHY_mcr20 <- subset_samples(resfinder_PHY_mcr20, sample_sums(resfinder_PHY_mcr20) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 20)
rfc <- plot_bar(resfinder_PHY_mcr20, fill = "Cluster_name") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARG Abundance normalized to 16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 9, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~hospital, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 11, family = "Times", hjust = 0, vjust = 0.5, angle = 90),
        strip.background = element_rect(colour = "white"))
rfc_plot

# Gamma
resfinder_PHY_Class_abund_WA <- subset_samples(resfinder_PHY_Class_abund, country == "Benin" | country == "Burkina Faso")
#resfinder_PHY_Class_abund_WA <- subset_samples(resfinder_PHY_Class_abund_WA, sample_type == "hospital effluent")
df<-data.frame(SUM=sample_sums(resfinder_PHY_Class_abund_WA),
               category_for_model=as.factor(sample_data(resfinder_PHY_Class_abund_WA)$category_for_model),
               SSU_counts=as.factor(sample_data(resfinder_PHY_Class_abund_WA)$SSU_counts),
               hospital=as.factor(sample_data(resfinder_PHY_Class_abund_WA)$hospital),
               connection_level=as.factor(sample_data(resfinder_PHY_Class_abund_WA)$connection_level),
               country=as.factor(sample_data(resfinder_PHY_Class_abund_WA)$country),
               alias=as.factor(sample_data(resfinder_PHY_Class_abund_WA)$alias))
df$SSU_counts <- as.character(df$SSU_counts)
df$SSU_counts <- as.numeric(df$SSU_counts)
## Explore data
#plot(df$SUM~df$hospital, cex.axis = 0.5, las = 2)

# Fit model with all variables  
fit <- glm(SUM~SSU_counts + hospital, data=df, family="Gamma"(link="log"))
summary(fit)

# Tukey's post hoc test
glht.mod <- glht(fit, mcp(hospital = "Tukey"))
summary(glht(glht.mod))

# Plot
fit <- glm(SUM~hospital, data=df, family="Gamma"(link="log"))
summary(fit)

dfA <- cbind(df, Mean = predict(fit, newdata = df, type = "response"), SE = predict(fit, 
    newdata = df, type = "response", se.fit = T)$se.fit)

cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#5F5E98"), 15)

resfinder_sum <- ggplot(dfA, aes(x = hospital, y = Mean, label = connection_level)) + 
  geom_jitter(data = dfA, aes(x = hospital, y = SUM, color = hospital), size = 5, alpha = 0.9, 
              position = position_jitter(seed = 1)) +
  geom_point() + scale_color_manual(values=cols) + 
  geom_text(aes(x = hospital, y = SUM, label = connection_level), data = dfA, size = 4, vjust = -1.5, hjust = 0, family = "Times",
            position = position_jitter(seed = 1)) + 
  geom_errorbar(aes(ymin = Mean - SE, ymax = Mean + SE), width = 0.4, lwd = 0.4) +
  theme_linedraw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 0.8, size = 16, family = "Times", face = "bold"), 
        axis.title.y = element_text(size = 24, family = "Times"),
        axis.text.y = element_text(size = 20, family = "Times"),
        plot.title = element_text(size = 30, family = "Times")) + 
  labs(y = "ARG sum/16s rRNA", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  scale_y_log10(breaks = c(10^-6, 10^-5, 10^-4, 10^-3, 10^-2, 10^-1), labels = trans_format("log10", math_format(10^.x)), limits=c(10^-6, 100)) +
  labs(title = "Relative sum of Polymyxin ARGs (ResFinder)")
resfinder_sum


## Only for koudougou
resfinder_PHY_Class_abund_WA <- subset_samples(resfinder_PHY_Class_abund, area_or_city == "Koudougou")
#resfinder_PHY_Class_abund_WA <- subset_samples(resfinder_PHY_Class_abund_WA, connection_level != "nd")

df<-data.frame(SUM=sample_sums(resfinder_PHY_Class_abund_WA),
               connection_level=as.factor(sample_data(resfinder_PHY_Class_abund_WA)$connection_level))

cols <- palette(c("#B2182B", "#D6604D", "#F4A582", "#4393C3", "#2166AC"))
resfinder_sum <- ggplot(df, aes(x = connection_level, y = SUM, color = connection_level)) + 
  geom_point(data = df, aes(x = connection_level, y = SUM, color = connection_level), size = 12, alpha = 1) +
  scale_color_manual(values=cols) +
  theme_linedraw() + 
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 0, vjust = 0.8, size = 30, hjust = 0.8, family = "Times", face = "bold"), 
        axis.text.y = element_text(size = 28, family = "Times"),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.title = element_blank(),
        legend.text = element_text(size = 28, family = "Times"),
        plot.title = element_text(size = 36, family = "Times")) +
  labs(y = "Abundance/16S", x = "") + 
  guides(color = FALSE, alpha = FALSE) + 
  labs(title = "Relative abundance of Polymyxin ARGs, \nConnected samples in Koudougou (ResFinder)")
resfinder_sum

### KPC
resfinder_PHY_sub <- subset_samples(resfinder_PHY, country == "Finland" | country == "Benin" | country == "Burkina Faso")
resfinder_PHY_sub <- subset_samples(resfinder_PHY_sub, sample_type != "wwtp influent")

#resfinder_PHY_Class <- tax_glom(resfinder_PHY_sub, taxrank = "Class")

resfinder_PHY_Class_abund <- subset_taxa(resfinder_PHY_sub, Cluster_name == "blaKPC-34_1_clust")

resfinder_PHY_mcr20 <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_Class_abund), TRUE)[1:39]), 
    resfinder_PHY_Class_abund)

resfinder_PHY_mcr20 <- subset_samples(resfinder_PHY_mcr20, sample_sums(resfinder_PHY_mcr20) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 39)
rfc <- plot_bar(resfinder_PHY_mcr20, fill = "Gene") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARG Abundance normalized to 16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 9, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~hospital, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 11, family = "Times", hjust = 0, vjust = 0.5, angle = 90),
        strip.background = element_rect(colour = "white"))
rfc_plot

### NDM
resfinder_PHY_sub <- subset_samples(resfinder_PHY, country == "Finland" | country == "Benin" | country == "Burkina Faso")
resfinder_PHY_sub <- subset_samples(resfinder_PHY_sub, sample_type != "wwtp influent")

#resfinder_PHY_Class <- tax_glom(resfinder_PHY_sub, taxrank = "Class")

resfinder_PHY_Class_abund <- subset_taxa(resfinder_PHY_sub, Cluster_name == "blaNDM-18_1_clust")

resfinder_PHY_mcr20 <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_Class_abund), TRUE)[1:24]), 
    resfinder_PHY_Class_abund)

resfinder_PHY_mcr20 <- subset_samples(resfinder_PHY_mcr20, sample_sums(resfinder_PHY_mcr20) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 39)
rfc <- plot_bar(resfinder_PHY_mcr20, fill = "Gene") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARG Abundance normalized to 16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 9, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~hospital, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 11, family = "Times", hjust = 0, vjust = 0.5, angle = 90),
        strip.background = element_rect(colour = "white"))
rfc_plot

### GES
resfinder_PHY_sub <- subset_samples(resfinder_PHY, country == "Finland" | country == "Benin" | country == "Burkina Faso")
resfinder_PHY_sub <- subset_samples(resfinder_PHY_sub, sample_type != "wwtp influent")

#resfinder_PHY_Class <- tax_glom(resfinder_PHY_sub, taxrank = "Class")

resfinder_PHY_Class_abund <- subset_taxa(resfinder_PHY_sub, Gene == "blaGES-5_1_DQ236171")

resfinder_PHY_mcr20 <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_Class_abund), TRUE)[1:28]), 
    resfinder_PHY_Class_abund)

resfinder_PHY_mcr20 <- subset_samples(resfinder_PHY_mcr20, sample_sums(resfinder_PHY_mcr20) != 0)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 39)
rfc <- plot_bar(resfinder_PHY_mcr20, fill = "Gene") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_fill_manual(values = cols) +
  labs(y = expression(atop(bold("ARG Abundance normalized to 16s rRNA"), atop("ResFinder")))) +
  ggtitle("Relative abundance of ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 9, family = "Times", angle = 90, hjust = 1),
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        plot.title = element_text(size = 30, family = "Times", face = "bold")) +
  facet_grid(~hospital, scales = "free", space = "free", labeller = label_wrap_gen(width = 30, multi_line = TRUE)) +
  theme(strip.text.x= element_text(size = 11, family = "Times", hjust = 0, vjust = 0.5, angle = 90),
        strip.background = element_rect(colour = "white"))
rfc_plot


```

```{r}
fit_meta_genera<-glm(SUM~Top_genus, data=df_temp, family="Gamma"(link="log"))
```
#### DESeq2 drafts
## DESeq2
# ResFinder
```{r}
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Divide by ARG gene lengths
## Get the lengths in terminal
# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
deseq_OTU_resfinder <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(deseq_OTU_resfinder))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], deseq_OTU_resfinder[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Deseq
deseq_OTU <- deseq_OTU_resfinder[, ] * 10^5 + 1

# Reload to get new row numbers
write.table(deseq_OTU, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU.txt", 
            row.names=T, sep = "\t", col.names = T)
deseq_OTU <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/deseq_OTU.txt", row.names=NULL)
deseq_OTU$row.names<-NULL

# Reload to get new row numbers
write.table(clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt", row.names=NULL)

resfinder_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_deseq = subset_samples(resfinder_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
resfinder_deseq <- subset_samples(resfinder_deseq, country != "UK")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
resfinder_deseq_stat <- subset_samples(resfinder_deseq, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | 	
                                    alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | 	
                                    alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | 	
                                    alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | 	
                                    alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | 	
                                    alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

## Exclude other than WA samples
resfinder_deseq_WA <- subset_samples(resfinder_deseq, continent_level_1 == "West Africa")

## Exclude biological / technical replicates
resfinder_deseq_WA_stat <- subset_samples(resfinder_deseq_WA, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH61")

## Exclude other than hospital WW
resfinder_deseq_WA_WW_stat <- subset_samples(resfinder_deseq_WA_stat, category == "WA hospital effluent")

##Exclude random samples to get equal groups by country
resfinder_deseq_WA_WW_stat <- subset_samples(resfinder_deseq_WA_WW_stat, alias != "BFH17" & alias != "BFH20" & alias != "BFH23" & 
                                          alias != "BFH29" & alias != "BFH31" & alias != "BFH34" & alias != "BFH37" &
                                          alias != "BFH40" & alias != "BFH8")

# Take pair wise comparisons
deseq_PHY = subset_samples(resfinder_deseq_stat, country == "Benin" | country == "Finland")

hist(log10(apply(otu_table(deseq_PHY), 1, var)), xlab = "log10(variance)")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_resfinder = res[which(res$padj < alpha), ]

sigtab_resfinder = cbind(as(sigtab_resfinder, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_resfinder), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_resfinder = merge(sigtab_resfinder, as.data.frame(n), by = 0)

kable(sigtab_resfinder, caption = "Taxonomy")

# Plot
# Class
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Class = factor(as.character(sigtab_resfinder$Class), levels = names(x))

# Gene cluster
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Cluster_name, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Cluster_name = factor(as.character(sigtab_resfinder$Cluster_name), levels = names(x))

# Gene 
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Gene, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Gene = factor(as.character(sigtab_resfinder$Gene), levels = names(x))

# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 13)

deseq_p <- ggplot(sigtab_resfinder, aes(x = Cluster_name, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 9, face = "italic"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 22, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 36, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  Benin         Finland") + 
    ggtitle("DESeq2 (ResFinder): Benin - Finland")
deseq_p

#ggsave(filename = "BEN_FIN_deseq_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

deseq_p <- ggplot(sigtab_resfinder, aes(x = Cluster_name, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 9, face = "italic"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 22, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 36, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  Burkina Faso         Finland") +
    ggtitle("DESeq2 (ResFinder): Burkina Faso - Finland")
deseq_p

#ggsave(filename = "BF_FIN_deseq_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

deseq_p <- ggplot(sigtab_resfinder, aes(x = Cluster_name, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 30, size = 15, face = "italic", family = "Times", hjust = 1, vjust = 1), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 28, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 36, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  Finland         Benin") + 
    ggtitle("DESeq2 (ResFinder): Finland - Benin")
deseq_p

#ggsave(filename = "BF_BEN_deseq_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Plot bar visualization for DESeq2 results
# Benin-Finland
```{r}
# Look at the data table
sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]

# Subset for eyeballiong the data
Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

# Take top 10 from each and combine back to one table
Fin25 <- Fin[1:25,]
Ben25 <- Ben[1:25,]
top_sigtab <- rbind(Fin25, Ben25)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 9)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + 
  ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Beninise / Finnish samples (25 with highest log 2 fold change values of each country)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 12, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

#ggsave(filename = "BEN_FIN_deseq50_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Plot bar visualization for DESeq2 results
# BF-Finland
```{r}
# Look at the data table
sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]

# Subset for eyeballiong the data
Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(Ben$log2FoldChange), ]

# Take top 10 from each and combine back to one table
Fin25 <- Fin[1:25,]
BF25 <- BF[1:25,]
top_sigtab <- rbind(Fin25, BF25)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 9)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Burkinabe / Finnish samples (25 with highest log 2 fold change values of each country)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 12, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "BF_FIN_deseq50_resfinder.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Plot bar visualization for DESeq2 results
# BF-Benin
```{r}
# Look at the data table
sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]

# Subset for eyeballiong the data
Ben <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(BF$log2FoldChange), ]

# Take top 10 from each and combine back to one table
Ben25 <- Ben[1:25,]
BF25 <- BF[1:25,]
top_sigtab <- rbind(Ben25, BF25)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 9)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Benin") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Burkinabe / Beninise samples (25 with highest log 2 fold change values of each country)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 12, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "BF_BEN_deseq50_resfinder.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

# With only samples from Burkina Faso (25) & Benin (25)
#ggsave(filename = "BF_BEN_WA_deseq50_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Plot bar visualization for DESeq2 results, ESKAPE approach, ARGs:
```{r}
## Clusters
#selected_sigtable <- subset(sigtab_resfinder, Cluster_name =="blaCMY-150_2_clust" | Cluster_name =="blaSHV-100_1_clust"
#                            | Cluster_name == "blaCTX-M-211_1_clust" |  Cluster_name == "blaKPC-34_1_clust"
#                            | Cluster_name == "blaNDM-18_1_clust" | Cluster_name == "blaIMP-1_1_clust"
#                            | Cluster_name == "blaVIM-48_1_clust" | Cluster_name == "blaOXA-368_1_clust"
#                            | Cluster_name == "mecA_1_clust" | Cluster_name == "VanHAX_1_clust"
#                            | Cluster_name == "VanHOX_1" | Cluster_name == "mcr-1.11_1_clust"
#                            | Cluster_name == "mcr-3.1_1_clust" | Cluster_name == "mcr-3.17_1"
#                            | Cluster_name == "mcr-5.1_1_clust" | Cluster_name == "mcr-7.1_1"
#                            | Cluster_name == "mcr-9_1" | Cluster_name == "mcr-10_1")

## Genes
selected_sigtable <- subset(sigtab_resfinder, Gene =="VanHOX_1_KF478993" | Gene =="VanHAX_2_M97297" | Gene =="VanHBX_1_AF192329"
                            | Gene == "mcr-7.1_1_MG267386" | Gene == "mcr-5.1_1_KY807921" | Gene == "mcr-3.17_1_MH332767"
                            | Gene == "mcr-9_1_NZ_NAAN01000063.1" | Gene == "mcr-3.15_1_MH332765" | Gene == "mcr-1.11_1_clust"
                            | Gene == "mcr-3.1_1_clust" | Gene == "mcr-10_1_MN179494"
                            | Gene == "nimA_1_X71444" | Gene == "	nimE_1_AJ244018" 
                            | Gene == "qnrVC4_1_GQ891757" | Gene == "qnrVC1_1_EU436855" | Gene == "qnrS1_1_AB187515"
                            | Gene == "qnrB60_1_AB734055"
                            | Gene == "blaAER-1_1_U14748" | Gene == "blaCARB-4_1_FJ785525" | Gene == "blaNPS_1_AY027589"
                            | Gene == "blaKPC-2_1_AY034847" | Gene == "blaNDM-1_1_FN396876" | Gene == "blaOXA-58_1_AY665723"
                            | Gene == "blaOXA-129_1_FJWZ01000025" | Gene == "blaOXA-10_1_J03427" | Gene == "blaOXA-14_1_KJ420612" 
                            | Gene == "blaOXA-16_1_AF043100" | Gene == "blaOXA-256_1_HE616889" | Gene == "blaCTX-M-15_1_AY044436"
                            | Gene == "blaCMY-4_1_LNHZ01000079" | Gene == "blaMOX-3_1_EU515248"
                            | Gene == "dfrA7_1_AB161450" | Gene == "dfrA14_1_KF921535"
                            | Gene == "cmlA1_1_M64556" | Gene == "catB3_1_AJ009818" | Gene == "catB8_1_AF227506"
                            | Gene == "erm(B)_1_JN899585" | Gene == "mef(A)_3_AF227520"
                            | Gene == "aadA2_1_NC_010870" | Gene == "aadA8b_2_AM040708" | Gene == "aph(3')-IIIa_3_AB247327"
                            | Gene == "lnu(F)_1_EU118119" | Gene == "lnu(C)_1_AY928180"
                            | Gene == "sul1_33_AJ564903" | Gene == "sul2_1_AF542061" | Gene == "sul3_2_AJ459418"
                            | Gene == "blaVEB-1_1_HM370393" | Gene == "blaVIM-2_1_AF302086")

sorted_selected_sigtab <- selected_sigtable[order(-selected_sigtable$log2FoldChange), ]

# Fin-Ben
# Subset for eyebolling the data
Fin <- subset(sorted_selected_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_selected_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

top_sigtab <- rbind(Fin, Ben)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#876BF3", "#FD8FD9"), 11)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Beninise / Finnish samples (selected ARGs)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 16, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 26),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 36, face = "bold"))
p

ggsave(filename = "BEN_FIN_selected_deseq_resfinder.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

# Fin-BF
# Subset for eyebolling the data
Fin <- subset(sorted_selected_sigtab,log2FoldChange>=0)
BF <- subset(sorted_selected_sigtab,log2FoldChange<=0)
BF <- Ben[order(BF$log2FoldChange), ]

top_sigtab <- rbind(Fin, BF)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#876BF3", "#FD8FD9"), 11)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Burkinabe / Finnish samples (selected ARGs)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 16, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 26),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 36, face = "bold"))
p

ggsave(filename = "BF_FIN_selected_deseq_resfinder.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## DESeq2
# ResFinder, with clusters
```{r}
OTU_resfinder <-as.matrix(read.table("ARG_genemat.txt", header= T, check.names = F, row.names = 1))

# Reorder to match metadata
match <- match(rownames(metadata), colnames(OTU_resfinder))
OTU_resfinder <- OTU_resfinder[,match]
all(colnames(OTU_resfinder) == rownames(metadata))

# Tax_table (Cluster names created using cd-hit and 90 % identity and added manually to the tax table in excel)
clusters_tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table.txt", header=FALSE, sep=";")
colnames(clusters_tax_table_resfinder) <- c("Gene",  "Cluster_name", "Class")
# Reorder columns
col_order <- c("Class", "Cluster_name", "Gene")
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[, col_order]

# Reorder tax_table to match OTU_resfinder
match <- match(rownames(OTU_resfinder), clusters_tax_table_resfinder$Gene)
clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
all(rownames(OTU_resfinder) == clusters_tax_table_resfinder$Gene)

# Divide by ARG gene lengths
## Get the lengths in terminal
# seqkit fx2tab --length --name --header-line resfinder.fasta > resfinder_lengths.txt
resfinder_lengths <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/resfinder_lengths.txt", header=FALSE, comment.char="#")
all(rownames(clusters_tax_table_resfinder$Gene) == resfinder_lengths$V1)
OTU_resfinder_length_norm <- OTU_resfinder/resfinder_lengths[, 2]

# Normalization with Metaxa2 SSU counts
deseq_OTU_resfinder <- t(t(OTU_resfinder_length_norm)/metadata$SSU_counts) * 1540
all(rownames(metadata) == colnames(deseq_OTU_resfinder))
identical(OTU_resfinder_length_norm[2025, 5]/metadata$SSU_counts[5], deseq_OTU_resfinder[2025, 5])
all(rownames(OTU_resfinder_length_norm) == clusters_tax_table_resfinder$Gene)

# Deseq
deseq_OTU <- deseq_OTU_resfinder[, ] * 10^5 + 1

# Merge ARGs so that cluster names instead of gene names can be used
clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/clusters_tax_table_resfinder.txt")
all(clusters_tax_table_resfinder$Gene == rownames(deseq_OTU))

Class <- as.data.frame(clusters_tax_table_resfinder$Class)
Cluster_name <- as.data.frame(clusters_tax_table_resfinder$Cluster_name)
Gene <- as.data.frame(clusters_tax_table_resfinder$Gene)

OTU_DESeq <- cbind(deseq_OTU, Class, Cluster_name, Gene)

names(OTU_DESeq)[names(OTU_DESeq) == "clusters_tax_table_resfinder$Class"] <- "Class"
names(OTU_DESeq)[names(OTU_DESeq) == "clusters_tax_table_resfinder$Cluster_name"] <- "Cluster_name"
names(OTU_DESeq)[names(OTU_DESeq) == "clusters_tax_table_resfinder$Gene"] <- "Gene"

all(OTU_DESeq$Gene == clusters_tax_table_resfinder$Gene)

# Merge
merged_deseq <- ddply(OTU_DESeq,"Cluster_name",numcolwise(sum))
# Remove duplicates
uniq_clust <- clusters_tax_table_resfinder[!duplicated(clusters_tax_table_resfinder[ , c("Cluster_name")]),]
# Reorder to match OTU
match <- match(uniq_clust$Cluster_name, merged_deseq$Cluster_name)
merged_deseq <- merged_deseq[match,]
all(merged_deseq$Cluster_name == uniq_clust$Cluster_name)

# Tax table
match <- match(merged_deseq$Cluster_name, clusters_tax_table_resfinder$Cluster_name)
heat_clusters_tax_table_resfinder <- clusters_tax_table_resfinder[match,]
identical(merged_deseq$Cluster_name, heat_clusters_tax_table_resfinder$Cluster_name)

# Reload to get new row numbers
write.table(merged_deseq, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/merged_deseq.txt", 
            row.names=T, sep = "\t", col.names = T)
merged_deseq <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/merged_deseq.txt", row.names=NULL)
merged_deseq$row.names<-NULL
merged_deseq$Cluster_name <- NULL

# Reload to get new row numbers
write.table(heat_clusters_tax_table_resfinder, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=F, sep = "\t", col.names = T)
heat_clusters_tax_table_resfinder <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_clusters_tax_table_resfinder.txt", row.names=NULL)

# Create phyloseq
merged_deseq_resfinder_PHY <- phyloseq(otu_table(merged_deseq, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(heat_clusters_tax_table_resfinder)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
merged_deseq_resfinder_PHY = subset_samples(merged_deseq_resfinder_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
merged_deseq_resfinder_PHY = subset_samples(merged_deseq_resfinder_PHY, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
merged_deseq_resfinder_PHY <- subset_samples(merged_deseq_resfinder_PHY, country != "UK")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
merged_deseq_resfinder_PHY_stat <- subset_samples(merged_deseq_resfinder_PHY, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | 	
                                    alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | 	
                                    alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | 	
                                    alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | 	
                                    alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | 	
                                    alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

## Exclude other than WA samples
merged_deseq_resfinder_PHY_WA_stat <- subset_samples(merged_deseq_resfinder_PHY, continent_level_1 == "West Africa")

## Exclude biological / technical replicates
merged_deseq_resfinder_PHY_WA_stat <- subset_samples(merged_deseq_resfinder_PHY_WA_stat, alias != "BH31" & alias != "BH33" 
                                     & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH61")

## Exclude other than hospital WW
merged_deseq_resfinder_PHY_WA_WW_stat <- subset_samples(merged_deseq_resfinder_PHY_WA_stat, category == "WA hospital effluent")

##Exclude random samples to get equal groups by country
merged_deseq_resfinder_PHY_WA_WW_stat <- subset_samples(merged_deseq_resfinder_PHY_WA_WW_stat, alias != "BFH17" & 
                                          alias != "BFH20" & alias != "BFH23" & 
                                          alias != "BFH29" & alias != "BFH31" & alias != "BFH34" & alias != "BFH37" &
                                          alias != "BFH40" & alias != "BFH8")

# Take pair wise comparisons
deseq_PHY = subset_samples(merged_deseq_resfinder_PHY_stat, country == "Burkina Faso" | country == "Finland")

hist(log10(apply(otu_table(deseq_PHY), 1, var)), xlab = "log10(variance)")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_resfinder = res[which(res$padj < alpha), ]

sigtab_resfinder = cbind(as(sigtab_resfinder, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_resfinder), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_resfinder = merge(sigtab_resfinder, as.data.frame(n), by = 0)

kable(sigtab_resfinder, caption = "Taxonomy")

# Plot
# Class
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Class = factor(as.character(sigtab_resfinder$Class), levels = names(x))

# Gene cluster
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Cluster_name, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Cluster_name = factor(as.character(sigtab_resfinder$Cluster_name), levels = names(x))

# Gene 
x = tapply(sigtab_resfinder$log2FoldChange, sigtab_resfinder$Gene, function(x) max(x))
x = sort(x, TRUE)
sigtab_resfinder$Gene = factor(as.character(sigtab_resfinder$Gene), levels = names(x))

# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 13)

deseq_p <- ggplot(sigtab_resfinder, aes(x = Cluster_name, y = log2FoldChange, color = Class, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 20, face = "italic", family = "Times"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 24, family = "Times"),
        legend.position = c(0.70, 0.70), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 30, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\nFinland         Benin") + 
    ggtitle("DESeq2 (ResFinder): Benin - Finland")
deseq_p

#ggsave(filename = "BEN_FIN_deseq_clusters_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

#ggsave(filename = "BF_FIN_deseq_clusters_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

#ggsave(filename = "BF_BEN_deseq_clusters_resfinder.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Plot bar visualization for DESeq2 results, with clusters
```{r}
# Benin-Finland
# Look at the data table
sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]

# Subset for eyebolling the data
Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

# Take top 10 from each and combine back to one table
Fin25 <- Fin[1:25,]
Ben25 <- Ben[1:25,]
top_sigtab <- rbind(Fin25, Ben25)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#FD8FD9", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 12)

p <- ggplot(data = top_sigtab,aes(x = Cluster_name, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Beninise / Finnish samples (25 ARG clusters with highest log 2 fold change values of each country)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 12, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 30, face = "bold"))
p

ggsave(filename = "BEN_FIN_deseq_clusters_bars_resfinder.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

# BF-Finland
# Look at the data table
sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]

# Subset for eyeballiong the data
Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(Ben$log2FoldChange), ]

# Take top 10 from each and combine back to one table
Fin25 <- Fin[1:25,]
BF25 <- BF[1:25,]
top_sigtab <- rbind(Fin25, BF25)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 13)
p <- ggplot(data = top_sigtab,aes(x = Cluster_name, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                                  Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Burkinabe / Finnish samples (25 with highest log 2 fold change values of each country)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 12, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "BF_FIN_deseq_clusters_bars_resfinder.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

# BF-Benin
# Look at the data table
sorted_sigtab <- sigtab_resfinder[order(-sigtab_resfinder$log2FoldChange), ]

# Subset for eyeballing the data
Ben <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(BF$log2FoldChange), ]

Ben10 <- Ben[1:10,]
BF10 <- BF[1:10,]
top_sigtab <- rbind(Ben10, BF10)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#876BF3", "#FD8FD9"), 10)

p <- ggplot(data = top_sigtab,aes(x = Cluster_name, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Benin") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched ARGs in Burkinabe / Beninise samples (10 with highest log 2 fold change values of each country)") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 10),
        axis.text.y = element_text(family = "Times", size = 18, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 28),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "BF_BEN_deseq_clusters_bars_resfinder.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## DESeq2
# Taxonomy (Metaxa2), Class
```{r}
metaxa_deseq_PHY <- prune_taxa(taxa_sums(metaxa_PHY_stat) > 0, metaxa_PHY_stat)

# rarefy
#metaxa_deseq_PHY <- rarefy_even_depth(metaxa_deseq_PHY, sample.size = min(sample_sums(metaxa_deseq_PHY)),
#                         replace = TRUE, trimOTUs = TRUE, verbose = TRUE, rngseed=1)

metaxa_deseq_PHY <- tax_glom(metaxa_deseq_PHY, taxrank = "Class")

# Take pair wise comparisons
#deseq_PHY = subset_samples(metaxa_deseq_PHY, country == "Benin" | country == "Finland")
deseq_PHY = subset_samples(metaxa_deseq_PHY, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

#dds$category <- relevel(dds$country, "Benin", "Finland")
dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaxa = res[which(res$padj < alpha), ]

sigtab_metaxa = cbind(as(sigtab_metaxa, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaxa), 
    ], "matrix"))

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaxa = merge(sigtab_metaxa, as.data.frame(n), by = 0)

kable(sigtab_metaxa, caption = "Taxonomy")

# Phylum
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Phylum = factor(as.character(sigtab_metaxa$Phylum), levels = names(x))

# Class
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Class = factor(as.character(sigtab_metaxa$Class), levels = names(x))

# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 17)

# Class
deseq_p <- ggplot(sigtab_metaxa, aes(x = Class, y = log2FoldChange, color = Phylum, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 20, face = "italic", family = "Times"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 20, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 30, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\nBurkina Faso         Finland") + 
    ggtitle("DESeq2 (Metaxa2): Burkina Faso - Finland")
deseq_p

#ggsave(filename = "BEN_FIN_class_deseq_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

ggsave(filename = "BF_FIN_class_deseq_metaxa.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## DESeq2
# Taxonomy (Metaxa2), Genus
```{r}
metaxa_deseq_PHY <- prune_taxa(taxa_sums(metaxa_PHY_stat) > 0, metaxa_PHY_stat)

# rarefy
#metaxa_deseq_PHY <- rarefy_even_depth(metaxa_deseq_PHY, sample.size = min(sample_sums(metaxa_deseq_PHY)),
#                         replace = TRUE, trimOTUs = TRUE, verbose = TRUE, rngseed=1)

metaxa_deseq_PHY <- tax_glom(metaxa_deseq_PHY, taxrank = "Genus")

# Take pair wise comparisons
deseq_PHY = subset_samples(metaxa_deseq_PHY, country == "Benin" | country == "Finland")
#deseq_PHY = subset_samples(metaxa_deseq_PHY, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")
#dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaxa = res[which(res$padj < alpha), ]

sigtab_metaxa = cbind(as(sigtab_metaxa, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaxa), 
    ], "matrix"))

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaxa = merge(sigtab_metaxa, as.data.frame(n), by = 0)

kable(sigtab_metaxa, caption = "Taxonomy")

# Phylum
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Phylum = factor(as.character(sigtab_metaxa$Phylum), levels = names(x))

# Class
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Class = factor(as.character(sigtab_metaxa$Class), levels = names(x))

# Genus
x = tapply(sigtab_metaxa$log2FoldChange, sigtab_metaxa$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaxa$Genus = factor(as.character(sigtab_metaxa$Genus), levels = names(x))

# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 17)

# Class
deseq_p <- ggplot(sigtab_metaxa, aes(x = Class, y = log2FoldChange, color = Phylum, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 20, face = "italic", family = "Times"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 24, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 30, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\nBenin         Finland") + 
    ggtitle("DESeq2 (Metaxa2): Benin - Finland")
deseq_p

# Genus
deseq_p <- ggplot(sigtab_metaxa, aes(x = Genus, y = log2FoldChange, color = Phylum, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 8, face = "italic", family = "Times"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 20, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 30, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-8, 5, 1)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\nBurkina Faso         Finland") + 
    ggtitle("DESeq2 (Metaxa2): Burkina Faso - Finland")
deseq_p

#ggsave(filename = "BEN_FIN_deseq_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

#ggsave(filename = "BF_FIN_deseq_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```
## Plot bar visualization of DESeq2 results, Metaxa2
```{r}
## Benin-Finland
# ESKAPE etc
selected_sigtable <- subset(sigtab_metaxa, Genus =="Staphylococcus" | Genus == "Escherichia-Shigella" | Genus == "Enterococcus"
                            | Genus == "Acinetobacter" | Genus == "Enterobacter" | Genus == "Pseudomonas" | Genus == "Clostridium"
                            | Genus == "Citrobacter" | Genus == "Streptococcus" | Genus == "Aeromonas"
                            | Genus == "Stenotrophomonas" | Genus == "Campylobacter"
                            | Genus == "Corynebacterium" | Genus == "Mycobacterium" | Genus == "Neisseria"
                            | Genus == "Salmonella" | Genus == "Vibrio" | Genus == "Klebsiella"
                            
                            | Genus =="Unclassified Staphylococcaceae" | Genus == "Unclassified Enterobacteriaceae"
                            | Genus == "Unclassified Enterococcaceae" | Genus == "Unclassified Pseudomonadaceae" 
                            | Genus =="Unclassified Clostridiaceae"| Genus == "Unclassified Streptococcaceae" 
                            | Genus == "Unclassified Aeromonadaceae"| Genus == "Unclassified Campylobacteraceae" 
                            | Genus == "Unclassified Corynebacteriaceae" | Genus == "Unclassified Mycobacteriaceae"
                            | Genus == "Unclassified Moraxellaceae" | Genus == "Unclassified Vibrionaceae"
                            | Genus == "Unclassified Xanthomonadaceae" | Genus == "Unclassified Neisseriaceae")

sorted_sigtab <- selected_sigtable[order(-selected_sigtable$log2FoldChange), ]
sorted_sigtab <- sigtab_metaxa[order(-sigtab_metaxa$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

Fin25 <- Fin[1:25,]
Ben25 <- Ben[1:25,]
top_sigtab <- rbind(Fin25, Ben25)

Fin8 <- Fin[1:8,]
Ben8 <- Ben[1:8,]
top_sigtab <- rbind(Fin8, Ben8)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 15)
p <- ggplot(data = top_sigtab,aes(x = Class, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Phylum), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched taxa in Finnish / Beninise samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 18, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

#ggsave(filename = "FIN_BEN_bars_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

#ggsave(filename = "FIN_BEN_pathogens_bars_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

#ggsave(filename = "FIN_BEN_raref_pathogens_bars_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

## BF-Finland
# ESKAPE etc
selected_sigtable <- subset(sigtab_metaxa, Genus =="Staphylococcus" | Genus == "Escherichia-Shigella" | Genus == "Enterococcus"
                            | Genus == "Acinetobacter" | Genus == "Enterobacter" | Genus == "Pseudomonas" | Genus == "Clostridium"
                            | Genus == "Citrobacter" | Genus == "Streptococcus" | Genus == "Aeromonas"
                            | Genus == "Stenotrophomonas" | Genus == "Campylobacter" | Genus == "Chlamydia"
                            | Genus == "Corynebacterium" | Genus == "Mycobacterium" | Genus == "Neisseria"
                            | Genus == "Salmonella" | Genus == "Vibrio" | Genus == "Klebsiella"

                            | Genus =="Unclassified Staphylococcaceae" | Genus == "Unclassified Enterobacteriaceae"
                            | Genus == "Unclassified Enterococcaceae" | Genus == "Unclassified Pseudomonadaceae" 
                            | Genus =="Unclassified Clostridiaceae"| Genus == "Unclassified Streptococcaceae" 
                            | Genus == "Unclassified Aeromonadaceae"| Genus == "Unclassified Campylobacteraceae" 
                            | Genus == "Unclassified Corynebacteriaceae" | Genus == "Unclassified Mycobacteriaceae"
                            | Genus == "Unclassified Moraxellaceae" | Genus == "Unclassified Vibrionaceae"
                            | Genus == "Unclassified Xanthomonadaceae" | Genus == "Unclassified Neisseriaceae")

sorted_sigtab <- selected_sigtable[order(-selected_sigtable$log2FoldChange), ]

# Subset for eyeballiong the data
Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(BF$log2FoldChange), ]

Fin3 <- Fin[1:3,]
BF8 <- BF[1:8,]
top_sigtab <- rbind(Fin3, BF8)

Fin25 <- Fin[1:25,]
BF25 <- BF[1:25,]
top_sigtab <- rbind(Fin25, BF25)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 6)
p <- ggplot(data = top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                    Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched taxa in Finnish / Burkinabe samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 18, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 24),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

#ggsave(filename = "FIN_BF_bars_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

#ggsave(filename = "FIN_BF_pathogens_bars_metaxa.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

ggsave(filename = "FIN_BF_raref_pathogens_bars_metaxa.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Test if the results are similar with Metaphlan3, Class
```{r}
# Deseq
deseq_OTU <- OTU_metaphlan[, ] * 10^5 + 1

metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_metaphlan)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_deseq = subset_samples(metaphlan_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaphlan_deseq <- subset_samples(metaphlan_deseq, country != "UK")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | 	
                                    alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | 	
                                    alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | 	
                                    alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | 	
                                    alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | 	
                                    alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

## Exclude other than WA samples
metaphlan_deseq_WA <- subset_samples(metaphlan_deseq, continent_level_1 == "West Africa")

## Exclude biological / technical replicates
metaphlan_deseq_WA_stat <- subset_samples(metaphlan_deseq_WA, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH61")

## Exclude other than hospital WW
metaphlan_deseq_WA_WW_stat <- subset_samples(metaphlan_deseq_WA_stat, category == "WA hospital effluent")

##Exclude random samples to get equal groups by country
metaphlan_deseq_WA_WW_stat <- subset_samples(metaphlan_deseq_WA_WW_stat, alias != "BFH17" & alias != "BFH20" & alias != "BFH23" & 
                                          alias != "BFH29" & alias != "BFH31" & alias != "BFH34" & alias != "BFH37" &
                                          alias != "BFH40" & alias != "BFH8")

#metaphlan_deseq_stat <- prune_taxa(taxa_sums(metaphlan_deseq_stat) > 0, metaphlan_deseq_stat)

# rarefy
#metaphlan_deseq_stat <- rarefy_even_depth(metaphlan_deseq_stat, sample.size = min(sample_sums(metaphlan_deseq_stat)),
#                         replace = TRUE, trimOTUs = TRUE, verbose = TRUE, rngseed=1)

metaphlan_deseq_stat <- tax_glom(metaphlan_deseq_stat, taxrank = "Class")

# Take pair wise comparisons
#deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Benin" | country == "Finland")
deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

#dds$category <- relevel(dds$country, "Benin", "Finland")
dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaphlan = res[which(res$padj < alpha), ]

sigtab_metaphlan = cbind(as(sigtab_metaphlan, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaphlan), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaphlan = merge(sigtab_metaphlan, as.data.frame(n), by = 0)

kable(sigtab_metaphlan, caption = "Taxonomy")

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Class = factor(as.character(sigtab_metaphlan$Class), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Phylum = factor(as.character(sigtab_metaphlan$Phylum), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Genus = factor(as.character(sigtab_metaphlan$Genus), levels = names(x))

# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 13)

deseq_p <- ggplot(sigtab_metaphlan, aes(x = Class, y = log2FoldChange, color = Phylum, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 16, face = "italic", family = "Times"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 22, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 36, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-20, 16, 2)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\nBurkina Faso         Finland") + 
    ggtitle("DESeq2 (Metaphlan3): Burkina Faso - Finland")
deseq_p

#ggsave(filename = "FIN_BEN_class_metaphlan.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

ggsave(filename = "FIN_BF_class_metaphlan.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Test if the results are similar with Metaphlan3, Genus
```{r}
# Deseq
deseq_OTU <- OTU_metaphlan[, ] * 10^5 + 1

metaphlan_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_metaphlan)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_deseq = subset_samples(metaphlan_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
metaphlan_deseq <- subset_samples(metaphlan_deseq, country != "UK")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
metaphlan_deseq_stat <- subset_samples(metaphlan_deseq, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | 	
                                    alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | 	
                                    alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | 	
                                    alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | 	
                                    alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | 	
                                    alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

## Exclude other than WA samples
metaphlan_deseq_WA <- subset_samples(metaphlan_deseq, continent_level_1 == "West Africa")

## Exclude biological / technical replicates
metaphlan_deseq_WA_stat <- subset_samples(metaphlan_deseq_WA, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH61")

## Exclude other than hospital WW
metaphlan_deseq_WA_WW_stat <- subset_samples(metaphlan_deseq_WA_stat, category == "WA hospital effluent")

##Exclude random samples to get equal groups by country
metaphlan_deseq_WA_WW_stat <- subset_samples(metaphlan_deseq_WA_WW_stat, alias != "BFH17" & alias != "BFH20" & alias != "BFH23" & 
                                          alias != "BFH29" & alias != "BFH31" & alias != "BFH34" & alias != "BFH37" &
                                          alias != "BFH40" & alias != "BFH8")

#metaphlan_deseq_stat <- prune_taxa(taxa_sums(metaphlan_deseq_stat) > 0, metaphlan_deseq_stat)

# rarefy
#metaphlan_deseq_stat <- rarefy_even_depth(metaphlan_deseq_stat, sample.size = min(sample_sums(metaphlan_deseq_stat)),
#                         replace = TRUE, trimOTUs = TRUE, verbose = TRUE, rngseed=1)

metaphlan_deseq_stat <- tax_glom(metaphlan_deseq_stat, taxrank = "Genus")

# Take pair wise comparisons
#deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Benin" | country == "Finland")
deseq_PHY = subset_samples(metaphlan_deseq_stat, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

#dds$category <- relevel(dds$country, "Benin", "Finland")
dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_metaphlan = res[which(res$padj < alpha), ]

sigtab_metaphlan = cbind(as(sigtab_metaphlan, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_metaphlan), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_metaphlan = merge(sigtab_metaphlan, as.data.frame(n), by = 0)

kable(sigtab_metaphlan, caption = "Taxonomy")

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Class = factor(as.character(sigtab_metaphlan$Class), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Phylum = factor(as.character(sigtab_metaphlan$Phylum), levels = names(x))

x = tapply(sigtab_metaphlan$log2FoldChange, sigtab_metaphlan$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_metaphlan$Genus = factor(as.character(sigtab_metaphlan$Genus), levels = names(x))

# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 13)

deseq_p <- ggplot(sigtab_metaphlan, aes(x = Genus, y = log2FoldChange, color = Phylum, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 9, face = "italic", family = "Times"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 22, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 36, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-20, 16, 2)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\n  Burkina Faso         Finland") + 
    ggtitle("DESeq2 (Metaphlan3): Burkina Faso - Finland")
deseq_p

#ggsave(filename = "FIN_BEN_metaphlan.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

#ggsave(filename = "FIN_BF_metaphlan.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Plot bar visualization of DESeq2 results, Metaphlan3
```{r}
## Benin-Finland
# ESKAPE etc
selected_sigtable <- subset(sigtab_metaphlan, Genus =="Staphylococcus" | Genus == "Escherichia" | Genus == "Enterococcus"
                            | Genus == "Acinetobacter" | Genus == "Enterobacter" | Genus == "Pseudomonas" | Genus == "Clostridium"
                            | Genus == "Citrobacter" | Genus == "Streptococcus" | Genus == "Aeromonas"
                            | Genus == "Stenotrophomonas" | Genus == "Campylobacter"
                            | Genus == "Corynebacterium" | Genus == "Mycobacterium" | Genus == "Neisseria"
                            | Genus == "Salmonella" | Genus == "Vibrio" | Genus == "Klebsiella"
                            
                            | Genus =="Staphylococcaceae_unclassified" | Genus == "Enterobacteriaceae_unclassified"
                            | Genus == "Enterococcaceae_unclassified" | Genus == "Pseudomonadaceae_unclassified" 
                            | Genus =="Clostridiaceae_unclassified"| Genus == "Streptococcaceae_unclassified" 
                            | Genus == "Aeromonadaceae_unclassified"| Genus == "Campylobacteraceae_unclassified" 
                            | Genus == "Corynebacteriaceae_unclassified" | Genus == "Mycobacteriaceae_unclassified"
                            | Genus == "Moraxellaceae_unclassified" | Genus == "Vibrionaceae_unclassified"
                            | Genus == "Xanthomonadaceae_unclassified" | Genus == "Neisseriaceae_unclassified")

sorted_sigtab <- selected_sigtable[order(-selected_sigtable$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

Fin3 <- Fin[1:3,]
Ben4 <- Ben[1:4,]
top_sigtab <- rbind(Fin3, Ben4)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
p <- ggplot(data = top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Genus), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched taxa in Finnish / Beninise samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 18, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p


sorted_sigtab <- sigtab_metaphlan[order(-sigtab_metaphlan$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

Fin25 <- Fin[1:25,]
Ben25 <- Ben[1:25,]
top_sigtab <- rbind(Fin25, Ben25)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 20)
p <- ggplot(data = top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched taxa in Finnish / Beninise samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 14, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

#ggsave(filename = "FIN_BEN_deseq50_metaphlan.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

#ggsave(filename = "FIN_BEN_pathogens_bars_metaphlan.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

## BF-Finland
# ESKAPE etc
selected_sigtable <- subset(sigtab_metaphlan, Genus =="Staphylococcus" | Genus == "Escherichia" | Genus == "Enterococcus"
                            | Genus == "Acinetobacter" | Genus == "Enterobacter" | Genus == "Pseudomonas" | Genus == "Clostridium"
                            | Genus == "Citrobacter" | Genus == "Streptococcus" | Genus == "Aeromonas"
                            | Genus == "Stenotrophomonas" | Genus == "Campylobacter"
                            | Genus == "Corynebacterium" | Genus == "Mycobacterium" | Genus == "Neisseria"
                            | Genus == "Salmonella" | Genus == "Vibrio" | Genus == "Klebsiella"
                            
                            | Genus =="Staphylococcaceae_unclassified" | Genus == "Enterobacteriaceae_unclassified"
                            | Genus == "Enterococcaceae_unclassified" | Genus == "Pseudomonadaceae_unclassified" 
                            | Genus =="Clostridiaceae_unclassified"| Genus == "Streptococcaceae_unclassified" 
                            | Genus == "Aeromonadaceae_unclassified"| Genus == "Campylobacteraceae_unclassified" 
                            | Genus == "Corynebacteriaceae_unclassified" | Genus == "Mycobacteriaceae_unclassified"
                            | Genus == "Moraxellaceae_unclassified" | Genus == "Vibrionaceae_unclassified"
                            | Genus == "Xanthomonadaceae_unclassified" | Genus == "Neisseriaceae_unclassified")

sorted_sigtab <- selected_sigtable[order(-selected_sigtable$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- Ben[order(BF$log2FoldChange), ]

Fin1 <- Fin[1:1,]
BF1 <- BF[1:1,]
top_sigtab <- rbind(Fin1, BF1)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
p <- ggplot(data = top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Genus), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched taxa in Finnish / Burkinabe samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 18, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

sorted_sigtab <- sigtab_metaphlan[order(-sigtab_metaphlan$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(BF$log2FoldChange), ]

Fin25 <- Fin[1:25,]
BF20 <- BF[1:20,]
top_sigtab <- rbind(Fin25, BF20)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 20)
p <- ggplot(data = top_sigtab,aes(x = Genus, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched taxa in Finnish / Burkinabe samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 14, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "FIN_BF_deseq50_metaphlan.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

#ggsave(filename = "FIN_BF_pathogens_metaphlan.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## DESeq2
# MGE
```{r}
# Deseq
deseq_OTU <- OTU_MGE_length_SSU_norm[, ] * 10^5 + 1

MGE_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(MGE_tax_table_trim)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
MGE_deseq = subset_samples(MGE_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
MGE_deseq <- subset_samples(MGE_deseq, country != "UK")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
MGE_deseq_stat <- subset_samples(MGE_deseq, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | 	
                                    alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | 	
                                    alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | 	
                                    alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | 	
                                    alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | 	
                                    alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

## Exclude other than WA samples
MGE_deseq_WA <- subset_samples(MGE_deseq, continent_level_1 == "West Africa")

## Exclude biological / technical replicates
MGE_deseq_WA_stat <- subset_samples(MGE_deseq_WA, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH61")

## Exclude other than hospital WW
MGE_deseq_WA_WW_stat <- subset_samples(MGE_deseq_WA_stat, category == "WA hospital effluent")

##Exclude random samples to get equal groups by country
MGE_deseq_WA_WW_stat <- subset_samples(MGE_deseq_WA_WW_stat, alias != "BFH17" & alias != "BFH20" & alias != "BFH23" & 
                                          alias != "BFH29" & alias != "BFH31" & alias != "BFH34" & alias != "BFH37" &
                                          alias != "BFH40" & alias != "BFH8")

#MGE_deseq_stat <- prune_taxa(taxa_sums(MGE_deseq_stat) > 0, MGE_deseq_stat)

# rarefy
#MGE_deseq_stat <- rarefy_even_depth(MGE_deseq_stat, sample.size = min(sample_sums(MGE_deseq_stat)),
#                         replace = TRUE, trimOTUs = TRUE, verbose = TRUE, rngseed=1)

#MGE_deseq_stat <- tax_glom(MGE_deseq_stat, taxrank = "Class")

# Take pair wise comparisons
deseq_PHY = subset_samples(MGE_deseq_stat, country == "Benin" | country == "Finland")
#deseq_PHY = subset_samples(MGE_deseq_stat, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

dds$category <- relevel(dds$country, "Benin", "Finland")
#dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_MGE = res[which(res$padj < alpha), ]

sigtab_MGE = cbind(as(sigtab_MGE, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_MGE), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_MGE = merge(sigtab_MGE, as.data.frame(n), by = 0)

kable(sigtab_MGE, caption = "Taxonomy")

x = tapply(sigtab_MGE$log2FoldChange, sigtab_MGE$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_MGE$Class = factor(as.character(sigtab_MGE$Class), levels = names(x))

x = tapply(sigtab_MGE$log2FoldChange, sigtab_MGE$Element, function(x) max(x))
x = sort(x, TRUE)
sigtab_MGE$Element = factor(as.character(sigtab_MGE$Element), levels = names(x))

# Plot
cols <- get_palette(c("#FD8FD9", "#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#B94ED9"), 18)

deseq_p <- ggplot(sigtab_MGE, aes(x = Class, y = log2FoldChange, color = Element, 
    size = n)) + 
  geom_point(alpha = 1.5) + 
  theme_minimal_vgrid() +
  theme(axis.text.x = element_text(angle = 60, hjust = 0.95, vjust = 1, size = 16, face = "italic", family = "Times"), 
        axis.text.y = element_text(size = 22, family = "Times"),
        axis.title.y.left = element_text(size = 22, family = "Times"),
        axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 22), 
        legend.text = element_text(size = 22, family = "Times"),
        legend.position = c(0.80, 0.80), 
        legend.key.width = unit(0.5, "lines"), 
        legend.key.height = unit(0.75, "lines"),
        legend.background = element_rect(fill = "white"),
        plot.title = element_text(size = 36, family = "Times")) + 
  guides(color = guide_legend(override.aes = list(size=5))) +
  scale_color_manual(values = cols) + 
  scale_y_continuous(breaks=seq(-20, 16, 2)) +
    labs(color = "") + guides(size = FALSE) +
  geom_hline(yintercept = 0, linetype = 2, color = "grey", alpha = 0.7) + ylab("log2 fold change\nBenin         Finland") + 
    ggtitle("DESeq2 (MGE): Benin - Finland")
deseq_p

ggsave(filename = "FIN_BEN_MGE.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

#ggsave(filename = "FIN_BF_MGE.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## Plot bar visualization of DESeq2 results, MGE
```{r}
## Benin-Finland
sorted_sigtab <- sigtab_MGE[order(-sigtab_MGE$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

Fin25 <- Fin[1:25,]
Ben25 <- Ben[1:25,]
top_sigtab <- rbind(Fin25, Ben25)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 15)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched MGEs in Finnish / Beninise samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 12),
        axis.text.y = element_text(family = "Times", size = 14, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "FIN_BEN_deseq50_MGE.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

## Benin-Finland
# Integrases
selected_sigtable <- subset(sigtab_MGE, Element =="integrase")
sorted_sigtab <- selected_sigtable[order(-selected_sigtable$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

Fin4 <- Fin[1:4,]
Ben12 <- Ben[1:12,]
top_sigtab <- rbind(Fin4, Ben12)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched taxa in Finnish / Beninise samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 18, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "FIN_BEN_selected_MGE.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 

# BF-Finland
sorted_sigtab <- sigtab_MGE[order(-sigtab_MGE$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(BF$log2FoldChange), ]

Fin25 <- Fin[1:25,]
BF25 <- BF[1:25,]
top_sigtab <- rbind(Fin25, BF25)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 20)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched MGEs in Finnish / Burkinabe samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 14, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

#ggsave(filename = "FIN_BF_deseq50_MGE.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

selected_sigtable <- subset(sigtab_MGE, Element =="integrase")
sorted_sigtab <- selected_sigtable[order(-selected_sigtable$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- Ben[order(Ben$log2FoldChange), ]

Fin1 <- Fin[1:1,]
BF12 <- BF[1:12,]
top_sigtab <- rbind(Fin1, BF12)

# Shorten gene names and order again
top_sigtab$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", top_sigtab$Gene)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 11)
p <- ggplot(data = top_sigtab,aes(x = Gene, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Class), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched MGEs in Finnish / Burkinabe samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 18, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

#ggsave(filename = "FIN_BF_selected_MGE.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1) 
```
## DESeq2
# VFDB
```{r}
# Deseq
deseq_OTU <- OTU_VFDB_SSU_norm[, ] * 10^5 + 1

VFDB_deseq <- phyloseq(otu_table(deseq_OTU, taxa_are_rows = T), sample_data(metadata), 
    tax_table(as.matrix(tax_table_VFDB)))

# Create phyloseq object with all samples
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
VFDB_deseq = subset_samples(VFDB_deseq, alias != "BFH24" & alias != "BH63" & alias != "FH10")
# Exclude samples from UK hospital whose sample collection does not match with other samples
VFDB_deseq <- subset_samples(VFDB_deseq, country != "UK")

# Create phyloseq object with equal group for the statistical analysis
## Include 8 samples per country
VFDB_deseq_stat <- subset_samples(VFDB_deseq, alias == "BFH1" | 	alias == "BFH19" | 	alias == "BFH25" | 	alias == "BFH31" | 	
                                    alias == "BFH6" | 	alias == "BFH15" | 	alias == "BFH39" |  	alias == "BFH10" | 	
                                    alias == "BH02" | 	alias == "BH09" | 	alias == "BH27" |  	alias == "BH34A" | 	
                                    alias == "BH44" | 	alias == "BH58" | 	alias == "BH61" | 	alias == "BH47" | 	
                                    alias == "FH1" | 	alias == "FH2" | 	alias == "FH3" | 	alias == "FH4" | 	
                                    alias == "FH5" | 	alias == "FH6" | 	alias == "FH7" | 	alias == "FH9")

## Exclude other than WA samples
VFDB_deseq_WA <- subset_samples(VFDB_deseq, continent_level_1 == "West Africa")

## Exclude biological / technical replicates
VFDB_deseq_WA_stat <- subset_samples(VFDB_deseq_WA, alias != "BH31" & alias != "BH33" & alias != "BH34B" & alias != "BH10"
                                     & alias != "BFH38B" & alias != "FH8" & alias != "BH45" & alias != "BH59" & alias != "BH61")

## Exclude other than hospital WW
VFDB_deseq_WA_WW_stat <- subset_samples(VFDB_deseq_WA_stat, category == "WA hospital effluent")

##Exclude random samples to get equal groups by country
VFDB_deseq_WA_WW_stat <- subset_samples(VFDB_deseq_WA_WW_stat, alias != "BFH17" & alias != "BFH20" & alias != "BFH23" & 
                                          alias != "BFH29" & alias != "BFH31" & alias != "BFH34" & alias != "BFH37" &
                                          alias != "BFH40" & alias != "BFH8")

#VFDB_deseq_stat <- prune_taxa(taxa_sums(VFDB_deseq_stat) > 0, VFDB_deseq_stat)

# rarefy
#VFDB_deseq_stat <- rarefy_even_depth(VFDB_deseq_stat, sample.size = min(sample_sums(VFDB_deseq_stat)),
#                         replace = TRUE, trimOTUs = TRUE, verbose = TRUE, rngseed=1)

#VFDB_deseq_stat <- tax_glom(VFDB_deseq_stat, taxrank = "Class")

# Take pair wise comparisons
#deseq_PHY = subset_samples(VFDB_deseq_stat, country == "Benin" | country == "Finland")
deseq_PHY = subset_samples(VFDB_deseq_stat, country == "Burkina Faso" | country == "Finland")

varianceThreshold = 50
keepOTUs = apply(otu_table(deseq_PHY), 1, var) > varianceThreshold
deseq_PHY = prune_taxa(keepOTUs, deseq_PHY)
deseq_PHY

dds = phyloseq_to_deseq2(deseq_PHY, ~country)

#dds$category <- relevel(dds$country, "Benin", "Finland")
dds$category <- relevel(dds$country, "Burkina Faso", "Finland")

dds = DESeq(dds, fitType = "mean", test = "Wald", betaPrior = FALSE)

res = results(dds, cooksCutoff = FALSE, alpha = 0.05)
resultsNames(dds)
plotDispEsts(dds)
head(res)
summary(res)

res = res[order(res$padj, na.last=NA), ]

alpha = 0.05
sigtab_VFDB = res[which(res$padj < alpha), ]

sigtab_VFDB = cbind(as(sigtab_VFDB, "data.frame"), as(tax_table(deseq_PHY)[rownames(sigtab_VFDB), 
    ], "matrix"))

otu_table(deseq_PHY)[otu_table(deseq_PHY) == 1] <- 0
otu_table(deseq_PHY)[otu_table(deseq_PHY) > 0] <- 1

n <- rowSums(otu_table(deseq_PHY))

sigtab_VFDB = merge(sigtab_VFDB, as.data.frame(n), by = 0)

kable(sigtab_VFDB, caption = "Taxonomy")

x = tapply(sigtab_VFDB$log2FoldChange, sigtab_VFDB$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab_VFDB$Class = factor(as.character(sigtab_VFDB$Class), levels = names(x))

x = tapply(sigtab_VFDB$log2FoldChange, sigtab_VFDB$Gene, function(x) max(x))
x = sort(x, TRUE)
sigtab_VFDB$Gene = factor(as.character(sigtab_VFDB$Gene), levels = names(x))
```
## Plot bar visualization of DESeq2 results, VFDB
```{r}
## Benin-Finland
sorted_sigtab <- sigtab_VFDB[order(-sigtab_VFDB$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
Ben <- subset(sorted_sigtab,log2FoldChange<=0)
Ben <- Ben[order(Ben$log2FoldChange), ]

Fin25 <- Fin[1:25,]
Ben25 <- Ben[1:25,]
top_sigtab <- rbind(Fin25, Ben25)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 18)
p <- ggplot(data = top_sigtab,aes(x = Reference, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Reference), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBenin                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched VFDBs in Finnish / Beninise samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 12),
        axis.text.y = element_text(family = "Times", size = 14, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 20),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

#ggsave(filename = "FIN_BEN_deseq50_VFDB.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# BF-Finland
sorted_sigtab <- sigtab_VFDB[order(-sigtab_VFDB$log2FoldChange), ]

Fin <- subset(sorted_sigtab,log2FoldChange>=0)
BF <- subset(sorted_sigtab,log2FoldChange<=0)
BF <- BF[order(BF$log2FoldChange), ]

Fin25 <- Fin[1:25,]
BF25 <- BF[1:25,]
top_sigtab <- rbind(Fin25, BF25)

cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#F22D3D", "#882255", "#5F5E98", "#E4C960", "#FD8FD9"), 25)
p <- ggplot(data = top_sigtab,aes(x = Reference, y = log2FoldChange)) + 
  geom_bar(stat = "identity",
    aes(fill = Reference), position = position_dodge(preserve = "single")) + ylab("Log2 Fold Change\nBurkina Faso                     Finland") + 
  scale_fill_manual(values = cols) +
  ggtitle("Enriched VFDBs in Finnish / Burkinabe samples") +
  coord_flip() +
  theme_minimal() +
  theme(axis.text.x = element_text(family = "Times", size = 14),
        axis.text.y = element_text(family = "Times", size = 14, face = "bold.italic"),
        axis.title.x = element_text(family = "Times",  size = 20, face = "bold"),
        axis.title.y = element_blank(),
        legend.text = element_text(family = "Times", size = 9),
        legend.title = element_blank(),
        plot.title = element_text(family = "Times", size = 20, face = "bold"))
p

ggsave(filename = "FIN_BF_deseq50_VFDB.png",
       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)
```










