---
title: "R_scripts"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Set working directory
```{r}
setwd("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles")
```
## Load libraries
```{r, warning=FALSE, error=FALSE, message=FALSE, results='hide'}
library(phyloseq)
library(dplyr)
library(viridis)
library(stringr)
library(vegan)
library(RColorBrewer)
library(BiocManager)
BiocManager::install("microbiome")
library(microbiome)
library(microbiomeutilities)
library(ggplot2)
library(knitr)
library(ggpubr)
library(pheatmap)
library(MASS)
library(multcomp)
library(gplots)
library(readr)
library(grid)
library(cowplot)
library(gridExtra)
```
## Load metadata
```{r}
metadata <-read.table("metadata.txt", sep="\t", header = T, row.names = 1, fill = 1)
```
# Metaxa2
## Load data, create phyloseq object and filter taxa
```{r}
metaxa_genus <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_metaxa_genus.txt")
# Create OTU table
OTU_metaxa <- metaxa_genus[,-1]
# Create tax table
tax_table_metaxa <- data.frame(str_split_fixed(data.frame(metaxa_genus) [,1], ";", 6))
colnames(tax_table_metaxa) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus")
# Check if samples are in order
identical(rownames(metadata), colnames(OTU_metaxa))

# Combine into phyloseq object
metaxa_PHY <- phyloseq(otu_table(OTU_metaxa, taxa_are_rows=TRUE), 
                       tax_table(as.matrix(tax_table_metaxa)), sample_data(metadata))
# Exclude taxa "Unknown"
metaxa_PHY <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown"))
```
## Add SSU counts into metadata file
```{r}
metadata$SSU_counts <- sample_sums(metaxa_PHY)

# Create extra column for Bacteria
metaxa_PHY_temp <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown", "Eukaryota", "Archaea", "Mitochondria", "Chloroplast", "Unclassified"))
metadata$SSU_bacteria <- sample_sums(metaxa_PHY_temp)
# Create extra column for Eucaryota
metaxa_PHY_temp <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown", "Bacteria", "Archaea", "Mitochondria", "Chloroplast", "Unclassified"))
metadata$SSU_eucaryota <- sample_sums(metaxa_PHY_temp)
# Create extra column for Mitochondria
metaxa_PHY_temp <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown", "Bacteria", "Archaea", "Chloroplast", "Unclassified", "Eucaryota"))
metadata$SSU_mitochondria <- sample_sums(metaxa_PHY_temp)
# Create extra column for Chloroplasts
metaxa_PHY_temp <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown", "Bacteria", "Archaea", "Mitochondria", "Unclassified", "Eucaryota"))
metadata$SSU_chloroplasts <- sample_sums(metaxa_PHY_temp)
# Create extra column for Archaea
metaxa_PHY_temp <- subset_taxa(metaxa_PHY, !Domain %in% c("Unknown", "Bacteria", "Chloroplast", "Mitochondria", "Unclassified", "Eucaryota"))
metadata$SSU_archaea <- sample_sums(metaxa_PHY_temp)
```
## Filter samples
```{r}
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaxa_PHY = subset_samples(metaxa_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Clean taxa that are not present in any of the remaining samples
any(taxa_sums(metaxa_PHY) == 0)
metaxa_PHY <- prune_taxa(taxa_sums(metaxa_PHY) > 0, metaxa_PHY)
any(taxa_sums(metaxa_PHY) == 0)
```
# Metaphlan3
```{bash}
# Modify data in command-line to match sample names in metadata file
#grep -E "s__" merged_abundance_table.txt > merged_abundance_table_species.txt
#tr '|' ';' <merged_abundance_table_species.txt > mod_merged_abundance_table_species.txt
#sed -i 's/_profile//g' mod_merged_abundance_table_species.txt
#sed -i 's/BFH38-A_S156/BFH38.A_S156/g' mod_merged_abundance_table_species.txt
#sed -i 's/BFH38-B_S157/BFH38.B_S157/g' mod_merged_abundance_table_species.txt
#sed -i 's/BH34-A_S98/BH34.A_S98/g' mod_merged_abundance_table_species.txt
#sed -i 's/BH34-B_S99/BH34.B_S99/g' mod_merged_abundance_table_species.txt

# Modify tax table
#awk '{print $1}' mod_merged_abundance_table.txt > tax_table_metaphlan
#sed '1,2d' -i tax_table_metaphlan
#grep -E "s__" tax_table_metaphlan > tax_table_metaphlan.txt
````
## Load data, create phyloseq object and filter taxa
```{r}
metaphlan <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_merged_abundance_table_species.txt", header=FALSE)
colnames(metaphlan) <- c("taxa", "accession", "BFH30_S148", 	"BH49_S110", 	"BH24_S90", 	"BH46_S107", 	"BFH42_S161", "FH2_S163", 	"BFH37_S155", 	"BH35_S100", 	"BFH26_S144", 	"BH27_S92", "BH62_S117", 	"BH60_S115", 	"BH39_S104", 	"BH29_S93", 	"BH02_S77", "FH4_S165", 	"BFH16_S134", 	"FH7_S168", 	"BFH29_S147", 	"BFH27_S145", "FH9_S170", 	"FH10_S171", 	"BH44_S105", 	"BFH11_S129", 	"BH07_S82", "BFH28_S146", 	"BFH34_S152", 	"BH48_S109", 	"BH63_S118", 	"FH3_S164", "BFH1_S123", 	"BH05_S80", 	"BSE74_S119", 	"BH14_S87", 	"BFH38.A_S156", "BH58_S113", 	"BFH23_S141", 	"BH01_S76", 	"BH25_S91", 	"BH04_S79", "FH8_S169", 	"BH37_S102", 	"BFH19_S137", 	"BFH39_S158", 	"BFH10_S128", "BFH4_S124", 	"BFH13_S131", 	"BH32_S96", 	"BH33_S97", 	"BFH40_S159", "BH38_S103", 	"BH03_S78", 	"BH13_S86", 	"BH59_S114", 	"BFH18_S136", "BFH41_S160", 	"BFH25_S143", 	"FH5_S166", 	"BH61_S116", 	"BH45_S106", "BFH38.B_S157", 	"BH22_S89", 	"BFH36_S154", 	"BH11_S85", 	"BH31_S95", "BFH17_S135", 	"BH20_S88", 	"BSE79_S120", 	"BH06_S81", 	"BH34.A_S98","BH34.B_S99", 	"FH1_S162", 	"BFH24_S142", 	"BFH14_S132", 	"BFH21_S139", "BFH20_S138", 	"BSE100_S122", 	"BFH8_S126", 	"BFH6_S125", 	"BFH32_S150", "BH30_S94", 	"BH50_S111", 	"BFH31_S149", 	"BFH22_S140", 	"BFH15_S133","BH10_S84", 	"FH6_S167", "BFH9_S127", "BFH33_S151", "BH36_S101", "BH52_S112","BH47_S108", "BSE93_S121", "BFH35_S153", "BH09_S83", "BFH12_S130")

# Reorder samples to match metadata
match <- match(rownames(metadata), colnames(metaphlan))
OTU_metaphlan  <- metaphlan[,match]
all(rownames(metadata) == colnames(OTU_metaphlan))

# Load tax_table
tax_table_metaphlan <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_metaphlan.txt", header=FALSE, sep=";")
colnames(tax_table_metaphlan) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Remove "*__"
tax_table_metaphlan <- apply(tax_table_metaphlan, 2, function(y) (gsub(".__", "", y)))
# Change into numbers
tax_table_metaphlan <- as.data.frame(unclass(tax_table_metaphlan)) 
# Check if samples are in order
identical(rownames(metadata), colnames(OTU_metaphlan))

# Combine into phyloseq object
metaphlan_PHY <- phyloseq(otu_table(OTU_metaphlan, taxa_are_rows = TRUE), tax_table(as.matrix(tax_table_metaphlan)), sample_data(metadata))
```
## Filter samples
```{r}
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
metaphlan_PHY = subset_samples(metaphlan_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Exclude virus
metaphlan_PHY <- subset_taxa(metaphlan_PHY, Kingdom != "Virus")

# Clean taxa that is not present in any of the remaining samples
any(taxa_sums(metaphlan_PHY) == 0)
metaphlan_PHY <- prune_taxa(taxa_sums(metaphlan_PHY) > 0, metaphlan_PHY)
any(taxa_sums(metaphlan_PHY) == 0)
```
# ResFinder
```{bash}
# Modify mapping output file "ARG_genemat.txt" in command line to match sample names in metadata file
## sed 's/BFH38-A_S156/BFH38.A_S156/g' ARG_genemat.txt > mod_ARG_genemat.txt
## sed -i 's/BFH38-B_S157/BFH38.B_S157/g' mod_ARG_genemat.txt
## sed -i 's/BH34-A_S98/BH34.A_S98/g' mod_ARG_genemat.txt
## sed -i 's/BH34-B_S99/BH34.B_S99/g' mod_ARG_genemat.txt
```
## Load data
```{r}
ARG_genemat <-as.matrix(read.table("mod_ARG_genemat.txt", header= T, check.names = F, row.names = 1))
identical(rownames(metadata), colnames(ARG_genemat))
```
## Normalize ARG counts to 16s counts
```{r}
ARG_genemat_norm <- t(t(ARG_genemat)/metadata$SSU_counts)

# Check if correct
identical(ARG_genemat[2020, 4]/metadata$SSU_counts[4], ARG_genemat_norm[2020, 4])

# Save and load again to drop row names
write.table(ARG_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)

ARG_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=NULL)
ARG_genemat_norm$row.names<-NULL
```
## Create phyloseq object
```{r}
# Create tax table (ordered in excel to match with the ARG hits)
tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_resfinder.txt", header=FALSE, sep=";")
colnames(tax_table_resfinder) <- c("Class", "Gene")
identical(rownames(tax_table_resfinder), rownames(ARG_genemat_norm))

# Combine to phyloseq object
resfinder_PHY <- phyloseq(otu_table(ARG_genemat_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_resfinder)))

# Have a quick look
plot_bar(otu_table(resfinder_PHY))
```
## Filter samples
```{r}
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
resfinder_PHY = subset_samples(resfinder_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Check if there are taxa that are not present in any of the remaining samples
any(taxa_sums(resfinder_PHY) == 0)
resfinder_PHY <- prune_taxa(taxa_sums(resfinder_PHY) > 0, resfinder_PHY)
any(taxa_sums(resfinder_PHY) == 0)
```
# CARD
## Load data (headers simplified & ";" cahenged into tab in excel)
```{r}
CARD_genemat <-as.matrix(read.table("mod_COUNT_TABLE.txt", header= T, check.names = F, row.names = 1))
# Check if samples are in order
identical(rownames(metadata), colnames(CARD_genemat))
```
## Normalize ARG counts to 16s counts
```{r}
CARD_genemat_norm <- t(t(CARD_genemat)/metadata$SSU_counts)

# Check if correct:
identical(CARD_genemat[1020, 4]/metadata$SSU_counts[4], CARD_genemat_norm[1020, 4])

# Save and load again to drop row names
write.table(CARD_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/CARD_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)

CARD_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/CARD_genemat_norm.txt", row.names=NULL)
CARD_genemat_norm$row.names<-NULL
```
## Create phyloseq object
```{r}
# Create tax table (ordered in excel to match with the ARG hits)
tax_table_CARD <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_CARD.txt", header=FALSE, sep="\t")
colnames(tax_table_CARD) <- c("Gene", "Class")
identical(rownames(tax_table_CARD), rownames(CARD_genemat_norm))

# Combine into phyloseq object
CARD_PHY <- phyloseq(otu_table(CARD_genemat_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_CARD)))

# Have a quick look
plot_bar(otu_table(CARD_PHY))
```
## Filter samples
```{r}
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
CARD_PHY = subset_samples(CARD_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Clean taxa that are not present in any of the remaining samples
any(taxa_sums(CARD_PHY) == 0)
CARD_PHY <- prune_taxa(taxa_sums(CARD_PHY) > 0, CARD_PHY)
any(taxa_sums(CARD_PHY) == 0)
```
# MGEs
```{r}
# Modify mapping output file "MGE_genemat.txt" in command line to match sample names in metadata file
## sed 's/BFH38-A_S156/BFH38.A_S156/g' MGE_genemat.txt > mod_MGE_genemat.txt
## sed -i 's/BFH38-B_S157/BFH38.B_S157/g' mod_MGE_genemat.txt
## sed -i 's/BH34-A_S98/BH34.A_S98/g' mod_MGE_genemat.txt
## sed -i 's/BH34-B_S99/BH34.B_S99/g' mod_MGE_genemat.txt
# MGEs reordered in genemat so that they are in alphabetical order
# Same is done for the tax_table obtained from MGE database in order to match gene hits & MGE elements and types
```
## Load data
```{r}
MGE_genemat <-as.matrix(read.table("mod_MGE_genemat.txt", header= T, check.names = F, row.names = 1))
identical(rownames(metadata), colnames(MGE_genemat))
```
## Normalize MGE counts to 16s counts
```{r}
MGE_genemat_norm <- t(t(MGE_genemat)/metadata$SSU_counts)

# Check if correct:
identical(MGE_genemat[2020, 4]/metadata$SSU_counts[4], MGE_genemat_norm[2020, 4])

# Save and load again to drop row names
write.table(MGE_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)

MGE_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/MGE_genemat_norm.txt", row.names=NULL)
MGE_genemat_norm$row.names<-NULL
```
## Create phyloseq object
```{r}
# Create tax table
tax_table_MGE <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_MGE.txt", header=FALSE, sep="\t")
colnames(tax_table_MGE) <- c("Element", "Class", "Gene")

# Combine to phyloseq object
MGE_PHY <- phyloseq(otu_table(MGE_genemat_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_MGE)))

# Have a quick look
plot_bar(otu_table(MGE_PHY))
```
## Filter samples
```{r}
# Exclude samples with low sequencing quality (BFH24_S142, BH63_S118, FH10_S171)
MGE_PHY = subset_samples(MGE_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# Clean taxa that are not present in any of the remaining samples
any(taxa_sums(MGE_PHY) == 0)
MGE_PHY <- prune_taxa(taxa_sums(MGE_PHY) > 0, MGE_PHY)
any(taxa_sums(MGE_PHY) == 0)
```
# Ordinations 
## Metxa2
```{r}
# Change into relative data
metaxa_rel_PHY <- transform_sample_counts(metaxa_PHY, function(x) x/sum(x))

# Ordinate and plot data
metaxa_rel_PHY_ord <- ordinate(metaxa_rel_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaxa_rel_PHY, metaxa_rel_PHY_ord, color = "country")
metaxa.p_ord <- p_ord + scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0")) + 
  geom_point(pch = 20, size = 5, alpha = 0.5) + stat_ellipse(level = 0.95, linetype = 1) +
    theme_minimal() + ggtitle("Taxonomy, Metaxa2") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 15)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))

# Test significance using pair-wise adonis
metaxa_temp <- subset_samples(metaxa_PHY, (country == "Benin" | country == "Finland"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

metaxa_temp <- subset_samples(metaxa_PHY, (country == "Burkina Faso" | country == "Finland"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))

metaxa_temp <- subset_samples(metaxa_PHY, (country == "Benin" | country == "Burkina Faso"))
metaxa_dist <- vegdist(t(otu_table(metaxa_temp)), dist = "horn")
adonis(metaxa_dist ~ country, data = data.frame(sample_data(metaxa_temp), permutations = 9999))
```
## Metaphlan3
```{r}
# Ordinate and plot data
metaphlan_PHY_ord <- ordinate(metaphlan_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(metaphlan_PHY, metaphlan_PHY_ord, color = "country")
metaphlan.p_ord <- p_ord + scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0")) + 
    geom_point(pch = 20, size = 5, alpha = 0.5) + stat_ellipse(level = 0.95, linetype = 1) +
    theme_minimal() + ggtitle("Taxonomy, Metaphlan3") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 15)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))

# Test significance using pair-wise adonis
metaphlan_temp <- subset_samples(metaphlan_PHY, (country == "Benin" | country == "Finland"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY, (country == "Burkina Faso" | country == "Finland"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))

metaphlan_temp <- subset_samples(metaphlan_PHY, (country == "Burkina Faso" | country == "Benin"))
metaphlan_dist <- vegdist(t(otu_table(metaphlan_temp)), dist = "horn")
adonis(metaphlan_dist ~ country, data = data.frame(sample_data(metaphlan_temp), permutations = 9999))
```
## ResFinder
```{r}
# Ordinate and plot data
resfinder_PHY_ord <- ordinate(resfinder_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(resfinder_PHY, resfinder_PHY_ord, color = "country")
resfinder.p_ord <- p_ord + scale_color_manual(values=c("#FF333F", "#EFAB15", "#785EF0")) + 
    geom_point(pch = 20, size = 5, alpha = 0.5) + stat_ellipse(level = 0.95, linetype = 1) +
    theme_minimal() + ggtitle("Resistome, ResFinder") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 15)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))

# Test significance between all pairs
resfinder_temp <- subset_samples(resfinder_PHY, (country == "Finland" | country == "Benin"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

resfinder_temp <- subset_samples(resfinder_PHY, (country == "Finland" | country == "Burkina Faso"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))

resfinder_temp <- subset_samples(resfinder_PHY, (country == "Burkina Faso" | country == "Benin"))
resfinder_dist <- vegdist(t(otu_table(resfinder_temp)), dist = "horn")
adonis(resfinder_dist ~ country, data = data.frame(sample_data(resfinder_temp), permutations = 9999))
```
## CARD
```{r}
# Ordinate and plot data
CARD_PHY_ord <- ordinate(CARD_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(CARD_PHY, CARD_PHY_ord, color = "country", label = "name")
CARD.p_ord <- p_ord + scale_color_manual(values=c("#C0392B", "#FFB000", "#785EF0")) + 
    geom_point(pch = 20, size = 5, alpha = 0.5) + stat_ellipse(level = 0.95, linetype = 1) +
    theme_minimal() + ggtitle("CARD") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 15)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))

# Test significance using pair-wise adonis
CARD_temp <- subset_samples(CARD_PHY, (country == "Finland" | country == "Benin"))
CARD_dist <- vegdist(t(otu_table(CARD_temp)), dist = "horn")
adonis(CARD_dist ~ country, data = data.frame(sample_data(CARD_temp), permutations = 9999))

CARD_temp <- subset_samples(CARD_PHY, (country == "Finland" | country == "Burkina Faso"))
CARD_dist <- vegdist(t(otu_table(CARD_temp)), dist = "horn")
adonis(CARD_dist ~ country, data = data.frame(sample_data(CARD_temp), permutations = 9999))

CARD_temp <- subset_samples(CARD_PHY, (country == "Burkina Faso" | country == "Benin"))
CARD_dist <- vegdist(t(otu_table(CARD_temp)), dist = "horn")
adonis(CARD_dist ~ country, data = data.frame(sample_data(CARD_temp), permutations = 9999))
```
## MGE
```{r}
# Ordinate and plot data
MGE_PHY_ord <- ordinate(MGE_PHY, method = "PCoA", distance = "horn")
p_ord <- plot_ordination(MGE_PHY, MGE_PHY_ord, color = "country", label = "name")
MGE.p_ord <- p_ord + scale_color_manual(values=c("#C0392B", "#FFB000", "#785EF0")) + 
    geom_point(pch = 20, size = 5, alpha = 0.5) + stat_ellipse(level = 0.95, linetype = 1) +
    theme_minimal() + ggtitle("MGEs") + 
    theme(plot.title = element_text(size = 25)) +
    theme(legend.text = element_text(size = 15)) +
    theme(legend.title = element_blank()) +
    theme(axis.title = element_text(size = 20)) +
    guides(fill = guide_legend(override.aes = list(linetype = 0)),
         color = guide_legend(override.aes = list(linetype = 0, size=5)))

# Test significance between all pairs
MGE_temp <- subset_samples(MGE_PHY, (country == "Finland" | country == "Benin"))
MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))

MGE_temp <- subset_samples(MGE_PHY, (country == "Finland" | country == "Burkina Faso"))
MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))

MGE_temp <- subset_samples(MGE_PHY, (country == "Burkina Faso" | country == "Benin"))
MGE_dist <- vegdist(t(otu_table(MGE_temp)), dist = "horn")
adonis(MGE_dist ~ country, data = data.frame(sample_data(MGE_temp), permutations = 9999))

# Plot all ordinations
plot_grid(metaxa.p_ord, metaphlan.p_ord)
plot_grid(resfinder.p_ord, CARD.p_ord)
plot_grid(MGE.p_ord)
```
# Diversity indexes
## Metaxa2
```{r, results='hide', fig.keep = 'none'}
# Take a look at the indexes
metaxa_tab <-microbiome::alpha(metaxa_PHY, index = "all")
kable(head(metaxa_tab))

# Create data table
metaxa.meta <- meta(metaxa_PHY)
kable(head(metaxa.meta))

# Add indexes to data table
## Shannon
metaxa.meta$diversity_shannon <- metaxa_tab$diversity_shannon
## Gini Simpson
metaxa.meta$diversity_gini_simpson <- metaxa_tab$diversity_gini_simpson
## Inverse Simpson
metaxa.meta$diversity_inverse_simpson <- metaxa_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(metaxa.meta$diversity_shannon)
shapiro.test(metaxa.meta$diversity_shannon)
qqnorm(metaxa.meta$diversity_shannon)

hist(metaxa.meta$diversity_gini_simpson)
shapiro.test(metaxa.meta$diversity_gini_simpson)
qqnorm(metaxa.meta$diversity_gini_simpson)

hist(metaxa.meta$diversity_inverse_simpson)
shapiro.test(metaxa.meta$diversity_inverse_simpson)
qqnorm(metaxa.meta$diversity_inverse_simpson)

# Create a list of pairwise comparisons, country
div.country <- levels(metaxa.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])

# Create a list of pairwise comparisons, location
div.location <- levels(metaxa.meta$location)
location.pairs <- combn(seq_along(div.location), 2, simplify = FALSE, FUN = function(i)div.location[i])
```
# Diversity plots, Metaxa2
```{r}
anova <- aov(metaxa.meta$diversity_shannon ~ country, data=metaxa.meta)
TukeyHSD(anova)
# sig:
#Benin - Finland

# Shannon
font <- 3
p_shannon_metaxa <- ggboxplot(metaxa.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "wilcox.test", size = font) +
#  stat_compare_means(label.y = 6.5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Gini Simpson
font <- 3
p_gsimp_metaxa <- ggboxplot(metaxa.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "wilcox.test", size = font) +
#  stat_compare_means(label.y = 0.5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Inverse Simpson
font <- 3
p_isimp_metaxa <- ggboxplot(metaxa.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "t.test") +
#  stat_compare_means(label.y = 5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Plot all
leg <- get_legend(p_shannon_metaxa + theme(legend.position="right"))
ggarrange(plot_grid(p_shannon_metaxa, p_gsimp_metaxa, p_isimp_metaxa,
                    labels = c("Shannon diversity, Metaxa2", 
                               "Gini Simpson diversity, Metaxa2", 
                               "Inverse Simpson diversity, Metaxa2"), ncol = 2, label_size = 10, leg))

# By location
# Shannon
anova <- aov(metaxa.meta$diversity_shannon ~ location, data=metaxa.meta)
TukeyHSD(anova)

p_shannon_metaxa <- ggboxplot(metaxa.meta, x = "location", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 10, angle = 60, hjust = 1),
        axis.title.y = element_text(size = 18),
        title = element_text(size = 14),
        legend.key.size = unit(1.5, "cm"),
        legend.position = "right") +
  guides(fill = guide_legend(title.theme = element_blank(), size = 7))
p_shannon_metaxa + ggtitle("Shannon diversity, Metaxa2")
```
## ResFinder
```{r, results='hide'}
# Shannon
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY, index = "diversity_shannon")
kable(head(resfinder_tab))

## Create data table
resfinder.meta <- meta(resfinder_PHY)
kable(head(resfinder.meta))

## Add indexes to data table
resfinder.meta$diversity_shannon <- resfinder_tab$diversity_shannon

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_shannon)
shapiro.test(resfinder.meta$diversity_shannon) 
qqnorm(resfinder.meta$diversity_shannon)

# Create a list of pairwise comparisons
div.country <- levels(resfinder.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])
print(country.pairs)

# Gini Simpson
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY, index = "diversity_gini_simpson")
kable(head(resfinder_tab))

## Add indexes to data table
resfinder.meta$diversity_gini_simpson <- resfinder_tab$diversity_gini_simpson

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_gini_simpson)
shapiro.test(resfinder.meta$diversity_gini_simpson)
qqnorm(resfinder.meta$diversity_gini_simpson)

# Inverse Simpson
## Take a look at the indexes
resfinder_tab <-microbiome::alpha(resfinder_PHY, index = "diversity_inverse_simpson")
kable(head(resfinder_tab))

## Add indexes to data table
resfinder.meta$diversity_inverse_simpson <- resfinder_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(resfinder.meta$diversity_inverse_simpson)
shapiro.test(resfinder.meta$diversity_inverse_simpson)
qqnorm(resfinder.meta$diversity_inverse_simpson)
```
## Diversity plots, ResFinder
```{r}
# Country
anova <- aov(resfinder.meta$diversity_shannon ~ country, data=resfinder.meta)
TukeyHSD(anova)

font <- 3
p_shannon_resfinder <- ggboxplot(resfinder.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "wilcox.test", size = font) +
#  stat_compare_means(label.y = 6.5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Gini Simpson
font <- 3
p_gsimp_resfinder <- ggboxplot(resfinder.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "wilcox.test", size = font) +
#  stat_compare_means(label.y = 0.5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Inverse Simpson
font <- 3
p_isimp_resfinder <- ggboxplot(resfinder.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "t.test") +
#  stat_compare_means(label.y = 5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Plot all
leg <- get_legend(p_shannon_resfinder + theme(legend.position="right"))
ggarrange(plot_grid(p_shannon_resfinder, p_gsimp_resfinder, p_isimp_resfinder,
                    labels = c("Shannon diversity, ResFinder", 
                               "Gini Simpson diversity, ResFinder", 
                               "Inverse Simpson diversity, ResFinder"), ncol = 2, label_size = 10, leg))
# Location
anova <- aov(resfinder.meta$diversity_shannon ~ location, data=resfinder.meta)
TukeyHSD(anova)

p_shannon_resfinder <- ggboxplot(resfinder.meta, x = "location", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 10, angle = 60, hjust = 1),
        axis.title.y = element_text(size = 18),
        title = element_text(size = 14),
        legend.key.size = unit(1.5, "cm"),
        legend.position = "right") +
  guides(fill = guide_legend(title.theme = element_blank(), size = 7))
p_shannon_resfinder + ggtitle("Shannon diversity, ResFinder")
```
## Diversity indexes, CARD
```{r, results='hide'}
# Shannon
## Take a look at the indexes
CARD_tab <-microbiome::alpha(CARD_PHY, index = "diversity_shannon")
kable(head(CARD_tab))

## Create data table
CARD.meta <- meta(CARD_PHY)
kable(head(CARD.meta))

## Add indexes to data table
CARD.meta$diversity_shannon <- CARD_tab$diversity_shannon

# Check whether the distribution of the diversity is normal
hist(CARD.meta$diversity_shannon)
shapiro.test(CARD.meta$diversity_shannon)
qqnorm(CARD.meta$diversity_shannon)

# Create a list of pairwise comparisons
div.country <- levels(CARD.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])
print(country.pairs)

# Gini Simpson
## Take a look at the indexes
CARD_tab <-microbiome::alpha(CARD_PHY, index = "diversity_gini_simpson")
kable(head(CARD_tab))

## Add indexes to data table
CARD.meta$diversity_gini_simpson <- CARD_tab$diversity_gini_simpson

# Check whether the distribution of the diversity is normal
hist(CARD.meta$diversity_gini_simpson)
shapiro.test(CARD.meta$diversity_gini_simpson)
qqnorm(CARD.meta$diversity_gini_simpson)

# Inverse Simpson
## Take a look at the indexes
CARD_tab <-microbiome::alpha(CARD_PHY, index = "diversity_inverse_simpson")
kable(head(CARD_tab))

## Add indexes to data table
CARD.meta$diversity_inverse_simpson <- CARD_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(CARD.meta$diversity_inverse_simpson)
shapiro.test(CARD.meta$diversity_inverse_simpson) 
qqnorm(CARD.meta$diversity_inverse_simpson)
```
## Diversity plots, CARD
```{r}
anova <- aov(CARD.meta$diversity_shannon ~ country, data=CARD.meta)
TukeyHSD(anova)

# Shannon
font <- 3
p_shannon_CARD <- ggboxplot(CARD.meta, x = "country", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "wilcox.test", size = font) +
#  stat_compare_means(label.y = 6.5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Gini Simpson
font <- 3
p_gsimp_CARD <- ggboxplot(CARD.meta, x = "country", y = "diversity_gini_simpson",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "wilcox.test", size = font) +
#  stat_compare_means(label.y = 0.5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Inverse Simpson
font <- 3
p_isimp_CARD <- ggboxplot(CARD.meta, x = "country", y = "diversity_inverse_simpson",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
#  stat_compare_means(comparisons = country.pairs, method = "t.test") +
#  stat_compare_means(label.y = 5, size = font) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 20))

# Plot all
leg <- get_legend(p_shannon_CARD + theme(legend.position="right"))
ggarrange(plot_grid(p_shannon_CARD, p_gsimp_CARD, p_isimp_CARD,
                    labels = c("Shannon diversity, CARD", 
                               "Gini Simpson diversity, CARD", 
                               "Inverse Simpson diversity, CARD"), ncol = 2, label_size = 10, leg))

anova <- aov(CARD.meta$diversity_shannon ~ location, data=CARD.meta)
TukeyHSD(anova)

p_shannon_CARD <- ggboxplot(CARD.meta, x = "location", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 10, angle = 60, hjust = 1),
        axis.title.y = element_text(size = 18),
        title = element_text(size = 14),
        legend.key.size = unit(1.5, "cm"),
        legend.position = "right") +
  guides(fill = guide_legend(title.theme = element_blank(), size = 7))
p_shannon_CARD + ggtitle("Shannon diversity, CARD")
```
## MGE
```{r, results='hide'}
# Shannon
## Take a look at the indexes
MGE_tab <-microbiome::alpha(MGE_PHY, index = "diversity_shannon")
kable(head(MGE_tab))

## Create data table
MGE.meta <- meta(MGE_PHY)
kable(head(MGE.meta))

## Add indexes to data table
MGE.meta$diversity_shannon <- MGE_tab$diversity_shannon

# Check whether the distribution of the diversity is normal
hist(MGE.meta$diversity_shannon)
shapiro.test(MGE.meta$diversity_shannon)
qqnorm(MGE.meta$diversity_shannon)

# Create a list of pairwise comparisons, country
div.country <- levels(MGE.meta$country)
country.pairs <- combn(seq_along(div.country), 2, simplify = FALSE, FUN = function(i)div.country[i])
print(country.pairs)

# Create a list of pairwise comparisons, location
div.location <- levels(MGE.meta$location)
location.pairs <- combn(seq_along(div.location), 2, simplify = FALSE, FUN = function(i)div.location[i])

# Gini Simpson
## Take a look at the indexes
MGE_tab <-microbiome::alpha(MGE_PHY, index = "diversity_gini_simpson")
kable(head(MGE_tab))

## Add indexes to data table
MGE.meta$diversity_gini_simpson <- MGE_tab$diversity_gini_simpson

# Check whether the distribution of the diversity is normal
hist(MGE.meta$diversity_gini_simpson)
shapiro.test(MGE.meta$diversity_gini_simpson)
qqnorm(MGE.meta$diversity_gini_simpson)

# Inverse Simpson
## Take a look at the indexes
MGE_tab <-microbiome::alpha(MGE_PHY, index = "diversity_inverse_simpson")
kable(head(MGE_tab))

## Add indexes to data table
MGE.meta$diversity_inverse_simpson <- MGE_tab$diversity_inverse_simpson

# Check whether the distribution of the diversity is normal
hist(MGE.meta$diversity_inverse_simpson)
shapiro.test(MGE.meta$diversity_inverse_simpson) 
qqnorm(MGE.meta$diversity_inverse_simpson)
```
## Diversity plots, MGE
```{r}
# country
anova <- aov(MGE.meta$diversity_shannon ~ location, data=MGE.meta)
TukeyHSD(anova)

p_shannon_MGE <- ggboxplot(MGE.meta, x = "location", y = "diversity_shannon",
 add = "boxplot", fill = "country", palette = c("#C0392B", "#FFB000", "#785EF0")) +
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 10, angle = 60, hjust = 1),
        axis.title.y = element_text(size = 18),
        title = element_text(size = 14),
        legend.key.size = unit(1.5, "cm"),
        legend.position = "right") +
  guides(fill = guide_legend(title.theme = element_blank(), size = 7))
p_shannon_MGE + ggtitle("Shannon diversity, MGE")
```
# Taxonomical composition
## Top 15 genera merged by location, Metaphlan3
```{r}
metaphlan_PHY_genus <- tax_glom(metaphlan_PHY, taxrank = "Genus")
metaphlan_PHY_genus_abun <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_genus), TRUE)[1:15]), 
    metaphlan_PHY_genus)
merged <- merge_samples(metaphlan_PHY_genus_abun, "location")

# Normalize by number of samples in each sample type
otu_table(merged) <- otu_table(merged)[, ]/as.matrix(table(sample_data(metaphlan_PHY)$location))[,1]

# Set colors
nb.cols <- 15
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)
# Facet labs
country.labs <- c("1" = "Benin", "2" = "Burkina Faso", "3" = "Finland")

metaphlan_top15genus <- plot_bar(merged, fill = "Genus", title = "15 most abundant genera, hospital waste waters, Metaphlan3") +
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country = country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=40, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
        ylab("Relative abundance (%)") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5,  title = "Genus"))
metaphlan_top15genus
```
# Result boxes per sample
```{r}
#MP
## set colors
mp_cols <- get_palette(c("#00798c", "#d1495b", "#edae49", "#66a182", "#2e4057", "#8d96a3"), 15)

metaphlan_PHY_genus <- tax_glom(metaphlan_PHY, taxrank = "Genus")
mp0 <- subset_samples(metaphlan_PHY_genus_abun, name=="BH50_S111")

mp0_abun <- prune_taxa(names(sort(taxa_sums(mp0), TRUE)[1:15]), mp0)

mp <- plot_bar(mp0_abun, fill = "Genus") +
  geom_bar(stat="identity") +
  labs(x="", y="Relative abundance %", title="Top 12 genera, Metaphlan3") +
  scale_fill_manual(values = mp_cols) +
  theme_classic() +
  theme(axis.text.x = element_blank(), 
        legend.text = element_text(size = 14, face = "italic", family = "Times"),
        plot.title = element_text(size = 20, family = "Times"),
        title = element_text(family = "Times", size = 16),
        axis.title = element_text(size = 16),
        axis.text = element_text(size = 12),
        panel.background = element_rect(fill = "#FFF9E7"))

## Pie chart, Metaphlan3
pie_rel <- as.data.frame(otu_table(mp0_abun))

pie_tax <- as.data.frame.table(tax_table(mp0_abun))
pie_tax <- subset(pie_tax, Var2!="Kingdom" & Var2!="Phylum" & Var2!="Class" & Var2!="Order" & Var2!="Family" & Var2!="Species")
pie_rel$Genus <- pie_tax$Freq


# Pie Chart with Percentages
df <- data.frame(group <- pie_rel$Genus,
                 value <- round(pie_rel$BH50_S111/sum(pie_rel$BH50_S111)*100))
colnames(df) <- c("Genus", "Rounded abundance of top 12 genera (%)")

bp<- ggplot(df, aes(x="", y=value, fill=group))+
geom_bar(width = 1, stat = "identity")

pie_mp <- bp + coord_polar("y", start=0) +
  scale_fill_manual(values = mp_cols) +
  labs(x="", y="Rounded abundance of top 16 genera (%)", title="Top 12 genera, Metaphlan3") +
  guides(fill=guide_legend(ncol=2)) +
  theme_classic() +
  theme(axis.text.x = element_blank(), 
        legend.text = element_text(size = 14, family = "Times", face = "italic"),
        legend.title = element_blank(),
        plot.title = element_text(size = 20, family = "Times"),
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.line = element_blank(),
        legend.key = element_rect(colour = "black"),
        legend.position = "bottom",
        panel.background = element_rect(fill = "#FFF9E7"),
        legend.background = element_rect(fill = "#FFF9E7"),
        legend.spacing.x = unit(0.55, 'cm'),
        legend.box.margin=margin(-20,-20,-20,-20))

# RF genes
## Shorten gene names
tax_table_resfinder$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", tax_table_resfinder$Gene)
## Phyloseq object with shortened names
resfinder_PHY <- phyloseq(otu_table(ARG_genemat_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_resfinder)))

# ARGs
## set colors
gen_cols <- get_palette(c("#FFFDF6", "#EABE94", "#0B775E", "#35274A" ,"#F2300F"), 10)

resfinder_PHY_gene <- tax_glom(resfinder_PHY, taxrank = "Gene")
rf0 <- subset_samples(resfinder_PHY_gene, name=="BFH31_S149")
rf0_abun <- prune_taxa(names(sort(taxa_sums(rf0), TRUE)[1:10]), rf0)

rf <- plot_bar(rf0_abun, fill = "Gene") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.25)) +
  geom_bar(stat="identity") +
  labs(x="", y="ARGs / 16s rRNA", title="Top 10 genes, ResFinder") +
  scale_fill_manual(values = gen_cols) +
  theme_classic() +
  theme(axis.text.x = element_blank(), 
        legend.text = element_text(size = 14, face = "italic", family = "Times"),
        plot.title = element_text(size = 20, family = "Times"),
        title = element_text(family = "Times", size = 16),
        axis.title = element_text(size = 16),
        axis.text = element_text(size = 12),
        panel.background = element_rect(fill = "#FFF9E7"))

## RF classes
## set colors
class_cols <- get_palette(c('#2A363B','#019875','#99B898','#FECEA8','#FF847C','#E84A5F','#C0392B','#96281B'), 17)

rfc0 <- subset_samples(resfinder_PHY, name=="BH50_S111")

rfc <- plot_bar(rfc0, fill = "Class") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,2)) +
  labs(x="", y="", title="All ARGs by classes, ResFinder") +
  scale_fill_manual(values = class_cols) +
  theme_classic() +
  theme(axis.text.x = element_blank(), 
        legend.text = element_text(size = 14, family = "Times"),
        plot.title = element_text(size = 20, family = "Times"),
        title = element_text(family = "Times", size = 16),
        axis.title = element_text(size = 16),
        axis.text = element_text(size = 12),
        panel.background = element_rect(fill = "#FFF9E7"))

## Add title and subtitle
title_gg <- ggplot() + 
  labs(title = "BH50_S111", subtitle = "mainly collecting toilet waters, sample from under floating stuff") +
  theme_minimal() +
  theme(title = element_text(family = "Times", face = "bold", size = 30),
        plot.subtitle = element_text(family = "Times", face = "italic", size = 20))

# PLOT
plot_all <- plot_grid(pie_mp, NULL, rf, rfc, ncol = 4, rel_widths = c(1.2, .15, 1, 1.2))
myplot <- plot_grid(title_gg, plot_all, ncol = 1, rel_heights = c(0.15, 1))
```
# Most abundant among all samples & both as pie charts
```{r}
metaphlan_PHY_genus <- tax_glom(metaphlan_PHY, taxrank = "Genus")
metaphlan_PHY_genus_abun <- prune_taxa(names(sort(taxa_sums(metaphlan_PHY_genus), TRUE)[1:12]), metaphlan_PHY_genus)
mp0 <- subset_samples(metaphlan_PHY_genus_abun, name=="FH2_S163")

mp_cols <- get_palette(c("#00798c", "#d1495b", "#edae49", "#66a182", "#2e4057", "#8d96a3"), 12)

## Pie chart, Metaphlan3
pie_mp_rel <- as.data.frame(otu_table(mp0))
pie_mp_tax <- as.data.frame.table(tax_table(mp0))
pie_mp_tax <- subset(pie_mp_tax, Var2!="Kingdom" & Var2!="Phylum" & Var2!="Class" & Var2!="Order" & Var2!="Family" & Var2!="Species")
pie_mp_rel$Genus <- pie_mp_tax$Freq

# Pie Chart with Percentages
df_mp <- data.frame(group <- pie_mp_rel$Genus,
                 value <- round(pie_mp_rel$FH2_S163/sum(pie_mp_rel$FH2_S163)*100))
colnames(df_mp) <- c("Genus", "Rounded abundance of top 12 genera (%)")

mp<- ggplot(df_mp, aes(x="", y=value, fill=group))+
geom_bar(width = 1, stat = "identity")

pie_mp <- mp + coord_polar("y", start=0) +
  scale_fill_manual(values = mp_cols) +
  labs(x="", fill="Genus", title="Top 12 genera, Metaphlan3") +
  guides(fill=guide_legend(ncol=1)) +
  theme_classic() +
  theme(axis.text.x = element_blank(), 
 #       legend.text = element_text(size = 22, family = "Times", face = "italic"),
 #       legend.title = element_text(size = 26, family = "Times", face = "bold"),
        plot.title = element_text(size = 36, family = "Times", hjust = 0.5),   #28
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.line = element_blank(),
#       legend.key = element_rect(colour = "black"),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFF9E7"),
#        legend.background = element_rect(fill = "#FFF9E7"),
#        legend.spacing.x = unit(0.7, 'cm'),
#        legend.box.margin=margin(-10, -10, -10, -10),
#        legend.key.size = unit(0.9, "cm")
)

# save to get the legend
#ggsave(filename = "mp_legend.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# RF Class
resfinder_PHY_class <- tax_glom(resfinder_PHY, taxrank = "Class")
resfinder_PHY_class_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_class), TRUE)[1:17]), resfinder_PHY_class)
rfc0 <- subset_samples(resfinder_PHY_class_abun, name=="FH2_S163")

## Pie chart, ResFinder
pie_rfc_rel <- as.data.frame(otu_table(rfc0))
pie_rfc_tax <- as.data.frame.table(tax_table(rfc0))
pie_rfc_tax <- subset(pie_rfc_tax, Var2!="Gene")
pie_rfc_rel$Class <- pie_rfc_tax$Freq

# Pie Chart with Percentages
df_rfc <- data.frame(Class <- pie_rfc_rel$Class,
                 Abundance <- round(pie_rfc_rel$FH2_S163/sum(pie_rfc_rel$FH2_S163)*100))
colnames(df_rfc) <- c("Class", "Abundance")

rfc<- ggplot(df_rfc, aes(x="", y=Abundance, fill=Class))+
geom_bar(width = 1, stat = "identity")

class_cols <- get_palette(c('#2A363B','#019875','#99B898','#FECEA8','#FF847C','#E84A5F','#C0392B','#96281B'), 17)

pie_rfc <- rfc + coord_polar("y", start=0) +
  scale_fill_manual(values = class_cols) +
  labs(x="", fill="Antibiotic class", title="ARGs by AB classes, ResFinder") +
  guides(fill=guide_legend(ncol=1)) +
  theme_classic() +
  theme(axis.text.x = element_blank(), 
#        legend.text = element_text(size = 22, family = "Times"),
#        legend.title = element_text(size = 26, family = "Times", face = "bold"),
        plot.title = element_text(size = 36, family = "Times", hjust = 0.5),       #28
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.line = element_blank(),
#       legend.key = element_rect(colour = "black"),
        legend.position = "none",
        panel.background = element_rect(fill = "#FFF9E7"),
#        legend.background = element_rect(fill = "#FFF9E7"),
#        legend.spacing.x = unit(0.7, 'cm'),
#        legend.box.margin=margin(-10, -10, -10, -10),
#        legend.key.size = unit(0.9, "cm")
)

# save to get the legend
#ggsave(filename = "rfc_legend.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

## Plot bar
samples_to_keep <- c("FH6_S167", "FH1_S162", "FH5_S166", "FH8_S169", "FH3_S164", "FH9_S170", "FH7_S168", "FH4_S165", "FH2_S163")
resfinder_PHY_bar <- prune_samples(samples_to_keep, resfinder_PHY)

sums <- as.table(sample_sums(resfinder_PHY_bar))
df_sums <- as.data.frame(sums)

colnames(df_sums) <- c("name","abundance")
  
p_bar <- ggplot(data=df_sums, aes(x=name, y=abundance, fill=name)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=c("black", "black", "black", "black", "black", "black", "black", "black", "#9986A5")) +
  labs(title="Realtive abundance of ARGs", y = "ARGs / 16s rRNA") +
  theme_classic() +
  theme(legend.position = "none",
        axis.title = element_blank(),
       axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background = element_rect(fill = "#FFF9E7"),
        axis.title.y = element_text(size = 24, family = "Times"),                      #18
        plot.title = element_text(size = 30, family = "Times", hjust = 0.5)) +         #24
  geom_text(aes(label = round(abundance, digits = 2)), position = position_dodge(0.9),
                  vjust = 1.5, color = "white", family="Times", size= 9)

## Add title and subtitle
title_gg <- ggplot() + 
  labs(title = "FH2_S163", subtitle = "Meilahti hospital main drain") +
  theme_minimal() +
  theme(title = element_text(family = "Times", face = "bold", size = 60),
        plot.subtitle = element_text(family = "Times", face = "italic", size = 30))

# plot
#plot_all <- plot_grid(title_gg, p_bar, pie_mp, pie_rfc, rel_widths = c(1, 1, 0.000001), rel_heights = c(0.2, 0.85, 0.85, 0.1)) # no vjust

plot_all <- plot_grid(title_gg, p_bar, pie_mp, pie_rfc, rel_widths = c(1, 1, 0.000001), rel_heights = c(0.3, 0.95, 0.95, 0.1))
```
## List 5 most abundant ARGs
```{r}
resfinder_PHY_gene <- tax_glom(resfinder_PHY, taxrank = "Gene")
rf <- subset_samples(resfinder_PHY_gene, name=="FH9_S170")
rf_abun <- prune_taxa(names(sort(taxa_sums(rf), TRUE)[1:5]), rf)

genes <- as.data.frame(rf_abun@tax_table, rownames=NULL)
View(genes$Gene)
```
# Heatmaps and bar plots
## Finnish hospitals
```{r}
## Shorten gene names
tax_table_resfinder$Gene <- gsub(pattern = "_[A-Z].*", replacement = "", tax_table_resfinder$Gene)

# Load ARG count data which has "1" instead of "0"
ARG_genemat <-as.matrix(read.table("mod_ARG_genemat.txt", header= T, check.names = F, row.names = 1))
ARG_genemat_heat <- ARG_genemat + 1
ARG_genemat_heat_norm <- t(t(ARG_genemat_heat)/metadata$SSU_counts)
identical(ARG_genemat_heat[2020, 4]/metadata$SSU_counts[4], ARG_genemat_heat_norm[2020, 4])
# Save and load again to exclude row.names
write.table(ARG_genemat_heat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_heat_norm.txt", row.names=T, sep = "\t", col.names = T)
ARG_genemat_heat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_heat_norm.txt", row.names=NULL)
ARG_genemat_heat_norm$row.names<-NULL
# Phyloseq
resfinder_PHY_heat <- phyloseq(otu_table(ARG_genemat_heat_norm, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_resfinder)))
# Filter
resfinder_PHY_heat = subset_samples(resfinder_PHY_heat, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")

# ARG class plot
rfc_cols <- get_palette(c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255", "#5F5E98", "#331225"), 17)

## Plot bar, RF
samples_to_keep <- c("FH1_S162",	"FH3_S164",	"FH2_S163", "FH4_S165","FH5_S166", "FH6_S167", "FH9_S170", "FH7_S168", "FH8_S169")

rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))

## Save plot
#ggsave(filename = "Finland_rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs, RF
## Plot
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))

## Save plot
#ggsave(filename = "Finland_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs, RF
## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("FH4_S165", "FH5_S166", "FH6_S167", "FH3_S164", "FH2_S163", "FH1_S162", "FH9_S170", "FH8_S169",  "FH7_S168")
heat.df <- heat.df[, col_order]
# Rename colnames
colnames(heat.df) = c("FH4_S165" = "Main drain",
                      "FH5_S166" = "Cancer, hematology & \nmaxillofacial surgery clinic",
                      "FH6_S167" = "Emergency & \nday hospital",
                      "FH1_S162" = "Main drain \n(Triangle Hospital)",
                      "FH2_S163" = "Main drain \n(Meilahti Hospital)",
                      "FH3_S164" = "Main drain \n(New Children's Hospital)",
                      "FH9_S170" = "H section of drain",
                      "FH7_S168" = "Main drain",
                      "FH8_S169" = "Main drain \n(technical replicate)")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 20, cellheight = 10, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 5, fontsize_col = 6, 
                        labels_row = as.expression(newnames), gaps_col = rep(c(3, 6, 7)),
                        filename = "Finland_heat.png")
custom_heat
```
## Feces, Menontin
```{r}
## Plot bar
samples_to_keep <- c("BH20_S88", "BH22_S89", "BH24_S90", "BH25_S91")
rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))

## Save plot
#ggsave(filename = "Feces_rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs
## Plot
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))

## Save plot
#ggsave(filename = "Feces_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs, RF
## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("BH20_S88", "BH22_S89", "BH24_S90", "BH25_S91")
heat.df <- heat.df[, col_order]

# Rename
colnames(heat.df) <- c("BH20_S88" = "50 y, 1 month hosp., \nfracture of thighbone",
                            "BH22_S89" = "33 y, 1 month hosp., \nfracture of thighbone",
                            "BH24_S90" = "24 y, 1 month hosp., \npolytrauma",
                            "BH25_S91" = "27 y, 2 months hosp., \nosteosynthesis")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

heat_num = as.matrix(heat.df[1:4,])

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 35, cellheight = 10, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 5, fontsize_col = 6, 
                        labels_row = as.expression(newnames),
                        filename = "Feces_heat.png")
custom_heat
```

## Savalou
```{r}
## Plot bar
samples_to_keep <- c("BSE74_S119",	"BSE79_S120",	"BSE93_S121",	"BSE100_S122")
rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))

## Save plot
#ggsave(filename = "Savalou_rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))
## Save plot
#ggsave(filename = "Savalou_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs, RF
## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("BSE100_S122", "BSE74_S119", "BSE79_S120", "BSE93_S121")
heat.df <- heat.df[, col_order]
# rename cols
colnames(heat.df) <- c("BSE100_S122" = "River Caiman, Ouesse village, \nDrinking water", 
                       "BSE74_S119" = "River Zoun, drinking water \nfor the village people \nwho live in the farm",
                       "BSE79_S120" = "Bedrock river, Aglomodjodji village, \nritual purposes and drinking",
                       "BSE93_S121" = "Aglomodjodji village, \ndrinking water")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 35, cellheight = 10, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 5, fontsize_col = 6, 
                        labels_row = as.expression(newnames),
                        filename = "Savalou_heat.png")
custom_heat
```

# Other Hospitals in BF outside Ougadougu
```{r}
## Plot bar
samples_to_keep <- c("BFH1_S123",	"BFH4_S124",	"BFH16_S134",	"BFH17_S135",	"BFH18_S136",	"BFH19_S137",	"BFH20_S138",	"BFH21_S139",	"BFH22_S140",	"BFH23_S141",	"BFH25_S143",	"BFH26_S144",	"BFH27_S145",	"BFH28_S146")
rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))

## Save plot
#ggsave(filename = "OtherBF_rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white")) 
## Save plot
#ggsave(filename = "OtherBF_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs, RF
## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("BFH16_S134", "BFH17_S135", "BFH18_S136", "BFH19_S137", "BFH20_S138", "BFH21_S139", "BFH22_S140", "BFH23_S141", "BFH25_S143", "BFH26_S144", "BFH27_S145", "BFH28_S146", "BFH1_S123", "BFH4_S124")
heat.df <- heat.df[, col_order]
# rename cols 
colnames(heat.df) <- c(
                            "BFH16_S134" = 	"Pediatric hospitalization",
                            "BFH17_S135" = 	"Medicine hospitalization",
                            "BFH18_S136" = 	"Delivery room",
                            "BFH19_S137" =	"Gynecology, operating room",
                            "BFH20_S138" = 	"Maternity hospitalization",
                            "BFH21_S139" =	"Surgery emergency, hospitalization",
                            "BFH22_S140" =	"Laboratory",
                            "BFH23_S141" =	"Emergency, external consultation",
                            "BFH25_S143" = 	"Big well",
                            "BFH26_S144" = 	"Exit after treatment \n(municipality channel)",
                            "BFH27_S145" =	"Marigot river",
                            "BFH28_S146" =  "Pediatric emergency",
                            "BFH1_S123" = "Big well no. 1",
                            "BFH4_S124" = "Pediatrics")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 20, cellheight = 10, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 5, fontsize_col = 6, 
                        labels_row = as.expression(newnames), gaps_col = 12,
                        filename = "OtherBF_heat.png")
custom_heat
```

# SOUKA & Sour de Vie, Ougadougou
```{r}
## Plot bar
samples_to_keep <- c("BFH6_S125",	"BFH8_S126",	"BFH9_S127",	"BFH10_S128",	"BFH11_S129",	"BFH12_S130",	"BFH13_S131",	"BFH14_S132",	"BFH15_S133")
rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))
                            
## Save plot
#ggsave(filename = "Ougadougou_rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE)) +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))
## Save plot
#ggsave(filename = "Ougadougou_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs, RF
## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("BFH10_S128", "BFH11_S129", "BFH12_S130", "BFH6_S125","BFH8_S126", "BFH9_S127", "BFH13_S131", "BFH14_S132", "BFH15_S133")
heat.df <- heat.df[, col_order]
#rename cols
colnames(heat.df) <- c("BFH10_S128" = "Delivery room, surgery",
                       "BFH11_S129" =	"Laboratory",
                       "BFH12_S130" =	"Pediatrics",
                       "BFH6_S125" = "Hospitalization, \nmaternity, surgery",
                       "BFH8_S126" = "Surgery",
                       "BFH9_S127" =	"Mortuary",
                       "BFH13_S131" =	"Delivery room",
                       "BFH14_S132" =	"Hospitalization, \nmaternity",
                       "BFH15_S133" =	"Hospitalization, \nmaternity, surgery")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 20, cellheight = 10, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 5, fontsize_col = 6, 
                        labels_row = as.expression(newnames), gaps_col = 6,
                        filename = "Ougadougou_heat.png")
custom_heat
```

# Yalgado
```{r}
## Plot bar
samples_to_keep <- c("BFH29_S147",	"BFH30_S148",	"BFH31_S149",	"BFH32_S150",	"BFH33_S151",	"BFH34_S152",	"BFH35_S153",	"BFH36_S154",	"BFH37_S155",	"BFH38.A_S156",	"BFH38.B_S157",	"BFH39_S158",	"BFH40_S159",	"BFH41_S160",	"BFH42_S161")
rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))
                            
## Save plot
#ggsave(filename = "Yalgado_rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free") +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))
## Save plot
#ggsave(filename = "Yalgado_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs, RF
## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("BFH29_S147", "BFH30_S148", "BFH31_S149", "BFH32_S150", "BFH33_S151", "BFH34_S152", "BFH35_S153", "BFH36_S154", "BFH37_S155", "BFH38.A_S156", "BFH38.B_S157", "BFH39_S158", "BFH40_S159", "BFH40_S159", "BFH41_S160", "BFH42_S161")
heat.df <- heat.df[, col_order]
# rename cols
colnames(heat.df) <- c("BFH29_S147" = "Urology, infectious diseases, pneumology",
                           "BFH30_S148" = "Urology, infectious diseases, \npneumology, daily hospital, \ndialysis, blood bank",	
                           "BFH31_S149" = "Anatomy pathology laboratory, \nophtalmology, laboratory of \nhematology-biochemestry, surgery",	
                           "BFH32_S150" = "Hospitalization, visceral surgery, \ngastrology, internal medicine, \npeadiatric emergency",	
                           "BFH33_S151" = "Kitchen, laboratory, trauma, \nendoscopy, peadiatrics 2, \ncardiology consultation",	
                           "BFH34_S152" = "Bacteriology lab",	
                           "BFH35_S153" = "Operating room, neurosurgery, \ntraumatology",	
                           "BFH36_S154" = "Operating room, maternity",	
                           "BFH37_S155" = "Mortuary",	
                           "BFH38.A_S156" = "Delivery room",	
                           "BFH38.B_S157" = "Delivery room (technical replicate)",	
                           "BFH39_S158" = "Emergency, resuscitation, \ndental surgery, \nnuclear medicine",	
                           "BFH40_S159" = "Mortuary, emergency, maternity, \nvisceral surgery emergency operating room",	
                           "BFH41_S160" = "Connection point to ONEA WWTP channel",	
                           "BFH42_S161" = "Kossodo river after ONEA WWTP treatment")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 30, cellheight = 15, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 8, fontsize_col = 7, 
                        labels_row = as.expression(newnames), filename = "Yalgado_heat.png")
custom_heat
```

# Calavi
```{r}
## Plot bar
samples_to_keep <- c("BH01_S76",	"BH02_S77",	"BH03_S78",	"BH04_S79",	"BH05_S80",	"BH06_S81",	"BH07_S82",	"BH09_S83",	"BH10_S84",	"BH11_S85",	"BH58_S113",	"BH59_S114",	"BH60_S115",	"BH61_S116",	"BH62_S117")
rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))
                            
## Save plot
#ggsave(filename = "Calavi_rfc.png",
#      width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free") +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))
## Save plot
#ggsave(filename = "Calavi_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("BH58_S113", "BH59_S114", "BH60_S115", "BH61_S116", "BH62_S117", "BH01_S76", "BH02_S77", "BH03_S78", "BH04_S79", "BH05_S80", "BH06_S81", "BH07_S82", "BH09_S83", "BH10_S84", "BH11_S85")
heat.df <- heat.df[, col_order]
#rename cols
colnames(heat.df) <- c("BH58_S113" = "Sump next to BH60 (2 m)",	
                           "BH59_S114" = "Sump next to BH60 (10 cm)",	
                           "BH60_S115" = "Emergency, surgery, ophtamology \nand ward sections, (50 cm)",	
                           "BH61_S116" = "Maternity, surgery and \nlaboratory sections (1 m)",	
                           "BH62_S117" = "Maternity, surgery and \nlaboratory sections (10  cm)",
                           "BH01_S76" = "Neonatology",
                           "BH02_S77" = "Pediatrics",	
                           "BH03_S78" = "Surgery room",	
                           "BH04_S79" = "Medicine ward, \nempty sump",	
                           "BH05_S80" = "Maternity ward",	
                           "BH06_S81" = "Mortuary ward",	
                           "BH07_S82" = "Laundry room, \nopen pit sump",	
                           "BH09_S83" = "Surgery room for \ndemanding operations, sump",	
                           "BH10_S84" = "Surgery room for \ndemanding operation",	
                           "BH11_S85" = "Well 100 m \naway from the hospital")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 20, cellheight = 10, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 6, fontsize_col = 7, 
                        labels_row = as.expression(newnames), gaps_col = 5,
                        filename = "Calavi_heat.png")
custom_heat
```

# Other Hospitals in Benin
```{r}
## Plot bar
samples_to_keep <- c("BH13_S86",	"BH14_S87",	"BH27_S92",	"BH29_S93",	"BH30_S94",	"BH31_S95",	"BH32_S96",	"BH33_S97",	"BH34.A_S98",	"BH34.B_S99",	"BH35_S100",	"BH36_S101",	"BH37_S102",	"BH38_S103",	"BH39_S104",	"BH44_S105",	"BH45_S106",	"BH46_S107",	"BH47_S108",	"BH48_S109",	"BH49_S110",	"BH50_S111",	"BH52_S112")
rfc0 <- prune_samples(samples_to_keep, resfinder_PHY_heat)

rfc <- plot_bar(rfc0, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))
                            
## Save plot
#ggsave(filename = "OtherBenin_rfc.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

# Selected ARGs
resfinder_PHY_gene <- tax_glom(rfc0, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaOXA-10_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free") +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))
## Save plot
#ggsave(filename = "OtherBenin_ARGs.png",
#       width = 16, height = 13, dpi = 300, units = "in", device='png', scale = 1)

## Heatmap
# Extract abundance matrix from the phyloseq object
heat_OTU = as(otu_table(rf_select), "matrix")
# Coerce to data.frame
heat.df = as.data.frame(heat_OTU)

# Change otu names from sp* into gene names
colnames(heat.df)
# reorder
col_order <- c("BH44_S105", "BH45_S106", "BH46_S107", "BH47_S108", "BH48_S109", "BH49_S110", "BH50_S111", "BH52_S112", "BH13_S86", "BH14_S87", "BH27_S92", "BH29_S93", "BH30_S94", "BH31_S95", "BH32_S96", "BH33_S97", "BH34.A_S98", "BH34.B_S99", "BH35_S100", "BH36_S101", "BH37_S102", "BH38_S103", "BH39_S104")
heat.df <- heat.df[, col_order]
# rename cols
colnames(heat.df) <- c("BH44_S105" = "Central ST (10 cm)",	
                           "BH45_S106" = "Central ST (1 m) ",	
                           "BH46_S107" = "Pediatric clinic (20 cm)",	
                           "BH47_S108" = "Hospitalization ward, \nvaccination room",	
                           "BH48_S109" = "Water pool surrounding \nthe surgery room",
                           "BH49_S110" = "Maternity clinic",
                           "BH50_S111" = "Peripheral ST collecting \nmainly toilet waters",	
                           "BH52_S112" = "Tank for washing hands",	
                           "BH13_S86" = "Street gutter 100 m \naway from the hospital, water",	
                           "BH14_S87" = "Street gutter 100 m \naway from the hospital, soil",	
                           "BH27_S92" = "Central ST for ICU, \nlaundry room, laboratory unit \nand maternal & child health unit",	
                           "BH29_S93" = "Surgery unit, open tank",	
                           "BH30_S94" = "Surgery unit 1st peripheral ST",	
                           "BH31_S95" = "Surgery unit 2nd pheripheral ST",	
                           "BH32_S96" = "Moist soil between \nBH31 & BH34",
                           "BH33_S97" = "Surgery unit 3rd \nperipheral ST",
                           "BH34.A_S98" = "Surgery unit central ST, (10cm)",
                           "BH34.B_S99" = "Surgery unit central ST, (10cm) \n(technical replicate)",
                           "BH35_S100" = "Surgery for minor operations",
                           "BH36_S101" = "Maternity ward",
                           "BH37_S102" = "Central ST for BH35 \nand BH36 (surface layer)",
                           "BH38_S103" = "Central ST for BH35 \nand BH36 (1,4 cm)",
                           "BH39_S104" = "ST collecting all laboratory \nwaters from the hospital")

# So same to the tax_table
# Extract abundance matrix from the phyloseq object
heat_tax = as(tax_table(rf_select), "matrix")
# Save to get it back without the current first header row
write.table(heat_tax, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=T, sep = "\t", col.names = F)
heat_tax <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/heat_tax", row.names=1, header = F)
# as dataframe
heat_tax.df = as.data.frame(heat_tax)
# Swap colnames
rownames(heat.df) <- heat_tax.df$V3

## Plot log: palette
newnames <- lapply(
  rownames(heat.df),
  function(x) bquote(italic(.(x))))

custom_heat <- pheatmap(log10(heat.df), cluster_rows = F, cluster_cols = F, 
                        cellwidth = 15, cellheight = 10, border_color = "white", fontsize=15,
                        colorRampPalette(brewer.pal(9, "Blues"))(100), main = "ARGs/16S rRNA, log10", 
                        angle_col = 90, legend = TRUE, fontsize_row = 7, fontsize_col = 5, 
                        labels_row = as.expression(newnames), gaps_col = 8,
                        filename = "OtherBenin_heat.png")
custom_heat
```

# All
```{r}
rfc <- plot_bar(resfinder_PHY, fill = "Class") 
rfc_plot <- rfc +
  geom_bar(stat="identity", color = NA, size = 0) +
  scale_y_continuous(labels = waiver(), limits=c(0,1.35)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = expression(atop(bold("ARG Abundance by Antibiotic Classes"), atop("ARGs / 16S rRNA")))) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 14, family = "Times", angle = 90), 
        axis.text.y = element_text(size = 28, family = "Times", face = "bold", angle = 90),
        axis.title.y = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFFCF3"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free", labeller = label_wrap_gen(width = 15, multi_line = TRUE)) +
  theme(strip.text.x= element_text(angle = 90, size = 30, family = "Times", hjust = 0, vjust = 0.75),
        strip.background = element_rect(colour = "white"))

# Selected ARGs
resfinder_PHY_gene <- tax_glom(resfinder_PHY, taxrank = "Gene")

rf_select <- subset_taxa(resfinder_PHY_gene, 
                         Gene == "blaCMY-2_1" | Gene == "blaKPC-3_1" | Gene == "blaOXA-48_1" | 
                         Gene == "blaOXA-58_1" | Gene == "blaOXA-181_1" | Gene == "blaVIM-5_1" |
                         Gene == "blaNDM-1_1" | Gene == "blaGES-5_1" | Gene == "blaCTX-M-15_1" |
                         Gene == "mcr-1.1_1" | Gene == "mcr-3.15_1" |
                         Gene == "mcr-3.1_1" | Gene == "blaNDM-5_1" | Gene == "blaMOX-6_1" |
                         Gene == "blaKPC-2_1" | Gene == "blaNPS_1" | Gene == "blaACT-4_2")

rf_selected_genes <- plot_bar(rf_select, "name", fill = "Gene") +
  geom_bar(stat="identity") +
  scale_y_continuous(labels = waiver(), limits=c(0,0.0075)) +
  scale_fill_manual(values = rfc_cols) +
  labs(y = "ARGs / 16S rRNA", title = "Abundance of selected ARGs") +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 8, family = "Times", angle = 30, hjust = 1), 
        axis.text.y = element_text(size = 20, family = "Times", face = "bold"),
        axis.title.y = element_text(size = 20, family = "Times"),
        axis.title.x = element_blank(),
        plot.title = element_text(size = 30, family = "Times"),
        legend.text = element_text(size = 18, family = "Times"),
        legend.title = element_blank(),
        legend.key = element_rect(size = 2, color = "white"),
        legend.key.size = unit(1, "cm"),
        legend.spacing.y = unit(2.2, "char"),
        panel.background = element_rect(fill = "#FFF9E7"),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  facet_grid(~location, scales = "free", space = "free") +
  theme(strip.text.x = element_text(size = 14, family = "Times", hjust = 0),
        strip.background = element_rect(colour = "white"))
```

## Heatmap of Metaphlan3 results
```{r, warning=FALSE, message=FALSE}
subs <- subset_samples(metaphlan_PHY, country == "Finland")
metaphlan_PHY_genus <- tax_glom(subs, "Genus")
heat.metaphlan <- plot_taxa_heatmap(metaphlan_PHY_genus,
                                 subset.top = 50,
                                 VariableA = "sample_type",
                                 heatcolors = rev(brewer.pal(12, "GnBu")),
                                 transformation = "log10",
                                 cluster_cols = T)
heat.metaphlan
```
## ARGs, ResFinder
```{r}
# Genes
resfinder_PHY_gene <- tax_glom(resfinder_PHY, taxrank = "Gene")
resfinder_PHY_gene_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_gene), TRUE)[1:15]), resfinder_PHY_gene)

nb.cols <- 15
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)

resfinder_top15genes <- plot_bar(resfinder_PHY_gene_abun, fill = "Gene", title = "Top 15 ARGs, ResFinder") +
  facet_grid(~country, scales = "free_x", space = "free_x") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
  ylab("ARGs/16s rRNA") + xlab("") +
  geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "Gene"))
resfinder_top15genes
```

```{r}
# Class
resfinder_PHY_class <- tax_glom(resfinder_PHY, taxrank = "Class")
resfinder_PHY_class_abun <- prune_taxa(names(sort(taxa_sums(resfinder_PHY_class), TRUE)), resfinder_PHY_class)

nb.cols <- 17
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)

resfinder_topclass <- plot_bar(resfinder_PHY_class_abun, fill = "Class", title = "All ARGs by classes, ResFinder") +
  facet_grid(~country, scales = "free_x", space = "free_x") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
        ylab("ARGs/16s rRNA") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "Class"))
resfinder_topclass
```

## Plot ARGs by CARD
```{r}
CARD_PHY_gene <- tax_glom(CARD_PHY, taxrank = "Gene")
CARD_PHY_gene_abun <- prune_taxa(names(sort(taxa_sums(CARD_PHY_gene), TRUE)[1:15]), CARD_PHY_gene)

nb.cols <- 15
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)
country.labs <- labeller(c(`1` = "Benin", `2` = "Burkina Faso", `3` = "Finland"))

CARD_top15genes <- plot_bar(CARD_PHY_gene_abun, fill = "Gene", title = "Top 15 ARGs, CARD") +
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
        ylab("ARGs/16s rRNA") + xlab("") +
       geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "Gene"))
CARD_top15genes

# By class
CARD_PHY_class <- tax_glom(CARD_PHY, taxrank = "Class")
CARD_PHY_class_abun <- prune_taxa(names(sort(taxa_sums(CARD_PHY_class), TRUE)), CARD_PHY_class)

nb.cols <- 32
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)
country.labs <- labeller(c(`1` = "Benin", `2` = "Burkina Faso", `3` = "Finland"))

CARD_classes <- plot_bar(CARD_PHY_class_abun, fill = "Class", title = "All ARGs by class, CARD") +
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
        ylab("ARGs/16s rRNA") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "Class"))
CARD_classes
```
## Plot MGEs
```{r}
MGE_PHY_gene <- tax_glom(MGE_PHY, taxrank = "Gene")
MGE_PHY_gene_abun <- prune_taxa(names(sort(taxa_sums(MGE_PHY_gene), TRUE)[1:15]), MGE_PHY_gene)

nb.cols <- 15
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)
country.labs <- labeller(c(`1` = "Benin", `2` = "Burkina Faso", `3` = "Finland"))

MGE_top15genes <- plot_bar(MGE_PHY_gene_abun, fill = "Gene", title = "Top 15 MGE genes") +
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
        ylab("MGEs/16s rRNA") + xlab("") +
  scale_fill_manual(values=mycolors) +
  geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "Gene"))
MGE_top15genes

# By Element (=type) (all)
MGE_PHY_element <- tax_glom(MGE_PHY, taxrank = "Element")
MGE_PHY_element_abun <- prune_taxa(names(sort(taxa_sums(MGE_PHY_element), TRUE)), MGE_PHY_element)

nb.cols <- 52
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)
country.labs <- labeller(c(`1` = "Benin", `2` = "Burkina Faso", `3` = "Finland"))

MGE_element.p <- plot_bar(MGE_PHY_element_abun, fill = "Element", title = "All MGEs by type") +
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
        ylab("MGEs/16s rRNA") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "MGE Type"))
MGE_element.p
```

```{r}
# Integrases
MGE_PHY_integrase = subset_taxa(MGE_PHY, Element == "integrase")

nb.cols <- 4
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)
country.labs <- labeller(c(`1` = "Benin", `2` = "Burkina Faso", `3` = "Finland"))

MGE_integrase.p <- plot_bar(MGE_PHY_integrase, fill = "Class", title = "Integrases") +
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank(),
        strip.text = element_text(size=20, color = "white"),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
        ylab("MGEs/16s rRNA") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "Integrase type"))
MGE_integrase.p

# Plasmids by MGE
MGE_PHY_plasmid = subset_taxa(MGE_PHY, Element == "plasmid")
MGE_PHY_plasmid_abun <- prune_taxa(names(sort(taxa_sums(MGE_PHY_plasmid), TRUE)[1:30]), MGE_PHY_plasmid)

nb.cols <- 30
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)
country.labs <- labeller(c(`1` = "Benin", `2` = "Burkina Faso", `3` = "Finland"))

MGE_plasmid.p <- plot_bar(MGE_PHY_plasmid_abun, fill = "Class", title = "Top 30 plasmid replicon types") +
  facet_grid(~country, scales = "free_x", space = "free_x", labeller = labeller(country.labs)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 10),
        axis.title.x = element_blank()) +
        ylab("MGEs/16s rRNA") + xlab("") +
  scale_fill_manual(values=mycolors) + geom_bar(stat="identity") +
  guides(fill = guide_legend(label.theme = element_text(size = 10), keywidth = 1.5, keyheight = 1.5, title = "Plasmid replicon type"))
MGE_plasmid.p
```
## crAssphage
```{r}
# Load data (modified in excel to match metadata order)
crassphage_table <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/mod_crassphage_table.txt", row.names=NULL)
identical(rownames(metadata), colnames(crassphage_table))

crassphage_table_norm <- t(t(crassphage_table)/metadata$total_seqs)

write.table(crassphage_table_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt", row.names=T, sep = "\t", col.names = T)

crassphage_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/crassphage_table_norm.txt", row.names=NULL)
crassphage_norm$row.names<-NULL

# log transform
crassphage_norm_log <- log10(crassphage_norm)

# Create phyloseq
crass_PHY<-phyloseq(otu_table(crassphage_norm, taxa_are_rows = TRUE), sample_data(metadata))
```

```{r}
# Filter samples
# With low quality (BFH24_S142, BH63_S118, FH10_S171)
crass_PHY = subset_samples(crass_PHY, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")
```

```{r}
# Plot
nb.cols <- 10
mycolors <- colorRampPalette(brewer.pal(11, "Spectral"))(nb.cols)

crassphage_plot <-plot_bar(crass_PHY, fill="sample_type", title = "crAssphage abundance normalized to library size, log10") +
  facet_grid(~ location, scales = "free", space = "free") +
  theme_classic() + scale_fill_manual(values = mycolors) +
  theme(axis.text.x = element_text(angle=90, hjust=1, size = 8),
        axis.title.x = element_blank(),
        strip.text = element_text(size=9, color = "white", angle = 90, hjust = 0),
        title = element_text(size = 15),
        strip.background = element_rect(fill="#4F567B")) +
  guides(fill = guide_legend(label.theme = element_text(size = 7), keywidth = 1.2, keyheight = 1.2)) + ylab("Abundance / total sequences (log10)")
crassphage_plot
```

```{r}
# Plot with ARG similarly, with a log change
ARG_genemat <-as.matrix(read.table("mod_ARG_genemat.txt", header= T, check.names = F, row.names = 1))
identical(rownames(metadata), colnames(ARG_genemat))
ARG_genemat_norm <- t(t(ARG_genemat)/metadata$SSU_counts)
identical(ARG_genemat[2020, 4]/metadata$SSU_counts[4], ARG_genemat_norm[2020, 4])
write.table(ARG_genemat_norm, "~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=T, sep = "\t", col.names = T)
ARG_genemat_norm <- read.delim("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/ARG_genemat_norm.txt", row.names=NULL)
ARG_genemat_norm$row.names<-NULL

# log transform
ARG_genemat_norm_log <- log10(ARG_genemat_norm)

tax_table_resfinder <- read.csv("~/Documents/Metagenomes_AMRIWA/R/AMRIWA/RFiles/tax_table_resfinder.txt", header=FALSE, sep=";")
colnames(tax_table_resfinder) <- c("Gene", "Class")

# Combine to phyloseq object
resfinder_PHY_log <- phyloseq(otu_table(ARG_genemat_norm_log, taxa_are_rows = TRUE), sample_data(metadata), 
    tax_table(as.matrix(tax_table_resfinder)))

resfinder_PHY_log = subset_samples(resfinder_PHY_log, name != "BFH24_S142" & name != "BH63_S118" & name != "FH10_S171")
```

# Mantel tests
```{r}
metaphlan_PHY_bacteria <- subset_taxa(metaphlan_PHY, Kingdom %in% c("Bacteria"))
metaphlan_PHY_bacteria <- subset_samples(metaphlan_PHY_bacteria, sample_sums(metaphlan_PHY_bacteria) != 0)

ARG_horn <- vegdist(t(as.matrix(otu_table(resfinder_PHY))), method = "bray")
MGE_horn <- vegdist(t(as.matrix(otu_table(MGE_PHY))), method = "bray")

# intersects, ResFinder & metaphlan
temp1 <- otu_table(metaphlan_PHY_bacteria)[, sample_names(metaphlan_PHY_bacteria) %in% c(intersect(sample_names(metaphlan_PHY_bacteria), 
    sample_names(resfinder_PHY)))]

temp2 <- otu_table(resfinder_PHY)[, sample_names(resfinder_PHY) %in% c(intersect(sample_names(resfinder_PHY), 
    sample_names(metaphlan_PHY_bacteria)))]

resfinder_metaphlan_dist <- vegdist(t(as.matrix(temp2)), method = "bray")


# intersects for MGE & metaphlan
temp3 <- otu_table(metaphlan_PHY_bacteria)[, sample_names(metaphlan_PHY_bacteria) %in% c(intersect(sample_names(metaphlan_PHY_bacteria), 
    sample_names(MGE_PHY)))]

temp4 <- otu_table(MGE_PHY)[, sample_names(MGE_PHY) %in% c(intersect(sample_names(MGE_PHY), 
    sample_names(metaphlan_PHY_bacteria)))]

MGE_metaphlan_dist <- vegdist(t(as.matrix(temp4)), method = "bray")

resfinder_DIST <- vegdist(t(as.matrix(temp1)), method = "bray")
MGE_DIST <- vegdist(t(as.matrix(temp3)), method = "bray")

# Between ARGs/MGEs and taxa
mantel(resfinder_metaphlan_dist, resfinder_DIST, method = "spearman") 

mantel(MGE_metaphlan_dist, MGE_DIST, method = "spearman") 

# Between ARGs and MGEs
temp5 <- otu_table(MGE_PHY)[, sample_names(MGE_PHY) %in% c(intersect(sample_names(MGE_PHY), 
    sample_names(resfinder_PHY)))]
temp6 <- otu_table(resfinder_PHY)[, sample_names(resfinder_PHY) %in% c(intersect(sample_names(resfinder_PHY), 
    sample_names(MGE_PHY)))]

MGE_AM_dist <- vegdist(t(as.matrix(temp5)), method = "bray")
ARG_MA_dist <- vegdist(t(as.matrix(temp6)), method = "bray")
mantel(MGE_AM_dist, ARG_MA_dist, method = "spearman") 
```


























